{
 "metadata": {
  "name": "",
  "signature": "sha256:a1b06a35799e8c22f98b0a8bfb62d4d0c8cf4ed20213ccaf1ff909de55fd8e07"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Setup"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%run _shared_setup.ipynb\n",
      "CACHE_DIR = os.path.join(CACHE_DIR, '20150918_assess_indel_filtering')\n",
      "PROCESSED_ASSEMBLED_SAMPLES_DIR = '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/validation_results'"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "python 3.4.1 |Anaconda 2.1.0 (x86_64)| (default, Sep 10 2014, 17:24:09) \n",
        "[GCC 4.2.1 (Apple Inc. build 5577)]\n",
        "numpy 1.9.2\n",
        "scipy 0.14.0\n",
        "pandas 0.14.1\n",
        "numexpr 2.3.1\n",
        "pysam 0.8.3\n",
        "petl 1.0.10\n",
        "petlx 1.0.3\n",
        "vcf 0.6.7\n",
        "h5py 2.3.1\n",
        "tables 3.1.1\n",
        "vcfplt 0.8\n",
        "mkdir: /nfs/team112_internal/production_files/Pf3k/methods/assembled_samples: File exists\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "tbl_pgv_metadata length = 5729\n",
        "tbl_pgv_locations length = 102\n",
        "tbl_pf3k_metadata length = 2512\n",
        "tbl_pf_solaris length = 10879"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "tbl_assembled_samples length = 11\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "PIPELINES_FN='%s/src/github/malariagen/methods-dev/pf3k_techbm/scripts/gatk_pipeline/pipeline_parameters_20150918.xlsx' % os.environ['HOME']\n",
      "OUTPUT_DIR = '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment'\n",
      "!mkdir -p {OUTPUT_DIR}\n",
      "NUCMER_DIR='~/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23'"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 213
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# CODING_EFFECTS = ['NON_SYNONYMOUS_CODING', 'SYNONYMOUS_CODING', 'STOP_GAINED']\n",
      "NONCODING_EFFECTS = ['INTERGENIC', 'INTRON', 'TRANSCRIPT']\n",
      " "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ref_dict=SeqIO.to_dict(SeqIO.parse(open(REF_GENOME), \"fasta\"))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 282
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ref_dict['Pf3D7_10_v3']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 283,
       "text": [
        "SeqRecord(seq=Seq('TAAACCCTGAACCCTAAACCCTGAACCCTAAACCACTAACCCTAAACCCTGAAC...GTT', SingleLetterAlphabet()), id='Pf3D7_10_v3', name='Pf3D7_10_v3', description='Pf3D7_10_v3', dbxrefs=[])"
       ]
      }
     ],
     "prompt_number": 283
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# location of CRT \"CVMNK\" etc haplotypes \n",
      "ref_dict['Pf3D7_07_v3'][403610:403633]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 285,
       "text": [
        "SeqRecord(seq=Seq('ATGTGTAATGAATAAAATTTTTG', SingleLetterAlphabet()), id='Pf3D7_07_v3', name='Pf3D7_07_v3', description='Pf3D7_07_v3', dbxrefs=[])"
       ]
      }
     ],
     "prompt_number": 285
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "NUCMER_DIR"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 214,
       "text": [
        "'~/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23'"
       ]
      }
     ],
     "prompt_number": 214
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Load data"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tbl_pipelines = etl.fromxlsx(PIPELINES_FN)\n",
      "tbl_pipelines.display(index_header=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>0|PIPELINE_NAME</th>\n",
        "<th>1|MAPPING_DIR</th>\n",
        "<th>2|BAMS_DIR</th>\n",
        "<th>3|UNFILTERED_DIR</th>\n",
        "<th>4|FILTERED_DIR</th>\n",
        "<th>5|ANNOTATED_DIR</th>\n",
        "<th>6|VQSR_ANNOTATIONS_SNP</th>\n",
        "<th>7|VQSR_ANNOTATIONS_INDEL</th>\n",
        "<th>8|MAX_GAUSSIANS_SNP</th>\n",
        "<th>9|MAX_GAUSSIANS_INDEL</th>\n",
        "<th>10|VQSR_TRAINING</th>\n",
        "<th>11|VQSR_PRIOR</th>\n",
        "<th>12|MIN_QUAL</th>\n",
        "<th>13|MAX_QUAL</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>default</td>\n",
        "<td>bwa_mem</td>\n",
        "<td>gatk_rec</td>\n",
        "<td>HaplotypeCaller</td>\n",
        "<td>crosses_truth</td>\n",
        "<td>SnpEff_region</td>\n",
        "<td>-an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -an DP</td>\n",
        "<td>-an QD -an DP -an FS -an SOR -an ReadPosRankSum -an MQRankSum</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>crosses</td>\n",
        "<td style='text-align: right'>15</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>no_QD_indel</td>\n",
        "<td>bwa_mem</td>\n",
        "<td>gatk_rec</td>\n",
        "<td>HaplotypeCaller</td>\n",
        "<td>no_QD_indel</td>\n",
        "<td>SnpEff_region</td>\n",
        "<td>-an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -an DP</td>\n",
        "<td>-an DP -an FS -an SOR -an ReadPosRankSum -an MQRankSum</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>crosses</td>\n",
        "<td style='text-align: right'>15</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>no_DP_indel</td>\n",
        "<td>bwa_mem</td>\n",
        "<td>gatk_rec</td>\n",
        "<td>HaplotypeCaller</td>\n",
        "<td>no_DP_indel</td>\n",
        "<td>SnpEff_region</td>\n",
        "<td>-an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -an DP</td>\n",
        "<td>-an QD -an FS -an SOR -an ReadPosRankSum -an MQRankSum</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>crosses</td>\n",
        "<td style='text-align: right'>15</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>no_FS_indel</td>\n",
        "<td>bwa_mem</td>\n",
        "<td>gatk_rec</td>\n",
        "<td>HaplotypeCaller</td>\n",
        "<td>no_FS_indel</td>\n",
        "<td>SnpEff_region</td>\n",
        "<td>-an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -an DP</td>\n",
        "<td>-an QD -an DP -an SOR -an ReadPosRankSum -an MQRankSum</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>crosses</td>\n",
        "<td style='text-align: right'>15</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>no_SOR_indel</td>\n",
        "<td>bwa_mem</td>\n",
        "<td>gatk_rec</td>\n",
        "<td>HaplotypeCaller</td>\n",
        "<td>no_SOR_indel</td>\n",
        "<td>SnpEff_region</td>\n",
        "<td>-an QD -an MQ -an MQRankSum -an ReadPosRankSum -an FS -an SOR -an DP</td>\n",
        "<td>-an QD -an DP -an FS -an ReadPosRankSum -an MQRankSum</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>crosses</td>\n",
        "<td style='text-align: right'>15</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n",
        "<p><strong>...</strong></p>"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 5
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Determine samples"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def determine_assembly_fasta(isolate_code='7G8'):\n",
      "    if isolate_code in ('7G8', 'GB4'):\n",
      "        return(\"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/Pf%s.Jul2015.fasta\" % isolate_code)\n",
      "    else:\n",
      "        return(\"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/Pf%s.Jun2015.fasta\" % isolate_code)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 189
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tbl_samples_to_process = (tbl_assembled_samples\n",
      "    .cutout('Notes')\n",
      "    .selectnotnone('bam_fn')\n",
      "    .selecteq('To be used for', 'Validation')\n",
      "    .addfield('truth_vcf_filestem', lambda rec: os.path.join(\n",
      "        '/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples',\n",
      "        'truth_vcfs_2',\n",
      "        \"truth_%s\" % rec['Isolate code']\n",
      "    ))\n",
      "    .cutout('To be used for')\n",
      "    .convert('ox_code', lambda v: 'ERS740937', where=lambda rec: rec['Isolate code'] == 'KE01')\n",
      "    .convert('ox_code', lambda v: 'ERS740936', where=lambda rec: rec['Isolate code'] == 'KH02')\n",
      "    .convert('ox_code', lambda v: 'ERS740940', where=lambda rec: rec['Isolate code'] == 'GN01')\n",
      "    .addfield('assembly_fasta', lambda rec: determine_assembly_fasta(rec['Isolate code']))\n",
      ")\n",
      "tbl_samples_to_process.displayall(index_header=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>0|Sample name</th>\n",
        "<th>1|Isolate code</th>\n",
        "<th>2|Cross parent?</th>\n",
        "<th>3|Origin</th>\n",
        "<th>4|Continent</th>\n",
        "<th>5|#contigs assemblies</th>\n",
        "<th>6|Both telomer repeats attached</th>\n",
        "<th>7|Icorned</th>\n",
        "<th>8|ox_code</th>\n",
        "<th>9|SRA accession</th>\n",
        "<th>10|coverage</th>\n",
        "<th>11|read length</th>\n",
        "<th>12|bam_fn</th>\n",
        "<th>13|truth_vcf_filestem</th>\n",
        "<th>14|assembly_fasta</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>7G8</td>\n",
        "<td>7G8</td>\n",
        "<td>Yes</td>\n",
        "<td>Brazil</td>\n",
        "<td>S America</td>\n",
        "<td style='text-align: right'>17</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td>75bp</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>ERR027099</td>\n",
        "<td>87X</td>\n",
        "<td style='text-align: right'>76</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf/4_0/PFprog3/PG0083_C/PG0083_C.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/truth_vcfs_2/truth_7G8</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/Pf7G8.Jul2015.fasta</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>GB4</td>\n",
        "<td>GB4</td>\n",
        "<td>Yes</td>\n",
        "<td>Ghana</td>\n",
        "<td>W Africa</td>\n",
        "<td style='text-align: right'>26</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td>75bp</td>\n",
        "<td>PG0084-C</td>\n",
        "<td>ERR027100</td>\n",
        "<td>104X</td>\n",
        "<td style='text-align: right'>76</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf/4_0/PFprog3/PG0084_C/PG0084_C.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/truth_vcfs_2/truth_GB4</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/PfGB4.Jul2015.fasta</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Kilifi</td>\n",
        "<td>KE01</td>\n",
        "<td>None</td>\n",
        "<td>Kenya</td>\n",
        "<td>E Africa</td>\n",
        "<td style='text-align: right'>22</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>250bp</td>\n",
        "<td>ERS740937</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>100/250</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/cram/16503_1#3.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/truth_vcfs_2/truth_KE01</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfKE01.Jun2015.fasta</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>KH2</td>\n",
        "<td>KH02</td>\n",
        "<td>None</td>\n",
        "<td>Camdodia</td>\n",
        "<td>SE Asia</td>\n",
        "<td style='text-align: right'>22</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>250bp</td>\n",
        "<td>ERS740936</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>250</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/cram/16503_1#2.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/truth_vcfs_2/truth_KH02</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfKH02.Jun2015.fasta</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Guinea</td>\n",
        "<td>GN01</td>\n",
        "<td>None</td>\n",
        "<td>Guinea</td>\n",
        "<td>W Africa</td>\n",
        "<td style='text-align: right'>51</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>250bp</td>\n",
        "<td>ERS740940</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>250</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/cram/16503_1#6.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/truth_vcfs_2/truth_GN01</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfGN01.Jun2015.fasta</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 190
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "assembly_fastas = etl.lookupone(tbl_samples_to_process, 'Isolate code', 'assembly_fasta')\n",
      "print(assembly_fastas['7G8'])\n",
      "print(assembly_fastas['KH02'])\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/Pf7G8.Jul2015.fasta\n",
        "/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfKH02.Jun2015.fasta\n"
       ]
      }
     ],
     "prompt_number": 191
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Functions"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def determine_consensus_region(rec, column1=4, column2=11):\n",
      "    if rec[column1] is None:\n",
      "        return rec[column2]\n",
      "    elif rec[column2] is None:\n",
      "        return rec[column1]\n",
      "    elif rec[column1] == rec[column2]:\n",
      "        return rec[column1]\n",
      "    else:\n",
      "        return \"ERROR__%s__%s\" % (rec[column1], rec[column2])\n",
      "    "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def determine_consensus_region2(rec, column1=4, column2=41, column3=79):\n",
      "    vals = []\n",
      "    if(rec[column1] is not None):\n",
      "        vals.append(str(rec[column1]))\n",
      "    if(rec[column2] is not None):\n",
      "        vals.append(str(rec[column2]))\n",
      "    if(rec[column3] is not None):\n",
      "        vals.append(str(rec[column3]))\n",
      "    unique_vals = np.unique(vals)\n",
      "    return('__'.join(unique_vals))\n",
      "    "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 8
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def upstream(prv, cur, nxt):\n",
      "    if prv is None or prv.CHROM != cur.CHROM:\n",
      "        return None\n",
      "    else:\n",
      "        return cur.POS - prv.POS\n",
      "\n",
      "def downstream(prv, cur, nxt):\n",
      "    if nxt is None or nxt.CHROM != cur.CHROM:\n",
      "        return None\n",
      "    else:\n",
      "        return nxt.POS - cur.POS\n",
      "\n",
      "def nearest(distance_previous, distance_next):\n",
      "    if distance_previous is None:\n",
      "        return(distance_next)\n",
      "    elif distance_next is None:\n",
      "        return(distance_previous)\n",
      "    else:\n",
      "        return(min(distance_previous, distance_next))\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 376
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# def nearest_consensus(prv, cur, nxt):\n",
      "    "
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def determine_mode(rec):\n",
      "    if len(rec['REF']) == len(str(rec['ALT'])):\n",
      "        return('SNP')\n",
      "    else:\n",
      "        return('INDEL')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 10
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def trim_variants(ref, alt):\n",
      "#     print(ref, alt)\n",
      "    while len(ref) > 1 and len(alt) > 1 and ref[-1] == alt[-1]:\n",
      "        ref = ref[0:-1]\n",
      "        alt = alt[0:-1]\n",
      "    return((ref, alt))\n",
      " \n",
      "trim_variants('AATC', 'TC')[1]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 11,
       "text": [
        "'T'"
       ]
      }
     ],
     "prompt_number": 11
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "fields_to_compare = ['QUAL', 'FILTER', 'AC', 'AF', 'AN', 'BaseQRankSum', 'ClippingRankSum', 'DP', 'FS', 'MLEAC', 'MLEAF', 'MQ',\n",
      "                     'MQRankSum', 'QD', 'ReadPosRankSum', 'RegionType', 'SNPEFF_EFFECT', 'SNPEFF_FUNCTIONAL_CLASS',\n",
      "                     'SNPEFF_IMPACT', 'SOR', 'VQSLOD', 'culprit', 'set', 'AD', 'GT']\n",
      "gatk_renames = dict(zip(fields_to_compare, ['gatk_%s' % field_name for field_name in fields_to_compare]))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 12
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def assess_validation(isolate_code='7G8', chromosome='Pf3D7_01_v3', MAPPING_DIR='bwa_mem', BAMS_DIR='gatk_rec',\n",
      "                      UNFILTERED_DIR='HaplotypeCaller', FILTERED_DIR='alistair_ann', ANNOTATED_DIR='SnpEff_region',\n",
      "                      rewrite=False):\n",
      "    ox_code = tbl_samples_to_process.selecteq('Isolate code', isolate_code).values('ox_code')[0]\n",
      "    \n",
      "    tbl_truth_cache_dir = os.path.join(CACHE_DIR, 'tbl_truth', MAPPING_DIR, BAMS_DIR, UNFILTERED_DIR, FILTERED_DIR, ANNOTATED_DIR)\n",
      "    tbl_gatk_cache_dir = os.path.join(CACHE_DIR, 'tbl_gatk', MAPPING_DIR, BAMS_DIR, UNFILTERED_DIR, FILTERED_DIR, ANNOTATED_DIR)\n",
      "    tbl_comparison_cache_dir = os.path.join(CACHE_DIR, 'tbl_comparison', MAPPING_DIR, BAMS_DIR, UNFILTERED_DIR, FILTERED_DIR, ANNOTATED_DIR)\n",
      "    if not os.path.exists(tbl_truth_cache_dir):\n",
      "        os.makedirs(tbl_truth_cache_dir)\n",
      "    if not os.path.exists(tbl_gatk_cache_dir):\n",
      "        os.makedirs(tbl_gatk_cache_dir)\n",
      "    if not os.path.exists(tbl_comparison_cache_dir):\n",
      "        os.makedirs(tbl_comparison_cache_dir)\n",
      "    tbl_truth_cache_fn = os.path.join(tbl_truth_cache_dir, \"tbl_truth_%s.%s\" % (isolate_code, chromosome))\n",
      "    tbl_gatk_cache_fn = os.path.join(tbl_gatk_cache_dir, \"tbl_gatk_%s.%s\" % (isolate_code, chromosome))\n",
      "    tbl_comparison_cache_fn = os.path.join(tbl_comparison_cache_dir, \"tbl_comparison_%s.%s\" % (isolate_code, chromosome))\n",
      "    \n",
      "#     print(tbl_truth_cache_fn)\n",
      "#     print(tbl_gatk_cache_fn)\n",
      "#     print(tbl_comparison_cache_fn)\n",
      "    \n",
      "    if not os.path.exists(tbl_truth_cache_fn) or rewrite:\n",
      "        truth_vcf_fn = \"%s.%s.annotated.vcf.gz\" % (\n",
      "            tbl_samples_to_process.selecteq('Isolate code', isolate_code).values('truth_vcf_filestem')[0],\n",
      "            chromosome\n",
      "        )\n",
      "        \n",
      "        tbl_truth = (etl.fromvcf(truth_vcf_fn)\n",
      "            .unpackdict('INFO', samplesize=1000000)\n",
      "            .rename('ALT', 'truth_ALTs')\n",
      "            .addfield('ALT', lambda rec: str(rec[4][0]))\n",
      "            .addfield('Truth_is_coding', lambda rec: not rec['SNPEFF_EFFECT'] is None and not rec['SNPEFF_EFFECT'] in NONCODING_EFFECTS)\n",
      "            .cut(['CHROM', 'POS', 'REF', 'ALT', 'truth_ALTs', 'RegionType', 'Truth_is_coding', 'SNPEFF_AMINO_ACID_CHANGE', 'SNPEFF_EFFECT',\n",
      "                  'SNPEFF_GENE_NAME', 'SNPEFF_IMPACT', 'SNPEFF_TRANSCRIPT_ID', 'RPA', 'RU', 'STR'])\n",
      "            .addfield('Truth_RPA_REF', lambda rec: None if rec['RPA'] is None else str(rec['RPA'][0]))\n",
      "            .addfield('Truth_RPA_ALT', lambda rec: None if rec['RPA'] is None else str(rec['RPA'][1]))\n",
      "            .rename('RegionType', 'Truth_RegionType')\n",
      "            .rename('SNPEFF_AMINO_ACID_CHANGE', 'Truth_AAChange')\n",
      "            .rename('SNPEFF_EFFECT', 'Truth_Effect')\n",
      "            .rename('SNPEFF_GENE_NAME', 'Truth_Gene')\n",
      "            .rename('SNPEFF_IMPACT', 'Truth_Impact')\n",
      "            .rename('SNPEFF_TRANSCRIPT_ID', 'Truth_Transcript')\n",
      "            .addfieldusingcontext('Truth_distance_previous', upstream)\n",
      "            .addfieldusingcontext('Truth_distance_next', downstream)\n",
      "            .addfield('Truth_distance_nearest', lambda rec: nearest(rec['Truth_distance_previous'], rec['Truth_distance_next']))\n",
      "            .addfield('Truth_mode', determine_mode)\n",
      "        )\n",
      "        etl.topickle(tbl_truth, tbl_truth_cache_fn)\n",
      "    else:\n",
      "        tbl_truth = etl.frompickle(tbl_truth_cache_fn)\n",
      "        \n",
      "    if not os.path.exists(tbl_gatk_cache_fn) or rewrite:\n",
      "        gatk_vcf_fn = \"%s.SNP_INDEL.%s.vcf.gz\" % (\n",
      "            os.path.join(PROCESSED_ASSEMBLED_SAMPLES_DIR, MAPPING_DIR, BAMS_DIR, UNFILTERED_DIR, FILTERED_DIR, ANNOTATED_DIR,\n",
      "                         '_annotated_vcfs', 'filtered.annotated'),\n",
      "            chromosome\n",
      "        )\n",
      "        print(gatk_vcf_fn)\n",
      "        \n",
      "        tbl_gatk = (etl.fromvcf(gatk_vcf_fn)\n",
      "            .sort(['CHROM', 'POS'])\n",
      "            .vcfmeltsamples()\n",
      "            .vcfunpackcall()\n",
      "            .unpackdict('INFO', samplesize=1000000)\n",
      "            .selecteq('SAMPLE', ox_code)\n",
      "            .selectnotnone('GT')\n",
      "            .selectne('GT', '0/0')\n",
      "            .rename('ALT', 'gatk_ALTs')\n",
      "            .addfield('ALT', lambda rec: str(rec['gatk_ALTs'][int(rec['GT'][2])-1]))\n",
      "            .addfield('gatk_RPA_REF', lambda rec: None if rec['RPA'] is None else str(rec['RPA'][0]))\n",
      "            .addfield('gatk_RPA_ALT', lambda rec: None if rec['RPA'] is None else str(rec['RPA'][int(rec['GT'][2])]))\n",
      "            .rename('RPA', 'gatk_RPA')\n",
      "            .rename('STR', 'gatk_STR')\n",
      "            .rename('RU', 'gatk_RU')\n",
      "            .addfield('GATK_is_coding', lambda rec: not rec['SNPEFF_EFFECT'] is None and not rec['SNPEFF_EFFECT'] in NONCODING_EFFECTS)\n",
      "            .addfield('is_het', lambda rec: None if rec['GT'] is None else rec['GT'][0] != rec['GT'][2])\n",
      "            .rename(gatk_renames)\n",
      "#             .select('ALT', lambda rec: str(rec) != '<*:DEL>')\n",
      "            .select('ALT', lambda rec: str(rec) != '*')\n",
      "            .addfield('temp_REF', lambda rec: rec['REF'])\n",
      "            .addfield('temp_ALT', lambda rec: rec['ALT'])\n",
      "            .convert('REF', lambda ref, rec: trim_variants(rec['temp_REF'], rec['temp_ALT'])[0], pass_row=True)\n",
      "            .convert('ALT', lambda ref, rec: trim_variants(rec['temp_REF'], rec['temp_ALT'])[1], pass_row=True)\n",
      "            .addfieldusingcontext('GATK_distance_previous', upstream)\n",
      "            .addfieldusingcontext('GATK_distance_next', downstream)\n",
      "            .addfield('GATK_distance_nearest', lambda rec: nearest(rec['GATK_distance_previous'], rec['GATK_distance_next']))\n",
      "            .addfield('GATK_mode', determine_mode)\n",
      "        )\n",
      "        etl.topickle(tbl_gatk, tbl_gatk_cache_fn)\n",
      "    else:\n",
      "        tbl_gatk = etl.frompickle(tbl_gatk_cache_fn)\n",
      "        \n",
      "    if not os.path.exists(tbl_comparison_cache_fn) or rewrite:\n",
      "        truth_coords_fn = \"%s/Pf%s.filter.coords\" % (\n",
      "            '/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/SNP/',\n",
      "            isolate_code\n",
      "        )\n",
      "        assembly_chrom = \"Pf%s_%s\" % (isolate_code, chromosome[6:8])\n",
      "        tbl_coords = (etl.fromtsv(truth_coords_fn)\n",
      "            .pushheader(['ref_start', 'ref_end', 'assembly_start', 'assembly_end', 'ref_length', 'assembly_length', 'identity',\n",
      "                 'ref_chrom_length', 'assembly_chrom_length', 'ref_pct', 'assembly_pct', 'ref_chrom', 'assembly_chrom',\n",
      "                 'ignore'])\n",
      "            .selecteq('ref_chrom', chromosome)\n",
      "            .selecteq('assembly_chrom', assembly_chrom)\n",
      "            .cut(['ref_start', 'ref_end', 'assembly_chrom'])\n",
      "            .convertnumbers()\n",
      "        )\n",
      "        lkp = etl.intervallookup(tbl_coords, 'ref_start', 'ref_end', include_stop=True, value='assembly_chrom')\n",
      "        \n",
      "        tbl_comparison = (tbl_truth\n",
      "            .outerjoin(tbl_gatk, key=['CHROM', 'POS', 'REF', 'ALT'])\n",
      "            .addfield('IsInTruth', lambda rec: rec['Truth_RegionType'] is not None)\n",
      "            .addfield('IsInGATK', lambda rec: rec['gatk_RegionType'] is not None)\n",
      "            .addfield('RegionType', lambda rec: determine_consensus_region(rec, 'Truth_RegionType', 'gatk_RegionType'))\n",
      "            .addfield('IsCoding', lambda rec: determine_consensus_region(rec, 'Truth_is_coding', 'GATK_is_coding'))\n",
      "            .addfield('Effect', lambda rec: determine_consensus_region(rec, 'Truth_Effect', 'gatk_SNPEFF_EFFECT'))\n",
      "            .addfield('mode', lambda rec: determine_consensus_region(rec, 'Truth_mode', 'GATK_mode'))\n",
      "            .addfield('IsAccessible', lambda rec: len(lkp.search(rec['POS'])) > 0)\n",
      "            .sort('gatk_VQSLOD', reverse=True)\n",
      "        )\n",
      "        etl.topickle(tbl_comparison, tbl_comparison_cache_fn)\n",
      "    else:\n",
      "        tbl_comparison = etl.frompickle(tbl_comparison_cache_fn)\n",
      "        \n",
      "    return(tbl_comparison)\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 13
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "pipelines_complete = ['default', 'alistair_ann']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 14
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "rewrite=False\n",
      "\n",
      "for rec in tbl_pipelines.data():\n",
      "#     if True:\n",
      "    if rec[0] in pipelines_complete:\n",
      "        annotated_vcfs_dir = \"%s/%s/%s/%s/%s/%s/\" % (PROCESSED_ASSEMBLED_SAMPLES_DIR, rec[1], rec[2], rec[3], rec[4], rec[5])\n",
      "\n",
      "        isolate_codes = list(tbl_samples_to_process.values('Isolate code'))\n",
      "        # isolate_codes = ['7G8']\n",
      "        chromosomes = ['Pf3D7_%02d_v3' % chrom for chrom in range(1,15)]\n",
      "\n",
      "        tbl_comparisons = collections.OrderedDict()\n",
      "        tbl_whole_genome_comparisons = collections.OrderedDict()\n",
      "\n",
      "    #     tbl_comparison_all_cache_dir = os.path.join(CACHE_DIR, 'tbl_comparison', rec[1], rec[2], rec[3], rec[4], rec[5])\n",
      "    #     tbl_comparison_all_cache_fn = os.path.join(tbl_comparison_all_cache_dir, \"tbl_comparison_all\")\n",
      "\n",
      "        # if not os.path.exists(tbl_comparison_all_cache_fn) or rewrite:\n",
      "        for isolate_code in isolate_codes:\n",
      "#             print(rec[0], isolate_code)\n",
      "            tbl_comparisons[isolate_code] = collections.OrderedDict()\n",
      "            for chromosome in chromosomes:\n",
      "                print(rec[0], isolate_code, chromosome)\n",
      "                tbl_comparisons[isolate_code][chromosome] = assess_validation(isolate_code, chromosome, MAPPING_DIR=rec[1],\n",
      "                                                                              BAMS_DIR=rec[2], UNFILTERED_DIR=rec[3],\n",
      "                                                                              FILTERED_DIR=rec[4], ANNOTATED_DIR=rec[5],\n",
      "                                                                              rewrite=rewrite)\n",
      "\n",
      "            tbl_comparison_cache_dir = os.path.join(CACHE_DIR, 'tbl_comparison', rec[1], rec[2], rec[3], rec[4], rec[5])\n",
      "            tbl_comparison_sample_cache_fn = os.path.join(tbl_comparison_cache_dir, \"tbl_comparison_%s\" % (isolate_code))\n",
      "            if not os.path.exists(tbl_comparison_sample_cache_fn) or rewrite:\n",
      "                tbl_whole_genome_comparisons[isolate_code] = (\n",
      "                    tbl_comparisons[isolate_code]['Pf3D7_01_v3']\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_02_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_03_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_04_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_05_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_06_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_07_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_08_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_09_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_10_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_11_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_12_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_13_v3'])\n",
      "                    .cat(tbl_comparisons[isolate_code]['Pf3D7_14_v3'])\n",
      "                    .sort('gatk_VQSLOD', reverse=True)\n",
      "                )\n",
      "                etl.topickle(tbl_whole_genome_comparisons[isolate_code], tbl_comparison_sample_cache_fn)\n",
      "            else:\n",
      "                tbl_whole_genome_comparisons[isolate_code] = etl.frompickle(tbl_comparison_sample_cache_fn)\n",
      "                \n",
      "        tbl_comparison_sample_cache_fn = os.path.join(tbl_comparison_cache_dir, \"tbl_comparison_All\")\n",
      "        if not os.path.exists(tbl_comparison_sample_cache_fn) or rewrite:\n",
      "            tbl_whole_genome_comparisons['All'] = (\n",
      "                tbl_whole_genome_comparisons['7G8'].addfield('Sample', '7G8')\n",
      "                .cat(tbl_whole_genome_comparisons['GB4'].addfield('Sample', 'GB4'))\n",
      "                .cat(tbl_whole_genome_comparisons['KE01'].addfield('Sample', 'KE01'))\n",
      "                .cat(tbl_whole_genome_comparisons['KH02'].addfield('Sample', 'KH02'))\n",
      "                .cat(tbl_whole_genome_comparisons['GN01'].addfield('Sample', 'GN01'))\n",
      "                .sort('gatk_VQSLOD', reverse=True)\n",
      "            )\n",
      "            etl.topickle(tbl_whole_genome_comparisons['All'], tbl_comparison_sample_cache_fn)\n",
      "        else:\n",
      "            tbl_whole_genome_comparisons['All'] = etl.frompickle(tbl_comparison_sample_cache_fn)\n",
      "\n",
      "\n",
      "        #     etl.topickle(tbl_whole_genome_comparisons, tbl_comparison_all_cache_fn)\n",
      "        # else:\n",
      "        #     tbl_whole_genome_comparisons = etl.frompickle(tbl_comparison_all_cache_fn)\n",
      "\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "default 7G8 Pf3D7_01_v3\n",
        "default 7G8 Pf3D7_02_v3\n",
        "default 7G8 Pf3D7_03_v3\n",
        "default 7G8 Pf3D7_04_v3\n",
        "default 7G8 Pf3D7_05_v3\n",
        "default 7G8 Pf3D7_06_v3\n",
        "default 7G8 Pf3D7_07_v3\n",
        "default 7G8 Pf3D7_08_v3\n",
        "default 7G8 Pf3D7_09_v3\n",
        "default 7G8 Pf3D7_10_v3\n",
        "default 7G8 Pf3D7_11_v3\n",
        "default 7G8 Pf3D7_12_v3\n",
        "default 7G8 Pf3D7_13_v3\n",
        "default 7G8 Pf3D7_14_v3\n",
        "default GB4 Pf3D7_01_v3\n",
        "default GB4 Pf3D7_02_v3\n",
        "default GB4 Pf3D7_03_v3\n",
        "default GB4 Pf3D7_04_v3\n",
        "default GB4 Pf3D7_05_v3\n",
        "default GB4 Pf3D7_06_v3\n",
        "default GB4 Pf3D7_07_v3\n",
        "default GB4 Pf3D7_08_v3\n",
        "default GB4 Pf3D7_09_v3\n",
        "default GB4 Pf3D7_10_v3\n",
        "default GB4 Pf3D7_11_v3\n",
        "default GB4 Pf3D7_12_v3\n",
        "default GB4 Pf3D7_13_v3\n",
        "default GB4 Pf3D7_14_v3\n",
        "default KE01 Pf3D7_01_v3\n",
        "default KE01 Pf3D7_02_v3\n",
        "default KE01 Pf3D7_03_v3\n",
        "default KE01 Pf3D7_04_v3\n",
        "default KE01 Pf3D7_05_v3\n",
        "default KE01 Pf3D7_06_v3\n",
        "default KE01 Pf3D7_07_v3\n",
        "default KE01 Pf3D7_08_v3\n",
        "default KE01 Pf3D7_09_v3\n",
        "default KE01 Pf3D7_10_v3\n",
        "default KE01 Pf3D7_11_v3\n",
        "default KE01 Pf3D7_12_v3\n",
        "default KE01 Pf3D7_13_v3\n",
        "default KE01 Pf3D7_14_v3\n",
        "default KH02 Pf3D7_01_v3\n",
        "default KH02 Pf3D7_02_v3\n",
        "default KH02 Pf3D7_03_v3\n",
        "default KH02 Pf3D7_04_v3\n",
        "default KH02 Pf3D7_05_v3\n",
        "default KH02 Pf3D7_06_v3\n",
        "default KH02 Pf3D7_07_v3\n",
        "default KH02 Pf3D7_08_v3\n",
        "default KH02 Pf3D7_09_v3\n",
        "default KH02 Pf3D7_10_v3\n",
        "default KH02 Pf3D7_11_v3\n",
        "default KH02 Pf3D7_12_v3\n",
        "default KH02 Pf3D7_13_v3\n",
        "default KH02 Pf3D7_14_v3\n",
        "default GN01 Pf3D7_01_v3\n",
        "default GN01 Pf3D7_02_v3\n",
        "default GN01 Pf3D7_03_v3\n",
        "default GN01 Pf3D7_04_v3\n",
        "default"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " GN01 Pf3D7_05_v3\n",
        "default GN01 Pf3D7_06_v3\n",
        "default GN01 Pf3D7_07_v3\n",
        "default GN01 Pf3D7_08_v3\n",
        "default GN01 Pf3D7_09_v3\n",
        "default GN01 Pf3D7_10_v3\n",
        "default GN01 Pf3D7_11_v3\n",
        "default GN01 Pf3D7_12_v3\n",
        "default GN01 Pf3D7_13_v3\n",
        "default GN01 Pf3D7_14_v3\n",
        "alistair_ann 7G8 Pf3D7_01_v3\n",
        "alistair_ann 7G8 Pf3D7_02_v3\n",
        "alistair_ann 7G8 Pf3D7_03_v3\n",
        "alistair_ann 7G8 Pf3D7_04_v3\n",
        "alistair_ann 7G8 Pf3D7_05_v3\n",
        "alistair_ann 7G8 Pf3D7_06_v3\n",
        "alistair_ann 7G8 Pf3D7_07_v3\n",
        "alistair_ann 7G8 Pf3D7_08_v3\n",
        "alistair_ann 7G8 Pf3D7_09_v3\n",
        "alistair_ann 7G8 Pf3D7_10_v3\n",
        "alistair_ann 7G8 Pf3D7_11_v3\n",
        "alistair_ann 7G8 Pf3D7_12_v3\n",
        "alistair_ann 7G8 Pf3D7_13_v3\n",
        "alistair_ann 7G8 Pf3D7_14_v3\n",
        "alistair_ann GB4 Pf3D7_01_v3\n",
        "alistair_ann GB4 Pf3D7_02_v3\n",
        "alistair_ann GB4 Pf3D7_03_v3\n",
        "alistair_ann GB4 Pf3D7_04_v3\n",
        "alistair_ann GB4 Pf3D7_05_v3\n",
        "alistair_ann GB4 Pf3D7_06_v3\n",
        "alistair_ann GB4 Pf3D7_07_v3\n",
        "alistair_ann GB4 Pf3D7_08_v3\n",
        "alistair_ann GB4 Pf3D7_09_v3\n",
        "alistair_ann GB4 Pf3D7_10_v3\n",
        "alistair_ann GB4 Pf3D7_11_v3\n",
        "alistair_ann GB4 Pf3D7_12_v3\n",
        "alistair_ann GB4 Pf3D7_13_v3\n",
        "alistair_ann GB4 Pf3D7_14_v3\n",
        "alistair_ann KE01 Pf3D7_01_v3\n",
        "alistair_ann KE01 Pf3D7_02_v3\n",
        "alistair_ann KE01 Pf3D7_03_v3\n",
        "alistair_ann KE01 Pf3D7_04_v3\n",
        "alistair_ann KE01 Pf3D7_05_v3\n",
        "alistair_ann KE01 Pf3D7_06_v3\n",
        "alistair_ann KE01 Pf3D7_07_v3\n",
        "alistair_ann KE01 Pf3D7_08_v3\n",
        "alistair_ann KE01 Pf3D7_09_v3\n",
        "alistair_ann KE01 Pf3D7_10_v3\n",
        "alistair_ann KE01 Pf3D7_11_v3\n",
        "alistair_ann KE01 Pf3D7_12_v3\n",
        "alistair_ann KE01 Pf3D7_13_v3\n",
        "alistair_ann KE01 Pf3D7_14_v3\n",
        "alistair_ann KH02 Pf3D7_01_v3\n",
        "alistair_ann KH02 Pf3D7_02_v3\n",
        "alistair_ann KH02 Pf3D7_03_v3\n",
        "alistair_ann KH02 Pf3D7_04_v3\n",
        "alistair_ann KH02 Pf3D7_05_v3\n",
        "alistair_ann KH02 Pf3D7_06_v3\n",
        "alistair_ann KH02 Pf3D7_07_v3\n",
        "alistair_ann KH02 Pf3D7_08_v3\n",
        "alistair_ann KH02 Pf3D7_09_v3\n",
        "alistair_ann KH02 Pf3D7_10_v3\n",
        "alistair_ann"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        " KH02 Pf3D7_11_v3\n",
        "alistair_ann KH02 Pf3D7_12_v3\n",
        "alistair_ann KH02 Pf3D7_13_v3\n",
        "alistair_ann KH02 Pf3D7_14_v3\n",
        "alistair_ann GN01 Pf3D7_01_v3\n",
        "alistair_ann GN01 Pf3D7_02_v3\n",
        "alistair_ann GN01 Pf3D7_03_v3\n",
        "alistair_ann GN01 Pf3D7_04_v3\n",
        "alistair_ann GN01 Pf3D7_05_v3\n",
        "alistair_ann GN01 Pf3D7_06_v3\n",
        "alistair_ann GN01 Pf3D7_07_v3\n",
        "alistair_ann GN01 Pf3D7_08_v3\n",
        "alistair_ann GN01 Pf3D7_09_v3\n",
        "alistair_ann GN01 Pf3D7_10_v3\n",
        "alistair_ann GN01 Pf3D7_11_v3\n",
        "alistair_ann GN01 Pf3D7_12_v3\n",
        "alistair_ann GN01 Pf3D7_13_v3\n",
        "alistair_ann GN01 Pf3D7_14_v3\n"
       ]
      }
     ],
     "prompt_number": 15
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(tbl_whole_genome_comparisons['alistair_ann']\n",
      ".selecteq('Sample', '7G8')\n",
      ".selecteq('IsInGATK', True)\n",
      ".selecteq('CHROM', 'Pf3D7_01_v3')\n",
      ".sort('POS')\n",
      ".selectgt('POS', 17816)\n",
      ".display(20)\n",
      ")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>CHROM</th>\n",
        "<th>POS</th>\n",
        "<th>REF</th>\n",
        "<th>ALT</th>\n",
        "<th>truth_ALTs</th>\n",
        "<th>Truth_RegionType</th>\n",
        "<th>Truth_is_coding</th>\n",
        "<th>Truth_AAChange</th>\n",
        "<th>Truth_Effect</th>\n",
        "<th>Truth_Gene</th>\n",
        "<th>Truth_Impact</th>\n",
        "<th>Truth_Transcript</th>\n",
        "<th>RPA</th>\n",
        "<th>RU</th>\n",
        "<th>STR</th>\n",
        "<th>Truth_RPA_REF</th>\n",
        "<th>Truth_RPA_ALT</th>\n",
        "<th>Truth_distance_previous</th>\n",
        "<th>Truth_distance_next</th>\n",
        "<th>Truth_distance_nearest</th>\n",
        "<th>Truth_mode</th>\n",
        "<th>ID</th>\n",
        "<th>gatk_ALTs</th>\n",
        "<th>gatk_QUAL</th>\n",
        "<th>gatk_FILTER</th>\n",
        "<th>SAMPLE</th>\n",
        "<th>gatk_AD</th>\n",
        "<th>gatk_DP</th>\n",
        "<th>GQ</th>\n",
        "<th>gatk_GT</th>\n",
        "<th>PGT</th>\n",
        "<th>PID</th>\n",
        "<th>PL</th>\n",
        "<th>gatk_AC</th>\n",
        "<th>gatk_AF</th>\n",
        "<th>gatk_AN</th>\n",
        "<th>gatk_BaseQRankSum</th>\n",
        "<th>gatk_ClippingRankSum</th>\n",
        "<th>gatk_FS</th>\n",
        "<th>GC</th>\n",
        "<th>gatk_MLEAC</th>\n",
        "<th>gatk_MLEAF</th>\n",
        "<th>gatk_MQ</th>\n",
        "<th>gatk_MQRankSum</th>\n",
        "<th>NEGATIVE_TRAIN_SITE</th>\n",
        "<th>POSITIVE_TRAIN_SITE</th>\n",
        "<th>gatk_QD</th>\n",
        "<th>gatk_RPA</th>\n",
        "<th>gatk_RU</th>\n",
        "<th>gatk_ReadPosRankSum</th>\n",
        "<th>gatk_RegionType</th>\n",
        "<th>SNPEFF_AMINO_ACID_CHANGE</th>\n",
        "<th>SNPEFF_CODON_CHANGE</th>\n",
        "<th>gatk_SNPEFF_EFFECT</th>\n",
        "<th>SNPEFF_EXON_ID</th>\n",
        "<th>gatk_SNPEFF_FUNCTIONAL_CLASS</th>\n",
        "<th>SNPEFF_GENE_BIOTYPE</th>\n",
        "<th>SNPEFF_GENE_NAME</th>\n",
        "<th>gatk_SNPEFF_IMPACT</th>\n",
        "<th>SNPEFF_TRANSCRIPT_ID</th>\n",
        "<th>gatk_SOR</th>\n",
        "<th>gatk_STR</th>\n",
        "<th>gatk_VQSLOD</th>\n",
        "<th>VariantType</th>\n",
        "<th>gatk_culprit</th>\n",
        "<th>gatk_set</th>\n",
        "<th>gatk_RPA_REF</th>\n",
        "<th>gatk_RPA_ALT</th>\n",
        "<th>GATK_is_coding</th>\n",
        "<th>is_het</th>\n",
        "<th>temp_REF</th>\n",
        "<th>temp_ALT</th>\n",
        "<th>GATK_distance_previous</th>\n",
        "<th>GATK_distance_next</th>\n",
        "<th>GATK_distance_nearest</th>\n",
        "<th>GATK_mode</th>\n",
        "<th>IsInTruth</th>\n",
        "<th>IsInGATK</th>\n",
        "<th>RegionType</th>\n",
        "<th>IsCoding</th>\n",
        "<th>Effect</th>\n",
        "<th>mode</th>\n",
        "<th>IsAccessible</th>\n",
        "<th>Sample</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>17855</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>165.84</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[8, 6]</td>\n",
        "<td style='text-align: right'>14</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>17855_T_A</td>\n",
        "<td>[196, 0, 318]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>-0.581</td>\n",
        "<td style='text-align: right'>0.581</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>28.92</td>\n",
        "<td style='text-align: right'>1.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>11.85</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-3.163</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.446</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-6.772</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>78</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>17857</td>\n",
        "<td>A</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C]</td>\n",
        "<td style='text-align: right'>197.84</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[7, 6]</td>\n",
        "<td style='text-align: right'>13</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>17857_A_C</td>\n",
        "<td>[228, 0, 300]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>-1.093</td>\n",
        "<td style='text-align: right'>-0.312</td>\n",
        "<td style='text-align: right'>3.358</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>28.59</td>\n",
        "<td style='text-align: right'>0.312</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>15.22</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.49</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.584</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-11.68</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>17861</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>115.98</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[8, 4]</td>\n",
        "<td style='text-align: right'>12</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>17855_T_A</td>\n",
        "<td>[144, 0, 324]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>-2.18</td>\n",
        "<td style='text-align: right'>1.24</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>33.33</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>28.24</td>\n",
        "<td style='text-align: right'>0.929</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>9.66</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-2.359</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.223</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-7.854</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>17867</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>297.2</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[2, 6]</td>\n",
        "<td style='text-align: right'>12</td>\n",
        "<td style='text-align: right'>66</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[246, 0, 66]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>1.18</td>\n",
        "<td style='text-align: right'>-1.611</td>\n",
        "<td style='text-align: right'>4.973</td>\n",
        "<td style='text-align: right'>33.33</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>28.24</td>\n",
        "<td style='text-align: right'>1.61</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>27.19</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-1.826</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.258</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-13.07</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>17868</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>205.97</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[5, 6]</td>\n",
        "<td style='text-align: right'>11</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>17857_A_C</td>\n",
        "<td>[234, 0, 234]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>0.143</td>\n",
        "<td style='text-align: right'>2.64</td>\n",
        "<td style='text-align: right'>7.404</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>27.57</td>\n",
        "<td style='text-align: right'>0.717</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>18.72</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.466</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.352</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-15.75</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>17872</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>214.41</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[3, 6]</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>17857_A_C</td>\n",
        "<td>[243, 0, 108]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.167]</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>-1.422</td>\n",
        "<td style='text-align: right'>-0.248</td>\n",
        "<td style='text-align: right'>4.771</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.167]</td>\n",
        "<td style='text-align: right'>27.11</td>\n",
        "<td style='text-align: right'>1.42</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>23.82</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.126</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-13.78</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>17873</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>214.67</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[3, 6]</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>17857_A_C</td>\n",
        "<td>[243, 0, 108]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.25]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>2.05</td>\n",
        "<td style='text-align: right'>-1.001</td>\n",
        "<td style='text-align: right'>4.771</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.25]</td>\n",
        "<td style='text-align: right'>27.11</td>\n",
        "<td style='text-align: right'>1.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>23.85</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.126</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-13.84</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>17876</td>\n",
        "<td>T</td>\n",
        "<td>TA</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[TA]</td>\n",
        "<td style='text-align: right'>214.02</td>\n",
        "<td>['VQSRTrancheINDEL99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[3, 4]</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>17857_A_C</td>\n",
        "<td>[243, 0, 108]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>1.75</td>\n",
        "<td style='text-align: right'>0.937</td>\n",
        "<td style='text-align: right'>3.68</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>26.16</td>\n",
        "<td style='text-align: right'>0.937</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>30.57</td>\n",
        "<td>[1, 2]</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.494</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>-17.61</td>\n",
        "<td>INSERTION.NumRepetitions_1.EventLength_1.RepeatExpansion_A</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>1</td>\n",
        "<td>2</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>TA</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>235</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>INDEL</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>INDEL</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18111</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>1320.67</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 25]</td>\n",
        "<td style='text-align: right'>25</td>\n",
        "<td style='text-align: right'>75</td>\n",
        "<td>1/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[1104, 75, 0]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.75]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>1.99</td>\n",
        "<td style='text-align: right'>-0.398</td>\n",
        "<td style='text-align: right'>19.951</td>\n",
        "<td style='text-align: right'>33.33</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.75]</td>\n",
        "<td style='text-align: right'>38.67</td>\n",
        "<td style='text-align: right'>2.06</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>28.71</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-3.44</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.8</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-8.489</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>235</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18113</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>1056.92</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 24]</td>\n",
        "<td style='text-align: right'>24</td>\n",
        "<td style='text-align: right'>75</td>\n",
        "<td>1/1</td>\n",
        "<td>1|1</td>\n",
        "<td>18113_A_G</td>\n",
        "<td>[1084, 75, 0]</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>33.33</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.75]</td>\n",
        "<td style='text-align: right'>38.34</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>30.59</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>4.977</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-2.372</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18123</td>\n",
        "<td>C</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>733.67</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 12]</td>\n",
        "<td style='text-align: right'>12</td>\n",
        "<td style='text-align: right'>45</td>\n",
        "<td>1/1</td>\n",
        "<td>1|1</td>\n",
        "<td>18113_A_G</td>\n",
        "<td>[675, 45, 0]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.75]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>0.296</td>\n",
        "<td style='text-align: right'>0.493</td>\n",
        "<td style='text-align: right'>17.649</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.75]</td>\n",
        "<td style='text-align: right'>38.8</td>\n",
        "<td style='text-align: right'>0.493</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>29.35</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-2.27</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.219</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-6.653</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>C</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18124</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>641.9</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 11]</td>\n",
        "<td style='text-align: right'>11</td>\n",
        "<td style='text-align: right'>45</td>\n",
        "<td>1/1</td>\n",
        "<td>1|1</td>\n",
        "<td>18113_A_G</td>\n",
        "<td>[675, 45, 0]</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>38.18</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>32.66</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.442</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-2.045</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18127</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>337.72</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 6]</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>21</td>\n",
        "<td>1/1</td>\n",
        "<td>1|1</td>\n",
        "<td>18113_A_G</td>\n",
        "<td>[315, 21, 0]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.75]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>-1.678</td>\n",
        "<td style='text-align: right'>-0.493</td>\n",
        "<td style='text-align: right'>7.425</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.75]</td>\n",
        "<td style='text-align: right'>31.15</td>\n",
        "<td style='text-align: right'>0.493</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>17.77</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-2.27</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.895</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-5.406</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18131</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>243.04</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 2]</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>18</td>\n",
        "<td>1/1</td>\n",
        "<td>1|1</td>\n",
        "<td>18113_A_G</td>\n",
        "<td>[270, 18, 0]</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>33.33</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.75]</td>\n",
        "<td style='text-align: right'>21.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>33.25</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.303</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-11.74</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18132</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>198.15</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 2]</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>15</td>\n",
        "<td>1/1</td>\n",
        "<td>1|1</td>\n",
        "<td>18113_A_G</td>\n",
        "<td>[225, 15, 0]</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>33.33</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.75]</td>\n",
        "<td style='text-align: right'>21.69</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>33.42</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.303</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-11.31</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>163</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18295</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>111.45</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[17, 5]</td>\n",
        "<td style='text-align: right'>22</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>18295_T_G</td>\n",
        "<td>[141, 0, 452]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>-3.369</td>\n",
        "<td style='text-align: right'>-0.627</td>\n",
        "<td style='text-align: right'>5.746</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>24.9</td>\n",
        "<td style='text-align: right'>3.29</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>5.07</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.74</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.584</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-15.3</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>163</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18298</td>\n",
        "<td>ACT</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>111.44</td>\n",
        "<td>['VQSRTrancheINDEL99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[4, 5]</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>18295_T_G</td>\n",
        "<td>[141, 0, 452]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>-0.361</td>\n",
        "<td style='text-align: right'>1.31</td>\n",
        "<td style='text-align: right'>13.222</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>25.46</td>\n",
        "<td style='text-align: right'>2.22</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>12.38</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.361</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.59</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-22.11</td>\n",
        "<td>DELETION.NumRepetitions_1.EventLength_2.RepeatExpansion_CT</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>ACT</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>INDEL</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>INDEL</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18302</td>\n",
        "<td>G</td>\n",
        "<td>GT</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[GT]</td>\n",
        "<td style='text-align: right'>131.84</td>\n",
        "<td>['VQSRTrancheINDEL99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[3, 4]</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td style='text-align: right'>96</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>18295_T_G</td>\n",
        "<td>[159, 0, 96]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>-0.55</td>\n",
        "<td style='text-align: right'>15.441</td>\n",
        "<td style='text-align: right'>23.81</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.25]</td>\n",
        "<td style='text-align: right'>26.31</td>\n",
        "<td style='text-align: right'>1.75</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>18.83</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-1.38</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.221</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-20.85</td>\n",
        "<td>INSERTION.NOVEL_1.Novel_T</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>G</td>\n",
        "<td>GT</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>INDEL</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>INDEL</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18304</td>\n",
        "<td>A</td>\n",
        "<td>AC</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[AC]</td>\n",
        "<td style='text-align: right'>192.64</td>\n",
        "<td>['VQSRTrancheINDEL99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 4]</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>38</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[159, 0, 38]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>27.19</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>29.59</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.329</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-19.18</td>\n",
        "<td>INSERTION.NOVEL_1.Novel_C</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>AC</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>INDEL</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>INDEL</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>18308</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>128.85</td>\n",
        "<td>['VQSRTrancheSNP99.90to100.00']</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[4, 3]</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>18295_T_G</td>\n",
        "<td>[156, 0, 116]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>1.75</td>\n",
        "<td style='text-align: right'>0.55</td>\n",
        "<td style='text-align: right'>8.451</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.25]</td>\n",
        "<td style='text-align: right'>31.56</td>\n",
        "<td style='text-align: right'>1.75</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>18.41</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-1.537</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.788</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-10.36</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>FilteredInAll</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>55</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>SubtelomericRepeat</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n",
        "<p><strong>...</strong></p>"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 418
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(tbl_whole_genome_comparisons['alistair_ann']\n",
      ".selecteq('Sample', '7G8')\n",
      ".selecteq('IsInGATK', True)\n",
      ".selecteq('CHROM', 'Pf3D7_01_v3')\n",
      ".sort('POS')\n",
      ".selectgt('POS', 120000)\n",
      ".cut(['CHROM', 'POS', 'REF', 'ALT', 'gatk_VQSLOD'])\n",
      ".display(20)\n",
      ")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>CHROM</th>\n",
        "<th>POS</th>\n",
        "<th>REF</th>\n",
        "<th>ALT</th>\n",
        "<th>gatk_VQSLOD</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>121289</td>\n",
        "<td>A</td>\n",
        "<td>AATAT</td>\n",
        "<td style='text-align: right'>3.23</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>121669</td>\n",
        "<td>C</td>\n",
        "<td>CATAT</td>\n",
        "<td style='text-align: right'>2.55</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>121693</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>4.64</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>121708</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>5.69</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>121710</td>\n",
        "<td>G</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>5.31</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>122104</td>\n",
        "<td>A</td>\n",
        "<td>AATATAT</td>\n",
        "<td style='text-align: right'>3.91</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>122494</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>0.012</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>122509</td>\n",
        "<td>A</td>\n",
        "<td>AT</td>\n",
        "<td style='text-align: right'>3.0</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>122796</td>\n",
        "<td>AT</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>2.38</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>123313</td>\n",
        "<td>C</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>-0.635</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>123315</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>-18.64</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>123317</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>-15.34</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>123589</td>\n",
        "<td>T</td>\n",
        "<td>TTATA</td>\n",
        "<td style='text-align: right'>3.98</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>123770</td>\n",
        "<td>CAAAATAAAAA</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>4.51</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>124599</td>\n",
        "<td>AAAAT</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>3.0</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>125521</td>\n",
        "<td>CAT</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>0.293</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>125704</td>\n",
        "<td>T</td>\n",
        "<td>TTA</td>\n",
        "<td style='text-align: right'>1.29</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>125757</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>4.39</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>125809</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>10.32</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>125815</td>\n",
        "<td>A</td>\n",
        "<td>AATAT</td>\n",
        "<td style='text-align: right'>3.83</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n",
        "<p><strong>...</strong></p>"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 423
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "rewrite=False\n",
      "\n",
      "tbl_whole_genome_comparisons = collections.OrderedDict()\n",
      "comparison_arrays = collections.OrderedDict()\n",
      "\n",
      "for rec in tbl_pipelines.sort('PIPELINE_NAME').data():\n",
      "#     print(rec[0])\n",
      "#     if not rec[0] in ['hrun']:\n",
      "    if rec[0] in pipelines_complete:\n",
      "#     if True:\n",
      "        print(rec[0])\n",
      "        tbl_comparison_cache_dir = os.path.join(CACHE_DIR, 'tbl_comparison', rec[1], rec[2], rec[3], rec[4], rec[5])\n",
      "        tbl_comparison_sample_cache_fn = os.path.join(tbl_comparison_cache_dir, \"tbl_comparison_All\")\n",
      "        tbl_whole_genome_comparisons[rec[0]] = etl.frompickle(tbl_comparison_sample_cache_fn)\n",
      "        comparison_array_cache_fn = os.path.join(tbl_comparison_cache_dir, \"comparison_array.npy\")\n",
      "        if not os.path.exists(comparison_array_cache_fn) or rewrite:\n",
      "            comparison_array = etl.toarray(\n",
      "                tbl_whole_genome_comparisons[rec[0]]\n",
      "                .addfield('num_alts', lambda rec: 0 if rec['gatk_ALTs'] is None else len(rec['gatk_ALTs']))\n",
      "                .addfield('num_filters', lambda rec: 0 if rec['gatk_FILTER'] is None else len(rec['gatk_FILTER']))\n",
      "                .replace('Truth_distance_nearest', None, 0)\n",
      "                .replace('STR', None, False)\n",
      "                .replace('Effect', None, '')\n",
      "                .replace('gatk_VQSLOD', None, np.nan)\n",
      "#                 .replace('gatk_ALTs', None, np.nan)\n",
      "#                 .replace('gatk_FILTER', None, '')\n",
      "                .replace('gatk_QUAL', None, np.nan)\n",
      "#                 .replace('gatk_AC', None, np.nan)\n",
      "#                 .replace('gatk_AF', None, np.nan)\n",
      "#                 .replace('gatk_AN', None, np.nan)\n",
      "                .replace('gatk_BaseQRankSum', None, np.nan)\n",
      "                .replace('gatk_ClippingRankSum', None, np.nan)\n",
      "                .replace('GC', None, np.nan)\n",
      "                .replace('gatk_DP', None, np.nan)\n",
      "                .replace('gatk_FS', None, np.nan)\n",
      "                .replace('gatk_MQ', None, np.nan)\n",
      "                .replace('gatk_MQRankSum', None, np.nan)\n",
      "                .replace('gatk_QD', None, np.nan)\n",
      "                .replace('gatk_ReadPosRankSum', None, np.nan)\n",
      "                .replace('gatk_SOR', None, np.nan)\n",
      "                .replace('GQ', None, 0)\n",
      "#                 .replace('is_het', None, False)\n",
      "#                 .replace('NEGATIVE_TRAIN_SITE', None, False)\n",
      "#                 .replace('POSITIVE_TRAIN_SITE', None, False)\n",
      "#                 .replace('gatk_RPA_REF', None, )\n",
      "                .convert('gatk_RPA_REF', lambda rec: np.nan if rec is None else int(rec))\n",
      "                .convert('gatk_RPA_ALT', lambda rec: np.nan if rec is None else int(rec))\n",
      "#                 .replace('gatk_RPA_ALT', None, '')\n",
      "                .replace('gatk_RU', None, '')\n",
      "                .replace('gatk_SNPEFF_IMPACT', None, '')\n",
      "                .replace('gatk_STR', None, False)\n",
      "                .replace('gatk_culprit', None, '')\n",
      "                .replace('gatk_set', None, '')\n",
      "                .cut(['CHROM', 'POS', 'IsInTruth', 'IsInGATK', 'RegionType', 'IsCoding', 'Effect', 'mode', 'IsAccessible',\n",
      "                    'Truth_distance_nearest', 'STR',\n",
      "                    'gatk_VQSLOD',\n",
      "                    'num_alts',\n",
      "                    'num_filters',\n",
      "                    'gatk_QUAL',\n",
      "# #                     #'gatk_AC', 'gatk_AF', 'gatk_AN',\n",
      "                    'gatk_BaseQRankSum', 'gatk_ClippingRankSum',\n",
      "                    'GQ', 'GC', 'gatk_MQ', 'gatk_MQRankSum', 'gatk_QD',\n",
      "                    'gatk_ReadPosRankSum', 'gatk_SOR',\n",
      "# 'GQ',\n",
      "#                     'NEGATIVE_TRAIN_SITE', 'POSITIVE_TRAIN_SITE',\n",
      "                    'gatk_RPA_REF',\n",
      "                    'gatk_RPA_ALT', 'gatk_RU',\n",
      "                    'gatk_SNPEFF_IMPACT', 'gatk_STR', 'gatk_culprit', 'gatk_set'\n",
      "                ])\n",
      "            )\n",
      "            np.save(comparison_array_cache_fn, comparison_array)\n",
      "        else:\n",
      "            comparison_array = np.load(comparison_array_cache_fn)\n",
      "        comparison_arrays[rec[0]] = comparison_array\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "alistair_ann\n",
        "default"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print(len(tbl_whole_genome_comparisons['default'].data()))\n",
      "print(len(tbl_whole_genome_comparisons['alistair_ann'].data()))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "582034\n",
        "582034"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n"
       ]
      }
     ],
     "prompt_number": 17
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print(shape(comparison_arrays['default']))\n",
      "print(shape(comparison_arrays['alistair_ann']))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "(582034,)\n",
        "(582034,)\n"
       ]
      }
     ],
     "prompt_number": 18
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "test_filters = collections.OrderedDict()\n",
      "test_variants = collections.OrderedDict()\n",
      "# for pipeline in pipelines_complete:\n",
      "for pipeline in pipelines_complete:\n",
      "    print(pipeline)\n",
      "    test_filters[pipeline] = (\n",
      "        (comparison_arrays[pipeline]['IsInGATK']) &\n",
      "        (comparison_arrays[pipeline]['RegionType'] == 'Core') &\n",
      "        (comparison_arrays[pipeline]['IsCoding'] == 'True') &\n",
      "        (comparison_arrays[pipeline]['IsAccessible'] == True) &\n",
      "        (np.in1d(comparison_arrays[pipeline]['mode'], ['IND', 'False__INDEL', 'False__True__IND', 'INDEL__True']))\n",
      "    #     (comparison_arrays['alistair_ann']['STR'] == True)\n",
      "    )\n",
      "    test_variants[pipeline] = comparison_arrays[pipeline][test_filters[pipeline]]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "default\n",
        "alistair_ann"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n"
       ]
      }
     ],
     "prompt_number": 19
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Create consensus alignments"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import copy\n",
      "def create_consensus(sample='7G8',\n",
      "                     pos_output_format = '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/pos/%s.pos',\n",
      "                     fasta_output_format = '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/fasta/%s.fasta'):\n",
      "    pos_dir = os.path.dirname(pos_output_format % sample)\n",
      "    if not os.path.exists(pos_dir):\n",
      "        os.makedirs(pos_dir)\n",
      "    fasta_dir = os.path.dirname(fasta_output_format % sample)\n",
      "    if not os.path.exists(fasta_dir):\n",
      "        os.makedirs(fasta_dir)\n",
      "    pos_fo = open(pos_output_format % sample, 'w')\n",
      "    fasta_fo = open(fasta_output_format % sample, 'w')\n",
      "    mutable_ref_dict = copy.deepcopy(ref_dict)\n",
      "    for chrom in mutable_ref_dict:\n",
      "        mutable_ref_dict[chrom].seq = mutable_ref_dict[chrom].seq.tomutable()\n",
      "    previous_new_pos = 1\n",
      "    current_offset = 0\n",
      "    current_chrom = ''\n",
      "    for i, rec in enumerate(tbl_whole_genome_comparisons['alistair_ann']\n",
      "        .selecteq('Sample', sample)\n",
      "        .selecteq('IsInGATK', True)\n",
      "        .selecteq('RegionType', 'Core')\n",
      "        .selectgt('gatk_VQSLOD', 0.0)\n",
      "        .sort(('CHROM', 'POS'))\n",
      "        .data()\n",
      "    ):\n",
      "        if current_chrom != rec[0]:\n",
      "            print(\"%s\\t%d\\t%d\\t%d\\t%d\" % (current_chrom, 0, 9999999, previous_new_pos, current_offset), file=pos_fo)  \n",
      "            current_chrom = rec[0]\n",
      "            current_offset = 0\n",
      "            previous_new_pos = 1\n",
      "    #     if current_chrom == '':\n",
      "    #         current_chrom = rec[0]\n",
      "        reflen = len(rec[2])\n",
      "        altlen = len(rec[3])\n",
      "        pos = int(rec[1])\n",
      "        new_pos = pos + current_offset\n",
      "        print(\"%s\\t%d\\t%d\\t%d\\t%d\" % (rec[0], pos, new_pos, previous_new_pos, current_offset), file=pos_fo)  \n",
      "        previous_new_pos = new_pos\n",
      "        startpos = pos + current_offset - 1\n",
      "        endpos = pos + current_offset + reflen - 1   \n",
      "        mutable_ref_dict[rec[0]].seq[startpos:endpos] = rec[3]\n",
      "        current_offset = current_offset + altlen - reflen\n",
      "#         if i%1000 == 0:\n",
      "#             print(current_chrom, current_offset)\n",
      "\n",
      "    print(\"%s\\t%d\\t%d\\t%d\\t%d\" % (current_chrom, 0, 9999999, previous_new_pos, current_offset), file=pos_fo)  \n",
      "    \n",
      "    for chrom in mutable_ref_dict:\n",
      "        SeqIO.write(mutable_ref_dict[chrom], fasta_fo, \"fasta\")\n",
      "    "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 318
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for sample in tbl_samples_to_process['Isolate code']:\n",
      "    print(sample)\n",
      "    create_consensus(sample)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "7G8\n",
        "GB4"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "KE01"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "KH02"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "GN01"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n"
       ]
      }
     ],
     "prompt_number": 319
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def run_nucmer(sample='7G8',\n",
      "               fasta_output_format='/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/fasta/%s.fasta',\n",
      "               nucmer_output_format='/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/%s',\n",
      "               delta_filter_i=99):\n",
      "    gatk_fasta = fasta_output_format % sample\n",
      "    assembly_fasta = assembly_fastas[sample]\n",
      "    output_filestem = nucmer_output_format % sample\n",
      "    \n",
      "    !{NUCMER_DIR}/nucmer -p {output_filestem} {gatk_fasta} {assembly_fasta}\n",
      "    !{NUCMER_DIR}/delta-filter -m -i {delta_filter_i} -l 1000 {output_filestem}.delta > {output_filestem}.filter.delta\n",
      "    !{NUCMER_DIR}/show-coords -THqcl -o {output_filestem}.filter.delta > {output_filestem}.filter.coords\n",
      "    !{NUCMER_DIR}/show-snps -CHTr {output_filestem}.filter.delta > {output_filestem}.Csnp\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 227
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "run_nucmer()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1: PREPARING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "2,3: RUNNING mummer AND CREATING CLUSTERS\r\n",
        "# reading input file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 23306599\r\n",
        "# construct suffix tree for sequence of length 23306599\r\n",
        "# (maximum reference length is 536870908)\r\n",
        "# (maximum query length is 4294967295)\r\n",
        "# process 233065 characters per dot\r\n",
        "#."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        ".\r\n",
        "# CONSTRUCTIONTIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref 8.96\r\n",
        "# reading input file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/Pf7G8.Jul2015.fasta\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 22831828\r\n",
        "# matching query-file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/Pf7G8.Jul2015.fasta\"\r\n",
        "# against subject-file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref\"\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# COMPLETETIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref 26.56\r\n",
        "# SPACE /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref 44.64\r\n",
        "4: FINISHING DATA\r\n"
       ]
      }
     ],
     "prompt_number": 234
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for sample in tbl_samples_to_process['Isolate code']:\n",
      "    print(sample)\n",
      "    run_nucmer(sample)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "7G8\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1: PREPARING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "2,3: RUNNING mummer AND CREATING CLUSTERS\r\n",
        "# reading input file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 23306599\r\n",
        "# construct suffix tree for sequence of length 23306599\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# (maximum reference length is 536870908)\r\n",
        "# (maximum query length is 4294967295)\r\n",
        "# process 233065 characters per dot\r\n",
        "#."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        ".\r\n",
        "# CONSTRUCTIONTIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref 8.98\r\n",
        "# reading input file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/Pf7G8.Jul2015.fasta\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 22831828\r\n",
        "# matching query-file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/Pf7G8.Jul2015.fasta\"\r\n",
        "# against subject-file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref\"\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# COMPLETETIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref 25.57\r\n",
        "# SPACE /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.ntref 44.64\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "4: FINISHING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "GB4\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1: PREPARING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "2,3: RUNNING mummer AND CREATING CLUSTERS\r\n",
        "# reading input file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.ntref\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 23317129\r\n",
        "# construct suffix tree for sequence of length 23317129\r\n",
        "# (maximum reference length is 536870908)\r\n",
        "# (maximum query length is 4294967295)\r\n",
        "# process 233171 characters per dot\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "#.."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        ".\r\n",
        "# CONSTRUCTIONTIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.ntref 9.01\r\n",
        "# reading input file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/PfGB4.Jul2015.fasta\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 23525156\r\n",
        "# matching query-file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/ReleaseJul/PfGB4.Jul2015.fasta\"\r\n",
        "# against subject-file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.ntref\"\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# COMPLETETIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.ntref 25.84\r\n",
        "# SPACE /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.ntref 45.32\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "4: FINISHING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "KE01\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1: PREPARING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "2,3: RUNNING mummer AND CREATING CLUSTERS\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# reading input file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.ntref\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 23321647\r\n",
        "# construct suffix tree for sequence of length 23321647\r\n",
        "# (maximum reference length is 536870908)\r\n",
        "# (maximum query length is 4294967295)\r\n",
        "# process 233216 characters per dot\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "#.."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        ".\r\n",
        "# CONSTRUCTIONTIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.ntref 9.00\r\n",
        "# reading input file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfKE01.Jun2015.fasta\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 22794355\r\n",
        "# matching query-file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfKE01.Jun2015.fasta\"\r\n",
        "# against subject-file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.ntref\"\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# COMPLETETIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.ntref 25.40\r\n",
        "# SPACE /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.ntref 44.62\r\n",
        "4: FINISHING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "KH02\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1: PREPARING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "2,3: RUNNING mummer AND CREATING CLUSTERS\r\n",
        "# reading input file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.ntref\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 23302910\r\n",
        "# construct suffix tree for sequence of length 23302910\r\n",
        "# (maximum reference length is 536870908)\r\n",
        "# (maximum query length is 4294967295)\r\n",
        "# process 233029 characters per dot\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "#."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        ".\r\n",
        "# CONSTRUCTIONTIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.ntref 9.01\r\n",
        "# reading input file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfKH02.Jun2015.fasta\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 23136990\r\n",
        "# matching query-file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfKH02.Jun2015.fasta\"\r\n",
        "# against subject-file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.ntref\"\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# COMPLETETIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.ntref 25.86\r\n",
        "# SPACE /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.ntref 44.94\r\n",
        "4: FINISHING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "GN01\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1: PREPARING DATA\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "2,3: RUNNING mummer AND CREATING CLUSTERS\r\n",
        "# reading input file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.ntref\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 23325939\r\n",
        "# construct suffix tree for sequence of length 23325939\r\n",
        "# (maximum reference length is 536870908)\r\n",
        "# (maximum query length is 4294967295)\r\n",
        "# process 233259 characters per dot\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "#."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "."
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        ".\r\n",
        "# CONSTRUCTIONTIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.ntref 9.05\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# reading input file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfGN01.Jun2015.fasta\" "
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "of length 24097878\r\n",
        "# matching query-file \"/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/PfGN01.Jun2015.fasta\"\r\n",
        "# against subject-file \"/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.ntref\"\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "# COMPLETETIME /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.ntref 26.46\r\n",
        "# SPACE /Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt_4/mummer/MUMmer3.23/mummer /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.ntref 45.89\r\n",
        "4: FINISHING DATA\r\n"
       ]
      }
     ],
     "prompt_number": 229
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ref_chroms = ['M76611', 'PFC10_API_IRAB'] + [\"Pf3D7_%02d_v3\" % i for i in range(1, 15)]\n",
      "query_chroms = [\"Pf%s_MT\" % '7G8', \"Pf%s_API\" % '7G8'] + [\"Pf%s_%02d\" % ('7G8', i) for i in range(1, 15)]\n",
      "chrom_dict = dict(zip(ref_chroms, query_chroms))\n",
      "print(chrom_dict['M76611'])\n",
      "print(chrom_dict['Pf3D7_08_v3'])\n",
      "\n",
      "temp = dict(zip(('M76611', 'PFC10_API_IRAB'), ('Pf7G8_MT', 'Pf7G8_API')))\n",
      "temp['M76611']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Pf7G8_MT\n",
        "Pf7G8_08\n"
       ]
      },
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 226,
       "text": [
        "'Pf7G8_MT'"
       ]
      }
     ],
     "prompt_number": 226
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def convert_Csnp(sample='7G8',\n",
      "                 fasta_output_format='/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/fasta/%s.fasta',\n",
      "                 nucmer_output_format='/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/%s'):\n",
      "    gatk_fasta = fasta_output_format % sample\n",
      "    assembly_fasta = assembly_fastas[sample]\n",
      "    output_filestem = nucmer_output_format % sample\n",
      "\n",
      "    Csnp_fn=output_filestem + \".Csnp\"\n",
      "    vcf_fn=output_filestem + \".vcf\"\n",
      "    ref_dict=SeqIO.to_dict(SeqIO.parse(open(gatk_fasta), \"fasta\"))\n",
      "\n",
      "#     Write VCF header\n",
      "    fo = open(vcf_fn, 'w')\n",
      "    fo.write(\"##fileformat=VCFv4.1\\n\")\n",
      "    fo.write(\"##description=This file created with convert_Csnp function in 20150929_consensus_alignment.ipynb\\n\")\n",
      "    fo.write(\"#CHROM\\tPOS\\tID\\tREF\\tALT\\tQUAL\\tFILTER\\tINFO\\n\")\n",
      "#     Write data\n",
      "    ref_chroms = ['M76611', 'PFC10_API_IRAB'] + [\"Pf3D7_%02d_v3\" % i for i in range(1, 15)]\n",
      "    query_chroms = [\"Pf%s_MT\" % sample, \"Pf%s_API\" % sample] + [\"Pf%s_%02d\" % (sample, i) for i in range(1, 15)]\n",
      "    chrom_dict = dict(zip(ref_chroms, query_chroms))\n",
      "\n",
      "    tbl_csnp = (etl.fromtsv(Csnp_fn).select(lambda rec: rec[9] == chrom_dict[rec[8]]))\n",
      "    \n",
      "    current_state = \"SNP\"\n",
      "    ins_sequence = ''\n",
      "    previous_chrom = ''\n",
      "    for rec in tbl_csnp:\n",
      "#         print('.', end='')\n",
      "        (pos, ref, alt, pos2, buff, dist, frm, frm2, chrom, query_chrom) = rec\n",
      "#         (chrom, bba, variant_type, pos, pos2, zero, strand, dot, note) = rec\n",
      "        pos = int(pos)\n",
      "        pos2 = int(pos2)\n",
      "        if (\n",
      "            frm != '1' or\n",
      "            not(ref in ['A', 'C', 'T', 'G', '.']) or\n",
      "            not(alt in ['A', 'C', 'T', 'G', '.'])\n",
      "        ):\n",
      "            return(\"error\")\n",
      "        if alt == '.':\n",
      "            variant_type = 'Del'\n",
      "        elif ref == '.':\n",
      "            variant_type = 'Ins'\n",
      "        else:\n",
      "            variant_type = 'SNP'\n",
      "        if variant_type == 'SNP':\n",
      "            if current_state == 'Del':\n",
      "                del_ref = str(ref_dict[previous_chrom][variant_start-1:variant_end].seq)\n",
      "                del_alt = ref_dict[previous_chrom][variant_start-1]\n",
      "                fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (previous_chrom, variant_start, del_ref, del_alt))\n",
      "            if current_state == 'Ins':\n",
      "                ins_ref = ref_dict[previous_chrom][variant_start-1]\n",
      "                ins_alt = ref_dict[previous_chrom][variant_start-1] + ins_sequence\n",
      "                fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (previous_chrom, variant_start, ins_ref, ins_alt))\n",
      "                ins_sequence = ''\n",
      "#             ref = ref_dict[chrom][pos-1]\n",
      "#             alt = note[20]\n",
      "            fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (chrom, pos, ref, alt))\n",
      "            current_state = \"SNP\"\n",
      "        elif variant_type == 'Del':\n",
      "            if current_state == 'Del':\n",
      "                if pos > (variant_end+1) or chrom != previous_chrom: # i.e. we have moved into a different del so need to print out previous\n",
      "                    if variant_start < 1 or variant_start > len(ref_dict[chrom]):\n",
      "                        print(chrom, variant_start, ref_dict[chrom], len(ref_dict[chrom]))\n",
      "                    del_ref = str(ref_dict[previous_chrom][variant_start-1:variant_end].seq)\n",
      "                    del_alt = ref_dict[previous_chrom][variant_start-1]\n",
      "                    fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (previous_chrom, variant_start, del_ref, del_alt))\n",
      "                    variant_start = pos-1\n",
      "            if current_state == 'Ins':\n",
      "                ins_ref = ref_dict[previous_chrom][variant_start-1]\n",
      "                ins_alt = ref_dict[previous_chrom][variant_start-1] + ins_sequence\n",
      "                fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (previous_chrom, variant_start, ins_ref, ins_alt))\n",
      "                variant_start = pos-1\n",
      "            if current_state == 'SNP':\n",
      "                variant_start = pos-1\n",
      "            variant_end = pos\n",
      "            current_state = \"Del\"\n",
      "        elif variant_type == 'Ins':\n",
      "            if current_state == 'Del':\n",
      "                del_ref = str(ref_dict[previous_chrom][variant_start-1:variant_end].seq)\n",
      "                del_alt = ref_dict[previous_chrom][variant_start-1]\n",
      "                fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (previous_chrom, variant_start, del_ref, del_alt))\n",
      "                variant_start = pos\n",
      "            if current_state == 'Ins':\n",
      "                if pos > (variant_end+1) or chrom != previous_chrom: # i.e. we have moved into a different del so need to print out previous\n",
      "                    ins_ref = ref_dict[previous_chrom][variant_start-1]\n",
      "                    ins_alt = ref_dict[previous_chrom][variant_start-1] + ins_sequence\n",
      "                    fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (previous_chrom, variant_start, ins_ref, ins_alt))\n",
      "                    variant_start = pos\n",
      "                    ins_sequence = ''\n",
      "            if current_state == 'SNP':\n",
      "                variant_start = pos\n",
      "#             ins_sequence = ins_sequence + note[26]\n",
      "            ins_sequence = ins_sequence + alt\n",
      "            variant_end = pos\n",
      "            current_state = \"Ins\"\n",
      "#         elif variant_type == 'Synteny':\n",
      "#             if current_state == 'Del':\n",
      "#                 ref = str(ref_dict[chrom][variant_start-1:variant_end].seq)\n",
      "#                 alt = ref_dict[chrom][variant_start]\n",
      "#                 fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (chrom, variant_start, ref, alt))\n",
      "#             if current_state == 'Ins':\n",
      "#                 ref = ref_dict[chrom][variant_start-1]\n",
      "#                 alt = ref_dict[chrom][variant_start-1] + ins_sequence\n",
      "#                 fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (chrom, variant_start, ref, alt))\n",
      "#                 ins_sequence = ''\n",
      "#             current_state = \"Synteny\"\n",
      "        else:\n",
      "            return(\"error\")\n",
      "        previous_chrom = chrom\n",
      "    fo.close()\n",
      "    \n",
      "    return(vcf_fn)\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 325
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "convert_Csnp()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 326,
       "text": [
        "'/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.vcf'"
       ]
      }
     ],
     "prompt_number": 326
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def tbl_converted_coords(sample='7G8',\n",
      "        pos_output_format = '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/pos/%s.pos',\n",
      "        nucmer_output_format='/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/%s'):\n",
      "    \n",
      "    pos_fn = pos_output_format % sample\n",
      "    output_filestem = nucmer_output_format % sample\n",
      "    vcf_fn=output_filestem + \".vcf\"\n",
      "    output_vcf_fn=output_filestem + \".3d7coordinates.vcf\"\n",
      "    \n",
      "    tbl_vcf = etl.fromvcf(vcf_fn).convert('POS', int).addfield('POS1', lambda rec: rec['POS']+1)\n",
      "    tbl_pos = (etl.fromtsv(pos_fn)\n",
      "        .setheader(('CHROM', 'var_pos', 'end', 'start', 'diff'))\n",
      "        .convert('var_pos', int)\n",
      "        .convert('start', int)\n",
      "        .convert('end', int)\n",
      "        .convert('diff', int)\n",
      "        .select(lambda rec: rec['end'] > rec['start'])\n",
      "    )\n",
      "    \n",
      "#     tbl_vcf.display()\n",
      "#     tbl_pos.display()\n",
      "    \n",
      "    tbl_vcf_converted = (tbl_vcf\n",
      "        .intervalleftjoin(tbl_pos, lstart='POS', lstop='POS1', lkey='CHROM',\n",
      "                          rstart='start', rstop='end', rkey='CHROM')\n",
      "        .addfield('NEW_POS', lambda rec: rec['POS'] if rec['diff'] is None else rec['POS'] - rec['diff'])\n",
      "    )\n",
      "#     tbl_vcf_converted.display()\n",
      "\n",
      "#     Write VCF header\n",
      "    fo = open(output_vcf_fn, 'w')\n",
      "    fo.write(\"##fileformat=VCFv4.1\\n\")\n",
      "    fo.write(\"##description=This file created with tbl_converted_coords function in 20150929_consensus_alignment.ipynb\\n\")\n",
      "    fo.write(\"#CHROM\\tPOS\\tID\\tREF\\tALT\\tQUAL\\tFILTER\\tINFO\\n\")\n",
      "\n",
      "    for rec in tbl_vcf_converted.data():\n",
      "        fo.write(\"%s\\t%s\\t.\\t%s\\t%s\\t.\\tPASS\\t.\\n\" % (rec[0], rec[13], rec[3], rec[4][0]))\n",
      "        \n",
      "    return tbl_vcf_converted\n",
      "    \n",
      "    \n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 360
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tbl_vcf_converted = tbl_converted_coords()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 361
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tbl_vcf_converted.display(50, index_header=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>0|CHROM</th>\n",
        "<th>1|POS</th>\n",
        "<th>2|ID</th>\n",
        "<th>3|REF</th>\n",
        "<th>4|ALT</th>\n",
        "<th>5|QUAL</th>\n",
        "<th>6|FILTER</th>\n",
        "<th>7|INFO</th>\n",
        "<th>8|POS1</th>\n",
        "<th>9|var_pos</th>\n",
        "<th>10|end</th>\n",
        "<th>11|start</th>\n",
        "<th>12|diff</th>\n",
        "<th>13|NEW_POS</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>M76611</td>\n",
        "<td style='text-align: right'>772</td>\n",
        "<td>None</td>\n",
        "<td>T</td>\n",
        "<td>[C]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>773</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>772</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>M76611</td>\n",
        "<td style='text-align: right'>1692</td>\n",
        "<td>None</td>\n",
        "<td>G</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>1693</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1692</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>M76611</td>\n",
        "<td style='text-align: right'>4179</td>\n",
        "<td>None</td>\n",
        "<td>C</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>4180</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>4179</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>PFC10_API_IRAB</td>\n",
        "<td style='text-align: right'>15420</td>\n",
        "<td>None</td>\n",
        "<td>AATTAAAAATATGTA</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>15421</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>15420</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>PFC10_API_IRAB</td>\n",
        "<td style='text-align: right'>16656</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[G]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>16657</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>16656</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>102905</td>\n",
        "<td>None</td>\n",
        "<td>AT</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>102906</td>\n",
        "<td style='text-align: right'>103840</td>\n",
        "<td style='text-align: right'>103835</td>\n",
        "<td style='text-align: right'>101264</td>\n",
        "<td style='text-align: right'>-5</td>\n",
        "<td style='text-align: right'>102910</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>102911</td>\n",
        "<td>None</td>\n",
        "<td>AT</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>102912</td>\n",
        "<td style='text-align: right'>103840</td>\n",
        "<td style='text-align: right'>103835</td>\n",
        "<td style='text-align: right'>101264</td>\n",
        "<td style='text-align: right'>-5</td>\n",
        "<td style='text-align: right'>102916</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>102921</td>\n",
        "<td>None</td>\n",
        "<td>T</td>\n",
        "<td>[TA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>102922</td>\n",
        "<td style='text-align: right'>103840</td>\n",
        "<td style='text-align: right'>103835</td>\n",
        "<td style='text-align: right'>101264</td>\n",
        "<td style='text-align: right'>-5</td>\n",
        "<td style='text-align: right'>102926</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>102925</td>\n",
        "<td>None</td>\n",
        "<td>T</td>\n",
        "<td>[TA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>102926</td>\n",
        "<td style='text-align: right'>103840</td>\n",
        "<td style='text-align: right'>103835</td>\n",
        "<td style='text-align: right'>101264</td>\n",
        "<td style='text-align: right'>-5</td>\n",
        "<td style='text-align: right'>102930</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>104436</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[ATATATATATATA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>104437</td>\n",
        "<td style='text-align: right'>104659</td>\n",
        "<td style='text-align: right'>104654</td>\n",
        "<td style='text-align: right'>104113</td>\n",
        "<td style='text-align: right'>-5</td>\n",
        "<td style='text-align: right'>104441</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>108423</td>\n",
        "<td>None</td>\n",
        "<td>C</td>\n",
        "<td>[CAAAAAAAAAA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>108424</td>\n",
        "<td style='text-align: right'>108579</td>\n",
        "<td style='text-align: right'>108550</td>\n",
        "<td style='text-align: right'>108340</td>\n",
        "<td style='text-align: right'>-29</td>\n",
        "<td style='text-align: right'>108452</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>116608</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[ATATGTTCATTTATTTATACATATATTTATTCATTTATTTGTTCATATA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>116609</td>\n",
        "<td style='text-align: right'>117080</td>\n",
        "<td style='text-align: right'>117045</td>\n",
        "<td style='text-align: right'>115824</td>\n",
        "<td style='text-align: right'>-35</td>\n",
        "<td style='text-align: right'>116643</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>116822</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[AT]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>116823</td>\n",
        "<td style='text-align: right'>117080</td>\n",
        "<td style='text-align: right'>117045</td>\n",
        "<td style='text-align: right'>115824</td>\n",
        "<td style='text-align: right'>-35</td>\n",
        "<td style='text-align: right'>116857</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>120993</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[AAAA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>120994</td>\n",
        "<td style='text-align: right'>121289</td>\n",
        "<td style='text-align: right'>121254</td>\n",
        "<td style='text-align: right'>118236</td>\n",
        "<td style='text-align: right'>-35</td>\n",
        "<td style='text-align: right'>121028</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>123290</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[AAAAAAA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>123291</td>\n",
        "<td style='text-align: right'>123589</td>\n",
        "<td style='text-align: right'>123568</td>\n",
        "<td style='text-align: right'>122776</td>\n",
        "<td style='text-align: right'>-21</td>\n",
        "<td style='text-align: right'>123311</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>123603</td>\n",
        "<td>None</td>\n",
        "<td>CT</td>\n",
        "<td>[C]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>123604</td>\n",
        "<td style='text-align: right'>123770</td>\n",
        "<td style='text-align: right'>123753</td>\n",
        "<td style='text-align: right'>123568</td>\n",
        "<td style='text-align: right'>-17</td>\n",
        "<td style='text-align: right'>123620</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>124449</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[AAAAAAATATATATATATATATA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>124450</td>\n",
        "<td style='text-align: right'>124599</td>\n",
        "<td style='text-align: right'>124572</td>\n",
        "<td style='text-align: right'>123753</td>\n",
        "<td style='text-align: right'>-27</td>\n",
        "<td style='text-align: right'>124476</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>125456</td>\n",
        "<td>None</td>\n",
        "<td>G</td>\n",
        "<td>[GA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>125457</td>\n",
        "<td style='text-align: right'>125521</td>\n",
        "<td style='text-align: right'>125490</td>\n",
        "<td style='text-align: right'>124572</td>\n",
        "<td style='text-align: right'>-31</td>\n",
        "<td style='text-align: right'>125487</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>128187</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[ATATATATATATATA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>128188</td>\n",
        "<td style='text-align: right'>128285</td>\n",
        "<td style='text-align: right'>128277</td>\n",
        "<td style='text-align: right'>128138</td>\n",
        "<td style='text-align: right'>-8</td>\n",
        "<td style='text-align: right'>128195</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>131372</td>\n",
        "<td>None</td>\n",
        "<td>AATATAAAAAT</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>131373</td>\n",
        "<td style='text-align: right'>132196</td>\n",
        "<td style='text-align: right'>132186</td>\n",
        "<td style='text-align: right'>130877</td>\n",
        "<td style='text-align: right'>-10</td>\n",
        "<td style='text-align: right'>131382</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>133180</td>\n",
        "<td>None</td>\n",
        "<td>ATATATATATATATATTT</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>133181</td>\n",
        "<td style='text-align: right'>133447</td>\n",
        "<td style='text-align: right'>133449</td>\n",
        "<td style='text-align: right'>132186</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>133178</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>134160</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[ATATATATATATATAAA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>134161</td>\n",
        "<td style='text-align: right'>136437</td>\n",
        "<td style='text-align: right'>136447</td>\n",
        "<td style='text-align: right'>134007</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>134150</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>136631</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[ACCATGAATATCTTCATTACTATGGATATCTTCTTTACCATGAATATCTTCATTA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>136632</td>\n",
        "<td style='text-align: right'>136681</td>\n",
        "<td style='text-align: right'>136709</td>\n",
        "<td style='text-align: right'>136447</td>\n",
        "<td style='text-align: right'>28</td>\n",
        "<td style='text-align: right'>136603</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>139516</td>\n",
        "<td>None</td>\n",
        "<td>C</td>\n",
        "<td>[CTTTT]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>139517</td>\n",
        "<td style='text-align: right'>139488</td>\n",
        "<td style='text-align: right'>139557</td>\n",
        "<td style='text-align: right'>139405</td>\n",
        "<td style='text-align: right'>69</td>\n",
        "<td style='text-align: right'>139447</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>140454</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[AA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>140455</td>\n",
        "<td style='text-align: right'>140411</td>\n",
        "<td style='text-align: right'>140468</td>\n",
        "<td style='text-align: right'>140263</td>\n",
        "<td style='text-align: right'>57</td>\n",
        "<td style='text-align: right'>140397</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>140567</td>\n",
        "<td>None</td>\n",
        "<td>TTTTTTTTTTTATTTTTTTATTTTTTT</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>140568</td>\n",
        "<td style='text-align: right'>141152</td>\n",
        "<td style='text-align: right'>141197</td>\n",
        "<td style='text-align: right'>140468</td>\n",
        "<td style='text-align: right'>45</td>\n",
        "<td style='text-align: right'>140522</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>140595</td>\n",
        "<td>None</td>\n",
        "<td>ATTTTTT</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>140596</td>\n",
        "<td style='text-align: right'>141152</td>\n",
        "<td style='text-align: right'>141197</td>\n",
        "<td style='text-align: right'>140468</td>\n",
        "<td style='text-align: right'>45</td>\n",
        "<td style='text-align: right'>140550</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>141231</td>\n",
        "<td>None</td>\n",
        "<td>CACAAACATAAACATAAACATAAACATAAACACAAAC</td>\n",
        "<td>[C]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>141232</td>\n",
        "<td style='text-align: right'>141594</td>\n",
        "<td style='text-align: right'>141633</td>\n",
        "<td style='text-align: right'>141197</td>\n",
        "<td style='text-align: right'>39</td>\n",
        "<td style='text-align: right'>141192</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>141275</td>\n",
        "<td>None</td>\n",
        "<td>TAA</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>141276</td>\n",
        "<td style='text-align: right'>141594</td>\n",
        "<td style='text-align: right'>141633</td>\n",
        "<td style='text-align: right'>141197</td>\n",
        "<td style='text-align: right'>39</td>\n",
        "<td style='text-align: right'>141236</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>141281</td>\n",
        "<td>None</td>\n",
        "<td>TAA</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>141282</td>\n",
        "<td style='text-align: right'>141594</td>\n",
        "<td style='text-align: right'>141633</td>\n",
        "<td style='text-align: right'>141197</td>\n",
        "<td style='text-align: right'>39</td>\n",
        "<td style='text-align: right'>141242</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>142509</td>\n",
        "<td>None</td>\n",
        "<td>TA</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>142510</td>\n",
        "<td style='text-align: right'>142559</td>\n",
        "<td style='text-align: right'>142589</td>\n",
        "<td style='text-align: right'>142425</td>\n",
        "<td style='text-align: right'>30</td>\n",
        "<td style='text-align: right'>142479</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>142513</td>\n",
        "<td>None</td>\n",
        "<td>AT</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>142514</td>\n",
        "<td style='text-align: right'>142559</td>\n",
        "<td style='text-align: right'>142589</td>\n",
        "<td style='text-align: right'>142425</td>\n",
        "<td style='text-align: right'>30</td>\n",
        "<td style='text-align: right'>142483</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>142648</td>\n",
        "<td>None</td>\n",
        "<td>ATATATATATATATATATATATATA</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>142649</td>\n",
        "<td style='text-align: right'>143309</td>\n",
        "<td style='text-align: right'>143339</td>\n",
        "<td style='text-align: right'>142589</td>\n",
        "<td style='text-align: right'>30</td>\n",
        "<td style='text-align: right'>142618</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>143372</td>\n",
        "<td>None</td>\n",
        "<td>T</td>\n",
        "<td>[TATATA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>143373</td>\n",
        "<td style='text-align: right'>145515</td>\n",
        "<td style='text-align: right'>145533</td>\n",
        "<td style='text-align: right'>143339</td>\n",
        "<td style='text-align: right'>18</td>\n",
        "<td style='text-align: right'>143354</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>147035</td>\n",
        "<td>None</td>\n",
        "<td>ATT</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>147036</td>\n",
        "<td style='text-align: right'>147296</td>\n",
        "<td style='text-align: right'>147301</td>\n",
        "<td style='text-align: right'>147033</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td style='text-align: right'>147030</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>147826</td>\n",
        "<td>None</td>\n",
        "<td>TTTTTTTTTTTT</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>147827</td>\n",
        "<td style='text-align: right'>148338</td>\n",
        "<td style='text-align: right'>148329</td>\n",
        "<td style='text-align: right'>147466</td>\n",
        "<td style='text-align: right'>-9</td>\n",
        "<td style='text-align: right'>147835</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>149611</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[AATATAT]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>149612</td>\n",
        "<td style='text-align: right'>150836</td>\n",
        "<td style='text-align: right'>150821</td>\n",
        "<td style='text-align: right'>149583</td>\n",
        "<td style='text-align: right'>-15</td>\n",
        "<td style='text-align: right'>149626</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>151175</td>\n",
        "<td>None</td>\n",
        "<td>T</td>\n",
        "<td>[C]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>151176</td>\n",
        "<td style='text-align: right'>152123</td>\n",
        "<td style='text-align: right'>152108</td>\n",
        "<td style='text-align: right'>150821</td>\n",
        "<td style='text-align: right'>-15</td>\n",
        "<td style='text-align: right'>151190</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>151187</td>\n",
        "<td>None</td>\n",
        "<td>T</td>\n",
        "<td>[TA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>151188</td>\n",
        "<td style='text-align: right'>152123</td>\n",
        "<td style='text-align: right'>152108</td>\n",
        "<td style='text-align: right'>150821</td>\n",
        "<td style='text-align: right'>-15</td>\n",
        "<td style='text-align: right'>151202</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>188259</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>188260</td>\n",
        "<td style='text-align: right'>188526</td>\n",
        "<td style='text-align: right'>188558</td>\n",
        "<td style='text-align: right'>187734</td>\n",
        "<td style='text-align: right'>32</td>\n",
        "<td style='text-align: right'>188227</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>188265</td>\n",
        "<td>None</td>\n",
        "<td>T</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>188266</td>\n",
        "<td style='text-align: right'>188526</td>\n",
        "<td style='text-align: right'>188558</td>\n",
        "<td style='text-align: right'>187734</td>\n",
        "<td style='text-align: right'>32</td>\n",
        "<td style='text-align: right'>188233</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>188268</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[G]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>188269</td>\n",
        "<td style='text-align: right'>188526</td>\n",
        "<td style='text-align: right'>188558</td>\n",
        "<td style='text-align: right'>187734</td>\n",
        "<td style='text-align: right'>32</td>\n",
        "<td style='text-align: right'>188236</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>188272</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[AT]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>188273</td>\n",
        "<td style='text-align: right'>188526</td>\n",
        "<td style='text-align: right'>188558</td>\n",
        "<td style='text-align: right'>187734</td>\n",
        "<td style='text-align: right'>32</td>\n",
        "<td style='text-align: right'>188240</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>188274</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>188275</td>\n",
        "<td style='text-align: right'>188526</td>\n",
        "<td style='text-align: right'>188558</td>\n",
        "<td style='text-align: right'>187734</td>\n",
        "<td style='text-align: right'>32</td>\n",
        "<td style='text-align: right'>188242</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>188280</td>\n",
        "<td>None</td>\n",
        "<td>T</td>\n",
        "<td>[TA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>188281</td>\n",
        "<td style='text-align: right'>188526</td>\n",
        "<td style='text-align: right'>188558</td>\n",
        "<td style='text-align: right'>187734</td>\n",
        "<td style='text-align: right'>32</td>\n",
        "<td style='text-align: right'>188248</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>188283</td>\n",
        "<td>None</td>\n",
        "<td>T</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>188284</td>\n",
        "<td style='text-align: right'>188526</td>\n",
        "<td style='text-align: right'>188558</td>\n",
        "<td style='text-align: right'>187734</td>\n",
        "<td style='text-align: right'>32</td>\n",
        "<td style='text-align: right'>188251</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>189014</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[ATATA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>189015</td>\n",
        "<td style='text-align: right'>189996</td>\n",
        "<td style='text-align: right'>190049</td>\n",
        "<td style='text-align: right'>188924</td>\n",
        "<td style='text-align: right'>53</td>\n",
        "<td style='text-align: right'>188961</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>189233</td>\n",
        "<td>None</td>\n",
        "<td>A</td>\n",
        "<td>[ATA]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>189234</td>\n",
        "<td style='text-align: right'>189996</td>\n",
        "<td style='text-align: right'>190049</td>\n",
        "<td style='text-align: right'>188924</td>\n",
        "<td style='text-align: right'>53</td>\n",
        "<td style='text-align: right'>189180</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>189383</td>\n",
        "<td>None</td>\n",
        "<td>ATATA</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>189384</td>\n",
        "<td style='text-align: right'>189996</td>\n",
        "<td style='text-align: right'>190049</td>\n",
        "<td style='text-align: right'>188924</td>\n",
        "<td style='text-align: right'>53</td>\n",
        "<td style='text-align: right'>189330</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>189501</td>\n",
        "<td>None</td>\n",
        "<td>TATATATATAT</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>[]</td>\n",
        "<td>{}</td>\n",
        "<td style='text-align: right'>189502</td>\n",
        "<td style='text-align: right'>189996</td>\n",
        "<td style='text-align: right'>190049</td>\n",
        "<td style='text-align: right'>188924</td>\n",
        "<td style='text-align: right'>53</td>\n",
        "<td style='text-align: right'>189448</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n",
        "<p><strong>...</strong></p>"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 355
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print(ref_dict['Pf3D7_01_v3'][102910 - 1]) #A\n",
      "print(ref_dict['Pf3D7_01_v3'][123620 - 1]) #C\n",
      "print(ref_dict['Pf3D7_01_v3'][139447 - 1]) #C\n",
      "print(ref_dict['Pf3D7_01_v3'][141192 - 1]) #C\n",
      "print(ref_dict['Pf3D7_01_v3'][147835 - 1]) #T\n",
      "print(ref_dict['Pf3D7_01_v3'][188251 - 1]) #T\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "A\n",
        "C\n",
        "C\n",
        "C\n",
        "T\n",
        "T\n"
       ]
      }
     ],
     "prompt_number": 352
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Install vcflib"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# On mac, simply:\n",
      "# brew install vcflib"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 364
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Annotate consensus alignment vcf"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "rewrite=True\n",
      "\n",
      "for sample in tbl_samples_to_process['Isolate code']:\n",
      "    print(sample)\n",
      "    nucmer_output_filestem='/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/%s' % sample\n",
      "    \n",
      "    unannotated_vcf_fn = \"%s.3d7coordinates.vcf\" % (nucmer_output_filestem)\n",
      "    sorted_vcf_fn = \"%s.sorted.vcf\" % (nucmer_output_filestem)\n",
      "    left_aligned_vcf_fn = \"%s.leftaligned.vcf\" % (nucmer_output_filestem)\n",
      "    snpeff_vcf_fn = \"%s.snpeff.vcf\" % (nucmer_output_filestem)\n",
      "    snpeff_annotated_vcf_fn = \"%s.snpeff_annotated.vcf\" % (nucmer_output_filestem)\n",
      "    annotated_vcf_fn = \"%s.annotated.vcf\" % (nucmer_output_filestem)\n",
      "\n",
      "    convert_Csnp(sample)\n",
      "    tbl_converted_coords(sample)\n",
      "    \n",
      "    !vcfstreamsort < {unannotated_vcf_fn} > {sorted_vcf_fn}\n",
      "\n",
      "    if (not os.path.isfile(left_aligned_vcf_fn) and not os.path.isfile(annotated_vcf_fn+'.gz')) or rewrite:\n",
      "        !{gatk_exe} -T LeftAlignAndTrimVariants \\\n",
      "        -R {REF_GENOME} \\\n",
      "        -V {sorted_vcf_fn} \\\n",
      "        -o {left_aligned_vcf_fn} \\\n",
      "#         2> /dev/null\n",
      "\n",
      "    if (not os.path.isfile(snpeff_vcf_fn) and not os.path.isfile(annotated_vcf_fn+'.gz')) or rewrite:\n",
      "        !{snpeff_exe} \\\n",
      "        -v -o gatk Pf3D7july2015 \\\n",
      "        {left_aligned_vcf_fn} \\\n",
      "        -no-downstream \\\n",
      "        -no-upstream \\\n",
      "        > {snpeff_vcf_fn} \\\n",
      "#         2> /dev/null\n",
      "\n",
      "    if (not os.path.isfile(snpeff_annotated_vcf_fn) and not os.path.isfile(annotated_vcf_fn+'.gz')) or rewrite:\n",
      "        !{gatk_exe} \\\n",
      "        -T VariantAnnotator \\\n",
      "        -R {REF_GENOME} \\\n",
      "        -A TandemRepeatAnnotator \\\n",
      "        -A SnpEff \\\n",
      "        --variant {left_aligned_vcf_fn} \\\n",
      "        --snpEffFile {snpeff_vcf_fn} \\\n",
      "        -o {snpeff_annotated_vcf_fn} \\\n",
      "#         2> /dev/null\n",
      "\n",
      "    if not os.path.isfile(annotated_vcf_fn+'.gz') or rewrite:\n",
      "        !cat {snpeff_annotated_vcf_fn} \\\n",
      "        | vcf-annotate -a {regions_fn} \\\n",
      "           -d key=INFO,ID=RegionType,Number=1,Type=String,Description='The type of genome region within which the variant is found. SubtelomericRepeat: repetitive regions at the ends of the chromosomes. SubtelomericHypervariable: subtelomeric region of poor conservation between the 3D7 reference genome and other samples. InternalHypervariable: chromosome-internal region of poor conservation between the 3D7 reference genome and other samples. Centromere: start and end coordinates of the centromere genome annotation. Core: everything else.' \\\n",
      "           -c CHROM,FROM,TO,INFO/RegionType \\\n",
      "        > {annotated_vcf_fn}\n",
      "\n",
      "        !bgzip -f {annotated_vcf_fn}\n",
      "        !tabix -p vcf -f {annotated_vcf_fn}.gz\n",
      "\n",
      "    if not os.path.isfile(annotated_vcf_fn.replace('.vcf', '.SNP.vcf')+'.gz') or rewrite:\n",
      "        !{gatk_exe} \\\n",
      "        -T SelectVariants \\\n",
      "        -R {REF_GENOME} \\\n",
      "        -V {annotated_vcf_fn}.gz \\\n",
      "        -selectType SNP \\\n",
      "        -o {annotated_vcf_fn.replace('.vcf', '.SNP.vcf')} 2> /dev/null\n",
      "\n",
      "        !bgzip -f {annotated_vcf_fn.replace('.vcf', '.SNP.vcf')}\n",
      "        !tabix -p vcf -f {annotated_vcf_fn.replace('.vcf', '.SNP.vcf')}.gz\n",
      "\n",
      "    if not os.path.isfile(annotated_vcf_fn.replace('.vcf', '.INDEL.vcf')+'.gz') or rewrite:\n",
      "        !{gatk_exe} \\\n",
      "        -T SelectVariants \\\n",
      "        -R {REF_GENOME} \\\n",
      "        -V {annotated_vcf_fn}.gz \\\n",
      "        -xlSelectType SNP \\\n",
      "        -o {annotated_vcf_fn.replace('.vcf', '.INDEL.vcf')} 2> /dev/null\n",
      "\n",
      "        !bgzip -f {annotated_vcf_fn.replace('.vcf', '.INDEL.vcf')}\n",
      "        !tabix -p vcf -f {annotated_vcf_fn.replace('.vcf', '.INDEL.vcf')}.gz\n",
      "\n",
      "#         !rm {unannotated_vcf_fn}\n",
      "#         !rm {left_aligned_vcf_fn}\n",
      "#         !rm {snpeff_vcf_fn}\n",
      "#         !rm {snpeff_annotated_vcf_fn}\n",
      "#         !rm {unannotated_vcf_fn}.idx\n",
      "#         !rm {left_aligned_vcf_fn}.idx\n",
      "#         !rm {snpeff_vcf_fn}.idx\n",
      "#         !rm {snpeff_annotated_vcf_fn}.idx\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "7G8\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:23,310 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:32:23,312 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:32:23,312 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:32:23,312 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:23,316 HelpFormatter - Program Args: -T LeftAlignAndTrimVariants -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -V /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.sorted.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.leftaligned.vcf \r\n",
        "INFO  16:32:23,322 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:32:23,322 HelpFormatter - Date/Time: 2015/10/02 16:32:23 \r\n",
        "INFO  16:32:23,323 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:32:23,323 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:23,674 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:23,729 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 1000 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:23,773 RMDTrackBuilder - Index file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.sorted.vcf.idx is out of date (index older than input file), deleting and updating the index file \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:24,088 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.sorted.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:24,167 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:32:24,168 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:32:24,168 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:32:24,169 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:32:24,169 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "7638 variants were aligned\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:25,348 ProgressMeter -            done     52845.0     1.0 s      22.0 s       99.6%     1.0 s       0.0 s \r\n",
        "INFO  16:32:25,349 ProgressMeter - Total runtime 1.18 secs, 0.02 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:26,666 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.000\tSnpEff version SnpEff 4.1g (build 2015-05-17), by Pablo Cingolani\r\n",
        "00:00:00.011\tCommand: 'ann'\r\n",
        "00:00:00.031\tReading configuration file 'snpEff.config'. Genome: 'Pf3D7july2015'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.527\tdone\r\n",
        "00:00:00.527\tReading database for genome version 'Pf3D7july2015' from file '/Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt/snpeff/snpEff/./data/Pf3D7july2015/snpEffectPredictor.bin' (this might take a while)\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.372\tdone\r\n",
        "00:00:01.382\tBuilding interval forest\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.773\tdone.\r\n",
        "00:00:01.773\tGenome stats :\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "#-----------------------------------------------\r\n",
        "# Genome name                : 'Plasmodium_falciparum'\r\n",
        "# Genome version             : 'Pf3D7july2015'\r\n",
        "# Has protein coding info    : true\r\n",
        "# Genes                      : 5645\r\n",
        "# Protein coding genes       : 5492\r\n",
        "#-----------------------------------------------\r\n",
        "# Transcripts                : 5886\r\n",
        "# Avg. transcripts per gene  : 1.04\r\n",
        "#-----------------------------------------------\r\n",
        "# Checked transcripts        : \r\n",
        "#               AA sequences :      0 ( 0.00% )\r\n",
        "#              DNA sequences :      0 ( 0.00% )\r\n",
        "#-----------------------------------------------\r\n",
        "# Protein coding transcripts : 5580\r\n",
        "#              Length errors :    126 ( 2.26% )\r\n",
        "#  STOP codons in CDS errors :    145 ( 2.60% )\r\n",
        "#         START codon errors :    200 ( 3.58% )\r\n",
        "#        STOP codon warnings :    205 ( 3.67% )\r\n",
        "#              UTR sequences :      0 ( 0.00% )\r\n",
        "#               Total Errors :    212 ( 3.80% )\r\n",
        "# WARNING                    : No protein coding transcript has UTR\r\n",
        "#-----------------------------------------------\r\n",
        "# Cds                        : 14366\r\n",
        "# Exons                      : 14366\r\n",
        "# Exons with sequence        : 14366\r\n",
        "# Exons without sequence     : 0\r\n",
        "# Avg. exons per transcript  : 2.44\r\n",
        "#-----------------------------------------------\r\n",
        "# Number of chromosomes      : 41\r\n",
        "# Chromosomes                : Format 'chromo_name size codon_table'\r\n",
        "#\t\t'Pf3D7_14_v3'\t3291936\tStandard\r\n",
        "#\t\t'Pf3D7_13_v3'\t2925236\tStandard\r\n",
        "#\t\t'Pf3D7_12_v3'\t2271494\tStandard\r\n",
        "#\t\t'Pf3D7_11_v3'\t2038340\tStandard\r\n",
        "#\t\t'Pf3D7_10_v3'\t1687656\tStandard\r\n",
        "#\t\t'Pf3D7_09_v3'\t1541735\tStandard\r\n",
        "#\t\t'Pf3D7_08_v3'\t1472805\tStandard\r\n",
        "#\t\t'Pf3D7_07_v3'\t1445207\tStandard\r\n",
        "#\t\t'Pf3D7_06_v3'\t1418242\tStandard\r\n",
        "#\t\t'Pf3D7_05_v3'\t1343557\tStandard\r\n",
        "#\t\t'Pf3D7_04_v3'\t1200490\tStandard\r\n",
        "#\t\t'Pf3D7_03_v3'\t1067971\tStandard\r\n",
        "#\t\t'Pf3D7_02_v3'\t947102\tStandard\r\n",
        "#\t\t'Pf3D7_01_v3'\t640851\tStandard\r\n",
        "#\t\t'Pf_M76611'\t5967\tProtozoan_Mitochondrial\r\n",
        "#\t\t'mal_mito_1'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_2'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_3'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_RNA19:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_SSUB:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_10:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_15:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_16:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_17:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_1:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_20:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_21:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_9:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUC:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUG:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA11:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA12:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA14:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA18:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA22:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA4:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA5:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA6:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA7:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_SSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#-----------------------------------------------\r\n",
        "\r\n",
        "00:00:02.436\tPredicting variants\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.leftaligned.vcf', line 22\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.leftaligned.vcf', line 23\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.leftaligned.vcf', line 24\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.leftaligned.vcf', line 25\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.leftaligned.vcf', line 26\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.476\tCreating sequences from exon information 'Pf3D7_01_v3'\r\n",
        "00:00:02.499\tBuilding sequence tree for chromosome 'Pf3D7_01_v3'\r\n",
        "00:00:02.501\tDone. Loaded 351 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.640\tCreating sequences from exon information 'Pf3D7_02_v3'\r\n",
        "00:00:02.654\tBuilding sequence tree for chromosome 'Pf3D7_02_v3'\r\n",
        "00:00:02.657\tDone. Loaded 558 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.802\tCreating sequences from exon information 'Pf3D7_03_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.815\tBuilding sequence tree for chromosome 'Pf3D7_03_v3'\r\n",
        "00:00:02.817\tDone. Loaded 631 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.926\tCreating sequences from exon information 'Pf3D7_04_v3'\r\n",
        "00:00:02.939\tBuilding sequence tree for chromosome 'Pf3D7_04_v3'\r\n",
        "00:00:02.940\tDone. Loaded 601 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.051\tCreating sequences from exon information 'Pf3D7_05_v3'\r\n",
        "00:00:03.064\tBuilding sequence tree for chromosome 'Pf3D7_05_v3'\r\n",
        "00:00:03.066\tDone. Loaded 776 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.229\tCreating sequences from exon information 'Pf3D7_06_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.248\tBuilding sequence tree for chromosome 'Pf3D7_06_v3'\r\n",
        "00:00:03.249\tDone. Loaded 831 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.384\tCreating sequences from exon information 'Pf3D7_07_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.404\tBuilding sequence tree for chromosome 'Pf3D7_07_v3'\r\n",
        "00:00:03.405\tDone. Loaded 728 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.518\tCreating sequences from exon information 'Pf3D7_08_v3'\r\n",
        "00:00:03.532\tBuilding sequence tree for chromosome 'Pf3D7_08_v3'\r\n",
        "00:00:03.534\tDone. Loaded 842 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.613\tCreating sequences from exon information 'Pf3D7_09_v3'\r\n",
        "00:00:03.628\tBuilding sequence tree for chromosome 'Pf3D7_09_v3'\r\n",
        "00:00:03.630\tDone. Loaded 976 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.698\tCreating sequences from exon information 'Pf3D7_10_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.718\tBuilding sequence tree for chromosome 'Pf3D7_10_v3'\r\n",
        "00:00:03.720\tDone. Loaded 1039 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.766\t\t10000 variants (10000 variants per second), 10000 VCF entries\r\n",
        "00:00:03.800\tCreating sequences from exon information 'Pf3D7_11_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.821\tBuilding sequence tree for chromosome 'Pf3D7_11_v3'\r\n",
        "00:00:03.840\tDone. Loaded 1315 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.944\tCreating sequences from exon information 'Pf3D7_12_v3'\r\n",
        "00:00:03.968\tBuilding sequence tree for chromosome 'Pf3D7_12_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.971\tDone. Loaded 1333 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.091\tCreating sequences from exon information 'Pf3D7_13_v3'\r\n",
        "00:00:04.121\tBuilding sequence tree for chromosome 'Pf3D7_13_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.125\tDone. Loaded 1779 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.249\tCreating sequences from exon information 'Pf3D7_14_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.285\tBuilding sequence tree for chromosome 'Pf3D7_14_v3'\r\n",
        "00:00:04.289\tDone. Loaded 2069 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "ERRORS: Some errors were detected\r\n",
        "Error type\tNumber of errors\r\n",
        "ERROR_CHROMOSOME_NOT_FOUND\t5\r\n",
        "\r\n",
        "\r\n",
        "\r\n",
        "WARNINGS: Some warning were detected\r\n",
        "Warning type\tNumber of warnings\r\n",
        "WARNING_REF_DOES_NOT_MATCH_GENOME\t23\r\n",
        "WARNING_TRANSCRIPT_INCOMPLETE\t178\r\n",
        "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\t1731\r\n",
        "WARNING_TRANSCRIPT_NO_START_CODON\t131\r\n",
        "\r\n",
        "\r\n",
        "00:00:04.436\tCreating summary file: snpEff_summary.html\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:05.201\tCreating genes file: snpEff_genes.txt\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:05.661\tdone.\r\n",
        "00:00:05.663\tLogging\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:06.669\tChecking for updates...\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:36,126 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:32:36,128 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:32:36,128 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:32:36,128 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:36,132 HelpFormatter - Program Args: -T VariantAnnotator -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -A TandemRepeatAnnotator -A SnpEff --variant /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.leftaligned.vcf --snpEffFile /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.snpeff.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.snpeff_annotated.vcf \r\n",
        "INFO  16:32:36,137 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:32:36,137 HelpFormatter - Date/Time: 2015/10/02 16:32:36 \r\n",
        "INFO  16:32:36,137 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:32:36,138 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:36,560 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:36,615 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 250 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:36,687 RMDTrackBuilder - Index file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.snpeff.vcf.idx is out of date (index older than input file), deleting and updating the index file \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:37,154 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/7G8.snpeff.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:37,284 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:32:37,286 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:32:37,286 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:32:37,286 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:32:37,287 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:37,392 SnpEff - Skipping malformed SnpEff effect field at M76611:772. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:32:37,415 SnpEff - Skipping malformed SnpEff effect field at M76611:1692. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:37,449 SnpEff - Skipping malformed SnpEff effect field at M76611:4179. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:37,549 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:15405. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:32:37,558 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:16656. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:37,694 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:219211. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F4|PF3D7_0105000|protein_coding|CODING|PF3D7_0105000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:37,742 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:283292. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gaa/gaTGTGTGa|E713DV*|AARP2|protein_coding|CODING|PF3D7_0106700.1|1)\" \r\n",
        "WARN  16:32:37,750 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:291541. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gAAAAAGGCATAATAGTAACAGTGATGATATAATGAAAAAAATGAAAAAAATGAAAAAAATGAAAAAAATGAAAAAAATAaa|E386EKGIIVTVMI**KK*KK*KK*KK*KK*?|IspD|protein_coding|CODING|PF3D7_0106900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:37,784 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:338313. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaT/gaT|D382|PF3D7_0108300|protein_coding|CODING|PF3D7_0108300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:37,792 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:349679. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cca/CCAGCAAATATACGAAGAACCAGCAAATGAAGAGcca|P171PANIRRTSK*RA?|PF3D7_0108500|protein_coding|CODING|PF3D7_0108500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:37,818 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:389954. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||agt/agATCATCTTAATATGATGTTAATAATAAt|S704RSS*YDVNNN|PF3D7_0110100|protein_coding|CODING|PF3D7_0110100.1|1)\" \r\n",
        "WARN  16:32:37,861 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:474666. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aat/Tat|N310Y|PF3D7_0112300|protein_coding|CODING|PF3D7_0112300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:37,863 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478408. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gat/gaATt|D377E?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:37,863 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478417. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N380S|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:37,864 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478423. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCc/aTc|T382I|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:37,864 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478424. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|acC/acG|T382|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:37,865 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:479314. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_DELETION(MODERATE||acttttttt/act|TFF679T|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:38,045 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:202663. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATAATGATAATATTAATTATAAt|N462K***Y*L*?|PF3D7_0204600|protein_coding|CODING|PF3D7_0204600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:38,370 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:798061. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cca/TGAAAATGTTGAAGAAAGTGTAGTGcca|P385*KC*RKCSA?|LSA3|protein_coding|CODING|PF3D7_0220000.1|2)\" \r\n",
        "WARN  16:32:38,371 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:798149. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||atc/aGTGAGTGAAAATGTTGAAGAAAGTtc|I414SE*KC*RKF|LSA3|protein_coding|CODING|PF3D7_0220000.1|2)\" \r\n",
        "WARN  16:32:38,380 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:806481. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/TAATGTATATATTATATAaaa|K579*CIYYIK|PF3D7_0220100|protein_coding|CODING|PF3D7_0220100.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:38,606 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:251923. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cac/cAAGGGTGATAAAATAAATGAATATGac|H228QG**NK*I*?|PF3D7_0305200|protein_coding|CODING|PF3D7_0305200.1|1)\" \r\n",
        "WARN  16:32:38,628 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:304997. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*117|PF3D7_0306700|protein_coding|CODING|PF3D7_0306700.1|4)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:38,664 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:386419. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||YVH1|protein_coding|CODING|PF3D7_0309000.1|4)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:38,764 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:657246. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_0316200|protein_coding|CODING|PF3D7_0316200.1|1)\" \r\n",
        "WARN  16:32:38,776 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:679678. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaAAAAATATATATAAAATACTACTACTACTAATAAt|N453KKIYIKYYYY**?|PF3D7_0316900|protein_coding|CODING|PF3D7_0316900.1|2)\" \r\n",
        "WARN  16:32:38,806 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:758163. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATAAATAATAATAat|N563I*III?|PF3D7_0318300|protein_coding|CODING|PF3D7_0318300.1|3)\" \r\n",
        "WARN  16:32:38,806 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:758175. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||tct/tAAAATAct|S559*NT|PF3D7_0318300|protein_coding|CODING|PF3D7_0318300.1|3)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:38,825 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:802582. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|RBX1|protein_coding|CODING|PF3D7_0319100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,051 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:352982. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_0406800|protein_coding|CODING|PF3D7_0406800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,084 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:435366. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAAAAAAAAAAAAAAAtg|M1KKKKKN?|PF3D7_0409000|protein_coding|CODING|PF3D7_0409000.1|1)\" \r\n",
        "WARN  16:32:39,104 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:487979. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||att/ATATAATATAAATAatt|I717I*YK*?|PF3D7_0410800|protein_coding|CODING|PF3D7_0410800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,167 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:636328. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtg/gtgAAAAATGTATTTATATATATATATATATAAA|V859VKNVFIYIYI*?|PF3D7_0414100|protein_coding|CODING|PF3D7_0414100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,182 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:676553. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||acc/acTCTTTGATAATAAAGATGTTATAAAATAAGAGAATc|T862TL***RCYKIRE?|PF3D7_0415200|protein_coding|CODING|PF3D7_0415200.1|1)\" \r\n",
        "WARN  16:32:39,224 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:760262. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/AATAAAATAAATCAAATGAATCAAATGAATATAAAATAaat|N1324NKINQMNQMNIK*?|PF3D7_0417400|protein_coding|CODING|PF3D7_0417400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,247 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:800569. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aca/ACATGGTCATAATTATaca|T1917TWS*LY?|PF3D7_0418000|protein_coding|CODING|PF3D7_0418000.1|6)\" \r\n",
        "WARN  16:32:39,279 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:842152. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||att/AAATGatt|I320K*?|PF3D7_0418700|protein_coding|CODING|PF3D7_0418700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,296 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:862141. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gat/gTTTTTTCTTTCTCTTTTTTTTTTTCTTTCTCTTTTTTTTTTTCTTTCTCTTTTTTTTTTTCTTTCTCTTTTTAAAat|D1382VFSFSFFFSFSFFFSFSFFFSFSF*N|PF3D7_0419400|protein_coding|CODING|PF3D7_0419400.1|1)\" \r\n",
        "WARN  16:32:39,320 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:898512. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaAAAATAATAAt|N2071KK**?|PF3D7_0420000|protein_coding|CODING|PF3D7_0420000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,660 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:183247. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tta/tTTAAAGGAAAAAAAAAGAACTAAAGAGGAAAAGAGATTAAAGGAAGAAGAAAGAAta|L943FKGKKKN*RGKEIKGRRKN?|CEP120|protein_coding|CODING|PF3D7_0504700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,905 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:511639. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/AGAGGAAGATGTAAACGATGAAgat|D8347RGRCKR*R?|PF3D7_0511500|protein_coding|CODING|PF3D7_0511500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,956 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:633219. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaC/gaC|D1413|PI3K|protein_coding|CODING|PF3D7_0515300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:39,985 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:705469. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taAt|Y596*?|ApiAP2|protein_coding|CODING|PF3D7_0516800.1|1)\" \r\n",
        "WARN  16:32:40,000 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:736646. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||acg/acTGGTAATGGGAGAGAATAACAAATg|T585TGNGRE*QM|PF3D7_0517500|protein_coding|CODING|PF3D7_0517500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:40,016 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:775896. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||atg/atAGGATAATATGAAAAATTATAATATg|M1288IG*YEKL*Y?|PUF1|protein_coding|CODING|PF3D7_0518700.1|5)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:40,204 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1243986. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_0530400|protein_coding|CODING|PF3D7_0530400.1|1)\" \r\n",
        "WARN  16:32:40,210 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1254189. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/|*1084|PF3D7_0530700|protein_coding|CODING|PF3D7_0530700.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:40,230 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1290376. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gct/|A259|PF3D7_0531600|protein_coding|CODING|PF3D7_0531600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,231 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291362. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ctt/cTtt|L588L?|PF3D7_0531600|protein_coding|CODING|PF3D7_0531600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,231 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291404. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cgt/Tgt|R602C|PF3D7_0531600|protein_coding|CODING|PF3D7_0531600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,232 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291707. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*6|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,232 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291718. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAg/aGg|K9R|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,232 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291722. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/TTtat|Y11F?|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,233 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291727. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAg/aGg|K12R|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,233 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291748. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aGa/aAa|R19K|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,233 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291753. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tAat|Y21*?|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,234 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291755. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|taT/taA|Y21*|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,234 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291757. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gGt/gAt|G22D|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,234 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291772. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tac/taTGc|Y27Y?|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,235 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291788. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F33|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,235 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291900. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cat/Tat|H70Y|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,236 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292024. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L111|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,236 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292033. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tTg/tCg|L114S|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,236 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292042. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/|I117|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,237 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292285. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/TGAaaa|K26*K|PF3D7_0531900|protein_coding|CODING|PF3D7_0531900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,237 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292387. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F60|PF3D7_0531900|protein_coding|CODING|PF3D7_0531900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,237 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292552. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cAt/cCt|H48P|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,238 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292595. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F63|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,238 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293105. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atT/atA|I232|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,239 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293615. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ctt/|L403|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,239 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293902. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/|Y498|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,240 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1294120. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||agg/aTGAGgg|R571MR?|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,240 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1294238. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tCg/tGg|S610W|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,241 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1294363. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||att/aTtt|I652I?|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,241 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1294485. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L693|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,242 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1295780. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tTt/tAt|F1124Y|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,242 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1295969. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cTt/cCt|L1187P|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:40,243 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1296007. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Acc/Gcc|T1200A|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:40,373 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:231367. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ata/aAGAAGTGAta|I751KK*?|PF3D7_0605600|protein_coding|CODING|PF3D7_0605600.1|7)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:40,438 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:398332. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/GATGATGATAATATTCATACGTAATAATgat|D698DDDNIHT***?|PF3D7_0609300|protein_coding|CODING|PF3D7_0609300.1|1)\" \r\n",
        "WARN  16:32:40,448 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:422621. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|DHHC2|protein_coding|CODING|PF3D7_0609800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:40,522 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:609636. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aat/aaCAATAACAATAATAATAACAATAACAACAATAACAATAATAATAACAATAACAACAATAACAATAATAATAACAATAACAATTAATATATATTATATATATATATATATATATATATATAt|N489NNNNNNNNNNNNNNNNNNNNNNNNNNNN*YILYIYIYIYIY|PF3D7_0614600|protein_coding|CODING|PF3D7_0614600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:40,767 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:1238044. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N5368|SET1|protein_coding|CODING|PF3D7_0629700.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:40,784 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:1292672. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatGAAAATAATATATATATATATATATTTATTTATATAATATAATATAA|N100NENNIYIYIFIYII*Y?|HAS1|protein_coding|CODING|PF3D7_0630900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:40,881 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:165628. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gac/gacTGAAAATGAG|D256D*K*?|PF3D7_0703900|protein_coding|CODING|PF3D7_0703900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:40,968 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:426088. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gtg/Gtg|V383|PF3D7_0709400|protein_coding|CODING|PF3D7_0709400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:40,969 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:426097. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gtg/Gtg|V380|PF3D7_0709400|protein_coding|CODING|PF3D7_0709400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:40,975 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:444056. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tTTTTTTTATAATAATAACAATAAAATGTat|Y609FFL***Q*NV?|PF3D7_0709900|protein_coding|CODING|PF3D7_0709900.1|1)\" \r\n",
        "WARN  16:32:41,003 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508506. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aac/|N12|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:41,003 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508509. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aaC/aaG|N12K|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:41,004 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508556. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cCt/cAt|P28H|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:41,004 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508558. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Gag/Aag|E29K|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:41,005 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508571. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aGa/aAa|R33K|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:41,005 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508573. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ccg/Tcg|P34S|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:41,063 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:668610. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agc/agTTTTTAAAc|S209SF*?|PF3D7_0714600|protein_coding|CODING|PF3D7_0714600.1|1)\" \r\n",
        "WARN  16:32:41,086 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:731125. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/|*1006|PF3D7_0716700|protein_coding|CODING|PF3D7_0716700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:41,134 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:863479. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aaa/aaTTTTTTTTTTTTTTTTTTGGAAAAATAa|K445NFFFFFLEK*|PF3D7_0719800|protein_coding|CODING|PF3D7_0719800.1|1)\" \r\n",
        "WARN  16:32:41,152 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:905448. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gATAATTATAATGATGATAATTATAATGATGTGCATATATATATATAat|D436DNYNDDNYNDVHIYI*?|PF3D7_0721000|protein_coding|CODING|PF3D7_0721000.1|1)\" \r\n",
        "WARN  16:32:41,160 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:923747. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gGt/gGt|G488|PF3D7_0721300|protein_coding|CODING|PF3D7_0721300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:41,254 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1150082. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATATATATATATATAATAATAAAACCAACAGGACAGGAAATGAAAat|N326IYIYI***NQQDRK*K?|VPS53|protein_coding|CODING|PF3D7_0727000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:41,386 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1328204. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ttc/TGATGATGTAGGTCttc|F909**CRS?|PTP4|protein_coding|CODING|PF3D7_0730900.1|2)\" \r\n",
        "WARN  16:32:41,386 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1328206. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ttc/ttAAGAAAGAAGTATCTGTACCAAGATTAGTGTGAATGAATTTAAAATAc|F909LRKKYLYQD*CE*I*N?|PTP4|protein_coding|CODING|PF3D7_0730900.1|2)\" \r\n",
        "WARN  16:32:41,410 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1334840. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aag/aaAg|K33K?|PF3D7_0731000|protein_coding|CODING|PF3D7_0731000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:41,659 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:92806. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tgt/tTgt|C1589L?|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:41,660 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:94091. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/tatGTATATATATATAT|Y1160YVYIY?|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:41,661 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:95108. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F821|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:41,661 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:95148. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||tgt/tgATATATGTATATATGt|C808*YMYIC|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:41,693 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:155008. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATATTTATAATATAAATAat|N1554IYL*YK*?|ApiAP2|protein_coding|CODING|PF3D7_0802100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:41,809 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:468337. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aGTAATAATAATAATAATAATAATAATAATAATAATAATAATTATATATAAATAATAATAATAATAat|N1186SNNNNNNNNNNNNNYI*IIIII?|pfa55-14|protein_coding|CODING|PF3D7_0809200.1|1)\" \r\n",
        "WARN  16:32:41,823 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:506358. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tGACATAAAAAGTTat|Y114*HKKL?|PF3D7_0809800|protein_coding|CODING|PF3D7_0809800.1|1)\" \r\n",
        "WARN  16:32:41,823 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:506368. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaaTAAAAAATTACGAA|K110K*KIT?|PF3D7_0809800|protein_coding|CODING|PF3D7_0809800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:41,908 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:736661. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATAATAATAATAAt|N1507K****?|VPS9|protein_coding|CODING|PF3D7_0815800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:42,010 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1010239. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aac/aacTGAAAATTTTGATGAAAACCAAAT|N356N*KF**KPN|PF3D7_0822900|protein_coding|CODING|PF3D7_0822900.1|1)\" \r\n",
        "WARN  16:32:42,011 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1010466. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION(HIGH||tttgatgaa/taa|FDE279*|PF3D7_0822900|protein_coding|CODING|PF3D7_0822900.1|1)\" \r\n",
        "WARN  16:32:42,011 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1010514. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION(HIGH||tttgatgaa/taa|FDE263*|PF3D7_0822900|protein_coding|CODING|PF3D7_0822900.1|1)\" \r\n",
        "WARN  16:32:42,046 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1110467. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||tca/tAAAATGAAAAATTTGTGAca|S398*NEKFVT|PF3D7_0826000|protein_coding|CODING|PF3D7_0826000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:42,151 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1282182. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F417|PF3D7_0830100|protein_coding|CODING|PF3D7_0830100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:42,153 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1283721. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tttttttcttatacatagacttttattgttatacatagat/|FFSYT*TFIVIHR?930|PF3D7_0830100|protein_coding|CODING|PF3D7_0830100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:42,155 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1285508. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||att/|I311|PF3D7_0830200|protein_coding|CODING|PF3D7_0830200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:42,182 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1362577. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||taC/taC|Y1048|CLAG8|protein_coding|CODING|PF3D7_0831600.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:42,183 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1362580. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaA/gaA|E1049|CLAG8|protein_coding|CODING|PF3D7_0831600.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:42,183 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1362595. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaA/aaA|K1054|CLAG8|protein_coding|CODING|PF3D7_0831600.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:42,183 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1362610. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||ttG/ttG|L1059|CLAG8|protein_coding|CODING|PF3D7_0831600.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:42,184 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1362898. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaA/aaA|K1155|CLAG8|protein_coding|CODING|PF3D7_0831600.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:42,184 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1363145. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Cta/Cta|L1238|CLAG8|protein_coding|CODING|PF3D7_0831600.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:42,396 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:349291. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*1134|PF3D7_0907200|protein_coding|CODING|PF3D7_0907200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:42,570 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:761842. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tct/|S211|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:42,570 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:761920. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gga/|G237|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:42,571 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762390. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tgt/|C394|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:42,600 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:835726. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaGTAACATAAATa|K658K*HK?|PF3D7_0920400|protein_coding|CODING|PF3D7_0920400.1|1)\" \r\n",
        "WARN  16:32:42,634 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:924305. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aaa/aaTTTTTTTTAa|K3585NFF*|PF3D7_0922800|protein_coding|CODING|PF3D7_0922800.1|3)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:42,893 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:103600. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aca/acaGAGAGTTAATTAAAAAAGAATCTATATATAAAAATAAA|T309TES*LKKNLYIKI?|PF3D7_1002000|protein_coding|CODING|PF3D7_1002000.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:42,893 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:103603. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatTGATGAAA|N308N**?|PF3D7_1002000|protein_coding|CODING|PF3D7_1002000.1|2)\" \r\n",
        "WARN  16:32:42,917 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:170668. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|PF3D7_1003700|protein_coding|CODING|PF3D7_1003700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,022 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:488661. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/AATAATTACAATAATAATTAAATAAATAAATAAATAAATAAATAAAaat|N471NNYNNN*INK*INK*K?|NIF4|protein_coding|CODING|PF3D7_1012700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,099 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:725581. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||ggt/ggAGCAGATGTAGAAGGAGCAGATGTAGAAGGAGCAGATGTAGAAGGAGCAGATGTAGAAGGTTTACAATAATAAAAACAATCATTACAt|G917GADVEGADVEGADVEGADVEGLQ**KQSLH|PPP8|protein_coding|CODING|PF3D7_1018200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,100 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:727177. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aat/aaAAAGGTAAGTGATGATGATGATAAAGCAGATGTAGAAGGAGCAGATGTAGAAGGAGCAGATGTAGAAGGAGCAGATGTAGAAGGTTTACAATAATAAAAACAATCATTACAt|N385KKVSDDDDKADVEGADVEGADVEGADVEGLQ**KQSLH|PPP8|protein_coding|CODING|PF3D7_1018200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,157 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:898636. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gTAAAGAGGTATATATGGACAAACATGAGATGGATTTGAACAATGaa|E1658VKRYIWTNMRWI*TM?|SEA1|protein_coding|CODING|PF3D7_1021800.1|1)\" \r\n",
        "WARN  16:32:43,191 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:995784. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aAATATATATATATATATATATATTaa|K424KYIYIYIY*|PF3D7_1023800|protein_coding|CODING|PF3D7_1023800.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,207 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1037283. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cat/cTATGATGATAACCATGATGATGATAACCATGat|H867L***P****P*?|PF3D7_1024800|protein_coding|CODING|PF3D7_1024800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,371 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1521658. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aCa|K842T|Pf11-1|protein_coding|CODING|PF3D7_1038400.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,485 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:192450. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taATATGAGAACAATAAAATATATATATAAt|Y592*YENNKIYI*?|PF3D7_1104300|protein_coding|CODING|PF3D7_1104300.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,614 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:565795. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtt/gtATTGATGATATGAAAGAAATGAATAATGTTGATGATATGAAAGAAATGAATAATTCt|V1260VLMI*KK*IMLMI*KK*II?|PF3D7_1114900|protein_coding|CODING|PF3D7_1114900.1|1)\" \r\n",
        "WARN  16:32:43,640 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:627749. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/aatGAATGGTAATATATAACAATT|N510NEW*YITI|PF3D7_1116600|protein_coding|CODING|PF3D7_1116600.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,681 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:757136. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaC/gaC|D315|PF3D7_1120000|protein_coding|CODING|PF3D7_1120000.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,748 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:953359. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaT/gaT|D995|PF3D7_1124100|protein_coding|CODING|PF3D7_1124100.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:43,774 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1020931. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gTATAAAAATAATGATAATAATAATTATGATGat|D522V*K*****L**?|ATG7|protein_coding|CODING|PF3D7_1126100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,815 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1147484. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aAATaa|K911K*|PF3D7_1129600|protein_coding|CODING|PF3D7_1129600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,859 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1284918. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/aatGTACAAACTCAAAATGCCCAACCTGACAATATCATCTAATTGTTCATCGTCACC|N905NVQTQNAQPDNII*LFIVT|PF3D7_1133200|protein_coding|CODING|PF3D7_1133200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:43,940 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1541072. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tct/tcTt|S90S?|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:43,940 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1541573. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||tca/tcATATATa|S257SYI|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:43,941 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1542865. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||caa/cAaa|Q688Q?|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:43,952 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1569921. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||ata/ATATATATATATata|I107IYIYI|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:43,953 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1572602. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||attata/|II1000|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,006 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1741827. Error was: \"START_LOST+CODON_INSERTION is not a recognized effect type\". Field was: \"START_LOST+CODON_INSERTION(HIGH||atg/aAAATTTtg|M1KIL|PF3D7_1143600|protein_coding|CODING|PF3D7_1143600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,045 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1853371. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1146800|protein_coding|CODING|PF3D7_1146800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,065 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1917647. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/ttTTTATATATATATATAt|F419FLYIYI?|PF3D7_1148100|protein_coding|CODING|PF3D7_1148100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,067 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920294. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||ata/ATATATATATATATATATata|I10IYIYIYI|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,067 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920424. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tgc/Agc|C53S|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,067 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920431. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|tGa/tAa|*55|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,067 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920434. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/|Y56|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,068 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920462. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gta/Tgta|V66C?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,068 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920809. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N181S|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,161 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:154670. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATAATAAAAATAATGAAAGTAat|N270I**K**K*?|PF3D7_1202600|protein_coding|CODING|PF3D7_1202600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,173 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190106. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/aAAga|R1003K?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,174 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190958. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_CHANGE_PLUS_CODON_DELETION(MODERATE||atatat/aat|IY718N|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,175 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190962. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/|Y717|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,175 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190964. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/|Y717|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,175 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190966. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/|I716|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,175 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190969. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aaT/aaG|N715K|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,176 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190972. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_CHANGE_PLUS_CODON_DELETION(MODERATE||aagaga/aaa|KR713K|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,176 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190993. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaA/aaG|K707|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,177 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:191958. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/ataAATATATATATATATA|I385INIYIY?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,177 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:192370. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tac/taTAc|Y248Y?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,178 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:192538. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||atg/|M192|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,203 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:258571. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaC/aaC|N735|HMGB3|protein_coding|CODING|PF3D7_1205800.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:44,222 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:319574. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Aat/Aat|N712|PF3D7_1207000|protein_coding|CODING|PF3D7_1207000.1|3|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,223 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:319579. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTGAAAAAAAATATATAat|N710MKKNI*?|PF3D7_1207000|protein_coding|CODING|PF3D7_1207000.1|3)\" \r\n",
        "WARN  16:32:44,269 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:468533. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gta/gtaTGACA|V258V*?|PF3D7_1210400|protein_coding|CODING|PF3D7_1210400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,398 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:873810. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aca/acGACCGATAAAGAAAATGGAATGACAGATAAAGAAAGTGGAACATAATAATCACTCCTATTAATATGAATAATATTATTATTATTATTATCACTCTTATTAATa|T377TTDKENGMTDKESGT**SLLLI*IILLLLLSLLLI|PF3D7_1221900|protein_coding|CODING|PF3D7_1221900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,472 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1075414. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tGat|Y986*?|PF3D7_1226500|protein_coding|CODING|PF3D7_1226500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,606 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1499999. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaTATATATATATATATATATATATATATATAAa|K471NIYIYIYIYI*?|XAB2|protein_coding|CODING|PF3D7_1235900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,683 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1745338. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aGg/aAg|R29K|PF3D7_1241000|protein_coding|CODING|PF3D7_1241000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:44,683 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1745389. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tGc/tAc|C12Y|PF3D7_1241000|protein_coding|CODING|PF3D7_1241000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,736 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1907488. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/Tttt|F149F?|PF3D7_1245700|protein_coding|CODING|PF3D7_1245700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,784 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:2052102. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ttt/ttAGTGCCAAAAAAATGAAGGTCCTTTGAATCATGAAGAAGATAATTTt|F776LVPKK*RSFES*RR*F?|G377|protein_coding|CODING|PF3D7_1250100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:44,928 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:194552. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/tgtTGATAATATTAATTATTGT|C6036C**Y*LL?|PF3D7_1303800|protein_coding|CODING|PF3D7_1303800.1|1)\" \r\n",
        "WARN  16:32:44,932 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:202803. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aat/aaAATATAAATCAATAATAATAAt|N3286KI*INNNN|PF3D7_1303800|protein_coding|CODING|PF3D7_1303800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,018 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:474049. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATATTTATAATAATGGAAATATTCATAATAATGAAAATATTTATAATAATGAAAAt|N318KYL**WKYS***KYL***K?|PF3D7_1310800|protein_coding|CODING|PF3D7_1310800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,065 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:610051. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*753|PF3D7_1314100|protein_coding|CODING|PF3D7_1314100.1|1)\" \r\n",
        "WARN  16:32:45,081 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:661261. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|MYB1|protein_coding|CODING|PF3D7_1315800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,153 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:886263. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_1321300|protein_coding|CODING|PF3D7_1321300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,178 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:960644. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*116|PF3D7_1322800|protein_coding|CODING|PF3D7_1322800.1|2)\" \r\n",
        "WARN  16:32:45,222 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1082129. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gtg/gCCAATCAATAATAACAATAATGATAATAATAATGAACATAATGATCATATGtg|V820ANQ**Q******T**SYV|PF3D7_1325900|protein_coding|CODING|PF3D7_1325900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,240 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1122538. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*4781|PF3D7_1326600|protein_coding|CODING|PF3D7_1326600.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,335 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1391352. Error was: \"START_LOST+CODON_INSERTION is not a recognized effect type\". Field was: \"START_LOST+CODON_INSERTION(HIGH||atg/aAAAtg|M1KM|PF3D7_1334000|protein_coding|CODING|PF3D7_1334000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,428 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1687897. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gaaTGGTAATAATATTATTGAAAATAATATGAATGGTAATAATATTATTGAAAATAATATGAATGGTATATATATAT|E2024EW**YY*K*YEW**YY*K*YEWYIY?|ApiAP2|protein_coding|CODING|PF3D7_1342900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,557 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2068339. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gaCGAATAAAATATGTGATGATAAAATATGTGAt|D143DE*NM***NM*?|PF3D7_1351900|protein_coding|CODING|PF3D7_1351900.1|1)\" \r\n",
        "WARN  16:32:45,584 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2157541. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1354000|protein_coding|CODING|PF3D7_1354000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,714 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2562862. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/aATTTaa|K249NL?|PF3D7_1363900|protein_coding|CODING|PF3D7_1363900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,722 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2586985. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtg/gTATATATATATGATAAATAAGGATAATAtg|V652VYIYDK*G*Y?|PF3D7_1364400|protein_coding|CODING|PF3D7_1364400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,786 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794272. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gcc/|A13|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,786 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794278. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ttG/ttA|L14|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,786 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794288. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||cat/cATATATATat|H18HIY?|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,787 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794404. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atT/atA|I56|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,787 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794407. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|taA/taG|*57|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,787 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794436. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/taTt|Y67Y?|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,788 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796123. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cTc/cAc|L2H|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,788 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796135. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|tTg/tAg|L6*|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,788 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796149. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tca/Aca|S11T|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,789 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796161. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|Tga/Aga|*15R|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,789 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796179. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||att/aTtt|I21I?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:32:45,789 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796202. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|acA/acC|T28|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:45,878 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:127020. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||PF3D7_1403400|protein_coding|CODING|PF3D7_1403400.1|4)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,037 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:621320. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaT/gaT|D808|PF3D7_1415400|protein_coding|CODING|PF3D7_1415400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,093 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:789883. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaC/gaC|D192|PF3D7_1419000|protein_coding|CODING|PF3D7_1419000.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,121 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:872338. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aAATAATCat|N447K*S?|PF3D7_1421100|protein_coding|CODING|PF3D7_1421100.1|1)\" \r\n",
        "WARN  16:32:46,131 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:902668. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/TAAAAaaa|K762*K?|PF3D7_1422400|protein_coding|CODING|PF3D7_1422400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,180 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1047258. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aAt/aAt|N1306|PF3D7_1426800|protein_coding|CODING|PF3D7_1426800.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:46,200 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1110765. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaAAAAATATAAATAAATAAATATATATATATATAAATGATAATAATAATGATAATAATAAt|N1344KKI*INKYIYI*MIIIMIII?|PF3D7_1428400|protein_coding|CODING|PF3D7_1428400.1|6)\" \r\n",
        "WARN  16:32:46,219 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1162755. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaC/gaC|D2467|PF3D7_1429400|protein_coding|CODING|PF3D7_1429400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,246 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1231663. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aATGTTATAGGTTGat|N210NVIG*?|PF3D7_1431300|protein_coding|CODING|PF3D7_1431300.1|1)\" \r\n",
        "WARN  16:32:46,246 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1231782. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||atg/TGatg|M250*?|PF3D7_1431300|protein_coding|CODING|PF3D7_1431300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,292 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1381494. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gac/Gac|D7058|PF3D7_1434500|protein_coding|CODING|PF3D7_1434500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:32:46,293 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1384278. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gat/Gat|D6130|PF3D7_1434500|protein_coding|CODING|PF3D7_1434500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,507 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2032798. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tTATATAATATATATATATTATAAATAATAATAATAATAATAat|Y517LYNIYIL*IIIIII?|ApiAP2|protein_coding|CODING|PF3D7_1449500.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,647 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2462332. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||UCHL3|protein_coding|CODING|PF3D7_1460400.1|5)\" \r\n",
        "WARN  16:32:46,671 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2540522. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaAATAAAATAAAATAAAATAAAAATGATGTAAAt|N293KIK*NKIKMM*?|PF3D7_1462400|protein_coding|CODING|PF3D7_1462400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,742 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2748786. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||atg/atAAATGAAGAAATAAATGAAGAAATAGATGAAGAAATAAATGAAGAAATAAATGAAGAAATAGATGAAAAATAg|M132INEEINEEIDEEINEEINEEIDEK*|PF3D7_1467200|protein_coding|CODING|PF3D7_1467200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:32:46,899 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3198390. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gATATATATATAAATAAGTGGAAATGAAATAAGTGGAAATAaa|E1592DIYINKWK*NKWK*?|SURF14.1|protein_coding|CODING|PF3D7_1477600.1|3)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:46,926 VariantAnnotator - Processed 18688 loci.\r\n",
        " \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:46,985 ProgressMeter -            done   2.3332831E7     9.0 s       0.0 s      100.0%     9.0 s       0.0 s \r\n",
        "INFO  16:32:46,986 ProgressMeter - Total runtime 9.70 secs, 0.16 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:32:48,287 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "GB4\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:05,437 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:33:05,438 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:33:05,439 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:33:05,439 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:05,442 HelpFormatter - Program Args: -T LeftAlignAndTrimVariants -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -V /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.sorted.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf \r\n",
        "INFO  16:33:05,447 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:33:05,447 HelpFormatter - Date/Time: 2015/10/02 16:33:05 \r\n",
        "INFO  16:33:05,447 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:33:05,447 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:05,798 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:05,844 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 1000 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:06,194 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.sorted.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:06,287 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:33:06,288 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:33:06,288 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:33:06,288 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:33:06,289 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "7543 variants were aligned\r\n",
        "INFO  16:33:07,399 ProgressMeter -            done     49755.0     1.0 s      22.0 s       99.7%     1.0 s       0.0 s \r\n",
        "INFO  16:33:07,399 ProgressMeter - Total runtime 1.11 secs, 0.02 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:08,695 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.000\tSnpEff version SnpEff 4.1g (build 2015-05-17), by Pablo Cingolani\r\n",
        "00:00:00.012\tCommand: 'ann'\r\n",
        "00:00:00.027\tReading configuration file 'snpEff.config'. Genome: 'Pf3D7july2015'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.538\tdone\r\n",
        "00:00:00.538\tReading database for genome version 'Pf3D7july2015' from file '/Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt/snpeff/snpEff/./data/Pf3D7july2015/snpEffectPredictor.bin' (this might take a while)\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.378\tdone\r\n",
        "00:00:01.387\tBuilding interval forest\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.744\tdone.\r\n",
        "00:00:01.744\tGenome stats :\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "#-----------------------------------------------\r\n",
        "# Genome name                : 'Plasmodium_falciparum'\r\n",
        "# Genome version             : 'Pf3D7july2015'\r\n",
        "# Has protein coding info    : true\r\n",
        "# Genes                      : 5645\r\n",
        "# Protein coding genes       : 5492\r\n",
        "#-----------------------------------------------\r\n",
        "# Transcripts                : 5886\r\n",
        "# Avg. transcripts per gene  : 1.04\r\n",
        "#-----------------------------------------------\r\n",
        "# Checked transcripts        : \r\n",
        "#               AA sequences :      0 ( 0.00% )\r\n",
        "#              DNA sequences :      0 ( 0.00% )\r\n",
        "#-----------------------------------------------\r\n",
        "# Protein coding transcripts : 5580\r\n",
        "#              Length errors :    126 ( 2.26% )\r\n",
        "#  STOP codons in CDS errors :    145 ( 2.60% )\r\n",
        "#         START codon errors :    200 ( 3.58% )\r\n",
        "#        STOP codon warnings :    205 ( 3.67% )\r\n",
        "#              UTR sequences :      0 ( 0.00% )\r\n",
        "#               Total Errors :    212 ( 3.80% )\r\n",
        "# WARNING                    : No protein coding transcript has UTR\r\n",
        "#-----------------------------------------------\r\n",
        "# Cds                        : 14366\r\n",
        "# Exons                      : 14366\r\n",
        "# Exons with sequence        : 14366\r\n",
        "# Exons without sequence     : 0\r\n",
        "# Avg. exons per transcript  : 2.44\r\n",
        "#-----------------------------------------------\r\n",
        "# Number of chromosomes      : 41\r\n",
        "# Chromosomes                : Format 'chromo_name size codon_table'\r\n",
        "#\t\t'Pf3D7_14_v3'\t3291936\tStandard\r\n",
        "#\t\t'Pf3D7_13_v3'\t2925236\tStandard\r\n",
        "#\t\t'Pf3D7_12_v3'\t2271494\tStandard\r\n",
        "#\t\t'Pf3D7_11_v3'\t2038340\tStandard\r\n",
        "#\t\t'Pf3D7_10_v3'\t1687656\tStandard\r\n",
        "#\t\t'Pf3D7_09_v3'\t1541735\tStandard\r\n",
        "#\t\t'Pf3D7_08_v3'\t1472805\tStandard\r\n",
        "#\t\t'Pf3D7_07_v3'\t1445207\tStandard\r\n",
        "#\t\t'Pf3D7_06_v3'\t1418242\tStandard\r\n",
        "#\t\t'Pf3D7_05_v3'\t1343557\tStandard\r\n",
        "#\t\t'Pf3D7_04_v3'\t1200490\tStandard\r\n",
        "#\t\t'Pf3D7_03_v3'\t1067971\tStandard\r\n",
        "#\t\t'Pf3D7_02_v3'\t947102\tStandard\r\n",
        "#\t\t'Pf3D7_01_v3'\t640851\tStandard\r\n",
        "#\t\t'Pf_M76611'\t5967\tProtozoan_Mitochondrial\r\n",
        "#\t\t'mal_mito_1'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_2'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_3'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_RNA19:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_SSUB:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_10:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_15:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_16:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_17:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_1:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_20:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_21:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_9:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUC:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUG:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA11:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA12:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA14:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA18:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA22:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA4:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA5:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA6:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA7:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_SSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#-----------------------------------------------\r\n",
        "\r\n",
        "00:00:02.430\tPredicting variants\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 22\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 23\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 24\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 25\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 26\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 27\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 28\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 29\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 30\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 31\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 32\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 33\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 34\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 35\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 36\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 37\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 38\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 39\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf', line 40\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.497\tCreating sequences from exon information 'Pf3D7_01_v3'\r\n",
        "00:00:02.516\tBuilding sequence tree for chromosome 'Pf3D7_01_v3'\r\n",
        "00:00:02.520\tDone. Loaded 351 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.677\tCreating sequences from exon information 'Pf3D7_02_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.690\tBuilding sequence tree for chromosome 'Pf3D7_02_v3'\r\n",
        "00:00:02.694\tDone. Loaded 558 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.836\tCreating sequences from exon information 'Pf3D7_03_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.848\tBuilding sequence tree for chromosome 'Pf3D7_03_v3'\r\n",
        "00:00:02.849\tDone. Loaded 631 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.959\tCreating sequences from exon information 'Pf3D7_04_v3'\r\n",
        "00:00:02.972\tBuilding sequence tree for chromosome 'Pf3D7_04_v3'\r\n",
        "00:00:02.973\tDone. Loaded 601 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.111\tCreating sequences from exon information 'Pf3D7_05_v3'\r\n",
        "00:00:03.124\tBuilding sequence tree for chromosome 'Pf3D7_05_v3'\r\n",
        "00:00:03.125\tDone. Loaded 776 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.255\tCreating sequences from exon information 'Pf3D7_06_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.271\tBuilding sequence tree for chromosome 'Pf3D7_06_v3'\r\n",
        "00:00:03.272\tDone. Loaded 831 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.377\tCreating sequences from exon information 'Pf3D7_07_v3'\r\n",
        "00:00:03.393\tBuilding sequence tree for chromosome 'Pf3D7_07_v3'\r\n",
        "00:00:03.394\tDone. Loaded 728 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.494\tCreating sequences from exon information 'Pf3D7_08_v3'\r\n",
        "00:00:03.508\tBuilding sequence tree for chromosome 'Pf3D7_08_v3'\r\n",
        "00:00:03.510\tDone. Loaded 842 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.589\tCreating sequences from exon information 'Pf3D7_09_v3'\r\n",
        "00:00:03.604\tBuilding sequence tree for chromosome 'Pf3D7_09_v3'\r\n",
        "00:00:03.606\tDone. Loaded 976 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.686\tCreating sequences from exon information 'Pf3D7_10_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.702\tBuilding sequence tree for chromosome 'Pf3D7_10_v3'\r\n",
        "00:00:03.704\tDone. Loaded 1039 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.769\t\t10000 variants (10000 variants per second), 10000 VCF entries\r\n",
        "00:00:03.791\tCreating sequences from exon information 'Pf3D7_11_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.827\tBuilding sequence tree for chromosome 'Pf3D7_11_v3'\r\n",
        "00:00:03.829\tDone. Loaded 1315 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.937\tCreating sequences from exon information 'Pf3D7_12_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.961\tBuilding sequence tree for chromosome 'Pf3D7_12_v3'\r\n",
        "00:00:03.964\tDone. Loaded 1333 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.079\tCreating sequences from exon information 'Pf3D7_13_v3'\r\n",
        "00:00:04.108\tBuilding sequence tree for chromosome 'Pf3D7_13_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.112\tDone. Loaded 1779 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.231\tCreating sequences from exon information 'Pf3D7_14_v3'\r\n",
        "00:00:04.270\tBuilding sequence tree for chromosome 'Pf3D7_14_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.274\tDone. Loaded 2069 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "ERRORS: Some errors were detected\r\n",
        "Error type\tNumber of errors\r\n",
        "ERROR_CHROMOSOME_NOT_FOUND\t19\r\n",
        "\r\n",
        "\r\n",
        "\r\n",
        "WARNINGS: Some warning were detected\r\n",
        "Warning type\tNumber of warnings\r\n",
        "WARNING_REF_DOES_NOT_MATCH_GENOME\t26\r\n",
        "WARNING_TRANSCRIPT_INCOMPLETE\t266\r\n",
        "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\t1692\r\n",
        "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS&WARNING_REF_DOES_NOT_MATCH_GENOME\t1\r\n",
        "WARNING_TRANSCRIPT_NO_START_CODON\t98\r\n",
        "\r\n",
        "\r\n",
        "00:00:04.416\tCreating summary file: snpEff_summary.html\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:05.325\tCreating genes file: snpEff_genes.txt\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:05.837\tdone.\r\n",
        "00:00:05.839\tLogging\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:06.846\tChecking for updates...\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:18,296 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:33:18,298 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:33:18,298 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:33:18,298 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n",
        "INFO  16:33:18,301 HelpFormatter - Program Args: -T VariantAnnotator -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -A TandemRepeatAnnotator -A SnpEff --variant /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.leftaligned.vcf --snpEffFile /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.snpeff.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.snpeff_annotated.vcf \r\n",
        "INFO  16:33:18,306 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:33:18,306 HelpFormatter - Date/Time: 2015/10/02 16:33:18 \r\n",
        "INFO  16:33:18,306 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:33:18,306 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:18,656 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:18,703 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 250 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:19,103 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GB4.snpeff.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:19,215 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:33:19,216 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:33:19,216 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:33:19,217 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:33:19,217 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:19,361 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,370 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:987. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,373 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:1205. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,393 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:2732. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,395 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:2772. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,395 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:2784. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,396 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:2787. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:19,446 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:9004. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:19,463 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:13289. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,470 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:15405. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,476 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:17907. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,478 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18694. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,479 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18828. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,479 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18860. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,487 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:22669. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,490 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23944. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,490 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23947. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,494 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:25587. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:33:19,495 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:25869. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:19,592 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:136672. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N862|VPS51|protein_coding|CODING|PF3D7_0103100.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:19,637 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:192647. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gaT/gaG|D793E|UBP1|protein_coding|CODING|PF3D7_0104300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:19,658 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:219211. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F4|PF3D7_0105000|protein_coding|CODING|PF3D7_0105000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:19,752 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:338313. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaT/gaT|D382|PF3D7_0108300|protein_coding|CODING|PF3D7_0108300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:19,784 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:402143. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N1851|PF3D7_0110500|protein_coding|CODING|PF3D7_0110500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:19,814 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:474666. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aat/Tat|N310Y|PF3D7_0112300|protein_coding|CODING|PF3D7_0112300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,815 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478408. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gat/gaATt|D377E?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,816 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478417. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N380S|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,816 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478423. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCc/aTc|T382I|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,817 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478424. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|acC/acG|T382|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,817 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478793. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|agG/agA|R505|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,818 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478794. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|Tag/Aag|*506K|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,818 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478795. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|tAg/tCg|*506S|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,819 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478796. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|taG/taA|*506|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,819 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478799. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L508|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,820 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478801. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|tTa/tAa|L508*|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,820 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478808. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|ttT/ttA|F510L|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,821 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478810. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttaatt/|LI511|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,821 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478816. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAc/tTc|Y513F|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,821 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478817. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taC/taT|Y513|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,822 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478818. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ctc/Ttc|L514F|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,822 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478820. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ctC/ctT|L514|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,823 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478822. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/agAa|R515R?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:19,823 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:479254. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tca/tcAa|S659S?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,824 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:479314. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||act/acTt|T679T?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:19,828 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:487993. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gatTGAAGAAGATGATGAT|D640D*RR**?|GARP|protein_coding|CODING|PF3D7_0113000.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:19,964 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:122170. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/ATGAAGTGaaa|K550MK*?|PF3D7_0202400|protein_coding|CODING|PF3D7_0202400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:20,069 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:355152. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gaa/gTTTTTTAAGATGATGAAAATTATAACGATGATGAAAATTATAaa|E178VF*DDENYNDDENYK|RRF1|protein_coding|CODING|PF3D7_0208600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:20,218 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:635775. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aga/aAATga|R578K*|PF3D7_0215400|protein_coding|CODING|PF3D7_0215400.1|1)\" \r\n",
        "WARN  16:33:20,224 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:642334. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/Tgat|D646*?|PF3D7_0215500|protein_coding|CODING|PF3D7_0215500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:20,305 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:766866. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tttttttttttc/|FFFF47|PF3D7_0218800|protein_coding|CODING|PF3D7_0218800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:20,332 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:798271. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gta/TGAAGAAATCgta|V455*RNR?|LSA3|protein_coding|CODING|PF3D7_0220000.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:20,655 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:571050. Error was: \"START_LOST+CODON_INSERTION is not a recognized effect type\". Field was: \"START_LOST+CODON_INSERTION(HIGH||atg/aAAAtg|M1KM|PF3D7_0314100|protein_coding|CODING|PF3D7_0314100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:20,794 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:940172. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|DGAT|protein_coding|CODING|PF3D7_0322300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:20,926 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:220487. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gag/gTAGGAGGAAAATAATGATGATAag|E404VGGK****?|SET8|protein_coding|CODING|PF3D7_0403900.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:20,980 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:352982. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_0406800|protein_coding|CODING|PF3D7_0406800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,023 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:466436. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||atA/atA|I266|EVP1|protein_coding|CODING|PF3D7_0410000.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,134 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:723769. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taAAAAAATTAATGATAATAATAAATAt|Y158*KN*****I?|PF3D7_0416500|protein_coding|CODING|PF3D7_0416500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,168 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:781048. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaG/gaG|E542|PF3D7_0417600|protein_coding|CODING|PF3D7_0417600.1|4|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,244 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:901359. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATAATATAAATAATAAt|N1122K*YK**?|PF3D7_0420000|protein_coding|CODING|PF3D7_0420000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,264 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:927556. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/AATTAGaat|N3190N*N|ApiAP2|protein_coding|CODING|PF3D7_0420300.1|1)\" \r\n",
        "WARN  16:33:21,301 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979922. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ccT/ccC|P3|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:21,302 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979935. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ggt/Cgt|G8R|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:21,302 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979996. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cAt/cCt|H28P|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:21,302 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979998. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aag/Gag|K29E|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:21,303 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:980013. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tcg/Ccg|S34P|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:21,303 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:980023. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|tAa/tGa|*37|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:21,304 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:980041. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gCg/gTg|A43V|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,486 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:44544. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Caa/Caa|Q1030|MESA|protein_coding|CODING|PF3D7_0500800.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:21,487 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:44546. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gTa/gTa|V1029|MESA|protein_coding|CODING|PF3D7_0500800.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,582 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:201125. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtg/gGATAATATTAACATTGtg|V1203G*Y*HC?|PF3D7_0504800|protein_coding|CODING|PF3D7_0504800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,629 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:274340. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gta/ATTAATAAAATATAAgta|V2647INKI*V|PF3D7_0506500|protein_coding|CODING|PF3D7_0506500.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,744 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:514314. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaCTAACAATAATATTAATAGTAACAATAATATTAATAGt|N9238N*Q*Y***Q*Y**?|PF3D7_0511500|protein_coding|CODING|PF3D7_0511500.1|1)\" \r\n",
        "WARN  16:33:21,763 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:561915. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/Taat|N706*?|PF3D7_0513200|protein_coding|CODING|PF3D7_0513200.1|2)\" \r\n",
        "WARN  16:33:21,787 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:633219. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaC/gaC|D1413|PI3K|protein_coding|CODING|PF3D7_0515300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,906 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:990562. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/AATTATATATATATATAaat|N187NYIYI*?|PF3D7_0523800|protein_coding|CODING|PF3D7_0523800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:21,997 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1254189. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/|*1084|PF3D7_0530700|protein_coding|CODING|PF3D7_0530700.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,009 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1289139. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gta/gtAa|V194V?|PF3D7_0531500|protein_coding|CODING|PF3D7_0531500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,009 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1289160. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTt/aAt|I201N|PF3D7_0531500|protein_coding|CODING|PF3D7_0531500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,010 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1289161. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||aaa/AAAaaa|K202KK|PF3D7_0531500|protein_coding|CODING|PF3D7_0531500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,010 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1289232. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCa/aTa|T225I|PF3D7_0531500|protein_coding|CODING|PF3D7_0531500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,011 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1289817. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ggc/Agc|G73S|PF3D7_0531600|protein_coding|CODING|PF3D7_0531600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,011 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1290376. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gct/|A259|PF3D7_0531600|protein_coding|CODING|PF3D7_0531600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,012 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291900. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cat/Tat|H70Y|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,012 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291953. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttg/|L88|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,013 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291955. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tTg/tCg|L88S|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,013 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291956. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ttG/ttA|L88|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,013 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291964. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/taTt|Y91Y?|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,014 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291968. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|atG/atT|M92I|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,014 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291971. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/TTata|I94L?|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,014 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291975. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|Tta/Cta|L95|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,015 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291981. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|Tta/Cta|L97|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,015 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291993. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Gtt/Att|V101I|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,015 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292024. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L111|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,016 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292033. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tTg/tCg|L114S|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,016 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292042. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/|I117|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,016 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292270. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/TAtat|Y21Y?|PF3D7_0531900|protein_coding|CODING|PF3D7_0531900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,017 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292387. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F60|PF3D7_0531900|protein_coding|CODING|PF3D7_0531900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,017 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292552. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cAt/cCt|H48P|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,018 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292595. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F63|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,018 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292895. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|ttA/ttT|L162F|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,018 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293105. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atT/atA|I232|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,019 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293178. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ctt/Gtt|L257V|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,019 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293223. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_CHANGE_PLUS_CODON_INSERTION(MODERATE||gtt/gTATtt|V272VF|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,019 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293240. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tgt/|C278|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,020 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293616. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ctt/|L403|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,020 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293902. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/taTATTTt|Y498YI?|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,021 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1294452. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aac/Aaac|N682K?|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,021 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1294485. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L693|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,022 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1295321. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tCt/tAt|S971Y|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,022 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1295329. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||caa/|Q974|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:22,022 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1295969. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cTt/cCt|L1187P|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,128 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:243000. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gac/gacTGATAATAAAAGGAATAAAACAACAACAATTTTGATGACGATAATAAAAGGAATAACAACAACAACCATTTTGATGAT|D507D***KE*NNNNFDDDNKRNNNNNHFDD|RAD50|protein_coding|CODING|PF3D7_0605800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,195 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:422621. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|DHHC2|protein_coding|CODING|PF3D7_0609800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,253 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:591869. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aga/agGATAAATGATGAATCAAAATAATAATGACATTAGa|R132RINDESK***H*?|PF3D7_0614100|protein_coding|CODING|PF3D7_0614100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,330 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:793843. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATGAAACAAAATGAAACAAATATAAAAATGATATGGAAAAt|N434K*NKMKQI*K*YGK?|PF3D7_0619000|protein_coding|CODING|PF3D7_0619000.2|6)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,390 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:968668. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTTATCATATAAATAGTATAAat|N763III*IV*?|TKL4|protein_coding|CODING|PF3D7_0623800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,465 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:1162862. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aat/aTTTAAAATGat|N1001I*ND|PK4|protein_coding|CODING|PF3D7_0628200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,491 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:1238065. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/TATATATATATTTATAAATAATaat|N5376YIYIYK**?|SET1|protein_coding|CODING|PF3D7_0629700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,587 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:118138. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtt/gtATATATATATATATATTAATCCGAAGGATAACATAAATGTATGTGCTAATCCGAAGGATAACATAAATGTATGTGCt|V967VYIYIY*SEG*HKCMC*SEG*HKCMC?|PF3D7_0703000|protein_coding|CODING|PF3D7_0703000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,700 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:465793. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ata/aAATAATTTGGta|I897K*FG?|PF3D7_0710200|protein_coding|CODING|PF3D7_0710200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,760 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:648363. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taATATATATATATATATTATAGACAATATGAATGATAATTAt|Y135*YIYIYYRQYE**L?|PF3D7_0714100|protein_coding|CODING|PF3D7_0714100.1|1)\" \r\n",
        "WARN  16:33:22,788 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:731125. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/|*1006|PF3D7_0716700|protein_coding|CODING|PF3D7_0716700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,827 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:849242. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATATATATATATATAat|N1451IYIYI*?|PF3D7_0719400|protein_coding|CODING|PF3D7_0719400.1|1)\" \r\n",
        "WARN  16:33:22,834 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:870675. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ggg/ATGGGGAAGATGTAAAGggg|G1584MGKM*R?|PF3D7_0719900|protein_coding|CODING|PF3D7_0719900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,873 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:989064. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||tGc/tGc|C307|PF3D7_0723700|protein_coding|CODING|PF3D7_0723700.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,902 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1066003. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATTTTGATAATAACAAAAGt|N251KF***QK?|PF3D7_0725100|protein_coding|CODING|PF3D7_0725100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:22,989 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1222886. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/TAGATGAGtat|Y1488*MS?|PF3D7_0728600|protein_coding|CODING|PF3D7_0728600.1|1)\" \r\n",
        "WARN  16:33:23,000 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1242785. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaTTAATAATAATAATAATAATAATAATATTAATAATAATAATAATATTAATAATAATAATATt|N1447N********Y*****Y****Y?|PF3D7_0729100|protein_coding|CODING|PF3D7_0729100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:23,178 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:92850. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|ttG/ttT|L1574F|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,178 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:92859. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS&WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||ttT/ttT|F1571|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS&WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:23,179 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:92871. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ttC/ttT|F1567|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,179 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:92887. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tCt/tTt|S1562F|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,181 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:95108. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F821|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,181 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:95147. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ttt/Gtt|F809V|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,182 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96378. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||atg/atATATg|M398IY?|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,183 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96386. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ata/Gta|I396V|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,183 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96390. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||atg/atATATg|M394IY?|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,183 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96416. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||tat/tatGTATGT|Y385YVC|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,184 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96422. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/tatGT|Y383Y?|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:23,221 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:186826. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aag/aAGAGGAAGAAGTAGAATAATGAACag|K1973KRKK*NNEQ|ACbeta|protein_coding|CODING|PF3D7_0802600.1|3)\" \r\n",
        "WARN  16:33:23,245 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:254311. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||tat/taTGAAGATGGACATGGTGAATAATATATATATATATATATATATATt|Y1410YEDGHGE*YIYIYIYI|PF3D7_0804500|protein_coding|CODING|PF3D7_0804500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:23,390 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:672763. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aCTTTTTTTTaa|K885TFF*|ABCF1|protein_coding|CODING|PF3D7_0813700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:23,483 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:941842. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaTATAACATGAAGAAAAAAATAAa|K883NIT*RKK*?|PF3D7_0820900|protein_coding|CODING|PF3D7_0820900.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:23,586 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1238271. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aac/aacTAA|N410N*|GAMA|protein_coding|CODING|PF3D7_0828800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:23,597 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1271720. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tct/tcTt|S148S?|PF3D7_0829700|protein_coding|CODING|PF3D7_0829700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,601 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1282182. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F417|PF3D7_0830100|protein_coding|CODING|PF3D7_0830100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:23,802 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:349291. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*1134|PF3D7_0907200|protein_coding|CODING|PF3D7_0907200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:23,872 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:530853. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaG/gaG|E314|PF3D7_0911600|protein_coding|CODING|PF3D7_0911600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:23,903 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:613791. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatTATATATAATATTAAAAAT|N567NYI*Y*K?|PF3D7_0914300|protein_coding|CODING|PF3D7_0914300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:23,959 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:761842. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tct/tcTTt|S211S?|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,960 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762374. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F389|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:23,960 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762509. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||ttt/TTTttt|F434FF|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:24,074 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1054052. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aat/aaAATAGTATGAGTAATATATAGTATGAAt|N1255KIV*VIYSMN|PF3D7_0926100|protein_coding|CODING|PF3D7_0926100.1|1)\" \r\n",
        "WARN  16:33:24,106 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1152725. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||atc/aATCAAAACATATGATGATAtc|I810NQNI**Y?|PF3D7_0928700|protein_coding|CODING|PF3D7_0928700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:24,146 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1271644. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aGt/aGt|S434|PF3D7_0931700|protein_coding|CODING|PF3D7_0931700.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:24,158 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1301534. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aca/acaGGTGAAAAGTGACAAA|T347TGEK*Q?|PF3D7_0932700|protein_coding|CODING|PF3D7_0932700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:24,173 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1345106. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/Taat|N575*?|XPD|protein_coding|CODING|PF3D7_0934100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:24,324 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:266127. Error was: \"SPLICE_SITE_ACCEPTOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_ACCEPTOR+INTRON(HIGH||||PF3D7_1006300|protein_coding|CODING|PF3D7_1006300.1|3)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:24,371 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:407743. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gat/Gat|D932|PF3D7_1010200|protein_coding|CODING|PF3D7_1010200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:24,371 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:407745. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gGt/gGt|G931|PF3D7_1010200|protein_coding|CODING|PF3D7_1010200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:24,587 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1082468. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ggg/ggATAGGCACACCAAAATg|G1443G*AHQN?|PF3D7_1025500|protein_coding|CODING|PF3D7_1025500.1|2)\" \r\n",
        "WARN  16:33:24,587 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1082471. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||ggt/ggATAAGCACACCAAAAt|G1442G*AHQN|PF3D7_1025500|protein_coding|CODING|PF3D7_1025500.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:24,680 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1344714. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/tgATGAAAAGAGTGATAATATTGATGGTTGt|C728**KE**Y*WL?|Myb2|protein_coding|CODING|PF3D7_1033600.1|1)\" \r\n",
        "WARN  16:33:24,681 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1344729. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||tct/tcCTAATATTATGGGTGTGATGAAAAGAGTGAt|S723S*YYGCDEKSD|Myb2|protein_coding|CODING|PF3D7_1033600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:24,730 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1483860. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aac/aaTAATAATAATAATATTTGCAATAATAATAATATTTGCAATAATAATAATATTTGATTAc|N498NNNNNICNNNNICNNNNI*L?|PF3D7_1037400|protein_coding|CODING|PF3D7_1037400.1|1)\" \r\n",
        "WARN  16:33:24,742 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1521658. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aCa|K842T|Pf11-1|protein_coding|CODING|PF3D7_1038400.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:24,938 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:479597. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaAAAAAAAAAAATTTAATAAt|N450KKKKI**?|PF3D7_1112600|protein_coding|CODING|PF3D7_1112600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,008 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:690406. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cac/cacAAATGAACTGAATCATCAC|H384HK*TESS?|PF3D7_1118200|protein_coding|CODING|PF3D7_1118200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,199 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1278131. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gac/GACGATGAACAATTAGATGATAGGTgac|D263DDEQLDDR*?|PF3D7_1133100|protein_coding|CODING|PF3D7_1133100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,242 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1412405. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ata/atAGAAGAAAGGAAAGAAAAAAATTGAAGACATGAATATGTTAAAAGAATATGATATAAGAAGAAGAAGGAAAGAAAAGGAAAGAACa|I2767IEERKEKN*RHEYVKRI*YKKKKERKGKN?|PF3D7_1136000|protein_coding|CODING|PF3D7_1136000.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,282 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1541662. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/aTATGTATGTATGTATCta|I287ICMYVS?|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,283 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1542865. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||caa/cAAaa|Q688Q?|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,293 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1571787. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/Tttt|F729F?|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,293 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1572017. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/ttAa|L805L?|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,293 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1572158. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/agAa|R852R?|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,294 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1572602. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||attata/|II1000|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,310 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1621385. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aga/aAATATATATATATATATATATga|R169KYIYIYI*|PF3D7_1140600|protein_coding|CODING|PF3D7_1140600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,384 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1853371. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1146800|protein_coding|CODING|PF3D7_1146800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,401 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1903508. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gct/gGCCGATGAATTAAAAAAATct|A1369GR*IKKI?|MAEBL|protein_coding|CODING|PF3D7_1147800.1|2)\" \r\n",
        "WARN  16:33:25,406 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1917646. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/tTTTATATATATATAtt|F419FYIYI?|PF3D7_1148100|protein_coding|CODING|PF3D7_1148100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,407 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920294. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/ATATata|I10IY?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,407 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920424. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tgc/Agc|C53S|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,408 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920431. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|tGa/tAa|*55|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,408 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920434. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/|Y56|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,408 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920462. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gta/Tgta|V66C?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,409 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920756. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tac/Atac|Y164I?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,409 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920809. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N181S|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,412 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1926922. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aat/Tat|N310Y|PF3D7_1148600|protein_coding|CODING|PF3D7_1148600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,413 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1931180. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/Tttt|F750F?|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:25,414 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1932043. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTt/aCt|I1037T|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,500 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:154617. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatTTAAAATAAAAGTAATAAAAAT|N287NLK*K**K?|PF3D7_1202600|protein_coding|CODING|PF3D7_1202600.1|1)\" \r\n",
        "WARN  16:33:25,516 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:201517. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaCTAATCATATACATCAAACAAATCATATACATCAAAAa|K241N*SYTSNKSYTSK?|PF3D7_1204100|protein_coding|CODING|PF3D7_1204100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,592 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:430508. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gaGGTAAGCCGATATATATATAAGAAGTGAt|D791EVSRYIYKK*?|KROX1|protein_coding|CODING|PF3D7_1209300.1|1)\" \r\n",
        "WARN  16:33:25,595 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:436709. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaT/gaT|D206|CIA1|protein_coding|CODING|PF3D7_1209400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,608 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:468533. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gta/gtaTGACA|V258V*?|PF3D7_1210400|protein_coding|CODING|PF3D7_1210400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,661 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:617385. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*110|PF3D7_1214800|protein_coding|CODING|PF3D7_1214800.1|3)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:25,838 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1149682. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tta/TTAGAAAAGGATCGATTGGATAGGTTAGAAAAGGAACGATTGGAATAGAtta|L525LEKDRLDRLEKERLE*I?|NEK1|protein_coding|CODING|PF3D7_1228300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,082 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1903451. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||ctt/cATTAATGATAAAtt|L539H***I|PF3D7_1245600|protein_coding|CODING|PF3D7_1245600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,084 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1907405. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_DELETION(MODERATE||attttttttttttttttt/att|IFFFFF121I|PF3D7_1245700|protein_coding|CODING|PF3D7_1245700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:26,084 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1907488. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/AATTttt|F149NF?|PF3D7_1245700|protein_coding|CODING|PF3D7_1245700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:26,085 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1908169. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||att/AATTAatt|I8N*?|PF3D7_1245800|protein_coding|CODING|PF3D7_1245800.1|1)\" \r\n",
        "WARN  16:33:26,104 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1960984. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agc/agTATATATATATATATATATATATATATGGTAAATTAAAAATGATAAAGTTAGTc|S458SIYIYIYIYMVN*K**S*?|PF3D7_1247500|protein_coding|CODING|PF3D7_1247500.1|1)\" \r\n",
        "WARN  16:33:26,133 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:2042841. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gat/gaGTGCCAAAAAAATGAAGGTGACCAAAAAAATGAAGGTGACCAAAAAAATGAAGGTTTTGAATAATACAAACATTTTGAATAATACAAACAATGTTAATAATAATGACCATATTAAAAATGATAATATTAAAAATGACAt|D124ECQKNEGDQKNEGDQKNEGFE*YKHFE*YKQC****PY*K**Y*K*H|PF3D7_1249900|protein_coding|CODING|PF3D7_1249900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,272 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:208289. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gaa/gaaTCGCATAATTACCAAAATAAACATAATTTACCATATAAATCAAAATATAAATCATAATATGTTGAA|E1457ESHNYQNKHNLPYKSKYKS*YVE|PF3D7_1303800|protein_coding|CODING|PF3D7_1303800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,411 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:661261. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|MYB1|protein_coding|CODING|PF3D7_1315800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,485 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:886263. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1321300|protein_coding|CODING|PF3D7_1321300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,556 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1093640. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aga/aTATATATATATATATAGAGATAGAGAAAGAGATCGAGATAGAGAAAga|R113IYIYI*R*RKRSR*RK?|PF3D7_1326300|protein_coding|CODING|PF3D7_1326300.1|2)\" \r\n",
        "WARN  16:33:26,563 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1114133. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gat/Gat|D2087|PF3D7_1326600|protein_coding|CODING|PF3D7_1326600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:26,566 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1122538. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*4781|PF3D7_1326600|protein_coding|CODING|PF3D7_1326600.1|2)\" \r\n",
        "WARN  16:33:26,574 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1148969. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||att/atATGATATGGATAATATGAAt|I1084I*YG*YE?|PF3D7_1327300|protein_coding|CODING|PF3D7_1327300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,651 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1391352. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1334000|protein_coding|CODING|PF3D7_1334000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,691 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1517385. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatAAATATATATATATATATATATAACATAAATAAAAATAAT|N736NKYIYIYI*HK*K*?|PF3D7_1337500|protein_coding|CODING|PF3D7_1337500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,831 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1938005. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gaaTCAAAGTAATGATATA|E3125ESK**Y?|PF3D7_1348400|protein_coding|CODING|PF3D7_1348400.1|1)\" \r\n",
        "WARN  16:33:26,834 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1946155. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cca/cTATAAATGAAAca|P409L*MK?|PF3D7_1348400|protein_coding|CODING|PF3D7_1348400.1|1)\" \r\n",
        "WARN  16:33:26,850 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1994610. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N335|PF3D7_1349500|protein_coding|CODING|PF3D7_1349500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:26,850 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1994621. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Cat/Cat|H332|PF3D7_1349500|protein_coding|CODING|PF3D7_1349500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:26,850 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1994622. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaC/aaC|N331|PF3D7_1349500|protein_coding|CODING|PF3D7_1349500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:33:26,866 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2035376. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/AAAAATATATaaa|K514KNI*?|PF3D7_1351000|protein_coding|CODING|PF3D7_1351000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,907 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2157541. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1354000|protein_coding|CODING|PF3D7_1354000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:26,983 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2389283. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/aATAATGGAAGTAAATATGAATGATGATATAAAAAAGGATGATATAAAAAAGGATGATATAAAAAAGat|N2707NNGSKYE**YKKG*YKKG*YKKD|PF3D7_1359700|protein_coding|CODING|PF3D7_1359700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,033 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2535720. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*1174|PF3D7_1363000|protein_coding|CODING|PF3D7_1363000.1|1)\" \r\n",
        "WARN  16:33:27,042 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2562845. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/agAAa|R243R?|PF3D7_1363900|protein_coding|CODING|PF3D7_1363900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:27,053 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2593702. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aATAATAATAATATTATTAAATAATGat|N2891NNNNIIK**?|PF3D7_1364400|protein_coding|CODING|PF3D7_1364400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,084 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2677111. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/AATATGAACAATAATATGAACAATATGAACAATTAaat|N402NMNNNMNNMNN*?|PF3D7_1366900|protein_coding|CODING|PF3D7_1366900.1|3)\" \r\n",
        "WARN  16:33:27,094 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2703137. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gac/gaATGAGGAGAGc|D637E*GE?|PSOP2|protein_coding|CODING|PF3D7_1367800.1|1)\" \r\n",
        "WARN  16:33:27,095 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2703182. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/tgACGAt|C652*R?|PSOP2|protein_coding|CODING|PF3D7_1367800.1|1)\" \r\n",
        "WARN  16:33:27,098 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2711715. Error was: \"START_LOST+CODON_DELETION is not a recognized effect type\". Field was: \"START_LOST+CODON_DELETION(HIGH||atg/|M1|RPN11|protein_coding|CODING|PF3D7_1368100.1|1)\" \r\n",
        "WARN  16:33:27,119 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2775403. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaC/aaC|N494|PF3D7_1369600|protein_coding|CODING|PF3D7_1369600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,223 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:184052. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Aat/Aat|N700|PF3D7_1405300|protein_coding|CODING|PF3D7_1405300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,334 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:518192. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|NMT|protein_coding|CODING|PF3D7_1412800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,342 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:542234. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gaa/Gaa|E877|PF3D7_1413700|protein_coding|CODING|PF3D7_1413700.1|8|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,522 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1057940. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/ATATATATATATATATATATATATTgaa|E594IYIYIYIY*?|PF3D7_1427100|protein_coding|CODING|PF3D7_1427100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,576 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1232560. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aGATGat|N509R*?|PF3D7_1431300|protein_coding|CODING|PF3D7_1431300.1|1)\" \r\n",
        "WARN  16:33:27,591 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1281349. Error was: \"SPLICE_SITE_ACCEPTOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_ACCEPTOR+INTRON(HIGH||||LRR5|protein_coding|CODING|PF3D7_1432400.1|5)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,646 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1462191. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N268|PF3D7_1436100|protein_coding|CODING|PF3D7_1436100.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,673 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1550446. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtg/gTGAATAATATGAATAATGTGAATAATATATATATATATATATATATATATATGTGAATAATAtg|V145VNNMNNVNNIYIYIYIYM*II?|MCA2|protein_coding|CODING|PF3D7_1438400.1|2)\" \r\n",
        "WARN  16:33:27,700 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1628076. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gaaTAATGTGAATT|E1097E*CE?|PF3D7_1440000|protein_coding|CODING|PF3D7_1440000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,737 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1739027. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tGat|Y2915*?|TREP|protein_coding|CODING|PF3D7_1442600.1|2)\" \r\n",
        "WARN  16:33:27,737 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1739056. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||caa/TGAGAAGcaa|Q2925*EA?|TREP|protein_coding|CODING|PF3D7_1442600.1|2)\" \r\n",
        "WARN  16:33:27,738 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1739169. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ata/atTTGAAGAa|I2962I*R?|TREP|protein_coding|CODING|PF3D7_1442600.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,769 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1836771. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/aAAAAAATAATATATATGTGAAAATATAAAAAgt|S563KKIIYM*KYKK?|IK1|protein_coding|CODING|PF3D7_1444500.1|2)\" \r\n",
        "WARN  16:33:27,810 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1947824. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/GGGAAAAAATGATgaa|E194GKK**?|PF3D7_1447800|protein_coding|CODING|PF3D7_1447800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,876 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2150424. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaT/gaT|D668|PF3D7_1452400|protein_coding|CODING|PF3D7_1452400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,944 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2373743. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1457800|protein_coding|CODING|PF3D7_1457800.1|1)\" \r\n",
        "WARN  16:33:27,966 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2443894. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*802|PF3D7_1459600|protein_coding|CODING|PF3D7_1459600.1|2)\" \r\n",
        "WARN  16:33:27,972 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2462332. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||UCHL3|protein_coding|CODING|PF3D7_1460400.1|5)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:27,998 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2540522. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aat/aaAATAAAATAAAATAATTATAAAAATGATGTAAAt|N293KIK*NNYKNDVN|PF3D7_1462400|protein_coding|CODING|PF3D7_1462400.1|1)\" \r\n",
        "WARN  16:33:28,013 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2582521. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*327|CYC1|protein_coding|CODING|PF3D7_1463700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:28,063 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2737568. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aTt/aTt|I165|PF3D7_1466900|protein_coding|CODING|PF3D7_1466900.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:28,107 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2881011. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|MPC2|protein_coding|CODING|PF3D7_1470400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:28,151 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3015662. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gaAAAAAATGATGGAAAAAATGAATAAAAAATGAt|D746EKNDGKNE*KM?|PF3D7_1474000|protein_coding|CODING|PF3D7_1474000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:33:28,211 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3196895. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/TATATATATATATATATATATATATATATATATATATAAAAaaa|K1094YIYIYIYIYIYI*K?|SURF14.1|protein_coding|CODING|PF3D7_1477600.1|3)\" \r\n",
        "WARN  16:33:28,228 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248049. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Att/Gtt|I340V|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,228 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248057. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F343|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,229 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248063. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|tgA/tgT|*344C|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,229 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248065. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tTt|Y345F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,229 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248067. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cat/Aat|H346N|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,229 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248074. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aGa|K348R|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,230 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248077. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aag/|K349|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,230 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248080. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTa/aAa|I350K|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,230 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248098. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tCt/tTt|S356F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,230 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248104. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tTt|Y358F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,231 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248109. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aca/Gca|T360A|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,231 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248110. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCa/aTa|T360I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,231 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248118. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/|I363|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,231 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248120. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atA/atC|I363|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,231 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248125. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|tAa/tTa|*365L|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,232 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248126. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|taA/taT|*365Y|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,232 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248133. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/tTTTGTTTGtt|F368FCL?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,232 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248169. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gtt/gTtt|V380V?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,232 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248175. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/aTTta|I382I?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,233 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248180. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|atA/atG|I383M|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,233 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248183. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|taA/taT|*384Y|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,233 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248186. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taC/taT|Y385|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,233 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248190. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tac/Cac|Y387H|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,233 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248192. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taC/taT|Y387|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,234 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248225. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|taA/taT|*398Y|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,234 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248226. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gaa/gAaa|E399E?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,234 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248263. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCt/aTt|T411I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,234 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248267. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taT/taC|Y412|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,235 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248269. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCa/aTa|T413I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,235 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248287. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tGt/tTt|C419F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,235 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248288. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tgT/tgG|C419W|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,235 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248293. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tTt/tAt|F421Y|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,236 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248294. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|ttT/ttA|F421L|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,236 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248295. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*422|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,236 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248303. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaG/aaA|K424|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:33:28,236 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248309. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaA/aaG|K426|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:28,248 VariantAnnotator - Processed 18088 loci.\r\n",
        " \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:28,329 ProgressMeter -            done   2.3332831E7     9.0 s       0.0 s      100.0%     9.0 s       0.0 s \r\n",
        "INFO  16:33:28,330 ProgressMeter - Total runtime 9.11 secs, 0.15 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:29,609 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "KE01\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:47,101 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:33:47,102 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:33:47,103 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:33:47,103 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n",
        "INFO  16:33:47,106 HelpFormatter - Program Args: -T LeftAlignAndTrimVariants -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -V /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.sorted.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf \r\n",
        "INFO  16:33:47,110 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:33:47,111 HelpFormatter - Date/Time: 2015/10/02 16:33:47 \r\n",
        "INFO  16:33:47,111 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:33:47,111 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:47,450 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:47,497 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 1000 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:47,831 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.sorted.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:47,912 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:33:47,913 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:33:47,913 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:33:47,914 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:33:47,914 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "6007 variants were aligned\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:48,967 ProgressMeter -            done     43534.0     1.0 s      24.0 s       99.8%     1.0 s       0.0 s \r\n",
        "INFO  16:33:48,968 ProgressMeter - Total runtime 1.05 secs, 0.02 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:50,239 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.000\tSnpEff version SnpEff 4.1g (build 2015-05-17), by Pablo Cingolani\r\n",
        "00:00:00.011\tCommand: 'ann'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.029\tReading configuration file 'snpEff.config'. Genome: 'Pf3D7july2015'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.511\tdone\r\n",
        "00:00:00.511\tReading database for genome version 'Pf3D7july2015' from file '/Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt/snpeff/snpEff/./data/Pf3D7july2015/snpEffectPredictor.bin' (this might take a while)\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.321\tdone\r\n",
        "00:00:01.331\tBuilding interval forest\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.729\tdone.\r\n",
        "00:00:01.730\tGenome stats :\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "#-----------------------------------------------\r\n",
        "# Genome name                : 'Plasmodium_falciparum'\r\n",
        "# Genome version             : 'Pf3D7july2015'\r\n",
        "# Has protein coding info    : true\r\n",
        "# Genes                      : 5645\r\n",
        "# Protein coding genes       : 5492\r\n",
        "#-----------------------------------------------\r\n",
        "# Transcripts                : 5886\r\n",
        "# Avg. transcripts per gene  : 1.04\r\n",
        "#-----------------------------------------------\r\n",
        "# Checked transcripts        : \r\n",
        "#               AA sequences :      0 ( 0.00% )\r\n",
        "#              DNA sequences :      0 ( 0.00% )\r\n",
        "#-----------------------------------------------\r\n",
        "# Protein coding transcripts : 5580\r\n",
        "#              Length errors :    126 ( 2.26% )\r\n",
        "#  STOP codons in CDS errors :    145 ( 2.60% )\r\n",
        "#         START codon errors :    200 ( 3.58% )\r\n",
        "#        STOP codon warnings :    205 ( 3.67% )\r\n",
        "#              UTR sequences :      0 ( 0.00% )\r\n",
        "#               Total Errors :    212 ( 3.80% )\r\n",
        "# WARNING                    : No protein coding transcript has UTR\r\n",
        "#-----------------------------------------------\r\n",
        "# Cds                        : 14366\r\n",
        "# Exons                      : 14366\r\n",
        "# Exons with sequence        : 14366\r\n",
        "# Exons without sequence     : 0\r\n",
        "# Avg. exons per transcript  : 2.44\r\n",
        "#-----------------------------------------------\r\n",
        "# Number of chromosomes      : 41\r\n",
        "# Chromosomes                : Format 'chromo_name size codon_table'\r\n",
        "#\t\t'Pf3D7_14_v3'\t3291936\tStandard\r\n",
        "#\t\t'Pf3D7_13_v3'\t2925236\tStandard\r\n",
        "#\t\t'Pf3D7_12_v3'\t2271494\tStandard\r\n",
        "#\t\t'Pf3D7_11_v3'\t2038340\tStandard\r\n",
        "#\t\t'Pf3D7_10_v3'\t1687656\tStandard\r\n",
        "#\t\t'Pf3D7_09_v3'\t1541735\tStandard\r\n",
        "#\t\t'Pf3D7_08_v3'\t1472805\tStandard\r\n",
        "#\t\t'Pf3D7_07_v3'\t1445207\tStandard\r\n",
        "#\t\t'Pf3D7_06_v3'\t1418242\tStandard\r\n",
        "#\t\t'Pf3D7_05_v3'\t1343557\tStandard\r\n",
        "#\t\t'Pf3D7_04_v3'\t1200490\tStandard\r\n",
        "#\t\t'Pf3D7_03_v3'\t1067971\tStandard\r\n",
        "#\t\t'Pf3D7_02_v3'\t947102\tStandard\r\n",
        "#\t\t'Pf3D7_01_v3'\t640851\tStandard\r\n",
        "#\t\t'Pf_M76611'\t5967\tProtozoan_Mitochondrial\r\n",
        "#\t\t'mal_mito_1'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_2'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_3'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_RNA19:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_SSUB:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_10:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_15:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_16:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_17:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_1:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_20:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_21:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_9:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUC:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUG:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA11:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA12:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA14:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA18:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA22:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA4:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA5:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA6:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA7:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_SSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#-----------------------------------------------\r\n",
        "\r\n",
        "00:00:02.399\tPredicting variants\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 22\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 23\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 24\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 25\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 26\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 27\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 28\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 29\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 30\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 31\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 32\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 33\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 34\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 35\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 36\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 37\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 38\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 39\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 40\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 41\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 42\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 43\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 44\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf', line 45\r\n",
        "00:00:02.440\tCreating sequences from exon information 'Pf3D7_01_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.462\tBuilding sequence tree for chromosome 'Pf3D7_01_v3'\r\n",
        "00:00:02.464\tDone. Loaded 351 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.594\tCreating sequences from exon information 'Pf3D7_02_v3'\r\n",
        "00:00:02.608\tBuilding sequence tree for chromosome 'Pf3D7_02_v3'\r\n",
        "00:00:02.611\tDone. Loaded 558 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.724\tCreating sequences from exon information 'Pf3D7_03_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.737\tBuilding sequence tree for chromosome 'Pf3D7_03_v3'\r\n",
        "00:00:02.739\tDone. Loaded 631 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.886\tCreating sequences from exon information 'Pf3D7_04_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.899\tBuilding sequence tree for chromosome 'Pf3D7_04_v3'\r\n",
        "00:00:02.900\tDone. Loaded 601 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.008\tCreating sequences from exon information 'Pf3D7_05_v3'\r\n",
        "00:00:03.022\tBuilding sequence tree for chromosome 'Pf3D7_05_v3'\r\n",
        "00:00:03.023\tDone. Loaded 776 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.148\tCreating sequences from exon information 'Pf3D7_06_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.163\tBuilding sequence tree for chromosome 'Pf3D7_06_v3'\r\n",
        "00:00:03.165\tDone. Loaded 831 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.276\tCreating sequences from exon information 'Pf3D7_07_v3'\r\n",
        "00:00:03.292\tBuilding sequence tree for chromosome 'Pf3D7_07_v3'\r\n",
        "00:00:03.293\tDone. Loaded 728 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.395\tCreating sequences from exon information 'Pf3D7_08_v3'\r\n",
        "00:00:03.409\tBuilding sequence tree for chromosome 'Pf3D7_08_v3'\r\n",
        "00:00:03.411\tDone. Loaded 842 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.476\tCreating sequences from exon information 'Pf3D7_09_v3'\r\n",
        "00:00:03.492\tBuilding sequence tree for chromosome 'Pf3D7_09_v3'\r\n",
        "00:00:03.493\tDone. Loaded 976 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.553\tCreating sequences from exon information 'Pf3D7_10_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.573\tBuilding sequence tree for chromosome 'Pf3D7_10_v3'\r\n",
        "00:00:03.575\tDone. Loaded 1039 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.632\tCreating sequences from exon information 'Pf3D7_11_v3'\r\n",
        "00:00:03.655\tBuilding sequence tree for chromosome 'Pf3D7_11_v3'\r\n",
        "00:00:03.658\tDone. Loaded 1315 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.738\t\t10000 variants (10000 variants per second), 10000 VCF entries\r\n",
        "00:00:03.759\tCreating sequences from exon information 'Pf3D7_12_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.783\tBuilding sequence tree for chromosome 'Pf3D7_12_v3'\r\n",
        "00:00:03.785\tDone. Loaded 1333 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.881\tCreating sequences from exon information 'Pf3D7_13_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.915\tBuilding sequence tree for chromosome 'Pf3D7_13_v3'\r\n",
        "00:00:03.919\tDone. Loaded 1779 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.050\tCreating sequences from exon information 'Pf3D7_14_v3'\r\n",
        "00:00:04.088\tBuilding sequence tree for chromosome 'Pf3D7_14_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.092\tDone. Loaded 2069 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "ERRORS: Some errors were detected\r\n",
        "Error type\tNumber of errors\r\n",
        "ERROR_CHROMOSOME_NOT_FOUND\t24\r\n",
        "\r\n",
        "\r\n",
        "\r\n",
        "WARNINGS: Some warning were detected\r\n",
        "Warning type\tNumber of warnings\r\n",
        "WARNING_REF_DOES_NOT_MATCH_GENOME\t19\r\n",
        "WARNING_TRANSCRIPT_INCOMPLETE\t213\r\n",
        "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\t1659\r\n",
        "WARNING_TRANSCRIPT_NO_START_CODON\t68\r\n",
        "\r\n",
        "\r\n",
        "00:00:04.223\tCreating summary file: snpEff_summary.html\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:05.128\tCreating genes file: snpEff_genes.txt\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:05.590\tdone.\r\n",
        "00:00:05.591\tLogging\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:06.598\tChecking for updates...\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:59,614 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:33:59,615 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:33:59,615 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:33:59,615 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n",
        "INFO  16:33:59,619 HelpFormatter - Program Args: -T VariantAnnotator -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -A TandemRepeatAnnotator -A SnpEff --variant /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.leftaligned.vcf --snpEffFile /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.snpeff.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.snpeff_annotated.vcf \r\n",
        "INFO  16:33:59,623 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:33:59,624 HelpFormatter - Date/Time: 2015/10/02 16:33:59 \r\n",
        "INFO  16:33:59,624 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:33:59,624 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:33:59,967 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:00,014 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 250 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:00,399 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KE01.snpeff.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:00,493 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:34:00,494 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:34:00,494 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:34:00,495 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:34:00,495 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:00,576 SnpEff - Skipping malformed SnpEff effect field at M76611:772. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:00,610 SnpEff - Skipping malformed SnpEff effect field at M76611:2712. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,620 SnpEff - Skipping malformed SnpEff effect field at M76611:3003. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,621 SnpEff - Skipping malformed SnpEff effect field at M76611:3008. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:00,634 SnpEff - Skipping malformed SnpEff effect field at M76611:4126. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,636 SnpEff - Skipping malformed SnpEff effect field at M76611:4237. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,637 SnpEff - Skipping malformed SnpEff effect field at M76611:4260. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,651 SnpEff - Skipping malformed SnpEff effect field at M76611:5504. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:00,734 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:13289. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:00,738 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:15405. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,743 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:17915. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,745 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18694. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,745 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18828. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,746 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18860. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,750 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:22669. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,751 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23943. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,752 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23947. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,754 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:25411. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,754 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:25587. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,756 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:27761. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,757 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:28008. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,758 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:28127. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,758 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:29000. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:00,759 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:29019. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:00,810 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:101981. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taGAAGAAAATGTt|Y986*KKM?|RESA|protein_coding|CODING|PF3D7_0102200.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:00,890 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:219211. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F4|PF3D7_0105000|protein_coding|CODING|PF3D7_0105000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:00,961 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:309256. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aATtg|M1N?|PF3D7_0107500|protein_coding|CODING|PF3D7_0107500.1|1)\" \r\n",
        "WARN  16:34:00,987 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:361987. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_CHANGE_PLUS_CODON_INSERTION(MODERATE||aaa/aaTAATATATATATATATATAAAAAAAAAa|K200NNIYIYIKKK|PF3D7_0108900|protein_coding|CODING|PF3D7_0108900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:01,038 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:474666. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aat/Tat|N310Y|PF3D7_0112300|protein_coding|CODING|PF3D7_0112300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:01,039 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:477167. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||cta/cTATATATATATATATAta|L247LYIYIY?|PF3D7_0112600|protein_coding|CODING|PF3D7_0112600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:01,040 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478408. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gat/gaATt|D377E?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:01,040 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478417. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N380S|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:01,041 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478423. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCc/aTc|T382I|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:01,041 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478424. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|acC/acG|T382|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:01,042 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:479254. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tca/tcAa|S659S?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:01,220 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:202518. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tatTAATAATGATAATATTAATTAT|Y510Y****Y*L?|PF3D7_0204600|protein_coding|CODING|PF3D7_0204600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:01,462 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:696841. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N907|PF3D7_0216800|protein_coding|CODING|PF3D7_0216800.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:01,466 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:698222. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gTATATATATAATAATCATACCTGTGat|D447VYI**SYL*?|PF3D7_0216800|protein_coding|CODING|PF3D7_0216800.1|1)\" \r\n",
        "WARN  16:34:01,476 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:709392. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/|*552|PF3D7_0217100|protein_coding|CODING|PF3D7_0217100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:01,544 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:798060. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gct/gcATAAAAATTAAAAATTAAAAATACAAATGAAAATGTTGAAGAAAGTGt|A384A*KLKIKNTNENVEES?|LSA3|protein_coding|CODING|PF3D7_0220000.1|2)\" \r\n",
        "WARN  16:34:01,580 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:848882. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/TAAAtgt|C225*M?|PF3D7_0221000|protein_coding|CODING|PF3D7_0221000.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:01,772 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:268346. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtg/gtTGAATAACGTGAATAACGTg|V1647VE*RE*R?|PF3D7_0305500|protein_coding|CODING|PF3D7_0305500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:01,822 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:386419. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||YVH1|protein_coding|CODING|PF3D7_0309000.1|4)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:01,860 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:451382. Error was: \"STOP_LOST+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_LOST+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||taa/tTATATATATATATATATATAAaa|*1295LYIYIYIK|PIESP1|protein_coding|CODING|PF3D7_0310400.1|1)\" \r\n",
        "WARN  16:34:01,878 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:491986. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||att/ATTAATTTTAATAAGACAGATGGTTAatt|I1009INFNKTDG*?|PF3D7_0311400|protein_coding|CODING|PF3D7_0311400.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:01,935 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:609560. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gat/ATATATTATTTATTAAAATAAAAATAAATTAATGATGATAATATGAGTGATGATAATATGAGTGATGATAATATAAGTGATGATAATATAAGCgat|D227IYYLLK*K*INDDNMSDDNMSDDNISDDNISD|PF3D7_0314900|protein_coding|CODING|PF3D7_0314900.1|1)\" \r\n",
        "WARN  16:34:01,952 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:656759. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aac/aaAAAAAAAAAATTACCATTGTGATGATAATAATAATAACAAc|N163KKKKLPL******Q?|PF3D7_0316200|protein_coding|CODING|PF3D7_0316200.1|1)\" \r\n",
        "WARN  16:34:01,952 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:657246. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_0316200|protein_coding|CODING|PF3D7_0316200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:01,974 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:719056. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/aACGGTGAAGAATATATATGATAATATGTGAAAAACAAAgt|S776NGEEYI**YVKNK?|EG5|protein_coding|CODING|PF3D7_0317500.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,089 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:983965. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|RIPR|protein_coding|CODING|PF3D7_0323400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,219 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:227486. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||atg/atAAAATATATATGAAATATGAAGATAAATAATATGAATAATATg|M1416IKYI*NMKINNMNNM|ApiAP2|protein_coding|CODING|PF3D7_0404100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,268 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:352982. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_0406800|protein_coding|CODING|PF3D7_0406800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,283 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:392544. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaC/aaC|N269|PF3D7_0407900|protein_coding|CODING|PF3D7_0407900.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:02,300 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:435366. Error was: \"START_LOST+CODON_INSERTION is not a recognized effect type\". Field was: \"START_LOST+CODON_INSERTION(HIGH||atg/aAAAAAAAAAAAAAAAAAAtg|M1KKKKKKM|PF3D7_0409000|protein_coding|CODING|PF3D7_0409000.1|1)\" \r\n",
        "WARN  16:34:02,314 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:466266. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTGATGATAAGGTGAACCATACAAATGATGATAAGGTGAACCATACGATATat|N323MMIR*TIQMMIR*TIRY?|EVP1|protein_coding|CODING|PF3D7_0410000.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,326 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:488000. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||att/TAATAATatt|I724**Y?|PF3D7_0410800|protein_coding|CODING|PF3D7_0410800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,443 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:767841. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Ggt/Ggt|G3850|PF3D7_0417400|protein_coding|CODING|PF3D7_0417400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:02,463 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:802241. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gTa/gTa|V2421|PF3D7_0418000|protein_coding|CODING|PF3D7_0418000.1|7|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,489 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:842169. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/agATATTAAt|S325RY*?|PF3D7_0418700|protein_coding|CODING|PF3D7_0418700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,546 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:928217. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gat/gATGATAATAATAATGAAGGTATTAATAATGATGATAATAATAATGGATGACat|D3410DDNNNEGINNDDNNNG*H|ApiAP2|protein_coding|CODING|PF3D7_0420300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,596 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:1006980. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATAATAATAATAACAATAATAATAATAATAACAATAat|N3423I****Q*****Q*?|PF3D7_0422200|protein_coding|CODING|PF3D7_0422200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,657 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:1101269. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||tat/tAACat|Y487*H|SURF4.2|protein_coding|CODING|PF3D7_0424400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,744 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:44310. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gaaTAAAA|E1107E*?|MESA|protein_coding|CODING|PF3D7_0500800.1|2)\" \r\n",
        "WARN  16:34:02,747 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:46927. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||ggt/ggAATCTAAGAAACAGAt|G235GI*ETD|MESA|protein_coding|CODING|PF3D7_0500800.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,817 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:158855. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gac/gACAAATATGATATATAAac|D411DKYDI*?|ATPase3|protein_coding|CODING|PF3D7_0504000.1|1)\" \r\n",
        "WARN  16:34:02,833 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:183949. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aag/aGTTAAGGag|K1177S*G?|CEP120|protein_coding|CODING|PF3D7_0504700.1|1)\" \r\n",
        "WARN  16:34:02,835 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:184002. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/TAaaa|K1195*?|CEP120|protein_coding|CODING|PF3D7_0504700.1|1)\" \r\n",
        "WARN  16:34:02,849 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:201654. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ata/ataTTCCAACATTTGTGACAATATA|I1026IFQHL*QY?|PF3D7_0504800|protein_coding|CODING|PF3D7_0504800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,855 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:210323. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tatAGGTGACAAATATGGTGACAAATATGGTGACAAATAT|Y2622YR*QIW*QIW*QI?|PF3D7_0505000|protein_coding|CODING|PF3D7_0505000.1|1)\" \r\n",
        "WARN  16:34:02,857 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:212080. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gAt/gAt|D2037|PF3D7_0505000|protein_coding|CODING|PF3D7_0505000.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:02,975 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:410638. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ggt/ggTGATGATGATGATAGGTGATGATGATGATGATGAt|G124GDDDDR******?|PI4K|protein_coding|CODING|PF3D7_0509800.1|1)\" \r\n",
        "WARN  16:34:03,012 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:514009. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||agc/AGCAATAATATGATAAGCTAAagc|S9137SNNMIS*S|PF3D7_0511500|protein_coding|CODING|PF3D7_0511500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,028 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:552726. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cct/ccAAACAAATTATATGAACAGACAGGATAATCTAt|P205PNKLYEQTG*S?|PF3D7_0512800|protein_coding|CODING|PF3D7_0512800.1|1)\" \r\n",
        "WARN  16:34:03,028 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:552729. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||tat/taACTTACAAAt|Y204*LTN|PF3D7_0512800|protein_coding|CODING|PF3D7_0512800.1|1)\" \r\n",
        "WARN  16:34:03,067 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:648390. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gTATAaa|E304V*?|PF3D7_0515600|protein_coding|CODING|PF3D7_0515600.1|6)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,082 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:688978. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gat/Gat|D889|MB2|protein_coding|CODING|PF3D7_0516600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:03,083 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:688982. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gtA/gtA|V887|MB2|protein_coding|CODING|PF3D7_0516600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:03,089 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:705469. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taAt|Y596*?|ApiAP2|protein_coding|CODING|PF3D7_0516800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,230 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1093099. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aATATATATATTATATTATTATGTAATTATTTATATATATAATGGTAATAAATGTGGAGACAATGATGATGat|N724NIYIILLCNYLYI*W**MWRQ***?|PF3D7_0526500|protein_coding|CODING|PF3D7_0526500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,289 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1242817. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||atc/atcTAATACAATATGATCAAAACATC|I390I*YNMIKT?|PF3D7_0530400|protein_coding|CODING|PF3D7_0530400.1|1)\" \r\n",
        "WARN  16:34:03,290 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1243986. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_0530400|protein_coding|CODING|PF3D7_0530400.1|1)\" \r\n",
        "WARN  16:34:03,294 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1254189. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/|*1084|PF3D7_0530700|protein_coding|CODING|PF3D7_0530700.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,410 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:190876. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gta/gtGAATAATGTGAATATGTGa|V622VNNVNM*|PF3D7_0604500|protein_coding|CODING|PF3D7_0604500.1|1)\" \r\n",
        "WARN  16:34:03,431 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:243010. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||cat/caATTTTGATGAGATAATAAAAGGAATAACAACAACAACAAt|H504QF**DNKRNNNNNN|RAD50|protein_coding|CODING|PF3D7_0605800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,534 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:533184. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*687|PF3D7_0612900|protein_coding|CODING|PF3D7_0612900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,547 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:566763. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gaa/Gaa|E209|ApiAP2|protein_coding|CODING|PF3D7_0613800.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:03,547 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:566781. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aaa/Gaa|K215E|ApiAP2|protein_coding|CODING|PF3D7_0613800.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:03,554 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:582443. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatAAGGAGGCGTAAAA|N1151NKEA*?|myoE|protein_coding|CODING|PF3D7_0613900.1|1)\" \r\n",
        "WARN  16:34:03,555 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:582456. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaGTAGATGAAAAGATAGGt|N1147K*MKR*?|myoE|protein_coding|CODING|PF3D7_0613900.1|1)\" \r\n",
        "WARN  16:34:03,571 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:629319. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION(HIGH||tgtgatgaaaaagaa/tga|CDEKE329*|GWT1|protein_coding|CODING|PF3D7_0615300.1|1)\" \r\n",
        "WARN  16:34:03,578 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:648273. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtc/gGTCATCATAATTGATGATCAAAAGAGTGtc|V2076GHHN**SKEC?|PF3D7_0615600|protein_coding|CODING|PF3D7_0615600.1|6)\" \r\n",
        "WARN  16:34:03,590 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:682238. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/TGaaa|K1285*?|PF3D7_0616400|protein_coding|CODING|PF3D7_0616400.1|1)\" \r\n",
        "WARN  16:34:03,591 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:682353. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTGTTGATAATATAAACGTGat|N1323MLII*T*?|PF3D7_0616400|protein_coding|CODING|PF3D7_0616400.1|1)\" \r\n",
        "WARN  16:34:03,593 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:687544. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aAATGATGGTGATAATAATGATGATGATGATAATAATGATGATGGTGATAATAATGat|N935K*W***********W****?|TLP|protein_coding|CODING|PF3D7_0616500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,878 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:126370. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gca/Gca|A256|PF3D7_0703200|protein_coding|CODING|PF3D7_0703200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:03,878 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:126382. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gca/Gca|A252|PF3D7_0703200|protein_coding|CODING|PF3D7_0703200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:03,892 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:165693. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gac/gGATGATAACTACGATAATTATAATAACTACGATAATTATAATACAAAAATGATGAAAATGACAAAAATGATGAAAATGac|D235G**LR*L**LR*L*YKNDENDKNDEND|PF3D7_0703900|protein_coding|CODING|PF3D7_0703900.1|1)\" \r\n",
        "WARN  16:34:03,898 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:185199. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aAACAAAAATGAACAAAATAAAGATAATAAACAAAATGaa|K1722KQK*TK*R**TK*?|PF3D7_0704100|protein_coding|CODING|PF3D7_0704100.1|1)\" \r\n",
        "WARN  16:34:03,901 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:191629. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaaTTTTGTTCATTTTGTTTATTATCTTTATTTTGTTCATAATGATAATTGTACAAATAATAATATTACAAATAATAATATTAAA|K1087KFCSFCLLSLFCS***LYK**YYK**Y*?|PF3D7_0704200|protein_coding|CODING|PF3D7_0704200.1|1)\" \r\n",
        "WARN  16:34:03,902 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:192811. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Ata/Ata|I694|PF3D7_0704200|protein_coding|CODING|PF3D7_0704200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,931 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:258575. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/AATAATTATATTTTCCTTGTTCAAATCTGACTTGTTCAATAATAATATAAATAATAATaat|N831NNYIFLVQI*LVQ**YK***?|PF3D7_0705200|protein_coding|CODING|PF3D7_0705200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:03,964 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:359397. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/Tagt|S567*?|PF3D7_0707700|protein_coding|CODING|PF3D7_0707700.1|1)\" \r\n",
        "WARN  16:34:04,003 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:465255. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aac/TGTTGACGAAAGAAACGACGAAAGAAACGACGAAAGAACGaac|N718C*RKKRRKKRRKNE?|PF3D7_0710200|protein_coding|CODING|PF3D7_0710200.1|1)\" \r\n",
        "WARN  16:34:04,003 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:465709. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aca/aTGTCAATTAAATAATTCCTCATGTCAATTAAATAATTCCTca|T869MSIK*FLMSIK*FL?|PF3D7_0710200|protein_coding|CODING|PF3D7_0710200.1|1)\" \r\n",
        "WARN  16:34:04,006 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:469484. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gag/gaTGTCAATTAAATAATTCCTCATGTCAATTAAATAATTCCTGAACACCAAAAATGTGATGATCCAATTGGATAAAGAAAAGg|E2127DVN*IIPHVN*IIPEHQKCDDPIG*RK?|PF3D7_0710200|protein_coding|CODING|PF3D7_0710200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:04,140 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:850980. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gat/gatAATTTGAGTGGTGTAGATCATTTTTTTTTTTATTATTTATTTTTTATATAATTGAGTGGTGTAGAT|D871DNLSGVDHFFFYYLFFI*LSGVD|PF3D7_0719400|protein_coding|CODING|PF3D7_0719400.1|1)\" \r\n",
        "WARN  16:34:04,160 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:906626. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaC/aaC|N828|PF3D7_0721000|protein_coding|CODING|PF3D7_0721000.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:04,160 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:906627. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Aaa/Aaa|K829|PF3D7_0721000|protein_coding|CODING|PF3D7_0721000.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:04,160 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:906629. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaA/aaA|K829|PF3D7_0721000|protein_coding|CODING|PF3D7_0721000.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:04,161 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:906642. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Tca/Tca|S834|PF3D7_0721000|protein_coding|CODING|PF3D7_0721000.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:04,246 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1136576. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_0726500|protein_coding|CODING|PF3D7_0726500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:04,359 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1327287. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ggt/gTAAGGTATTAATAAAGAATCATATACTACCAAAAGTTATGTAAgt|G603VRY**RIIYYQKLCK?|PTP4|protein_coding|CODING|PF3D7_0730900.1|2)\" \r\n",
        "WARN  16:34:04,362 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1327786. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taGt|Y769*?|PTP4|protein_coding|CODING|PF3D7_0730900.1|2)\" \r\n",
        "WARN  16:34:04,363 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1328116. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taGt|Y879*?|PTP4|protein_coding|CODING|PF3D7_0730900.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:04,498 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96413. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tat/Cat|Y387H|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:04,499 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96418. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tGt|Y385C|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:04,500 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96422. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ata/Gta|I384V|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:04,500 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96426. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|atG/atA|M382I|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:04,500 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:96434. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ata/Gta|I380V|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:04,526 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:129817. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/GAATAGAGAAAAGGAAAAAgaa|E324E*RKGKR?|PF3D7_0801900|protein_coding|CODING|PF3D7_0801900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:04,576 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:261398. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aac/TAGACATGGTGAATATAAAGTAACAATAATAATAATAACaac|N3773*TW*I*SNNNNNNN|PF3D7_0804500|protein_coding|CODING|PF3D7_0804500.1|1)\" \r\n",
        "WARN  16:34:04,580 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:273093. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aaaGACAAAATAGAAGAGAAATATAATATAGAAGAGAAATTATTGTAAATATTGTCCCCATTATTGTAAATATTGTCCCCA|K1440KDKIEEKYNIEEKLL*ILSPLL*ILSP|PF3D7_0804700|protein_coding|CODING|PF3D7_0804700.1|1)\" \r\n",
        "WARN  16:34:04,582 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:276590. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/aGACAAAATAGAAGAGAAATATAATATAGAAGAGAAATTATTGTAAATATTGTCCCCATTATTGTAAATATTGTCCCCAATATTAATCATCATAGTAATAATCACAat|N275RQNRREI*YRREIIVNIVPIIVNIVPNINHHSNNHN|PF3D7_0804700|protein_coding|CODING|PF3D7_0804700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:04,708 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:639147. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aTTAAATGACAAATGACAaa|K771IK*QMT?|PF3D7_0812900|protein_coding|CODING|PF3D7_0812900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:04,823 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:961739. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_0821300|protein_coding|CODING|PF3D7_0821300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,108 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:285588. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||ATG3|protein_coding|CODING|PF3D7_0905700.1|5)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,164 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:430985. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaA/aaA|K136|SPM1|protein_coding|CODING|PF3D7_0909500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:05,165 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:430991. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||ttG/ttG|L138|SPM1|protein_coding|CODING|PF3D7_0909500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,229 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:604303. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||atg/atgGAAATATTGATGATGAAAAAATGAATGTTGATGATGAAAAAATG|M91MEILMMKK*MLMMKK?|PF3D7_0914100|protein_coding|CODING|PF3D7_0914100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,287 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:761677. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_CHANGE_PLUS_CODON_DELETION(MODERATE||atatat/att|IY156I|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:05,288 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:761842. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tct/tcTt|S211S?|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:05,288 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762325. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tct/|S372|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:05,288 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762374. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F389|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:05,289 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762398. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|tcG/tcT|S396|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,374 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:992087. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aac/AACAATGTGAGTAGTATAAATATATaac|N971NNVSSINI*?|PF3D7_0924400|protein_coding|CODING|PF3D7_0924400.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,430 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1157973. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||atg/atGGAAAAAATATCATTATTATTATCATTATTATTATCACAATAATTATGg|M576MEKISLLLSLLLSQ*LW|PF3D7_0928800|protein_coding|CODING|PF3D7_0928800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,474 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1272162. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||PF3D7_0931700|protein_coding|CODING|PF3D7_0931700.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,683 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:387462. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aat/aTATATATATAACAATATGAACAATATGAATAATat|N163IYI*QYEQYE*Y|PF3D7_1009600|protein_coding|CODING|PF3D7_1009600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,846 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:877838. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Aat/Aat|N5492|PF3D7_1021700|protein_coding|CODING|PF3D7_1021700.1|3|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:05,849 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:885215. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatGATAAAAATAATGATAAAAATAATATATAAATGTAAATATAAATGTAAATATAAATGTAAAT|N3032NDKNNDKNNI*M*I*M*I*M*?|PF3D7_1021700|protein_coding|CODING|PF3D7_1021700.1|3)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,896 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1018973. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gatTGATAAACAAAGTAGTGAT|D514D**TK**?|PF3D7_1024100|protein_coding|CODING|PF3D7_1024100.1|1)\" \r\n",
        "WARN  16:34:05,916 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1075282. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aac/aTATATGAAAAAATATATATGAAAAAATATATATGAAAAAATATATATGAAAAAATAAATATGAAAAAATAACAAAATTGAGGGTGATAATAAAAac|N3550IYEKIYMKKYI*KNIYEKINMKK*QN*G***K?|PF3D7_1025500|protein_coding|CODING|PF3D7_1025500.1|5)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:05,938 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1135242. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gaAGATAAATGATGAt|D305EDK**?|MPP10|protein_coding|CODING|PF3D7_1027100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,028 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1401951. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/AGATgat|D920R*?|GLURP|protein_coding|CODING|PF3D7_1035300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,066 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1520788. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||att/atATATATATATATATATATATATATATATTGAAGAGGAAAAATTGACTGAAGAGGAAAAATTGACt|I552IYIYIYIYIY*RGKID*RGKID?|Pf11-1|protein_coding|CODING|PF3D7_1038400.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,174 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:192450. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||tat/taATATGAGAACAATAAAATTTATATATATATATATATATATATATATAAt|Y592*YENNKIYIYIYIYIYN|PF3D7_1104300|protein_coding|CODING|PF3D7_1104300.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,537 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1234262. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aaa/aaTTAa|K710N*|PF3D7_1131900|protein_coding|CODING|PF3D7_1131900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,638 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1529369. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/aATATGAACAATGTGAATAGTATGAACAATGATGAATAgt|S633NMNNVNSMNNDE*?|PF3D7_1138700|protein_coding|CODING|PF3D7_1138700.1|4)\" \r\n",
        "WARN  16:34:06,643 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1542865. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||caa/cAAaa|Q688Q?|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,644 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1542894. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaatacaaa/|KYK698|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,653 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1571787. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F729|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,654 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1572062. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaatat/|KY820|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,671 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1624734. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gaGGAGGAAAAATTGGAAGAAGATATATAAAAAAAGATAAAAATTGGAa|E1847EEEKLEEDI*KKIKIG?|PF3D7_1140700|protein_coding|CODING|PF3D7_1140700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,710 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1741827. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1143600|protein_coding|CODING|PF3D7_1143600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,768 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1903508. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gct/gGCCGATGAATTAAAAAAATct|A1369GR*IKKI?|MAEBL|protein_coding|CODING|PF3D7_1147800.1|2)\" \r\n",
        "WARN  16:34:06,773 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1917620. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||tgt/tgTGTt|C410CV|PF3D7_1148100|protein_coding|CODING|PF3D7_1148100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,775 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920279. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||cct/Gcct|P5A?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,775 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920284. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|tAa/tGa|*6|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,775 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920294. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||ata/ATATATata|I10IYI|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,776 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920402. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atA/atT|I45|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,776 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920405. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|taG/taA|*46|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,776 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920462. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gta/Tgta|V66C?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,777 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922738. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gta/|V28|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,778 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922759. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gtaaaa/|VK35|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,778 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922762. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aTa|K36I|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,778 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922780. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gta/|V42|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,779 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922782. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|Aaa/Taa|K43*|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,779 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922801. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tta/ttATATCTTATTTGAGGTATGTATACTCATATCTTATTTGAGGTATGTATACATATATCTTATTTGAGGTATGTATACTCAa|L49LYLI*GMYTHILFEVCIHISYLRYVYS?|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,779 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922820. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|gtA/gtT|V55|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,779 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922839. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION(HIGH||ttaaaa/taa|LK62*|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,780 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922849. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||aagtaa/|K*65|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,780 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922858. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gttaaaaaa/|VKK68|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,780 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922877. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaaaaa/|KK75|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,781 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922896. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ttt/Gtt|F81V|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,781 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922897. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tttaaaaaa/|FKK81|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,781 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1922924. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N90S|PF3D7_1148500|protein_coding|CODING|PF3D7_1148500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,783 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1927580. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tTt/tAt|F529Y|PF3D7_1148600|protein_coding|CODING|PF3D7_1148600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,784 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1929744. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/aTta|I271I?|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,784 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1929748. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|tTg/tAg|L272*|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,785 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1929749. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|ttG/ttT|L272F|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,785 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1929751. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tGt|Y273C|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,786 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1931289. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||att/|I786|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,786 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1932043. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTt/aCt|I1037T|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,787 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1932676. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N1248S|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,787 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1932808. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCc/aGc|T1292S|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,787 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1932993. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||cat/cATat|H1354H?|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,802 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1979392. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gta/gtGGAAGAAAATGTGGAAGAAATGTGa|V954VEENVEEM*|PF3D7_1149200|protein_coding|CODING|PF3D7_1149200.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,889 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190964. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/|Y717|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,889 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190973. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/aAAga|R714K?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,890 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:191119. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aat/aaAAATAt|N665KN?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:06,890 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:192538. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||atg/atATATATATATATATg|M192IYIYI?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:06,985 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:468533. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gta/gtaTGACA|V258V*?|PF3D7_1210400|protein_coding|CODING|PF3D7_1210400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,104 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:826548. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gatTGATGATAATAATAAAAAGTATGATGAT|D388D*****KV**?|PF3D7_1220700|protein_coding|CODING|PF3D7_1220700.1|1)\" \r\n",
        "WARN  16:34:07,110 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:842412. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tca/tAAAAca|S497*N?|SET10|protein_coding|CODING|PF3D7_1221000.1|1)\" \r\n",
        "WARN  16:34:07,110 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:842420. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||ggt/ggATAAAACATCAACAAt|G494G*NINN|SET10|protein_coding|CODING|PF3D7_1221000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,208 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1149682. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tta/TTAGAAAAGGATCGATTGGATAGGTTAGAAAAGGAACGATTGGTATTATATTATATTATATTATATTAATAGAtta|L525LEKDRLDRLEKERLVLYYIILY**I?|NEK1|protein_coding|CODING|PF3D7_1228300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,224 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1197279. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtg/gtgGATGATTATGTGAATGATTATGTGGATGATTATGTGAATGATTATGTGGATTGATTATGTG|V645VDDYVNDYVDDYVNDYVD*LC?|MRP2|protein_coding|CODING|PF3D7_1229100.1|1)\" \r\n",
        "WARN  16:34:07,227 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1204433. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||aat/ATTaat|N285IN|PF3D7_1229200|protein_coding|CODING|PF3D7_1229200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,290 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1393519. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tca/tGAca|S2605*?|AARP1|protein_coding|CODING|PF3D7_1233600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,318 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1477116. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gatAATGATTGATGATAATGATGATAATGATAATGAT|D966DND*********?|PF3D7_1235400|protein_coding|CODING|PF3D7_1235400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,439 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1835511. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ggg/gggAGGAACAAAAAATGAGG|G1525GRNKK*?|DOC2|protein_coding|CODING|PF3D7_1243900.1|4)\" \r\n",
        "WARN  16:34:07,466 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1907463. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tCt/tTt|S140F|PF3D7_1245700|protein_coding|CODING|PF3D7_1245700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:07,466 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1907468. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ctt/Ttt|L142F|PF3D7_1245700|protein_coding|CODING|PF3D7_1245700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:07,467 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1907792. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ata/aTATAAATATATATATCta|I250I*IYIS?|PF3D7_1245700|protein_coding|CODING|PF3D7_1245700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,498 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1996086. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gaa/gAAGTTTTTTTTTaa|E438EVFF*|PF3D7_1248700|protein_coding|CODING|PF3D7_1248700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,662 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:234446. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1304400|protein_coding|CODING|PF3D7_1304400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,777 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:610051. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*753|PF3D7_1314100|protein_coding|CODING|PF3D7_1314100.1|1)\" \r\n",
        "WARN  16:34:07,794 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:661261. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|MYB1|protein_coding|CODING|PF3D7_1315800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,806 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:699921. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1316800|protein_coding|CODING|PF3D7_1316800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:07,926 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1074404. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tTATAAAGAGGAAGATAat|Y1087L*RGR*?|PF3D7_1325800|protein_coding|CODING|PF3D7_1325800.1|1)\" \r\n",
        "WARN  16:34:07,951 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1148969. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||att/atAATGATATGGATAATATGAATGATATGGATAATATGAATGATATGGATAATATGAAt|I1084IMIWII*MIWII*MIWII*?|PF3D7_1327300|protein_coding|CODING|PF3D7_1327300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,037 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1401731. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gaAAAGAAAAAGAAATGAACAAAAAGAAACAGAAGAt|D182EKKKK*TKRNRR?|MSRP5|protein_coding|CODING|PF3D7_1334300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,130 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1670955. Error was: \"START_LOST+CODON_INSERTION is not a recognized effect type\". Field was: \"START_LOST+CODON_INSERTION(HIGH||atg/aCAAAAAAAAAAAtg|M1TKKKM|SPECT1|protein_coding|CODING|PF3D7_1342500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,425 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2562595. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||cat/cATATATATATATATATATATat|H160HIYIYIY?|PF3D7_1363900|protein_coding|CODING|PF3D7_1363900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,426 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2562845. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/agAa|R243R?|PF3D7_1363900|protein_coding|CODING|PF3D7_1363900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,459 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2656398. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatAATAATAATAATGATAGTAAATAATAAT|N889NNNNNDSK**?|Rhop148|protein_coding|CODING|PF3D7_1366400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,486 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2725946. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|TIM9|protein_coding|CODING|PF3D7_1368600.1|1)\" \r\n",
        "WARN  16:34:08,507 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794272. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gcc/|A13|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,508 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794278. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ttG/ttA|L14|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,508 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794306. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tATATATATATAAAAATATGAAAATATGAat|Y24YIYIKI*KYE?|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,508 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794436. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/taTt|Y67Y?|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,509 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794463. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||agt/|S76|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,509 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794758. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/|I175|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,509 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2794812. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaG/aaA|K192|PF3D7_1370500|protein_coding|CODING|PF3D7_1370500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,510 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796744. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aaAAAATATATATATATATAAGTa|K209KKYIYI*V|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,511 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796751. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/AAAAAAAaaa|K212KKK?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,511 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796765. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||gta/gtAAAAAAa|V216VKK|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,511 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796780. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||gta/gtAAAAAAa|V221VKK|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,512 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796795. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||gta/gtAAAAAAa|V226VKK|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,512 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796802. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/|K229|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,512 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796818. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/aAAAAAAAAAGTaa|K234KKKS?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,512 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796838. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aaT/aaG|N240K|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,513 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796843. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/aaAGTTa|K242KV?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,513 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796853. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/Taaa|K246*?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,513 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796861. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/aaGTa|K248K?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,513 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796866. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/aGTaa|K250S?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,514 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796878. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/aGTaa|K254S?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,514 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796884. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aaa/Gaa|K256E|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,514 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796885. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aTa|K256I|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,514 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796903. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/aaAGTTa|K262KV?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,515 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796922. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/GTaaa|K269V?|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,515 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796929. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*271|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,515 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796967. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaA/aaG|K283|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,515 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796968. Error was: \"STOP_LOST+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_LOST+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||taa/tGTAaa|*284CK|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,516 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796987. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aGa|K290R|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,516 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796988. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aaA/aaT|K290N|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,516 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796994. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaA/aaG|K292|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,517 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796995. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|Aaa/Taa|K293*|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,517 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2796998. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaaagt/|KS294|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,517 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2797014. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gta/|V299|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,518 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2797021. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaA/aaG|K301|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,518 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2797022. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|Aaa/Taa|K302*|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,518 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2797028. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aaa/Gaa|K304E|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,518 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2797029. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aTa|K304I|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,519 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2797042. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaA/aaG|K308|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,519 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2797043. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|Aaa/Taa|K309*|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,519 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2797060. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaA/aaG|K314|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,519 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2797061. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|Aaa/Taa|K315*|PF3D7_1370800|protein_coding|CODING|PF3D7_1370800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,521 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2800979. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tct/Gct|S326A|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,521 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2800980. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tct/|S326|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,522 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2800988. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Gca/Aca|A329T|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,522 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2801021. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|Aag/Tag|K340*|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,522 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2801038. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|cgG/cgA|R345|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,522 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2801041. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|acT/acA|T346|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,523 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2801058. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gCc/gTc|A352V|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,523 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2801063. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tct/Act|S354T|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,523 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2801064. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tCt/tAt|S354Y|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,523 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2801067. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTt/aCt|I355T|PF3D7_1371000|protein_coding|CODING|PF3D7_1371000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,525 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2805302. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F787|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,526 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806668. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_DELETION(MODERATE||ttatatata/tta|LYI1242L|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,526 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806761. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|Taa/Caa|*1273Q|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,526 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806764. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Gcc/Acc|A1274T|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,527 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806798. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gAc/gGc|D1285G|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,527 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806815. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tca/Cca|S1291P|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,527 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806821. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ccc/Acc|P1293T|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,527 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806826. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|atA/atG|I1294M|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,528 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806834. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aGa/aAa|R1297K|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,528 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806840. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tTt/tAt|F1299Y|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,528 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806841. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ttT/ttC|F1299|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,528 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806845. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Att/Ttt|I1301F|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,528 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806847. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atT/atC|I1301|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,529 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2806850. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|gtA/gtT|V1302|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:08,529 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2807006. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/ATata|I1355I?|PF3D7_1371300|protein_coding|CODING|PF3D7_1371300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,599 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:102121. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/ACATATATATATATATATATTTAAAGATGgaa|E426TYIYIYI*RW?|PF3D7_1402800|protein_coding|CODING|PF3D7_1402800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,682 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:354410. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/tgATAATTAt|C244**L?|PF3D7_1409000|protein_coding|CODING|PF3D7_1409000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,734 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:518192. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|NMT|protein_coding|CODING|PF3D7_1412800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,755 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:583273. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||cac/cacTAATATCACCATAATCATCACCATAATCATCACCATAAT|H1244H*YHHNHHHNHHHN|ABCk2|protein_coding|CODING|PF3D7_1414500.1|4)\" \r\n",
        "WARN  16:34:08,779 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:652903. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaTGCACCTACAAATATATTATATATTATATATTATATATGTTTTTTTTAAAa|K1481NAPTNILYIIYYICFF*?|MCA3|protein_coding|CODING|PF3D7_1416200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,815 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:766156. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||ata/ataCAATGAAGGTGAATGCAAAATAATAATAATATA|I2417IQ*R*MQNNNNI|LISP1|protein_coding|CODING|PF3D7_1418100.1|1)\" \r\n",
        "WARN  16:34:08,821 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:780077. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ggC/ggT|G24|PF3D7_1418600|protein_coding|CODING|PF3D7_1418600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,863 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:904687. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/AATAATAATGATGATAATAATAATGATGATTAaat|N1435NNNDDNNNDD*?|PF3D7_1422400|protein_coding|CODING|PF3D7_1422400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:08,967 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1228016. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAATTATGGTGACAACAATATAAATGGTGACAACAtg|M1KIMVTTI*MVTT?|PF3D7_1431200|protein_coding|CODING|PF3D7_1431200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:09,010 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1356019. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaTTAAa|K787N*?|PF3D7_1433800|protein_coding|CODING|PF3D7_1433800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:09,109 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1630997. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATATAAATAATGACAat|N124IYK**Q?|PF3D7_1440000|protein_coding|CODING|PF3D7_1440000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:09,195 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1880474. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaaTATATATATATATATATATATATATATATAAATATAATATAATGTATATAATATAATGTATATAATATAATGTAATATATATATATATATATATAATAATGATATACTTAAA|K701KYIYIYIYIYINII*CI*YNVYNIM*YIYIYI***YT*?|PF3D7_1445800|protein_coding|CODING|PF3D7_1445800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:09,232 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1989546. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aTTAAATGAGAACACAACAACAACAACAAaa|K2010IK*EHNNNNK?|PF3D7_1448500|protein_coding|CODING|PF3D7_1448500.1|1)\" \r\n",
        "WARN  16:34:09,244 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2018824. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aaa/aaTTCATGa|K440NS*|PF3D7_1449300|protein_coding|CODING|PF3D7_1449300.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:09,403 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2483285. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gaa/gATAAATGATaa|E171DK**|PF3D7_1461000|protein_coding|CODING|PF3D7_1461000.1|1)\" \r\n",
        "WARN  16:34:09,404 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2483291. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tAat|Y173*?|PF3D7_1461000|protein_coding|CODING|PF3D7_1461000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:09,415 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2510893. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tATAATAGGTAGATAATAat|Y2249YNR*II?|PF3D7_1461800|protein_coding|CODING|PF3D7_1461800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:09,601 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3050549. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aATAATAATAATAATTGATAATGATAATGat|N1644NNNNN*****?|PF3D7_1474400|protein_coding|CODING|PF3D7_1474400.1|12)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:09,667 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248049. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Att/Gtt|I340V|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,667 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248057. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F343|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,667 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248063. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|tgA/tgT|*344C|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,668 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248065. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tTt|Y345F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,668 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248067. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cat/Aat|H346N|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,668 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248074. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aGa|K348R|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,669 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248077. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aag/|K349|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,669 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248080. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTa/aAa|I350K|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,669 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248098. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tCt/tTt|S356F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,669 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248104. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tTt|Y358F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,670 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248109. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aca/Gca|T360A|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,670 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248110. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCa/aTa|T360I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,670 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248118. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/|I363|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,670 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248120. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atA/atC|I363|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,671 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248125. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|tAa/tTa|*365L|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,671 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248126. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|taA/taT|*365Y|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,672 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248133. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/tTTGTTTGtt|F368FVC?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,672 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248165. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|caC/caT|H378|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,672 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248246. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/Tttt|F406F?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,673 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248252. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/TTtaa|*408F?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,673 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248257. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tGt|Y409C|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,673 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248260. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aTt|N410I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,674 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248263. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCt/aTt|T411I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:09,674 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248267. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taT/taC|Y412|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,674 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248269. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCa/aTa|T413I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,675 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248287. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tGt/tTt|C419F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,675 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248288. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tgT/tgG|C419W|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,675 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248293. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tTt/tAt|F421Y|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,676 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248294. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|ttT/ttA|F421L|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,676 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248295. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*422|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,676 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248303. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaG/aaA|K424|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,677 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248309. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaA/aaG|K426|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:09,677 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248314. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||acg/acAg|T428T?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "INFO  16:34:09,694 VariantAnnotator - Processed 15593 loci.\r\n",
        " \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:09,747 ProgressMeter -            done   2.3332831E7     9.0 s       0.0 s      100.0%     9.0 s       0.0 s \r\n",
        "INFO  16:34:09,747 ProgressMeter - Total runtime 9.25 secs, 0.15 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:11,045 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "KH02\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:27,691 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:27,692 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:34:27,692 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:34:27,693 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n",
        "INFO  16:34:27,696 HelpFormatter - Program Args: -T LeftAlignAndTrimVariants -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -V /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.sorted.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf \r\n",
        "INFO  16:34:27,700 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:34:27,701 HelpFormatter - Date/Time: 2015/10/02 16:34:27 \r\n",
        "INFO  16:34:27,701 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:34:27,701 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:28,040 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:28,088 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 1000 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:28,425 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.sorted.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:28,520 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:34:28,521 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:34:28,521 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:34:28,521 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:34:28,522 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "6619 variants were aligned\r\n",
        "INFO  16:34:29,576 ProgressMeter -            done     44936.0     1.0 s      23.0 s       99.8%     1.0 s       0.0 s \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:29,577 ProgressMeter - Total runtime 1.06 secs, 0.02 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:30,883 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.000\tSnpEff version SnpEff 4.1g (build 2015-05-17), by Pablo Cingolani\r\n",
        "00:00:00.010\tCommand: 'ann'\r\n",
        "00:00:00.025\tReading configuration file 'snpEff.config'. Genome: 'Pf3D7july2015'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.508\tdone\r\n",
        "00:00:00.509\tReading database for genome version 'Pf3D7july2015' from file '/Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt/snpeff/snpEff/./data/Pf3D7july2015/snpEffectPredictor.bin' (this might take a while)\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.315\tdone\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.325\tBuilding interval forest\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.693\tdone.\r\n",
        "00:00:01.693\tGenome stats :\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "#-----------------------------------------------\r\n",
        "# Genome name                : 'Plasmodium_falciparum'\r\n",
        "# Genome version             : 'Pf3D7july2015'\r\n",
        "# Has protein coding info    : true\r\n",
        "# Genes                      : 5645\r\n",
        "# Protein coding genes       : 5492\r\n",
        "#-----------------------------------------------\r\n",
        "# Transcripts                : 5886\r\n",
        "# Avg. transcripts per gene  : 1.04\r\n",
        "#-----------------------------------------------\r\n",
        "# Checked transcripts        : \r\n",
        "#               AA sequences :      0 ( 0.00% )\r\n",
        "#              DNA sequences :      0 ( 0.00% )\r\n",
        "#-----------------------------------------------\r\n",
        "# Protein coding transcripts : 5580\r\n",
        "#              Length errors :    126 ( 2.26% )\r\n",
        "#  STOP codons in CDS errors :    145 ( 2.60% )\r\n",
        "#         START codon errors :    200 ( 3.58% )\r\n",
        "#        STOP codon warnings :    205 ( 3.67% )\r\n",
        "#              UTR sequences :      0 ( 0.00% )\r\n",
        "#               Total Errors :    212 ( 3.80% )\r\n",
        "# WARNING                    : No protein coding transcript has UTR\r\n",
        "#-----------------------------------------------\r\n",
        "# Cds                        : 14366\r\n",
        "# Exons                      : 14366\r\n",
        "# Exons with sequence        : 14366\r\n",
        "# Exons without sequence     : 0\r\n",
        "# Avg. exons per transcript  : 2.44\r\n",
        "#-----------------------------------------------\r\n",
        "# Number of chromosomes      : 41\r\n",
        "# Chromosomes                : Format 'chromo_name size codon_table'\r\n",
        "#\t\t'Pf3D7_14_v3'\t3291936\tStandard\r\n",
        "#\t\t'Pf3D7_13_v3'\t2925236\tStandard\r\n",
        "#\t\t'Pf3D7_12_v3'\t2271494\tStandard\r\n",
        "#\t\t'Pf3D7_11_v3'\t2038340\tStandard\r\n",
        "#\t\t'Pf3D7_10_v3'\t1687656\tStandard\r\n",
        "#\t\t'Pf3D7_09_v3'\t1541735\tStandard\r\n",
        "#\t\t'Pf3D7_08_v3'\t1472805\tStandard\r\n",
        "#\t\t'Pf3D7_07_v3'\t1445207\tStandard\r\n",
        "#\t\t'Pf3D7_06_v3'\t1418242\tStandard\r\n",
        "#\t\t'Pf3D7_05_v3'\t1343557\tStandard\r\n",
        "#\t\t'Pf3D7_04_v3'\t1200490\tStandard\r\n",
        "#\t\t'Pf3D7_03_v3'\t1067971\tStandard\r\n",
        "#\t\t'Pf3D7_02_v3'\t947102\tStandard\r\n",
        "#\t\t'Pf3D7_01_v3'\t640851\tStandard\r\n",
        "#\t\t'Pf_M76611'\t5967\tProtozoan_Mitochondrial\r\n",
        "#\t\t'mal_mito_1'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_2'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_3'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_RNA19:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_SSUB:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_10:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_15:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_16:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_17:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_1:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_20:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_21:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_9:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUC:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUG:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA11:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA12:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA14:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA18:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA22:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA4:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA5:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA6:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA7:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_SSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#-----------------------------------------------\r\n",
        "\r\n",
        "00:00:02.349\tPredicting variants\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 22\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 23\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 24\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 25\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 26\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 27\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 28\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 29\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 30\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf', line 31\r\n",
        "00:00:02.386\tCreating sequences from exon information 'Pf3D7_01_v3'\r\n",
        "00:00:02.410\tBuilding sequence tree for chromosome 'Pf3D7_01_v3'\r\n",
        "00:00:02.413\tDone. Loaded 351 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.571\tCreating sequences from exon information 'Pf3D7_02_v3'\r\n",
        "00:00:02.585\tBuilding sequence tree for chromosome 'Pf3D7_02_v3'\r\n",
        "00:00:02.589\tDone. Loaded 558 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.734\tCreating sequences from exon information 'Pf3D7_03_v3'\r\n",
        "00:00:02.746\tBuilding sequence tree for chromosome 'Pf3D7_03_v3'\r\n",
        "00:00:02.747\tDone. Loaded 631 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.913\tCreating sequences from exon information 'Pf3D7_04_v3'\r\n",
        "00:00:02.925\tBuilding sequence tree for chromosome 'Pf3D7_04_v3'\r\n",
        "00:00:02.926\tDone. Loaded 601 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.035\tCreating sequences from exon information 'Pf3D7_05_v3'\r\n",
        "00:00:03.048\tBuilding sequence tree for chromosome 'Pf3D7_05_v3'\r\n",
        "00:00:03.049\tDone. Loaded 776 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.162\tCreating sequences from exon information 'Pf3D7_06_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.178\tBuilding sequence tree for chromosome 'Pf3D7_06_v3'\r\n",
        "00:00:03.179\tDone. Loaded 831 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.286\tCreating sequences from exon information 'Pf3D7_07_v3'\r\n",
        "00:00:03.303\tBuilding sequence tree for chromosome 'Pf3D7_07_v3'\r\n",
        "00:00:03.304\tDone. Loaded 728 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.417\tCreating sequences from exon information 'Pf3D7_08_v3'\r\n",
        "00:00:03.431\tBuilding sequence tree for chromosome 'Pf3D7_08_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.433\tDone. Loaded 842 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.497\tCreating sequences from exon information 'Pf3D7_09_v3'\r\n",
        "00:00:03.513\tBuilding sequence tree for chromosome 'Pf3D7_09_v3'\r\n",
        "00:00:03.515\tDone. Loaded 976 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.571\tCreating sequences from exon information 'Pf3D7_10_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.588\tBuilding sequence tree for chromosome 'Pf3D7_10_v3'\r\n",
        "00:00:03.590\tDone. Loaded 1039 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.662\t\t10000 variants (10000 variants per second), 10000 VCF entries\r\n",
        "00:00:03.671\tCreating sequences from exon information 'Pf3D7_11_v3'\r\n",
        "00:00:03.691\tBuilding sequence tree for chromosome 'Pf3D7_11_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.693\tDone. Loaded 1315 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.809\tCreating sequences from exon information 'Pf3D7_12_v3'\r\n",
        "00:00:03.834\tBuilding sequence tree for chromosome 'Pf3D7_12_v3'\r\n",
        "00:00:03.836\tDone. Loaded 1333 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.927\tCreating sequences from exon information 'Pf3D7_13_v3'\r\n",
        "00:00:03.961\tBuilding sequence tree for chromosome 'Pf3D7_13_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.965\tDone. Loaded 1779 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.063\tCreating sequences from exon information 'Pf3D7_14_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.100\tBuilding sequence tree for chromosome 'Pf3D7_14_v3'\r\n",
        "00:00:04.105\tDone. Loaded 2069 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "ERRORS: Some errors were detected\r\n",
        "Error type\tNumber of errors\r\n",
        "ERROR_CHROMOSOME_NOT_FOUND\t10\r\n",
        "\r\n",
        "\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "WARNINGS: Some warning were detected\r\n",
        "Warning type\tNumber of warnings\r\n",
        "WARNING_REF_DOES_NOT_MATCH_GENOME\t25\r\n",
        "WARNING_TRANSCRIPT_INCOMPLETE\t182\r\n",
        "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\t1927\r\n",
        "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS&WARNING_REF_DOES_NOT_MATCH_GENOME\t1\r\n",
        "WARNING_TRANSCRIPT_NO_START_CODON\t83\r\n",
        "\r\n",
        "\r\n",
        "00:00:04.222\tCreating summary file: snpEff_summary.html\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:05.525\tCreating genes file: snpEff_genes.txt\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:06.187\tdone.\r\n",
        "00:00:06.188\tLogging\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:07.192\tChecking for updates...\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:40,817 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:34:40,819 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:34:40,819 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:34:40,819 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n",
        "INFO  16:34:40,822 HelpFormatter - Program Args: -T VariantAnnotator -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -A TandemRepeatAnnotator -A SnpEff --variant /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.leftaligned.vcf --snpEffFile /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.snpeff.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.snpeff_annotated.vcf \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:40,826 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:34:40,827 HelpFormatter - Date/Time: 2015/10/02 16:34:40 \r\n",
        "INFO  16:34:40,827 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:34:40,827 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:41,176 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:41,227 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 250 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:41,613 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/KH02.snpeff.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:41,726 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:34:41,727 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:34:41,727 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:34:41,727 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:34:41,728 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:41,792 SnpEff - Skipping malformed SnpEff effect field at M76611:74. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:41,869 SnpEff - Skipping malformed SnpEff effect field at M76611:5244. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:41,972 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:13289. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:41,980 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:15405. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:41,991 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18694. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:41,993 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18828. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:41,993 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18860. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:42,007 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23944. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:42,008 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23947. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:34:42,012 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:25869. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:42,124 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:192508. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gat/gaATGATAATAATGTGATAATAAAGGTGGt|D747E***CDNKGG|UBP1|protein_coding|CODING|PF3D7_0104300.1|1)\" \r\n",
        "WARN  16:34:42,142 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:219211. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F4|PF3D7_0105000|protein_coding|CODING|PF3D7_0105000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:42,235 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:349589. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaG/gaG|E140|PF3D7_0108500|protein_coding|CODING|PF3D7_0108500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:42,236 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:349601. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaA/gaA|E144|PF3D7_0108500|protein_coding|CODING|PF3D7_0108500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:42,240 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:354004. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaA/gaA|E736|PSOP24|protein_coding|CODING|PF3D7_0108700.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:42,246 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:361994. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aTAAATGGATAGTTATAATAAaa|K198ING*L**?|PF3D7_0108900|protein_coding|CODING|PF3D7_0108900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:42,264 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:393220. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttacacatatat/|LHIY87|PF3D7_0110300|protein_coding|CODING|PF3D7_0110300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,273 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:407049. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aag/ATATTTCCTTTTCTTTTTTTTTTTTTTTTTTATATATATATATATATATaag|K438IFPFLFFFFFYIYIYI*?|PIP5K|protein_coding|CODING|PF3D7_0110600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:42,303 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:474666. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aat/Tat|N310Y|PF3D7_0112300|protein_coding|CODING|PF3D7_0112300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,303 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:475318. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/|Y527|PF3D7_0112300|protein_coding|CODING|PF3D7_0112300.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,305 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478408. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gat/gaATt|D377E?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,305 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478417. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N380S|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,306 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478423. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCc/aTc|T382I|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,306 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478424. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|acC/acG|T382|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,307 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478793. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|agG/agA|R505|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,307 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478794. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|Tag/Aag|*506K|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,308 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478795. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|tAg/tCg|*506S|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,309 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478796. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|taG/taA|*506|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,309 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478799. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L508|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,310 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478809. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tta/Ata|L511I|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,310 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478810. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L511|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,311 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478812. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Att/Ttt|I512F|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,311 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478816. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAc/tTc|Y513F|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,312 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478817. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taC/taT|Y513|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,312 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478818. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ctc/Ttc|L514F|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,313 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478820. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ctC/ctT|L514|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,313 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478822. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/agAa|R515R?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,314 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:479254. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tca/tcAa|S659S?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,315 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:482617. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_0112800|protein_coding|CODING|PF3D7_0112800.1|1)\" \r\n",
        "WARN  16:34:42,319 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:487993. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gatAAGAGAAGAAGATGATGAT|D640DKRRR**?|GARP|protein_coding|CODING|PF3D7_0113000.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:42,478 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:169373. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gTAATGAGGATGTAGATATGGATGATGAAGat|D255VMRM*IWMMK?|PF3D7_0203700|protein_coding|CODING|PF3D7_0203700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:42,532 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:266354. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatTTTTAATGACAAGTAGTATGACAAATAAT|N331NF**QVV*QI?|PF3D7_0206500|protein_coding|CODING|PF3D7_0206500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:42,588 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:408276. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gaa/gaAAAATCCACATTTTATATATATCCACATTTTATATATAAGAAAAAAGa|E126EKSTFYIYPHFIYKKK?|PF3D7_0209900|protein_coding|CODING|PF3D7_0209900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:42,595 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:419653. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cct/ccAATATAAATAAAAATGATGAAATATAAATAAAAATGATGAAAAAt|P495PI*IKMMKYK*K**K?|PF3D7_0210200|protein_coding|CODING|PF3D7_0210200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:42,744 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:659340. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/TATATATAAAGGATGATGAAATGGAAAATAAaaa|K236YI*RMMKWKIK?|PF3D7_0216000|protein_coding|CODING|PF3D7_0216000.1|1)\" \r\n",
        "WARN  16:34:42,771 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:696844. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N906|PF3D7_0216800|protein_coding|CODING|PF3D7_0216800.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,005 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:133485. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aat/|N6|PF3D7_0302400|protein_coding|CODING|PF3D7_0302400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:43,016 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:137422. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agg/Tagg|R527*?|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|5)\" \r\n",
        "WARN  16:34:43,020 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:137817. Error was: \"FRAME_SHIFT+SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"FRAME_SHIFT+SPLICE_SITE_DONOR+INTRON(HIGH||aaagaa/|KE615|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|6)\" \r\n",
        "WARN  16:34:43,027 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:139770. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Gct/Tct|A1122S|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:43,028 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:139771. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gCt/gCt|A1122|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:43,028 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:139773. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aat/Cat|N1123H|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:43,029 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:139775. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N1123|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:43,029 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:139776. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Ggt/Ggt|G1124|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:43,030 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:139795. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gGt/gCt|G1130A|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:43,030 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:139795. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gGt/gGt|G1130|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:43,031 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:139796. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||ggT/ggT|G1130|CLAG3.1|protein_coding|CODING|PF3D7_0302500.1|9|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,223 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:657246. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_0316200|protein_coding|CODING|PF3D7_0316200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,241 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:707278. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaCCAAAATAAAAATATATATAAATAAAAAATATATATAAATAAAt|N817NQNKNIYK*KIYINK?|PF3D7_0317300|protein_coding|CODING|PF3D7_0317300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,322 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:907920. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N733|PF3D7_0321700|protein_coding|CODING|PF3D7_0321700.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:43,323 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:909210. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taGTCATAATTTt|Y303*S*F?|PF3D7_0321700|protein_coding|CODING|PF3D7_0321700.1|1)\" \r\n",
        "WARN  16:34:43,324 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:909246. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taAt|Y291*?|PF3D7_0321700|protein_coding|CODING|PF3D7_0321700.1|1)\" \r\n",
        "WARN  16:34:43,337 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:943829. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aAATaa|K429K*|PF3D7_0322400|protein_coding|CODING|PF3D7_0322400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,494 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:227462. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/tgAt|C1424*?|ApiAP2|protein_coding|CODING|PF3D7_0404100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,524 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:309982. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATGATGATAATAATAAt|N1004K*****?|PF3D7_0405700|protein_coding|CODING|PF3D7_0405700.1|1)\" \r\n",
        "WARN  16:34:43,528 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:319177. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|ASP|protein_coding|CODING|PF3D7_0405900.1|1)\" \r\n",
        "WARN  16:34:43,540 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:352982. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_0406800|protein_coding|CODING|PF3D7_0406800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,598 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:514525. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATGATGATGTCACGATTAAACAAAAt|N848K**CHD*TK?|PF3D7_0411400|protein_coding|CODING|PF3D7_0411400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,682 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:695682. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gATCCTTACTATTATCATTTTTTATGTCCTTACTATTATCATTTTTTATGTCCTTACTATTATCATTTTTTATGATCAAAATAat|D593DPYYYHFLCPYYYHFLCPYYYHFL*SK*?|PF3D7_0415800|protein_coding|CODING|PF3D7_0415800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,807 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:898514. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatTAATAAT|N2070N**?|PF3D7_0420000|protein_coding|CODING|PF3D7_0420000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,860 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979922. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ccT/ccC|P3|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:43,861 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979935. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ggt/Cgt|G8R|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:43,861 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979996. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cAt/cCt|H28P|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:43,862 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979998. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aag/Gag|K29E|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:43,862 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:980013. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tcg/Ccg|S34P|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:43,863 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:980023. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|tAa/tGa|*37|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:43,863 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:980041. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gCg/gTg|A43V|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:43,888 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:1014701. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaAAAATAATGATGTAAAt|N849KK**CK?|PF3D7_0422200|protein_coding|CODING|PF3D7_0422200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,122 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:183875. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gag/gaTTAAAGGAGGAGg|E1152D*RRR|CEP120|protein_coding|CODING|PF3D7_0504700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,205 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:369479. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aAATGaa|K820K*?|PF3D7_0508900|protein_coding|CODING|PF3D7_0508900.1|1)\" \r\n",
        "WARN  16:34:44,221 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:410656. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gaTGATGATGATGATAGATGATGGt|D130DDDDDR*W?|PI4K|protein_coding|CODING|PF3D7_0509800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,286 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:592164. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATATATATGGTCATAATAAAAATGAGGACAat|N1167IYIWS**K*GQ?|UvrD|protein_coding|CODING|PF3D7_0514100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,363 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:805618. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATGATATAAATAACAAt|N1170K*YK*Q?|CCR4|protein_coding|CODING|PF3D7_0519500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,475 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1134853. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_0527200|protein_coding|CODING|PF3D7_0527200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,523 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1253061. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/tgTAATAATAATAATAATAATAAATTGTAATAATAATAATAAt|C708CNNNNNNKL*****?|PF3D7_0530700|protein_coding|CODING|PF3D7_0530700.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,612 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:190886. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/AAATAATGTAAATAATGTAaat|N626K*CK*CK?|PF3D7_0604500|protein_coding|CODING|PF3D7_0604500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,767 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:629340. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION(HIGH||tgtgatgaaaaagaa/tga|CDEKE322*|GWT1|protein_coding|CODING|PF3D7_0615300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,770 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:634992. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gTAGTATGat|D1661VV*?|PF3D7_0615400|protein_coding|CODING|PF3D7_0615400.1|2)\" \r\n",
        "WARN  16:34:44,791 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:687641. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tct/tcTATATATATATATATATATATATAGt|S967SIYIYIYI*?|TLP|protein_coding|CODING|PF3D7_0616500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:44,882 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:924026. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*1448|PF3D7_0622800|protein_coding|CODING|PF3D7_0622800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,099 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:155939. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaCTAAAAAAAAAAAAAAATGATAATAATAATTATGATAATAATAAt|N3486N*KKKKMIIIIMIII?|PF3D7_0703900|protein_coding|CODING|PF3D7_0703900.1|1)\" \r\n",
        "WARN  16:34:45,108 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:185178. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/aATGAACAAAATAAAGATAATAAACAAAATTTTTTTTTTTTTTAGTTATTATTATCATAATTATTATTATCATATCATCat|N1715NEQNKDNKQNFFFF*LLLS*LLLSYHH|PF3D7_0704100|protein_coding|CODING|PF3D7_0704100.1|1)\" \r\n",
        "WARN  16:34:45,111 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:191671. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/aatAATAATATTAAATTTTGTTTATTATCTTTATTTTGTTCATGATGATATGATAATAATAATTATGATAATAATAACTAAAAAAAAAAAAAAAAT|N1073NNNIKFCLLSLFCS**YDNNNYDNNN*KKKKN|PF3D7_0704200|protein_coding|CODING|PF3D7_0704200.1|1)\" \r\n",
        "WARN  16:34:45,111 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:192811. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Ata/Ata|I694|PF3D7_0704200|protein_coding|CODING|PF3D7_0704200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,206 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:469937. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaTTAAAAAGt|N2278N*K?|PF3D7_0710200|protein_coding|CODING|PF3D7_0710200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,280 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:678685. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gca/gcaCCCACAAATGTTATGAAACAATACATCAAATAGTTTAATGAATAATGCA|A1332APTNVMKQYIK*FNE*C?|PF3D7_0715200|protein_coding|CODING|PF3D7_0715200.1|1)\" \r\n",
        "WARN  16:34:45,292 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:713061. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gAa/gAa|E198|PF3D7_0716200|protein_coding|CODING|PF3D7_0716200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:45,292 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:713097. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aAt/aAt|N210|PF3D7_0716200|protein_coding|CODING|PF3D7_0716200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:45,293 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:713157. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTt/aAt|I230N|PF3D7_0716200|protein_coding|CODING|PF3D7_0716200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:45,293 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:713169. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTt/aAt|I234N|PF3D7_0716200|protein_coding|CODING|PF3D7_0716200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,299 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:731125. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/|*1006|PF3D7_0716700|protein_coding|CODING|PF3D7_0716700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,387 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:923747. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gGt/gGt|G488|PF3D7_0721300|protein_coding|CODING|PF3D7_0721300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:45,400 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:947715. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/GAGAGATGACAACAAAATGgaa|E452ER*QQNG?|RALP1|protein_coding|CODING|PF3D7_0722200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,457 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1040585. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taAAAAAAAATAATGATGATATTCGAAATAATAACTAt|Y1760*KKIMMIFEIIT?|PF3D7_0724700|protein_coding|CODING|PF3D7_0724700.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,483 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1081613. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ttt/tttGCATTTTATTTCATTCTTTGACTTTATCGTT|F95FAFYFIL*LYR?|PF3D7_0725500|protein_coding|CODING|PF3D7_0725500.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,529 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1150960. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||ccG/ccG|P33|VPS53|protein_coding|CODING|PF3D7_0727000.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:45,530 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1150962. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Ccg/Ccg|P33|VPS53|protein_coding|CODING|PF3D7_0727000.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,579 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1222899. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||gat/gAAATAAATAAATAAATAAATATATATTATTat|D1492EINK*INIYYY|PF3D7_0728600|protein_coding|CODING|PF3D7_0728600.1|1)\" \r\n",
        "WARN  16:34:45,580 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1222980. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/aATGATGATTATAATGATGAAAATAAATAAATAAATAAATATATATTATTTGATAATGat|N1519NDDYNDENK*INKYILFDND|PF3D7_0728600|protein_coding|CODING|PF3D7_0728600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,637 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1327786. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taGt|Y769*?|PTP4|protein_coding|CODING|PF3D7_0730900.1|2)\" \r\n",
        "WARN  16:34:45,638 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1328116. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taGt|Y879*?|PTP4|protein_coding|CODING|PF3D7_0730900.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,708 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:95108. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F821|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:45,708 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:95149. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS&WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||tGt/tGt|C808|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS&WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:45,822 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:427538. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aGa/aAa|R33K|PF3D7_0808500|protein_coding|CODING|PF3D7_0808500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:45,822 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:427544. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aGt/aTt|S31I|PF3D7_0808500|protein_coding|CODING|PF3D7_0808500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:45,823 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:427562. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cGa/cAa|R25Q|PF3D7_0808500|protein_coding|CODING|PF3D7_0808500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:45,823 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:427599. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ctt/Ttt|L13F|PF3D7_0808500|protein_coding|CODING|PF3D7_0808500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:45,823 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:427605. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tgt/Ggt|C11G|PF3D7_0808500|protein_coding|CODING|PF3D7_0808500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:46,011 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:941842. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaA/aaA|K883|PF3D7_0820900|protein_coding|CODING|PF3D7_0820900.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:46,141 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1283721. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tttttttcttatacatagacttttattgttatacatagat/|FFSYT*TFIVIHR?930|PF3D7_0830100|protein_coding|CODING|PF3D7_0830100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:46,144 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1288705. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tatata/|YI1377|PF3D7_0830200|protein_coding|CODING|PF3D7_0830200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:46,321 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:285581. Error was: \"FRAME_SHIFT+SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"FRAME_SHIFT+SPLICE_SITE_DONOR+INTRON(HIGH||aat/|N140|ATG3|protein_coding|CODING|PF3D7_0905700.1|5)\" \r\n",
        "WARN  16:34:46,328 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:303869. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gTATAATAATATGAAGGGTAGAAATACAAAATGTGATAATATAAAAGat|D104V**YEG*KYKM**YKR?|RNaseII|protein_coding|CODING|PF3D7_0906000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:46,477 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:714246. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aga/aTATATATATATATATATTAAAATAAAAAATGGAGAGGGTAga|R415IYIYIY*NKKWRG*?|OGG1|protein_coding|CODING|PF3D7_0917100.1|1)\" \r\n",
        "WARN  16:34:46,495 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:761252. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/Tttt|F15F?|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:46,496 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762835. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tga/tgATATATATATATATATa|*542*YIYIY?|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:46,592 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1047792. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*1278|PF3D7_0926000|protein_coding|CODING|PF3D7_0926000.1|4)\" \r\n",
        "WARN  16:34:46,615 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1116001. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*682|FH|protein_coding|CODING|PF3D7_0927300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:46,702 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1370074. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aTTATATATATATATATATAaa|K655IIYIYI*?|VPS33|protein_coding|CODING|PF3D7_0935200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:46,991 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:710822. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaATGACGACATAAAAAATAAGGACGACATAAAAAATAAGGACGACATAAAAAATAAGGACGACATAAAAAATAAGTATATATATATATATATATATATa|K439K*RHKK*GRHKK*GRHKK*GRHKK*VYIYIYIY?|PF3D7_1017700|protein_coding|CODING|PF3D7_1017700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,085 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:995784. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aAATTATATATATaa|K424KLYI*|PF3D7_1023800|protein_coding|CODING|PF3D7_1023800.1|2)\" \r\n",
        "WARN  16:34:47,094 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1018974. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gaATGATAAACAAAGTAGTGAt|D514E**TK**?|PF3D7_1024100|protein_coding|CODING|PF3D7_1024100.1|1)\" \r\n",
        "WARN  16:34:47,111 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1068664. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||PF3D7_1025500|protein_coding|CODING|PF3D7_1025500.1|9)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,112 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1068666. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aTGTAaa|K5567M*?|PF3D7_1025500|protein_coding|CODING|PF3D7_1025500.1|9)\" \r\n",
        "WARN  16:34:47,117 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1082468. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ggg/ggATAGGCACACCAAAATg|G1443G*AHQN?|PF3D7_1025500|protein_coding|CODING|PF3D7_1025500.1|2)\" \r\n",
        "WARN  16:34:47,117 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1082471. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||ggt/ggATAAGCACACCAAAAt|G1442G*AHQN|PF3D7_1025500|protein_coding|CODING|PF3D7_1025500.1|2)\" \r\n",
        "WARN  16:34:47,147 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1170781. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aaa/aaTGTGTGa|K281NV*|PF3D7_1028400|protein_coding|CODING|PF3D7_1028400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,195 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1311410. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gatTGATGAAAAAAATGATGATGAAAAAAATGAT|D224D**KK***KK*?|PF3D7_1032700|protein_coding|CODING|PF3D7_1032700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,228 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1404421. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||tta/ttATAa|L76L*|MSP3|protein_coding|CODING|PF3D7_1035400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,355 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:110641. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Cgt/Cgt|R212|PF3D7_1102300|protein_coding|CODING|PF3D7_1102300.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,386 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:192450. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||tat/taATATGAGAACAATAAAAATTATGAGAACAATAAAAAt|Y592*YENNKNYENNKN|PF3D7_1104300|protein_coding|CODING|PF3D7_1104300.1|2)\" \r\n",
        "WARN  16:34:47,395 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:217098. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/ATTTaat|N759I*?|PF3D7_1104900|protein_coding|CODING|PF3D7_1104900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,470 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:443827. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|GS27|protein_coding|CODING|PF3D7_1111300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,614 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:873116. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||att/ATGTTGACGACGAAGATGATATGatt|I363MLTTKMI*?|CDPK6|protein_coding|CODING|PF3D7_1122800.1|1)\" \r\n",
        "WARN  16:34:47,620 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:891527. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gaC/gaG|D4088E|PF3D7_1122900|protein_coding|CODING|PF3D7_1122900.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:47,630 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:922248. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/AATGATGATAATAACAGCTTGTTGTTGTTGTTGTTGTTGTAAAAAAAATAAaat|N143NDDNNSLLLLLLL*KK*N|PF3D7_1123300|protein_coding|CODING|PF3D7_1123300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,833 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1541662. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/aTATGTATGTATGTATCta|I287ICMYVS?|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,833 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1542865. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||caa/cAaa|Q688Q?|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,834 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1542894. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaatacaaa/|KYK698|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,842 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1570488. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/|K296|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,843 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1571787. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F729|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,844 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1572602. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||attata/|II1000|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,849 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1589377. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtg/gtTGGATGATAATAATGTg|V886VG***C?|PF3D7_1139800|protein_coding|CODING|PF3D7_1139800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:47,948 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1903547. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gct/gGCCGATGAATTAAAAAAATct|A1382GR*IKKI?|MAEBL|protein_coding|CODING|PF3D7_1147800.1|2)\" \r\n",
        "WARN  16:34:47,948 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1903658. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aaa/aGCTGAAGAAAAGaa|K1419S*RKE|MAEBL|protein_coding|CODING|PF3D7_1147800.1|2)\" \r\n",
        "WARN  16:34:47,948 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1903664. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gct/gGCCGATGAATTAAAAAAATct|A1421GR*IKKI?|MAEBL|protein_coding|CODING|PF3D7_1147800.1|2)\" \r\n",
        "WARN  16:34:47,954 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1917647. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/ttTTATATATATAt|F419FYIY?|PF3D7_1148100|protein_coding|CODING|PF3D7_1148100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,956 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920294. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||ata/ATATATata|I10IYI|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,957 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920424. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tgc/Agc|C53S|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,957 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920431. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|tGa/tAa|*55|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,957 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920434. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/|Y56|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,958 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920462. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gta/Tgta|V66C?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,958 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920756. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tac/Atac|Y164I?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,958 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920809. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N181S|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,962 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1927664. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gTt/gCt|V557A|PF3D7_1148600|protein_coding|CODING|PF3D7_1148600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,962 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1927671. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ttA/ttG|L559|PF3D7_1148600|protein_coding|CODING|PF3D7_1148600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,963 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1927672. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aat/|N560|PF3D7_1148600|protein_coding|CODING|PF3D7_1148600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,963 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1927694. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aGt/aTt|S567I|PF3D7_1148600|protein_coding|CODING|PF3D7_1148600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,963 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1927700. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tGt|Y569C|PF3D7_1148600|protein_coding|CODING|PF3D7_1148600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,965 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1932043. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTt/aCt|I1037T|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,965 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1932676. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N1248S|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,966 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1932827. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atT/atA|I1298|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:47,966 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1933087. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAc/aGc|N1385S|PF3D7_1148640|protein_coding|CODING|PF3D7_1148640.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,047 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:149518. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||PF3D7_1202600|protein_coding|CODING|PF3D7_1202600.1|1)\" \r\n",
        "WARN  16:34:48,060 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190106. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/aAga|R1003K?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:48,061 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190948. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||atatatatatatatatatata/|IYIYIYI716|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:48,061 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190968. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aat/|N715|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:48,062 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190975. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaG/aaA|K713|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:48,062 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:191119. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aat/aaAATAAt|N665KI?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:48,063 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:192775. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttc/ttTc|F113F?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,096 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:292746. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gta/gtTTAAAGAa|V108V*R?|PF3D7_1206500|protein_coding|CODING|PF3D7_1206500.1|1)\" \r\n",
        "WARN  16:34:48,109 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:334030. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||agt/agATAATATGATGAt|S1259R*YDD|PF3D7_1207200|protein_coding|CODING|PF3D7_1207200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,133 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:408074. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gac/gaTATGTGTAAATAATAATGAc|D359DMCK***?|PF3D7_1208800|protein_coding|CODING|PF3D7_1208800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,230 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:710266. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gaATAATAATAATGATAATAATAATGAt|D1800E********?|PF3D7_1218200|protein_coding|CODING|PF3D7_1218200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,352 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1075403. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/agAAAAAAAAGATAATACATATAGTGATAATACATAAGAt|S982RKKR*YI***YIR?|PF3D7_1226500|protein_coding|CODING|PF3D7_1226500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,396 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1204404. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||caa/cAAaa|Q275Q?|PF3D7_1229200|protein_coding|CODING|PF3D7_1229200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,458 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1393519. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tca/tGAca|S2605*?|AARP1|protein_coding|CODING|PF3D7_1233600.1|1)\" \r\n",
        "WARN  16:34:48,497 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1514229. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gga/gCAAATATCATGTAga|G763ANIM*?|PF3D7_1236200|protein_coding|CODING|PF3D7_1236200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,497 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1514235. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ata/aAATGATCATGta|I761K*SC?|PF3D7_1236200|protein_coding|CODING|PF3D7_1236200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,602 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1835511. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ggg/gggAAGAACAAAAAATGAGGGAAGAACAAAAAATGAGG|G1525GKNKK*GKNKK*?|DOC2|protein_coding|CODING|PF3D7_1243900.1|4)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,798 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:166356. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/agtAAGATGATAATAAATTAAATAGT|S575SKMIIN*I?|PF3D7_1303400|protein_coding|CODING|PF3D7_1303400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,866 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:379414. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||act/actTTTAAGAAACAATATAAATAATATTACAAATGATGATATTA|T4172TFKKQYK*YYK**Y?|PF3D7_1308400|protein_coding|CODING|PF3D7_1308400.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:48,940 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:610051. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*753|PF3D7_1314100|protein_coding|CODING|PF3D7_1314100.1|1)\" \r\n",
        "WARN  16:34:48,956 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:661261. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|MYB1|protein_coding|CODING|PF3D7_1315800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,053 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:960644. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*116|PF3D7_1322800|protein_coding|CODING|PF3D7_1322800.1|2)\" \r\n",
        "WARN  16:34:49,071 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1006505. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatCGTTATCAATAATAAT|N4266NRYQ**?|PF3D7_1324300|protein_coding|CODING|PF3D7_1324300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,100 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1078592. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATATAAATAATAATGAACATAat|N1999IYK***T*?|PF3D7_1325900|protein_coding|CODING|PF3D7_1325900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,160 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1259502. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aat/aaCAATAATAATGGTAATAATAAAATATTATTATGGTAAATATAt|N120NNNNGNNKILLW*IY|PF3D7_1329600|protein_coding|CODING|PF3D7_1329600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,204 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1391352. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1334000|protein_coding|CODING|PF3D7_1334000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,242 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1506021. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/|*904|PF3D7_1337300|protein_coding|CODING|PF3D7_1337300.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,305 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1688971. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/agtTAATAATATCAATAGTAATAATATCAATAGT|S1666S**YQ***YQ*?|ApiAP2|protein_coding|CODING|PF3D7_1342900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,394 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1945815. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aca/acTAAATGACCAATa|T522TK*PI|PF3D7_1348400|protein_coding|CODING|PF3D7_1348400.1|1)\" \r\n",
        "WARN  16:34:49,395 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1945911. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ata/atGACCCAATAAATGACCCAACa|I490MTQ*MTQ?|PF3D7_1348400|protein_coding|CODING|PF3D7_1348400.1|1)\" \r\n",
        "WARN  16:34:49,395 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1946045. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||caa/caaTAAATGACCCAATAAATGACCCAATAAATGACCCAATAAATGACCCA|Q445Q*MTQ*MTQ*MTQ*MT?|PF3D7_1348400|protein_coding|CODING|PF3D7_1348400.1|1)\" \r\n",
        "WARN  16:34:49,395 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1946055. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||aca/acTAAATGACCAATa|T442TK*PI|PF3D7_1348400|protein_coding|CODING|PF3D7_1348400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,602 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2562713. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||cat/caTt|H199H?|PF3D7_1363900|protein_coding|CODING|PF3D7_1363900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,751 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:127020. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||PF3D7_1403400|protein_coding|CODING|PF3D7_1403400.1|4)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,792 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:259169. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aga/ATATATATATATATATAGGAGGTGGACGAGGCGGAGGAGGAGGTGGCaga|R47IYIYI*EVDEAEEEVA?|NOP1|protein_coding|CODING|PF3D7_1407100.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,876 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:518192. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAAAAtg|M1KK?|NMT|protein_coding|CODING|PF3D7_1412800.1|1)\" \r\n",
        "WARN  16:34:49,880 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:530305. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aag/aagAAAAAAAAAAGATGAAAAGAAG|K362KKKKR*KE?|PF3D7_1413200|protein_coding|CODING|PF3D7_1413200.1|4)\" \r\n",
        "WARN  16:34:49,919 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:652903. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaTGAACACCTACAAATGTAAAAGCACCTACAAATGTAAAa|K1481NEHLQM*KHLQM*?|MCA3|protein_coding|CODING|PF3D7_1416200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:49,999 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:903872. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aga/aATCATATATAATATga|R1163NHI*Y?|PF3D7_1422400|protein_coding|CODING|PF3D7_1422400.1|1)\" \r\n",
        "WARN  16:34:50,022 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:972858. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||ttg/|L1|PF3D7_1423900|protein_coding|CODING|PF3D7_1423900.1|1|WARNING_TRANSCRIPT_INCOMPLETE)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:50,022 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:973221. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/tTTaa|*2F?|PF3D7_1424000|protein_coding|CODING|PF3D7_1424000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:50,110 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1228016. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_1431200|protein_coding|CODING|PF3D7_1431200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:50,274 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1734643. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tct/ACAAAATAGTGACCATAACATTATGACAGAAGAATATGAGAAAAATAAGAATtct|S1454TK**P*HYDRRI*EK*EF?|TREP|protein_coding|CODING|PF3D7_1442600.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:50,397 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2110559. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aac/TAACAACAATaac|N563*QQ*?|Sir2B|protein_coding|CODING|PF3D7_1451400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:50,665 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2948655. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/TCGTGAGAACAATATGGTCagt|S999S*EQYGQ?|HDA1|protein_coding|CODING|PF3D7_1472200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:50,745 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3196978. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gCc/gCc|A1121|SURF14.1|protein_coding|CODING|PF3D7_1477600.1|3|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:34:50,762 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248049. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Att/Gtt|I340V|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,762 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248057. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F343|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,763 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248063. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|tgA/tgT|*344C|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,763 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248065. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tTt|Y345F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:34:50,763 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248067. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cat/Aat|H346N|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,764 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248074. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aGa|K348R|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,764 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248077. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aag/|K349|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,764 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248080. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTa/aAa|I350K|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,764 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248098. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tCt/tTt|S356F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,764 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248104. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tTt|Y358F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,765 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248109. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aca/Gca|T360A|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,765 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248110. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCa/aTa|T360I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,765 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248116. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tttataatataagttttgttttgttttaatataagaaaaaacacacgtcacagtgtttttataatataa/|FII*VLFCFNIRKNTRHSVFII*362|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,766 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248186. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L386|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,766 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248190. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tac/|Y387|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,766 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248323. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tgt/tgTt|C431C?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,766 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248329. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/taTTt|Y433Y?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,767 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248334. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ata/Gta|I435V|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,767 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248337. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ata/Tta|I436L|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,767 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248340. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ctt/Ttt|L437F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,767 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248344. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTa/aCa|I438T|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,768 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248346. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cat/Tat|H439Y|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,768 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248400. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gtt/|V457|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,768 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248407. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tattta/|YL459|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,768 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248414. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|atG/atA|M461I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,769 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248417. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|taT/taA|Y462*|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,769 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248420. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taT/taC|Y463|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,769 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248424. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cat/Tat|H465Y|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:34:50,769 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248426. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|caT/caC|H465|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "INFO  16:34:50,781 VariantAnnotator - Processed 16594 loci.\r\n",
        " \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:50,859 ProgressMeter -            done   2.3332831E7     9.0 s       0.0 s      100.0%     9.0 s       0.0 s \r\n",
        "INFO  16:34:50,860 ProgressMeter - Total runtime 9.13 secs, 0.15 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:34:52,176 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "GN01\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:09,411 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:35:09,412 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:35:09,412 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:35:09,413 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n",
        "INFO  16:35:09,416 HelpFormatter - Program Args: -T LeftAlignAndTrimVariants -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -V /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.sorted.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf \r\n",
        "INFO  16:35:09,421 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:35:09,421 HelpFormatter - Date/Time: 2015/10/02 16:35:09 \r\n",
        "INFO  16:35:09,421 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:35:09,421 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:09,763 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:09,810 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 1000 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:10,163 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.sorted.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:10,260 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:35:10,261 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:35:10,262 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:35:10,262 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:35:10,262 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "9987 variants were aligned\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:11,435 ProgressMeter -            done     46322.0     1.0 s      25.0 s       99.9%     1.0 s       0.0 s \r\n",
        "INFO  16:35:11,436 ProgressMeter - Total runtime 1.17 secs, 0.02 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:12,738 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.000\tSnpEff version SnpEff 4.1g (build 2015-05-17), by Pablo Cingolani\r\n",
        "00:00:00.010\tCommand: 'ann'\r\n",
        "00:00:00.031\tReading configuration file 'snpEff.config'. Genome: 'Pf3D7july2015'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:00.516\tdone\r\n",
        "00:00:00.516\tReading database for genome version 'Pf3D7july2015' from file '/Users/rpearson/src/github/malariagen/methods-dev/pf3k_techbm/opt/snpeff/snpEff/./data/Pf3D7july2015/snpEffectPredictor.bin' (this might take a while)\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.335\tdone\r\n",
        "00:00:01.345\tBuilding interval forest\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:01.700\tdone.\r\n",
        "00:00:01.700\tGenome stats :\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "#-----------------------------------------------\r\n",
        "# Genome name                : 'Plasmodium_falciparum'\r\n",
        "# Genome version             : 'Pf3D7july2015'\r\n",
        "# Has protein coding info    : true\r\n",
        "# Genes                      : 5645\r\n",
        "# Protein coding genes       : 5492\r\n",
        "#-----------------------------------------------\r\n",
        "# Transcripts                : 5886\r\n",
        "# Avg. transcripts per gene  : 1.04\r\n",
        "#-----------------------------------------------\r\n",
        "# Checked transcripts        : \r\n",
        "#               AA sequences :      0 ( 0.00% )\r\n",
        "#              DNA sequences :      0 ( 0.00% )\r\n",
        "#-----------------------------------------------\r\n",
        "# Protein coding transcripts : 5580\r\n",
        "#              Length errors :    126 ( 2.26% )\r\n",
        "#  STOP codons in CDS errors :    145 ( 2.60% )\r\n",
        "#         START codon errors :    200 ( 3.58% )\r\n",
        "#        STOP codon warnings :    205 ( 3.67% )\r\n",
        "#              UTR sequences :      0 ( 0.00% )\r\n",
        "#               Total Errors :    212 ( 3.80% )\r\n",
        "# WARNING                    : No protein coding transcript has UTR\r\n",
        "#-----------------------------------------------\r\n",
        "# Cds                        : 14366\r\n",
        "# Exons                      : 14366\r\n",
        "# Exons with sequence        : 14366\r\n",
        "# Exons without sequence     : 0\r\n",
        "# Avg. exons per transcript  : 2.44\r\n",
        "#-----------------------------------------------\r\n",
        "# Number of chromosomes      : 41\r\n",
        "# Chromosomes                : Format 'chromo_name size codon_table'\r\n",
        "#\t\t'Pf3D7_14_v3'\t3291936\tStandard\r\n",
        "#\t\t'Pf3D7_13_v3'\t2925236\tStandard\r\n",
        "#\t\t'Pf3D7_12_v3'\t2271494\tStandard\r\n",
        "#\t\t'Pf3D7_11_v3'\t2038340\tStandard\r\n",
        "#\t\t'Pf3D7_10_v3'\t1687656\tStandard\r\n",
        "#\t\t'Pf3D7_09_v3'\t1541735\tStandard\r\n",
        "#\t\t'Pf3D7_08_v3'\t1472805\tStandard\r\n",
        "#\t\t'Pf3D7_07_v3'\t1445207\tStandard\r\n",
        "#\t\t'Pf3D7_06_v3'\t1418242\tStandard\r\n",
        "#\t\t'Pf3D7_05_v3'\t1343557\tStandard\r\n",
        "#\t\t'Pf3D7_04_v3'\t1200490\tStandard\r\n",
        "#\t\t'Pf3D7_03_v3'\t1067971\tStandard\r\n",
        "#\t\t'Pf3D7_02_v3'\t947102\tStandard\r\n",
        "#\t\t'Pf3D7_01_v3'\t640851\tStandard\r\n",
        "#\t\t'Pf_M76611'\t5967\tProtozoan_Mitochondrial\r\n",
        "#\t\t'mal_mito_1'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_2'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_3'\t1"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\tVertebrate_Mitochondrial\r\n",
        "#\t\t'mal_mito_RNA19:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_SSUB:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_10:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_15:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_16:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_17:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_1:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_20:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_21:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_9:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUC:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_LSUG:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA11:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA12:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA14:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA18:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA22:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA4:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA5:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA6:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_RNA7:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#\t\t'malmito_rna_SSUF:rRNA'\t1\tVertebrate_Mitochondrial\r\n",
        "#-----------------------------------------------\r\n",
        "\r\n",
        "00:00:02.365\tPredicting variants\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 22\r\n",
        "WARNING: Chromosome 'M76611' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 23\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 24\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 25\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 26\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 27\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 28\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 29\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 30\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 31\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 32\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 33\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 34\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 35\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 36\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 37\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 38\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 39\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 40\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 41\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 42\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 43\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 44\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 45\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 46\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 47\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 48\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 49\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 50\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 51\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 52\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 53\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 54\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 55\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 56\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 57\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 58\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 59\r\n",
        "WARNING: Chromosome 'PFC10_API_IRAB' not found. File '/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf', line 60\r\n",
        "00:00:02.403\tCreating sequences from exon information 'Pf3D7_01_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.425\tBuilding sequence tree for chromosome 'Pf3D7_01_v3'\r\n",
        "00:00:02.428\tDone. Loaded 351 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.586\tCreating sequences from exon information 'Pf3D7_02_v3'\r\n",
        "00:00:02.600\tBuilding sequence tree for chromosome 'Pf3D7_02_v3'\r\n",
        "00:00:02.603\tDone. Loaded 558 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.766\tCreating sequences from exon information 'Pf3D7_03_v3'\r\n",
        "00:00:02.779\tBuilding sequence tree for chromosome 'Pf3D7_03_v3'\r\n",
        "00:00:02.780\tDone. Loaded 631 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.944\tCreating sequences from exon information 'Pf3D7_04_v3'\r\n",
        "00:00:02.956\tBuilding sequence tree for chromosome 'Pf3D7_04_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:02.957\tDone. Loaded 601 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.095\tCreating sequences from exon information 'Pf3D7_05_v3'\r\n",
        "00:00:03.109\tBuilding sequence tree for chromosome 'Pf3D7_05_v3'\r\n",
        "00:00:03.110\tDone. Loaded 776 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.252\tCreating sequences from exon information 'Pf3D7_06_v3'\r\n",
        "00:00:03.268\tBuilding sequence tree for chromosome 'Pf3D7_06_v3'\r\n",
        "00:00:03.270\tDone. Loaded 831 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.384\tCreating sequences from exon information 'Pf3D7_07_v3'\r\n",
        "00:00:03.399\tBuilding sequence tree for chromosome 'Pf3D7_07_v3'\r\n",
        "00:00:03.400\tDone. Loaded 728 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.489\tCreating sequences from exon information 'Pf3D7_08_v3'\r\n",
        "00:00:03.504\tBuilding sequence tree for chromosome 'Pf3D7_08_v3'\r\n",
        "00:00:03.506\tDone. Loaded 842 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.588\tCreating sequences from exon information 'Pf3D7_09_v3'\r\n",
        "00:00:03.604\tBuilding sequence tree for chromosome 'Pf3D7_09_v3'\r\n",
        "00:00:03.606\tDone. Loaded 976 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.643\t\t10000 variants (10000 variants per second), 10000 VCF entries\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.704\tCreating sequences from exon information 'Pf3D7_10_v3'\r\n",
        "00:00:03.722\tBuilding sequence tree for chromosome 'Pf3D7_10_v3'\r\n",
        "00:00:03.724\tDone. Loaded 1039 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:03.853\tCreating sequences from exon information 'Pf3D7_11_v3'\r\n",
        "00:00:03.873\tBuilding sequence tree for chromosome 'Pf3D7_11_v3'\r\n",
        "00:00:03.876\tDone. Loaded 1315 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.004\tCreating sequences from exon information 'Pf3D7_12_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.028\tBuilding sequence tree for chromosome 'Pf3D7_12_v3'\r\n",
        "00:00:04.030\tDone. Loaded 1333 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.186\tCreating sequences from exon information 'Pf3D7_13_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.217\tBuilding sequence tree for chromosome 'Pf3D7_13_v3'\r\n",
        "00:00:04.220\tDone. Loaded 1779 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.371\tCreating sequences from exon information 'Pf3D7_14_v3'\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.409\tBuilding sequence tree for chromosome 'Pf3D7_14_v3'\r\n",
        "00:00:04.414\tDone. Loaded 2069 sequences.\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:04.464\t\t20000 variants (10000 variants per second), 20000 VCF entries\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\r\n",
        "ERRORS: Some errors were detected\r\n",
        "Error type\tNumber of errors\r\n",
        "ERROR_CHROMOSOME_NOT_FOUND\t39\r\n",
        "\r\n",
        "\r\n",
        "\r\n",
        "WARNINGS: Some warning were detected\r\n",
        "Warning type\tNumber of warnings\r\n",
        "WARNING_REF_DOES_NOT_MATCH_GENOME\t38\r\n",
        "WARNING_TRANSCRIPT_INCOMPLETE\t250\r\n",
        "WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\t2120\r\n",
        "WARNING_TRANSCRIPT_NO_START_CODON\t222\r\n",
        "\r\n",
        "\r\n",
        "00:00:04.527\tCreating summary file: snpEff_summary.html\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:05.160\tCreating genes file: snpEff_genes.txt\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:05.749\tdone.\r\n",
        "00:00:05.751\tLogging\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "00:00:06.758\tChecking for updates...\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:22,287 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:35:22,289 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.4-0-g7e26428, Compiled 2015/05/15 03:25:41 \r\n",
        "INFO  16:35:22,289 HelpFormatter - Copyright (c) 2010 The Broad Institute \r\n",
        "INFO  16:35:22,289 HelpFormatter - For support and documentation go to http://www.broadinstitute.org/gatk \r\n",
        "INFO  16:35:22,292 HelpFormatter - Program Args: -T VariantAnnotator -R /data/plasmodium/pfalciparum/recon/roamato/Pf3D7_v3/3D7_sorted.fa -A TandemRepeatAnnotator -A SnpEff --variant /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.leftaligned.vcf --snpEffFile /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.snpeff.vcf -o /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.snpeff_annotated.vcf \r\n",
        "INFO  16:35:22,296 HelpFormatter - Executing as rpearson@dhcp537.well.ox.ac.uk on Mac OS X 10.10.5 x86_64; Java HotSpot(TM) 64-Bit Server VM 1.7.0_60-b19. \r\n",
        "INFO  16:35:22,297 HelpFormatter - Date/Time: 2015/10/02 16:35:22 \r\n",
        "INFO  16:35:22,297 HelpFormatter - -------------------------------------------------------------------------------- \r\n",
        "INFO  16:35:22,297 HelpFormatter - -------------------------------------------------------------------------------- \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:22,643 GenomeAnalysisEngine - Strictness is SILENT \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:22,688 GenomeAnalysisEngine - Downsampling Settings: Method: BY_SAMPLE, Target Coverage: 250 \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:23,093 RMDTrackBuilder - Writing Tribble index to disk for file /lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/GN01.snpeff.vcf.idx \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:23,206 GenomeAnalysisEngine - Preparing for traversal \r\n",
        "INFO  16:35:23,207 GenomeAnalysisEngine - Done preparing for traversal \r\n",
        "INFO  16:35:23,207 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING] \r\n",
        "INFO  16:35:23,207 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining \r\n",
        "INFO  16:35:23,208 ProgressMeter -        Location |     sites | elapsed |     sites | completed | runtime |   runtime \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:23,286 SnpEff - Skipping malformed SnpEff effect field at M76611:772. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,302 SnpEff - Skipping malformed SnpEff effect field at M76611:1692. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:23,352 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:23,361 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:987. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,363 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:1205. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,377 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:2732. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,378 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:2772. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,379 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:2784. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,380 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:2787. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,404 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:5822. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,411 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:6632. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:23,443 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:14410. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,446 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:15405. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,449 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:16656. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,454 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18694. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,454 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18828. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,455 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:18860. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,456 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:19295. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,460 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:21322. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,461 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:21498. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:23,464 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:22669. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,466 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23313. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,467 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23943. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,468 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:23947. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,470 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:25377. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,471 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:25587. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,472 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:25869. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,476 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:27761. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,477 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:28008. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,477 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:28127. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,478 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:28183. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,479 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:29000. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,480 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:31446. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,481 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:31449. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,481 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:31461. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,482 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:31498. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,483 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:33026. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,483 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:33246. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n",
        "WARN  16:35:23,484 SnpEff - Skipping malformed SnpEff effect field at PFC10_API_IRAB:34210. Error was: \" is not a recognized effect type\". Field was: \"(MODIFIER|||||||||ERROR_CHROMOSOME_NOT_FOUND)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:23,609 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:219211. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F4|PF3D7_0105000|protein_coding|CODING|PF3D7_0105000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:23,758 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478408. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gat/gaATt|D377E?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:23,758 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478417. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aGt|N380S|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:23,759 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478423. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCc/aTc|T382I|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:23,759 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:478424. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|acC/acG|T382|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:23,760 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:479254. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tca/tcAa|S659S?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:23,760 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:479314. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||act/acTt|T679T?|PF3D7_0112700|protein_coding|CODING|PF3D7_0112700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:23,762 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:482617. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|PF3D7_0112800|protein_coding|CODING|PF3D7_0112800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:23,826 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_01_v3:587484. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||PF3D7_0115100|protein_coding|CODING|PF3D7_0115100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,066 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:466898. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*1246|PF3D7_0211500|protein_coding|CODING|PF3D7_0211500.1|1)\" \r\n",
        "WARN  16:35:24,069 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:471024. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATATAAGTAGt|N412KYK*?|TKL1|protein_coding|CODING|PF3D7_0211700.1|1)\" \r\n",
        "WARN  16:35:24,079 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:487627. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cac/Gac|H1398D|PF3D7_0212100|protein_coding|CODING|PF3D7_0212100.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:24,080 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:487639. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aac/Cac|N1402H|PF3D7_0212100|protein_coding|CODING|PF3D7_0212100.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,230 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_02_v3:709392. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/tgTTTAAa|*552CL?|PF3D7_0217100|protein_coding|CODING|PF3D7_0217100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,551 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:347810. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||atg/aCGTTTTATTGAAAATAATAtg|M1197TFY*K*Y?|PF3D7_0307900|protein_coding|CODING|PF3D7_0307900.1|2)\" \r\n",
        "WARN  16:35:24,567 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:386419. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||YVH1|protein_coding|CODING|PF3D7_0309000.1|4)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,599 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:451382. Error was: \"STOP_LOST+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_LOST+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||taa/tTTACAATAATAATAATCTTAAaa|*1295FTIIIILK|PIESP1|protein_coding|CODING|PF3D7_0310400.1|1)\" \r\n",
        "WARN  16:35:24,601 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:456924. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gTATGAAAAAATAAACGATGAAAAAATAAACGat|D728V*KNKR*KNKR?|PF3D7_0310500|protein_coding|CODING|PF3D7_0310500.1|1)\" \r\n",
        "WARN  16:35:24,602 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:457889. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taAt|Y406*?|PF3D7_0310500|protein_coding|CODING|PF3D7_0310500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,635 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:538491. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aca/aAACATATAATTTAAATAAAGATAATAca|T105KHII*IKII?|PF3D7_0313000|protein_coding|CODING|PF3D7_0313000.1|1)\" \r\n",
        "WARN  16:35:24,651 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:571050. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|PF3D7_0314100|protein_coding|CODING|PF3D7_0314100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,686 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:657246. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|PF3D7_0316200|protein_coding|CODING|PF3D7_0316200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,729 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:757995. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAt/aTt|N619I|PF3D7_0318300|protein_coding|CODING|PF3D7_0318300.1|3|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,793 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:907920. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aaT/aaT|N733|PF3D7_0321700|protein_coding|CODING|PF3D7_0321700.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:24,826 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_03_v3:983965. Error was: \"START_LOST+CODON_INSERTION is not a recognized effect type\". Field was: \"START_LOST+CODON_INSERTION(HIGH||atg/aAAAtg|M1KM|RIPR|protein_coding|CODING|PF3D7_0323400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,957 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:227462. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/tgAt|C1424*?|ApiAP2|protein_coding|CODING|PF3D7_0404100.1|1)\" \r\n",
        "WARN  16:35:24,967 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:252498. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTTTAATGATAATAATAATAATGat|N1590I*******?|PF3D7_0404600|protein_coding|CODING|PF3D7_0404600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:24,993 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:319177. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|ASP|protein_coding|CODING|PF3D7_0405900.1|1)\" \r\n",
        "WARN  16:35:25,000 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:336832. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||act/aGGAGAAGGAACAAATAAct|T2408RRRNK*?|PF3D7_0406500|protein_coding|CODING|PF3D7_0406500.1|1)\" \r\n",
        "WARN  16:35:25,010 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:352982. Error was: \"START_LOST+CODON_INSERTION is not a recognized effect type\". Field was: \"START_LOST+CODON_INSERTION(HIGH||atg/aAAAtg|M1KM|PF3D7_0406800|protein_coding|CODING|PF3D7_0406800.1|1)\" \r\n",
        "WARN  16:35:25,041 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:435366. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAAAAAAAAAAAAAAAAtg|M1KKKKKK?|PF3D7_0409000|protein_coding|CODING|PF3D7_0409000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:25,280 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:933504. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gtg/Gtg|V348|PF3D7_0420600|protein_coding|CODING|PF3D7_0420600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:25,281 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:933505. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gTg/gTg|V348|PF3D7_0420600|protein_coding|CODING|PF3D7_0420600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:25,311 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979922. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ccT/ccC|P3|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:25,311 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979935. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ggt/Cgt|G8R|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:25,311 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979996. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cAt/cCt|H28P|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:25,312 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:979998. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aag/Gag|K29E|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:25,312 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:980013. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tcg/Ccg|S34P|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:25,313 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:980023. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|tAa/tGa|*37|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:25,313 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_04_v3:980041. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gCg/gTg|A43V|PF3D7_0421400|protein_coding|CODING|PF3D7_0421400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:25,617 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:274340. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gta/ATTAATAAAATATAAgta|V2647INKI*V|PF3D7_0506500|protein_coding|CODING|PF3D7_0506500.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:25,663 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:348539. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taATACTATGGGGGGAAAt|Y2167*YYGGK?|RCC1|protein_coding|CODING|PF3D7_0508500.1|1)\" \r\n",
        "WARN  16:35:25,666 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:354544. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaaGAAAAATAATCCTATGGGGGGGAAAAAA|K165KEK*SYGGEK?|RCC1|protein_coding|CODING|PF3D7_0508500.1|1)\" \r\n",
        "WARN  16:35:25,672 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:369479. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aAATGaa|K820K*?|PF3D7_0508900|protein_coding|CODING|PF3D7_0508900.1|1)\" \r\n",
        "WARN  16:35:25,672 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:369491. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aAATGaa|K824K*?|PF3D7_0508900|protein_coding|CODING|PF3D7_0508900.1|1)\" \r\n",
        "WARN  16:35:25,673 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:369503. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aAATGaa|K828K*?|PF3D7_0508900|protein_coding|CODING|PF3D7_0508900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:25,737 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:552810. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cct/ccAAACAAATTATATGAACGGACAGGATAATCTt|P177PNKLYERTG*S?|PF3D7_0512800|protein_coding|CODING|PF3D7_0512800.1|1)\" \r\n",
        "WARN  16:35:25,737 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:552813. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||tat/taACTTACAAAt|Y176*LTN|PF3D7_0512800|protein_coding|CODING|PF3D7_0512800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:25,788 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:688978. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gat/Gat|D889|MB2|protein_coding|CODING|PF3D7_0516600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:25,788 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:688982. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gtA/gtA|V887|MB2|protein_coding|CODING|PF3D7_0516600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:25,982 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1243986. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_0530400|protein_coding|CODING|PF3D7_0530400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,000 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1289232. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCa/aTa|T225I|PF3D7_0531500|protein_coding|CODING|PF3D7_0531500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,001 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1290376. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gctttt/|AF259|PF3D7_0531600|protein_coding|CODING|PF3D7_0531600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,002 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1290491. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aaT/aaG|N297K|PF3D7_0531600|protein_coding|CODING|PF3D7_0531600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,002 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291707. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*6|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,003 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291718. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAg/aGg|K9R|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,003 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291722. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/TTtat|Y11F?|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,003 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291727. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAg/aGg|K12R|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,004 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291748. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aGa/aAa|R19K|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,004 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291753. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tAat|Y21*?|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,005 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291755. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|taT/taA|Y21*|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,005 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291757. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gGt/gAt|G22D|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,005 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291772. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tac/taTGc|Y27Y?|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,006 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291821. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aaT/aaG|N43K|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,006 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291829. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tTt|Y46F|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,006 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291830. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|taT/taG|Y46*|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,006 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291844. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cCt/cTt|P51L|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,007 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291878. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ttA/ttG|L62|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,007 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1291993. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Gtt/Att|V101I|PF3D7_0531700|protein_coding|CODING|PF3D7_0531700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,007 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292387. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F60|PF3D7_0531900|protein_coding|CODING|PF3D7_0531900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,008 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292552. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cAt/cCt|H48P|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,008 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292595. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F63|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,009 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292652. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|tcT/tcG|S81|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,009 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1292895. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|ttA/ttT|L162F|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,009 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293105. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atT/atA|I232|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,010 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293615. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ctt/|L403|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,010 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1293917. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTc/aAc|I503N|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,011 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1294485. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L693|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,011 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_05_v3:1295969. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cTt/cCt|L1187P|PF3D7_0532000|protein_coding|CODING|PF3D7_0532000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,148 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:316535. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||atG/atG|M116|SAS6|protein_coding|CODING|PF3D7_0607600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,148 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:316553. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||atG/atG|M110|SAS6|protein_coding|CODING|PF3D7_0607600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:26,185 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:422621. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|DHHC2|protein_coding|CODING|PF3D7_0609800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,223 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:533184. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*687|PF3D7_0612900|protein_coding|CODING|PF3D7_0612900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,264 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:648412. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ggt/ggtAAATGATGAAAATAATGGTCATAGTGGT|G2029GK**K*WS*W?|PF3D7_0615600|protein_coding|CODING|PF3D7_0615600.1|6)\" \r\n",
        "WARN  16:35:26,266 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:651171. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tGATGAAAATAATGGTCATAGTGGTAAATATGATGATAATACATat|Y1110**K*WS*W*I***YI?|PF3D7_0615600|protein_coding|CODING|PF3D7_0615600.1|6)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,458 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_06_v3:1187890. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTCAATTAGAGAat|N252IN*R?|GATB|protein_coding|CODING|PF3D7_0628800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,608 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:207245. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aAt/aAt|N843|PF3D7_0704400|protein_coding|CODING|PF3D7_0704400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,714 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508506. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aac/|N12|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,715 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508509. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aaC/aaG|N12K|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,715 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508580. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tGc/tAc|C36Y|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,716 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:508590. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|acC/acT|T39|PF3D7_0711600|protein_coding|CODING|PF3D7_0711600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,792 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:731125. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tga/|*1006|PF3D7_0716700|protein_coding|CODING|PF3D7_0716700.1|1)\" \r\n",
        "WARN  16:35:26,793 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:733495. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_0716800|protein_coding|CODING|PF3D7_0716800.1|1)\" \r\n",
        "WARN  16:35:26,812 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:788448. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/aatGATGATGATAATAATGATGATGATAATAATTGATGATAATAAAAACAAATG|N1955NDDDNNDDDNN****KQM|PF3D7_0718000|protein_coding|CODING|PF3D7_0718000.1|9)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,838 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:863478. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aaaTAA|K445K*|PF3D7_0719800|protein_coding|CODING|PF3D7_0719800.1|1)\" \r\n",
        "WARN  16:35:26,858 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:923747. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gGt/gGt|G488|PF3D7_0721300|protein_coding|CODING|PF3D7_0721300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,896 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1030750. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||tat/taATAATAATAATAATAATAATATAATAATAATAATTAt|Y776*******YNNNNY|PF3D7_0724600|protein_coding|CODING|PF3D7_0724600.1|1)\" \r\n",
        "WARN  16:35:26,924 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1084454. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ctt/cAGGAAGTAAAGAAATGAAATAAAATGAACGATtt|L232QEVKK*NKMND?|PF3D7_0725600|protein_coding|CODING|PF3D7_0725600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,925 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1085257. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||att/Aatt|I500N?|PF3D7_0725600|protein_coding|CODING|PF3D7_0725600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,926 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1085327. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ttt/Gtt|F523V|PF3D7_0725600|protein_coding|CODING|PF3D7_0725600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,926 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1085521. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||tgc/TGAtgc|C588*C|PF3D7_0725600|protein_coding|CODING|PF3D7_0725600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,926 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1085558. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cgt/Tgt|R600C|PF3D7_0725600|protein_coding|CODING|PF3D7_0725600.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,927 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1085980. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|Gaa/Taa|E45*|PF3D7_0725700|protein_coding|CODING|PF3D7_0725700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,927 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1085988. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|gtT/gtA|V47|PF3D7_0725700|protein_coding|CODING|PF3D7_0725700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,928 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1085989. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Gta/Tta|V48L|PF3D7_0725700|protein_coding|CODING|PF3D7_0725700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,928 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1086003. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|acT/acC|T52|PF3D7_0725700|protein_coding|CODING|PF3D7_0725700.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,929 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1086454. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gaa/gaATa|E28E?|PF3D7_0725900|protein_coding|CODING|PF3D7_0725900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,929 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1086710. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|aaA/aaG|K47|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,929 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1086811. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|cTt/cGt|L81R|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,930 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1086841. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gTc/gCc|V91A|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,930 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1086906. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Atg/Gtg|M113V|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,931 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1087265. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taC/taT|Y232|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,931 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1087337. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|ttC/ttT|F256|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,931 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1087339. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tTt/tGt|F257C|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,932 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1087382. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F272|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,933 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1088643. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ctt/|L692|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,934 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1089479. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atC/atT|I970|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,934 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1089487. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||cca/|P973|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:26,935 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1090209. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ctg/cTGTGAGCTGTGAGCTGtg|L1214L*AVSC?|PF3D7_0726000|protein_coding|CODING|PF3D7_0726000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:26,953 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1123070. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||cCa/cCa|P359|PF3D7_0726400|protein_coding|CODING|PF3D7_0726400.1|3|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:26,954 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1124491. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_0726400|protein_coding|CODING|PF3D7_0726400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,075 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_07_v3:1334840. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aag/aaAg|K33K?|PF3D7_0731000|protein_coding|CODING|PF3D7_0731000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,207 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:85892. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tGAAAGAAat|Y779*KK?|PF3D7_0801000|protein_coding|CODING|PF3D7_0801000.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,208 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:85895. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aga/aAGAAATTTTga|R778KKF*|PF3D7_0801000|protein_coding|CODING|PF3D7_0801000.1|2)\" \r\n",
        "WARN  16:35:27,211 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:92806. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tgt/tTgt|C1589L?|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:27,213 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:95108. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F821|PF3D7_0801100|protein_coding|CODING|PF3D7_0801100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:27,243 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:182360. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||acg/acAAAATGAAAGg|T484TK*K?|ACbeta|protein_coding|CODING|PF3D7_0802600.1|3)\" \r\n",
        "WARN  16:35:27,243 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:182378. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||acg/acAAAATGAAAGg|T490TK*K?|ACbeta|protein_coding|CODING|PF3D7_0802600.1|3)\" \r\n",
        "WARN  16:35:27,244 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:182396. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||acg/acAAAATGAAAGg|T496TK*K?|ACbeta|protein_coding|CODING|PF3D7_0802600.1|3)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,269 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:251362. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/taTGGTGATGATTAAAAAAATTTTTTTTTTTTTTTTTTTt|Y427YGDD*KNFFFFFF?|PF3D7_0804500|protein_coding|CODING|PF3D7_0804500.1|1)\" \r\n",
        "WARN  16:35:27,277 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:274437. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaaTAGAAGAGAAATATAATAAAAA|K992K*KRNIIK?|PF3D7_0804700|protein_coding|CODING|PF3D7_0804700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,343 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:465718. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||acc/TGATCTGCATTACTATCCATATTACacc|T1828*SALLSILH?|VAR|protein_coding|CODING|PF3D7_0809100.1|2)\" \r\n",
        "WARN  16:35:27,358 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:482512. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATAAt|N4957K*?|PF3D7_0809600|protein_coding|CODING|PF3D7_0809600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,472 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:788355. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/tAACATTAACGAAGGATGTGATAACATTAACGAAGGGTgt|C421*H*RRM**H*RRV?|PF3D7_0817300|protein_coding|CODING|PF3D7_0817300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,541 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:961739. Error was: \"START_LOST+CODON_INSERTION is not a recognized effect type\". Field was: \"START_LOST+CODON_INSERTION(HIGH||atg/aAAAtg|M1KM|PF3D7_0821300|protein_coding|CODING|PF3D7_0821300.1|1)\" \r\n",
        "WARN  16:35:27,561 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1010466. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION(HIGH||tttgatgaa/taa|FDE279*|PF3D7_0822900|protein_coding|CODING|PF3D7_0822900.1|1)\" \r\n",
        "WARN  16:35:27,562 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1010514. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_DELETION(HIGH||tttgatgaa/taa|FDE263*|PF3D7_0822900|protein_coding|CODING|PF3D7_0822900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,600 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1110461. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gtg/gAAAATTAATACAAAtg|V396EN*YK?|PF3D7_0826000|protein_coding|CODING|PF3D7_0826000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,634 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1203236. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgc/tgAAAACATTAAAAATGACGATGGTTACATTAAAAATGACGATGGTTAc|C1460*KH*K*RWLH*K*RWL?|SET3|protein_coding|CODING|PF3D7_0827800.1|1)\" \r\n",
        "WARN  16:35:27,667 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1283721. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||tttttttcttatacatagacttttattgttatacatagat/|FFSYT*TFIVIHR?930|PF3D7_0830100|protein_coding|CODING|PF3D7_0830100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,688 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_08_v3:1334649. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/GAATAATATTaaa|K413E*Y*?|PF3D7_0831200|protein_coding|CODING|PF3D7_0831200.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:27,909 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:468847. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aTa/aTa|I1430|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,911 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472141. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gAa/gAa|E2528|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,911 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472192. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gAt/gAt|D2545|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,912 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472193. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaT/gaT|D2545|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,912 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472196. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaA/gaA|E2546|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,912 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472197. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Tat/Tat|Y2547|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,912 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472202. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaC/gaC|D2548|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,913 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472206. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gat/Gat|D2550|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,913 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472211. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaT/gaT|D2551|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,913 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472214. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaT/gaT|D2552|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,913 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472224. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Tta/Tta|L2556|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,914 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472225. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||tTa/tTa|L2556|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,914 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472232. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gaC/gaC|D2558|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,914 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472233. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Tta/Tta|L2559|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:27,914 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:472234. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||tTa/tTa|L2559|PF3D7_0910200|protein_coding|CODING|PF3D7_0910200.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,000 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:684805. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aAg/aAg|K415|PF3D7_0916400|protein_coding|CODING|PF3D7_0916400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:28,000 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:684817. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aAg/aAg|K419|PF3D7_0916400|protein_coding|CODING|PF3D7_0916400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:28,031 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:761842. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tct/tcATTTt|S211SF?|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:28,031 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762341. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/Tttt|F378F?|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:28,032 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762374. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F389|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:28,032 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762509. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/TTttt|F434F?|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:28,032 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:762789. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aaa/Gaa|K527E|TR|protein_coding|CODING|PF3D7_0918500.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,110 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:982412. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cat/cTATAACAATGAAAATATTATTTATAATAATAACAATGAAAATATTATTTATAATAat|H421L*Q*KYYL***Q*KYYL**?|PF3D7_0924200|protein_coding|CODING|PF3D7_0924200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,202 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1256375. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aATAATAATAATAATAATAATAATTGat|N521NNNNNNNN*?|PF3D7_0931100|protein_coding|CODING|PF3D7_0931100.1|1)\" \r\n",
        "WARN  16:35:28,207 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1271773. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gga/ggAAAAAAAATAAAAATAAGGAAGAGGAAAATATTGTGATGGAAGGAAATAAAAATAAGGAAGAAGAAAACATTGTGATGGAAGAa|G477GKKIKIRKRKIL*WKEIKIRKKKTL*WK?|PF3D7_0931700|protein_coding|CODING|PF3D7_0931700.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,207 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1272162. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||PF3D7_0931700|protein_coding|CODING|PF3D7_0931700.1|2)\" \r\n",
        "WARN  16:35:28,229 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_09_v3:1338294. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||agt/agtTTTTAATAGT|S211SF**?|PF3D7_0933900|protein_coding|CODING|PF3D7_0933900.1|4)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,440 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:441954. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1011400|protein_coding|CODING|PF3D7_1011400.1|1)\" \r\n",
        "WARN  16:35:28,446 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:456003. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/TAATAATAATAATAATaat|N203******?|PREBP|protein_coding|CODING|PF3D7_1011800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,504 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:632060. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1015700|protein_coding|CODING|PF3D7_1015700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,530 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:707645. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gtC/gtC|V703|PF3D7_1017600|protein_coding|CODING|PF3D7_1017600.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,598 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:911052. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatTAAT|N431N*?|PF3D7_1021900|protein_coding|CODING|PF3D7_1021900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,652 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1075102. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||gat/gGGTGATAATAAAAAAAAATTGAGGat|D3610G***KKIED|PF3D7_1025500|protein_coding|CODING|PF3D7_1025500.1|5)\" \r\n",
        "WARN  16:35:28,659 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_10_v3:1096623. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aaaTAA|K218K*|PF3D7_1025900|protein_coding|CODING|PF3D7_1025900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:28,930 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:273940. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaaTATTAAGATGGATGAG|K204KY*DG*?|PF3D7_1106500|protein_coding|CODING|PF3D7_1106500.1|1)\" \r\n",
        "WARN  16:35:28,930 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:273958. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaaTATTAAGATGGATGAG|K198KY*DG*?|PF3D7_1106500|protein_coding|CODING|PF3D7_1106500.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,067 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:690406. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||cac/cacAAATGAACTGAATCATCAC|H384HK*TESS?|PF3D7_1118200|protein_coding|CODING|PF3D7_1118200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,175 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1020897. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||agt/agAAAATAATATAATAATAATTATGATGATAAAAATAATGATAATAATAATTATGATGATAAAAATAAt|S533RK*YNNNYDDKNNDNNNYDDKNN|ATG7|protein_coding|CODING|PF3D7_1126100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,260 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1278131. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gac/GACGATGAACAATTAGATGATAGGTgac|D263DDEQLDDR*?|PF3D7_1133100|protein_coding|CODING|PF3D7_1133100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,305 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1408651. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gTTTTATGACATGATAAATGATGACATGATAAATAat|D1516VL*HDK**HDK*?|PF3D7_1136000|protein_coding|CODING|PF3D7_1136000.1|2)\" \r\n",
        "WARN  16:35:29,324 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1465057. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|PF3D7_1137200|protein_coding|CODING|PF3D7_1137200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,350 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1542865. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||caa/cAAAaa|Q688QK|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,350 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1542894. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aaa/AAATACAaaa|K698KYK?|PF3D7_1138900|protein_coding|CODING|PF3D7_1138900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,360 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1572158. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/agAAa|R852R?|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,360 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1572461. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||act/acTt|T953T?|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,361 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1572602. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||att/atTATATATAt|I1000IIY?|PF3D7_1139400|protein_coding|CODING|PF3D7_1139400.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,379 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1628442. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATATTATATTTAATAATAATGATAATAATAATAGTAATAATAAt|N611KYYI***********?|PF3D7_1140700|protein_coding|CODING|PF3D7_1140700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,454 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1853371. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|PF3D7_1146800|protein_coding|CODING|PF3D7_1146800.1|1)\" \r\n",
        "WARN  16:35:29,476 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1917620. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tgt/tgTTt|C410C?|PF3D7_1148100|protein_coding|CODING|PF3D7_1148100.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,478 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920309. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tat/TATATAATATtat|Y15YIIL?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,478 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920402. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atA/atT|I45|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,478 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920405. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_STOP(LOW|SILENT|taG/taA|*46|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,479 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_11_v3:1920462. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gta/Tgta|V66C?|PF3D7_1148200|protein_coding|CODING|PF3D7_1148200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,586 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190106. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_INSERTION(MODERATE||aga/aAAAga|R1003KR|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,587 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190965. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tat/Aat|Y717N|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,587 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190971. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/agaG|R714R?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,587 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:190973. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aGa/aAa|R714K|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,587 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:191119. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"CODON_CHANGE_PLUS_CODON_INSERTION(MODERATE||aat/aaAATATAt|N665KIY|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,588 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:192538. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||atg/atATATATATATg|M192IYIY?|PF3D7_1203800|protein_coding|CODING|PF3D7_1203800.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,661 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:408040. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tatTGACAATATAAATAATTATGACAATATAAATAATTAT|Y370Y*QYK*L*QYK*L?|PF3D7_1208800|protein_coding|CODING|PF3D7_1208800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,749 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:683586. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/AGATGATgat|D251R**?|PF3D7_1217300|protein_coding|CODING|PF3D7_1217300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,800 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:839773. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tct/tctTATGGCGCTGATTAATTCAAACATTTCTTATGGCGCTGATAATTCAAACATTTCT|S1376SYGAD*FKHFLWR**FKHF?|SET10|protein_coding|CODING|PF3D7_1221000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,816 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:891244. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Atg/Atg|M222|ApiAP2|protein_coding|CODING|PF3D7_1222400.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:29,840 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:967371. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tatTGATGATAATAATAAATAT|Y40Y*****I?|PF3D7_1223600|protein_coding|CODING|PF3D7_1223600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,917 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1204404. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||caa/cAaa|Q275Q?|PF3D7_1229200|protein_coding|CODING|PF3D7_1229200.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:29,953 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1316738. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aaT/aaA|N681K|PF3D7_1231800|protein_coding|CODING|PF3D7_1231800.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:29,973 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1375760. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Ata/Ata|I139|PF3D7_1233300|protein_coding|CODING|PF3D7_1233300.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:29,979 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1393519. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tca/tGAca|S2605*?|AARP1|protein_coding|CODING|PF3D7_1233600.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:30,017 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1514208. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aat/aTGATCATGTAGAAACAAAat|N770MIM*KQN|PF3D7_1236200|protein_coding|CODING|PF3D7_1236200.1|1)\" \r\n",
        "WARN  16:35:30,017 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1514229. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gga/gCAAATATCATGTAga|G763ANIM*?|PF3D7_1236200|protein_coding|CODING|PF3D7_1236200.1|1)\" \r\n",
        "WARN  16:35:30,018 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1514235. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ata/aAATGATCATGta|I761K*SC?|PF3D7_1236200|protein_coding|CODING|PF3D7_1236200.1|1)\" \r\n",
        "WARN  16:35:30,045 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1598367. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||aAt/aAt|N1181|PF3D7_1238500|protein_coding|CODING|PF3D7_1238500.1|1|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:30,092 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1745296. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|gCg/gTg|A43V|PF3D7_1241000|protein_coding|CODING|PF3D7_1241000.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:30,093 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1747992. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1241100|protein_coding|CODING|PF3D7_1241100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:30,120 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1830594. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gaAAAGATGACAACAATATAAATGGTGATGACAACAATATAAATGGa|E312EKMTTI*MVMTTI*M?|PF3D7_1243700|protein_coding|CODING|PF3D7_1243700.1|1)\" \r\n",
        "WARN  16:35:30,122 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1835511. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ggg/gggAGGAACAAAAAATGAGG|G1525GRNKK*?|DOC2|protein_coding|CODING|PF3D7_1243900.1|4)\" \r\n",
        "WARN  16:35:30,141 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_12_v3:1892630. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aag/aaAAATGTTAGATATATTACGAATATAAAAAATGTTAGATATATTACGAATATAAAAATGTAGATAAATATATAGATAAAAg|K23KNVRYITNIKNVRYITNIKM*INI*IK?|PF3D7_1245200|protein_coding|CODING|PF3D7_1245200.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:30,471 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:661261. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|MYB1|protein_coding|CODING|PF3D7_1315800.1|1)\" \r\n",
        "WARN  16:35:30,484 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:699921. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1316800|protein_coding|CODING|PF3D7_1316800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:30,546 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:886263. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAAAAtg|M1KK?|PF3D7_1321300|protein_coding|CODING|PF3D7_1321300.1|1)\" \r\n",
        "WARN  16:35:30,576 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:970818. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||cga/cTATATATATATATATTAATATga|R181LYIYILI*|PF3D7_1323300|protein_coding|CODING|PF3D7_1323300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:30,607 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1060620. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tAAat|Y2624*?|PF3D7_1325400|protein_coding|CODING|PF3D7_1325400.1|1)\" \r\n",
        "WARN  16:35:30,616 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1082205. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aatGAATATATATGTGAATAATGTGAAT|N794NEYICE*CE?|PF3D7_1325900|protein_coding|CODING|PF3D7_1325900.1|1)\" \r\n",
        "WARN  16:35:30,629 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1120112. Error was: \"STOP_GAINED+CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_INSERTION(HIGH||aaa/aAATaa|K4080K*|PF3D7_1326600|protein_coding|CODING|PF3D7_1326600.1|1)\" \r\n",
        "WARN  16:35:30,630 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1122538. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*4781|PF3D7_1326600|protein_coding|CODING|PF3D7_1326600.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:30,708 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1359601. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTTATCAAACAAATCAAACAAATTTTATAAATCATATGAat|N240IIKQIKQIL*II*?|PF3D7_1333100|protein_coding|CODING|PF3D7_1333100.1|1)\" \r\n",
        "WARN  16:35:30,718 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1391352. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1334000|protein_coding|CODING|PF3D7_1334000.1|1)\" \r\n",
        "WARN  16:35:30,724 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1407511. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/|M1|MSRP6|protein_coding|CODING|PF3D7_1334500.1|1)\" \r\n",
        "WARN  16:35:30,734 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1436419. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aaATAAt|N740K*?|RH2b|protein_coding|CODING|PF3D7_1335300.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:30,905 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:1946250. Error was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"STOP_GAINED+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||cct/ccATATAAATGAAAt|P377PYK*N|PF3D7_1348400|protein_coding|CODING|PF3D7_1348400.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:30,999 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2226146. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/aaaTAATAATAGTAACAAT|K298K****Q?|PF3D7_1356100|protein_coding|CODING|PF3D7_1356100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,104 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2562757. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/tTAtt|F214L?|PF3D7_1363900|protein_coding|CODING|PF3D7_1363900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:31,104 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2562845. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/agAa|R243R?|PF3D7_1363900|protein_coding|CODING|PF3D7_1363900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,159 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_13_v3:2731423. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||ctg/cGTGGAAGTTAAAAAAATGACCACGTtg|L1574RGS*KNDHV?|ERCC4|protein_coding|CODING|PF3D7_1368800.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,268 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:156032. Error was: \"SPLICE_SITE_ACCEPTOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_ACCEPTOR+INTRON(HIGH||||ACalpha|protein_coding|CODING|PF3D7_1404600.1|17)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,286 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:206974. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/TGACgat|D473*R?|PF3D7_1405900|protein_coding|CODING|PF3D7_1405900.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,332 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:354410. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tgt/tgATAATTAt|C244**L?|PF3D7_1409000|protein_coding|CODING|PF3D7_1409000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,500 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:881449. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAtg|M1N?|PF3D7_1421300|protein_coding|CODING|PF3D7_1421300.1|1)\" \r\n",
        "WARN  16:35:31,531 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:972858. Error was: \"START_LOST+CODON_CHANGE_PLUS_CODON_INSERTION is not a recognized effect type\". Field was: \"START_LOST+CODON_CHANGE_PLUS_CODON_INSERTION(HIGH||ttg/ttTTTg|L1FL|PF3D7_1423900|protein_coding|CODING|PF3D7_1423900.1|1|WARNING_TRANSCRIPT_INCOMPLETE)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,561 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1057940. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/Tgaa|E594*?|PF3D7_1427100|protein_coding|CODING|PF3D7_1427100.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,604 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1195912. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gaa/gaCAACAATATAAATGGTGACAACAATATAAATGGTGAAAACAATATAAATGGTGAAATACAATATAAATGGTGAa|E1336DNNINGDNNINGENNINGEIQYKW*?|PF3D7_1430300|protein_coding|CODING|PF3D7_1430300.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,698 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1494213. Error was: \"FRAME_SHIFT+START_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+START_LOST(HIGH||atg/aAAtg|M1K?|PKG|protein_coding|CODING|PF3D7_1436600.1|1)\" \r\n",
        "WARN  16:35:31,744 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1630982. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/aTATATAAATAATGACAATATAAATAATGACAat|N129IYK**QYK**Q?|PF3D7_1440000|protein_coding|CODING|PF3D7_1440000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,758 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1673651. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||gat/gTATGat|D140V*?|PF3D7_1441100|protein_coding|CODING|PF3D7_1441100.1|1)\" \r\n",
        "WARN  16:35:31,780 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1739747. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||tat/tGAat|Y3155*?|TREP|protein_coding|CODING|PF3D7_1442600.1|2)\" \r\n",
        "WARN  16:35:31,781 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1739813. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||Gga/Gga|G3177|TREP|protein_coding|CODING|PF3D7_1442600.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:31,781 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1739814. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"SYNONYMOUS_CODING(LOW||gGa/gGa|G3177|TREP|protein_coding|CODING|PF3D7_1442600.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:31,781 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1739819. Error was: \"SnpEff issued the following warning or error: \"WARNING_REF_DOES_NOT_MATCH_GENOME\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cat/Aat|H3179N|TREP|protein_coding|CODING|PF3D7_1442600.1|2|WARNING_REF_DOES_NOT_MATCH_GENOME)\" \r\n",
        "WARN  16:35:31,782 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:1739845. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aaa/Taaa|K3188*?|TREP|protein_coding|CODING|PF3D7_1442600.1|2)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:31,908 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2136250. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/Taat|N652*?|RON2|protein_coding|CODING|PF3D7_1452000.1|1)\" \r\n",
        "WARN  16:35:31,909 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2136262. Error was: \"FRAME_SHIFT+STOP_GAINED is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_GAINED(HIGH||aat/TGaat|N656*?|RON2|protein_coding|CODING|PF3D7_1452000.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:32,012 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2462332. Error was: \"SPLICE_SITE_DONOR+INTRON is not a recognized effect type\". Field was: \"SPLICE_SITE_DONOR+INTRON(HIGH||||UCHL3|protein_coding|CODING|PF3D7_1460400.1|5)\" \r\n",
        "WARN  16:35:32,050 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:2582521. Error was: \"FRAME_SHIFT+STOP_LOST is not a recognized effect type\". Field was: \"FRAME_SHIFT+STOP_LOST(HIGH||taa/|*327|CYC1|protein_coding|CODING|PF3D7_1463700.1|1)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:32,265 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248049. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Att/Gtt|I340V|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,265 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248057. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/|F343|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,265 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248063. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|tgA/tgT|*344C|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,266 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248065. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tTt|Y345F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,266 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248067. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Cat/Aat|H346N|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,266 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248074. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aAa/aGa|K348R|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,266 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248077. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aag/|K349|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,267 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248080. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aTa/aAa|I350K|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,267 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248098. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tCt/tTt|S356F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "WARN  16:35:32,267 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248104. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|tAt/tTt|Y358F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,267 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248109. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aca/Gca|T360A|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,268 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248110. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|aCa/aTa|T360I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,268 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248118. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ata/|I363|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,268 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248120. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|atA/atC|I363|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,268 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248125. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|tAa/tTa|*365L|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,269 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248126. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_LOST(HIGH|MISSENSE|taA/taT|*365Y|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,269 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248133. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||ttt/tTTGTTTGtt|F368FVC?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,269 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248148. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"STOP_GAINED(HIGH|NONSENSE|Aga/Tga|R373*|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,269 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248149. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||aga/agAa|R373R?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,269 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248186. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taC/taT|Y385|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,270 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248190. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tac/Cac|Y387H|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,270 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248192. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|taC/taT|Y387|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,270 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248210. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|ttG/ttT|L393F|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,270 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248211. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Ttt/Gtt|F394V|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,271 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248216. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"SYNONYMOUS_CODING(LOW|SILENT|gtT/gtA|V395|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,271 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248217. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Tta/Ata|L396I|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,271 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248218. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||tta/|L396|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,271 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248226. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Gaa/Aaa|E399K|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,271 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248232. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"NON_SYNONYMOUS_CODING(MODERATE|MISSENSE|Aca/Gca|T401A|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "WARN  16:35:32,272 SnpEff - Skipping malformed SnpEff effect field at Pf3D7_14_v3:3248237. Error was: \"SnpEff issued the following warning or error: \"WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS\"\". Field was: \"FRAME_SHIFT(HIGH||gtc/Agtc|V403S?|PF3D7_1478900|protein_coding|CODING|PF3D7_1478900.1|1|WARNING_TRANSCRIPT_MULTIPLE_STOP_CODONS)\" \r\n",
        "INFO  16:35:32,285 VariantAnnotator - Processed 21539 loci.\r\n",
        " \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:32,363 ProgressMeter -            done   2.3332831E7     9.0 s       0.0 s      100.0%     9.0 s       0.0 s \r\n",
        "INFO  16:35:32,363 ProgressMeter - Total runtime 9.16 secs, 0.15 min, 0.00 hours \r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "INFO  16:35:33,703 GATKRunReport - Uploaded run statistics report to AWS S3 \r\n"
       ]
      }
     ],
     "prompt_number": 367
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def merge_consensus_alignment(sample='7G8',\n",
      "                      rewrite=False):\n",
      "    ox_code = tbl_samples_to_process.selecteq('Isolate code', sample).values('ox_code')[0]\n",
      "    nucmer_output_filestem='/lustre/scratch110/malaria/rp7/Pf3k/GATKbuild/consensus_alignment/nucmer/%s' % sample\n",
      "    annotated_vcf_fn = \"%s.annotated.vcf.gz\" % (nucmer_output_filestem)\n",
      "\n",
      "    \n",
      "    tbl_consensus = (etl.fromvcf(annotated_vcf_fn)\n",
      "        .unpackdict('INFO', samplesize=1000000)\n",
      "        .rename('ALT', 'consensus_ALTs')\n",
      "        .addfield('ALT', lambda rec: str(rec[4][0]))\n",
      "        .addfield('Consensus_is_coding', lambda rec: not rec['SNPEFF_EFFECT'] is None and not rec['SNPEFF_EFFECT'] in NONCODING_EFFECTS)\n",
      "        .cut(['CHROM', 'POS', 'REF', 'ALT', 'consensus_ALTs', 'RegionType', 'Consensus_is_coding', 'SNPEFF_AMINO_ACID_CHANGE', 'SNPEFF_EFFECT',\n",
      "              'SNPEFF_GENE_NAME', 'SNPEFF_IMPACT', 'SNPEFF_TRANSCRIPT_ID', 'RPA', 'RU', 'STR'])\n",
      "        .addfield('Consensus_RPA_REF', lambda rec: None if rec['RPA'] is None else str(rec['RPA'][0]))\n",
      "        .addfield('Consensus_RPA_ALT', lambda rec: None if rec['RPA'] is None else str(rec['RPA'][1]))\n",
      "        .rename('RegionType', 'Consensus_RegionType')\n",
      "        .rename('SNPEFF_AMINO_ACID_CHANGE', 'Consensus_AAChange')\n",
      "        .rename('SNPEFF_EFFECT', 'Consensus_Effect')\n",
      "        .rename('SNPEFF_GENE_NAME', 'Consensus_Gene')\n",
      "        .rename('SNPEFF_IMPACT', 'Consensus_Impact')\n",
      "        .rename('SNPEFF_TRANSCRIPT_ID', 'Consensus_Transcript')\n",
      "        .addfieldusingcontext('Consensus_distance_previous', upstream)\n",
      "        .addfieldusingcontext('Consensus_distance_next', downstream)\n",
      "        .addfield('Consensus_distance_nearest', lambda rec: nearest(rec['Consensus_distance_previous'], rec['Consensus_distance_next']))\n",
      "        .addfield('Consensus_mode', determine_mode)\n",
      "    )\n",
      "        \n",
      "    return(tbl_consensus)\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 397
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tbl_consensus_7G8 = merge_consensus_alignment()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 398
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tbl_consensus_7G8.display(10, index_header=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>0|CHROM</th>\n",
        "<th>1|POS</th>\n",
        "<th>2|REF</th>\n",
        "<th>3|ALT</th>\n",
        "<th>4|consensus_ALTs</th>\n",
        "<th>5|Consensus_RegionType</th>\n",
        "<th>6|Consensus_is_coding</th>\n",
        "<th>7|Consensus_AAChange</th>\n",
        "<th>8|Consensus_Effect</th>\n",
        "<th>9|Consensus_Gene</th>\n",
        "<th>10|Consensus_Impact</th>\n",
        "<th>11|Consensus_Transcript</th>\n",
        "<th>12|RPA</th>\n",
        "<th>13|RU</th>\n",
        "<th>14|STR</th>\n",
        "<th>15|Consensus_RPA_REF</th>\n",
        "<th>16|Consensus_RPA_ALT</th>\n",
        "<th>17|Consensus_distance_previous</th>\n",
        "<th>18|Consensus_distance_next</th>\n",
        "<th>19|Consensus_distance_nearest</th>\n",
        "<th>20|Consensus_mode</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>M76611</td>\n",
        "<td style='text-align: right'>772</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td>[C]</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>920</td>\n",
        "<td style='text-align: right'>920</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>M76611</td>\n",
        "<td style='text-align: right'>1692</td>\n",
        "<td>G</td>\n",
        "<td>A</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>920</td>\n",
        "<td style='text-align: right'>2487</td>\n",
        "<td style='text-align: right'>920</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>M76611</td>\n",
        "<td style='text-align: right'>4179</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2487</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2487</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>PFC10_API_IRAB</td>\n",
        "<td style='text-align: right'>15405</td>\n",
        "<td>AAATTAAAAATATGT</td>\n",
        "<td>A</td>\n",
        "<td>[A]</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[2, 1]</td>\n",
        "<td>AATTAAAAATATGT</td>\n",
        "<td>True</td>\n",
        "<td>2</td>\n",
        "<td>1</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1251</td>\n",
        "<td style='text-align: right'>1251</td>\n",
        "<td>INDEL</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>PFC10_API_IRAB</td>\n",
        "<td style='text-align: right'>16656</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>[G]</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1251</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1251</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>102910</td>\n",
        "<td>AT</td>\n",
        "<td>A</td>\n",
        "<td>[A]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td>INDEL</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>102916</td>\n",
        "<td>AT</td>\n",
        "<td>A</td>\n",
        "<td>[A]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td>INDEL</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>102926</td>\n",
        "<td>T</td>\n",
        "<td>TA</td>\n",
        "<td>[TA]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>[1, 2]</td>\n",
        "<td>A</td>\n",
        "<td>True</td>\n",
        "<td>1</td>\n",
        "<td>2</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>INDEL</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>102930</td>\n",
        "<td>T</td>\n",
        "<td>TA</td>\n",
        "<td>[TA]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>[1, 2]</td>\n",
        "<td>A</td>\n",
        "<td>True</td>\n",
        "<td>1</td>\n",
        "<td>2</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>1478</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>INDEL</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>104408</td>\n",
        "<td>A</td>\n",
        "<td>AATATATATATAT</td>\n",
        "<td>[AATATATATATAT]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>[17, 23]</td>\n",
        "<td>AT</td>\n",
        "<td>True</td>\n",
        "<td>17</td>\n",
        "<td>23</td>\n",
        "<td style='text-align: right'>1478</td>\n",
        "<td style='text-align: right'>4044</td>\n",
        "<td style='text-align: right'>1478</td>\n",
        "<td>INDEL</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n",
        "<p><strong>...</strong></p>"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 399
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(tbl_consensus_7G8\n",
      "    .addfield('is_clustered', lambda rec: rec['Consensus_distance_nearest'] <= 10)\n",
      "    .valuecounts('STR', 'is_clustered')\n",
      ")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>STR</th>\n",
        "<th>is_clustered</th>\n",
        "<th>count</th>\n",
        "<th>frequency</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>7659</td>\n",
        "<td style='text-align: right'>0.4093751670319098</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>5899</td>\n",
        "<td style='text-align: right'>0.315302795446042</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>4721</td>\n",
        "<td style='text-align: right'>0.2523384467368646</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>430</td>\n",
        "<td style='text-align: right'>0.0229835907851836</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 409,
       "text": [
        "+------+--------------+-------+--------------------+\n",
        "| STR  | is_clustered | count | frequency          |\n",
        "+======+==============+=======+====================+\n",
        "| True | False        |  7659 | 0.4093751670319098 |\n",
        "+------+--------------+-------+--------------------+\n",
        "| None | True         |  5899 |  0.315302795446042 |\n",
        "+------+--------------+-------+--------------------+\n",
        "| None | False        |  4721 | 0.2523384467368646 |\n",
        "+------+--------------+-------+--------------------+\n",
        "| True | True         |   430 | 0.0229835907851836 |\n",
        "+------+--------------+-------+--------------------+"
       ]
      }
     ],
     "prompt_number": 409
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(tbl_consensus_7G8\n",
      "    .addfield('is_clustered', lambda rec: rec['Consensus_distance_nearest'] <= 20)\n",
      "    .valuecounts('Consensus_mode', 'STR', 'is_clustered')\n",
      "    .displayall()\n",
      ")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>Consensus_mode</th>\n",
        "<th>STR</th>\n",
        "<th>is_clustered</th>\n",
        "<th>count</th>\n",
        "<th>frequency</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>7490</td>\n",
        "<td style='text-align: right'>0.40034208135122135</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>5097</td>\n",
        "<td style='text-align: right'>0.27243572612111816</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>2598</td>\n",
        "<td style='text-align: right'>0.1388636485114116</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>1740</td>\n",
        "<td style='text-align: right'>0.09300336736330109</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>1185</td>\n",
        "<td style='text-align: right'>0.06333850018707574</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>599</td>\n",
        "<td style='text-align: right'>0.03201667646587204</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 412
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(tbl_whole_genome_comparisons['alistair_ann']\n",
      ".selecteq('Sample', '7G8')\n",
      ".selecteq('IsInGATK', False)\n",
      ".addfield('is_clustered', lambda rec: rec['Truth_distance_nearest'] <= 20)\n",
      ".valuecounts('Truth_mode', 'STR', 'is_clustered')\n",
      ".display(20)\n",
      ")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>Truth_mode</th>\n",
        "<th>STR</th>\n",
        "<th>is_clustered</th>\n",
        "<th>count</th>\n",
        "<th>frequency</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>8441</td>\n",
        "<td style='text-align: right'>0.2780852605916848</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>8252</td>\n",
        "<td style='text-align: right'>0.27185873361006785</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>6649</td>\n",
        "<td style='text-align: right'>0.21904856032153916</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>4893</td>\n",
        "<td style='text-align: right'>0.16119786519074916</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>1300</td>\n",
        "<td style='text-align: right'>0.04282796336561903</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>819</td>\n",
        "<td style='text-align: right'>0.026981616920339988</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 420
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tbl_whole_genome_comparisons['alistair_ann']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>CHROM</th>\n",
        "<th>POS</th>\n",
        "<th>REF</th>\n",
        "<th>ALT</th>\n",
        "<th>truth_ALTs</th>\n",
        "<th>Truth_RegionType</th>\n",
        "<th>Truth_is_coding</th>\n",
        "<th>Truth_AAChange</th>\n",
        "<th>Truth_Effect</th>\n",
        "<th>Truth_Gene</th>\n",
        "<th>Truth_Impact</th>\n",
        "<th>Truth_Transcript</th>\n",
        "<th>RPA</th>\n",
        "<th>RU</th>\n",
        "<th>STR</th>\n",
        "<th>Truth_RPA_REF</th>\n",
        "<th>Truth_RPA_ALT</th>\n",
        "<th>Truth_distance_previous</th>\n",
        "<th>Truth_distance_next</th>\n",
        "<th>Truth_distance_nearest</th>\n",
        "<th>Truth_mode</th>\n",
        "<th>ID</th>\n",
        "<th>gatk_ALTs</th>\n",
        "<th>gatk_QUAL</th>\n",
        "<th>gatk_FILTER</th>\n",
        "<th>SAMPLE</th>\n",
        "<th>gatk_AD</th>\n",
        "<th>gatk_DP</th>\n",
        "<th>GQ</th>\n",
        "<th>gatk_GT</th>\n",
        "<th>PGT</th>\n",
        "<th>PID</th>\n",
        "<th>PL</th>\n",
        "<th>gatk_AC</th>\n",
        "<th>gatk_AF</th>\n",
        "<th>gatk_AN</th>\n",
        "<th>gatk_BaseQRankSum</th>\n",
        "<th>gatk_ClippingRankSum</th>\n",
        "<th>gatk_FS</th>\n",
        "<th>GC</th>\n",
        "<th>gatk_MLEAC</th>\n",
        "<th>gatk_MLEAF</th>\n",
        "<th>gatk_MQ</th>\n",
        "<th>gatk_MQRankSum</th>\n",
        "<th>NEGATIVE_TRAIN_SITE</th>\n",
        "<th>POSITIVE_TRAIN_SITE</th>\n",
        "<th>gatk_QD</th>\n",
        "<th>gatk_RPA</th>\n",
        "<th>gatk_RU</th>\n",
        "<th>gatk_ReadPosRankSum</th>\n",
        "<th>gatk_RegionType</th>\n",
        "<th>SNPEFF_AMINO_ACID_CHANGE</th>\n",
        "<th>SNPEFF_CODON_CHANGE</th>\n",
        "<th>gatk_SNPEFF_EFFECT</th>\n",
        "<th>SNPEFF_EXON_ID</th>\n",
        "<th>gatk_SNPEFF_FUNCTIONAL_CLASS</th>\n",
        "<th>SNPEFF_GENE_BIOTYPE</th>\n",
        "<th>SNPEFF_GENE_NAME</th>\n",
        "<th>gatk_SNPEFF_IMPACT</th>\n",
        "<th>SNPEFF_TRANSCRIPT_ID</th>\n",
        "<th>gatk_SOR</th>\n",
        "<th>gatk_STR</th>\n",
        "<th>gatk_VQSLOD</th>\n",
        "<th>VariantType</th>\n",
        "<th>gatk_culprit</th>\n",
        "<th>gatk_set</th>\n",
        "<th>gatk_RPA_REF</th>\n",
        "<th>gatk_RPA_ALT</th>\n",
        "<th>GATK_is_coding</th>\n",
        "<th>is_het</th>\n",
        "<th>temp_REF</th>\n",
        "<th>temp_ALT</th>\n",
        "<th>GATK_distance_previous</th>\n",
        "<th>GATK_distance_next</th>\n",
        "<th>GATK_distance_nearest</th>\n",
        "<th>GATK_mode</th>\n",
        "<th>IsInTruth</th>\n",
        "<th>IsInGATK</th>\n",
        "<th>RegionType</th>\n",
        "<th>IsCoding</th>\n",
        "<th>Effect</th>\n",
        "<th>mode</th>\n",
        "<th>IsAccessible</th>\n",
        "<th>Sample</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>Pf3D7_06_v3</td>\n",
        "<td style='text-align: right'>1030211</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>A316E</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>870</td>\n",
        "<td style='text-align: right'>1015</td>\n",
        "<td style='text-align: right'>870</td>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>8385.68</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 113]</td>\n",
        "<td style='text-align: right'>113</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>1/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[3518, 338, 0]</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>30.38</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>A316E</td>\n",
        "<td>gCa/gAa</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td style='text-align: right'>1.043</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>12.64</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>1666</td>\n",
        "<td style='text-align: right'>1015</td>\n",
        "<td style='text-align: right'>1015</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_06_v3</td>\n",
        "<td style='text-align: right'>1030211</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>A316E</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>742</td>\n",
        "<td style='text-align: right'>88</td>\n",
        "<td style='text-align: right'>88</td>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>8385.68</td>\n",
        "<td>[]</td>\n",
        "<td>PG0084-C</td>\n",
        "<td>[0, 121]</td>\n",
        "<td style='text-align: right'>121</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>1/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[3684, 362, 0]</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>30.38</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>A316E</td>\n",
        "<td>gCa/gAa</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td style='text-align: right'>1.043</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>12.64</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>1666</td>\n",
        "<td style='text-align: right'>88</td>\n",
        "<td style='text-align: right'>88</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>GB4</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_06_v3</td>\n",
        "<td style='text-align: right'>1030211</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>A316E</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>870</td>\n",
        "<td style='text-align: right'>1015</td>\n",
        "<td style='text-align: right'>870</td>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>8385.68</td>\n",
        "<td>[]</td>\n",
        "<td>ERS740937</td>\n",
        "<td>[0, 15]</td>\n",
        "<td style='text-align: right'>15</td>\n",
        "<td style='text-align: right'>45</td>\n",
        "<td>1/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[455, 45, 0]</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>30.38</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>A316E</td>\n",
        "<td>gCa/gAa</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td style='text-align: right'>1.043</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>12.64</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>834</td>\n",
        "<td style='text-align: right'>1015</td>\n",
        "<td style='text-align: right'>834</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>KE01</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_06_v3</td>\n",
        "<td style='text-align: right'>1030211</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>A316E</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>742</td>\n",
        "<td style='text-align: right'>595</td>\n",
        "<td style='text-align: right'>595</td>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>8385.68</td>\n",
        "<td>[]</td>\n",
        "<td>ERS740936</td>\n",
        "<td>[0, 11]</td>\n",
        "<td style='text-align: right'>11</td>\n",
        "<td style='text-align: right'>33</td>\n",
        "<td>1/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[309, 33, 0]</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>30.38</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>A316E</td>\n",
        "<td>gCa/gAa</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td style='text-align: right'>1.043</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>12.64</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>742</td>\n",
        "<td style='text-align: right'>595</td>\n",
        "<td style='text-align: right'>595</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>KH02</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_06_v3</td>\n",
        "<td style='text-align: right'>1030211</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>A316E</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>742</td>\n",
        "<td style='text-align: right'>595</td>\n",
        "<td style='text-align: right'>595</td>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>8385.68</td>\n",
        "<td>[]</td>\n",
        "<td>ERS740940</td>\n",
        "<td>[0, 16]</td>\n",
        "<td style='text-align: right'>16</td>\n",
        "<td style='text-align: right'>47</td>\n",
        "<td>1/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[445, 47, 0]</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[10]</td>\n",
        "<td>[1.0]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>30.38</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>A316E</td>\n",
        "<td>gCa/gAa</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0625300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0625300.1</td>\n",
        "<td style='text-align: right'>1.043</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>12.64</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>742</td>\n",
        "<td style='text-align: right'>595</td>\n",
        "<td style='text-align: right'>595</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>GN01</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n",
        "<p><strong>...</strong></p>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 419,
       "text": [
        "+---------------+---------+-----+-----+------------+------------------+-----------------+----------------+-------------------------+-----------------+--------------+-------------------+------+------+------+---------------+---------------+-------------------------+---------------------+------------------------+------------+------+-----------+-----------+-------------+-------------+----------+---------+----+---------+------+------+----------------+---------+---------+---------+-------------------+----------------------+---------+------+------------+------------+---------+----------------+---------------------+---------------------+---------+----------+---------+---------------------+-----------------+--------------------------+---------------------+-------------------------+----------------+------------------------------+---------------------+------------------+--------------------+----------------------+----------+----------+-------------+-------------+--------------+----------+--------------+--------------+----------------+--------+----------+----------+------------------------+--------------------+-----------------------+-----------+-----------+----------+------------+----------+-------------------------+-------+--------------+--------+\n",
        "| CHROM         | POS     | REF | ALT | truth_ALTs | Truth_RegionType | Truth_is_coding | Truth_AAChange | Truth_Effect            | Truth_Gene      | Truth_Impact | Truth_Transcript  | RPA  | RU   | STR  | Truth_RPA_REF | Truth_RPA_ALT | Truth_distance_previous | Truth_distance_next | Truth_distance_nearest | Truth_mode | ID   | gatk_ALTs | gatk_QUAL | gatk_FILTER | SAMPLE      | gatk_AD  | gatk_DP | GQ | gatk_GT | PGT  | PID  | PL             | gatk_AC | gatk_AF | gatk_AN | gatk_BaseQRankSum | gatk_ClippingRankSum | gatk_FS | GC   | gatk_MLEAC | gatk_MLEAF | gatk_MQ | gatk_MQRankSum | NEGATIVE_TRAIN_SITE | POSITIVE_TRAIN_SITE | gatk_QD | gatk_RPA | gatk_RU | gatk_ReadPosRankSum | gatk_RegionType | SNPEFF_AMINO_ACID_CHANGE | SNPEFF_CODON_CHANGE | gatk_SNPEFF_EFFECT      | SNPEFF_EXON_ID | gatk_SNPEFF_FUNCTIONAL_CLASS | SNPEFF_GENE_BIOTYPE | SNPEFF_GENE_NAME | gatk_SNPEFF_IMPACT | SNPEFF_TRANSCRIPT_ID | gatk_SOR | gatk_STR | gatk_VQSLOD | VariantType | gatk_culprit | gatk_set | gatk_RPA_REF | gatk_RPA_ALT | GATK_is_coding | is_het | temp_REF | temp_ALT | GATK_distance_previous | GATK_distance_next | GATK_distance_nearest | GATK_mode | IsInTruth | IsInGATK | RegionType | IsCoding | Effect                  | mode  | IsAccessible | Sample |\n",
        "+===============+=========+=====+=====+============+==================+=================+================+=========================+=================+==============+===================+======+======+======+===============+===============+=========================+=====================+========================+============+======+===========+===========+=============+=============+==========+=========+====+=========+======+======+================+=========+=========+=========+===================+======================+=========+======+============+============+=========+================+=====================+=====================+=========+==========+=========+=====================+=================+==========================+=====================+=========================+================+==============================+=====================+==================+====================+======================+==========+==========+=============+=============+==============+==========+==============+==============+================+========+==========+==========+========================+====================+=======================+===========+===========+==========+============+==========+=========================+=======+==============+========+\n",
        "| 'Pf3D7_06_v3' | 1030211 | 'G' | 'T' | [T]        | 'Core'           | True            | 'A316E'        | 'NON_SYNONYMOUS_CODING' | 'PF3D7_0625300' | 'MODERATE'   | 'PF3D7_0625300.1' | None | None | None | None          | None          |                     870 |                1015 |                    870 | 'SNP'      | None | [T]       |   8385.68 | []          | 'PG0083-C'  | [0, 113] |     113 | 99 | '1/1'   | None | None | [3518, 338, 0] | [10]    | [1.0]   |      10 | None              | None                 |     0.0 | 38.1 | [10]       | [1.0]      |    60.0 | None           | None                | True                |   30.38 | None     | None    | None                | 'Core'          | 'A316E'                  | 'gCa/gAa'           | 'NON_SYNONYMOUS_CODING' | '1'            | 'MISSENSE'                   | 'protein_coding'    | 'PF3D7_0625300'  | 'MODERATE'         | 'PF3D7_0625300.1'    |    1.043 | None     |       12.64 | 'SNP'       | 'FS'         | 'snp'    | None         | None         | True           | False  | 'G'      | 'T'      |                   1666 |               1015 |                  1015 | 'SNP'     | True      | True     | 'Core'     | True     | 'NON_SYNONYMOUS_CODING' | 'SNP' | True         | '7G8'  |\n",
        "+---------------+---------+-----+-----+------------+------------------+-----------------+----------------+-------------------------+-----------------+--------------+-------------------+------+------+------+---------------+---------------+-------------------------+---------------------+------------------------+------------+------+-----------+-----------+-------------+-------------+----------+---------+----+---------+------+------+----------------+---------+---------+---------+-------------------+----------------------+---------+------+------------+------------+---------+----------------+---------------------+---------------------+---------+----------+---------+---------------------+-----------------+--------------------------+---------------------+-------------------------+----------------+------------------------------+---------------------+------------------+--------------------+----------------------+----------+----------+-------------+-------------+--------------+----------+--------------+--------------+----------------+--------+----------+----------+------------------------+--------------------+-----------------------+-----------+-----------+----------+------------+----------+-------------------------+-------+--------------+--------+\n",
        "| 'Pf3D7_06_v3' | 1030211 | 'G' | 'T' | [T]        | 'Core'           | True            | 'A316E'        | 'NON_SYNONYMOUS_CODING' | 'PF3D7_0625300' | 'MODERATE'   | 'PF3D7_0625300.1' | None | None | None | None          | None          |                     742 |                  88 |                     88 | 'SNP'      | None | [T]       |   8385.68 | []          | 'PG0084-C'  | [0, 121] |     121 | 99 | '1/1'   | None | None | [3684, 362, 0] | [10]    | [1.0]   |      10 | None              | None                 |     0.0 | 38.1 | [10]       | [1.0]      |    60.0 | None           | None                | True                |   30.38 | None     | None    | None                | 'Core'          | 'A316E'                  | 'gCa/gAa'           | 'NON_SYNONYMOUS_CODING' | '1'            | 'MISSENSE'                   | 'protein_coding'    | 'PF3D7_0625300'  | 'MODERATE'         | 'PF3D7_0625300.1'    |    1.043 | None     |       12.64 | 'SNP'       | 'FS'         | 'snp'    | None         | None         | True           | False  | 'G'      | 'T'      |                   1666 |                 88 |                    88 | 'SNP'     | True      | True     | 'Core'     | True     | 'NON_SYNONYMOUS_CODING' | 'SNP' | True         | 'GB4'  |\n",
        "+---------------+---------+-----+-----+------------+------------------+-----------------+----------------+-------------------------+-----------------+--------------+-------------------+------+------+------+---------------+---------------+-------------------------+---------------------+------------------------+------------+------+-----------+-----------+-------------+-------------+----------+---------+----+---------+------+------+----------------+---------+---------+---------+-------------------+----------------------+---------+------+------------+------------+---------+----------------+---------------------+---------------------+---------+----------+---------+---------------------+-----------------+--------------------------+---------------------+-------------------------+----------------+------------------------------+---------------------+------------------+--------------------+----------------------+----------+----------+-------------+-------------+--------------+----------+--------------+--------------+----------------+--------+----------+----------+------------------------+--------------------+-----------------------+-----------+-----------+----------+------------+----------+-------------------------+-------+--------------+--------+\n",
        "| 'Pf3D7_06_v3' | 1030211 | 'G' | 'T' | [T]        | 'Core'           | True            | 'A316E'        | 'NON_SYNONYMOUS_CODING' | 'PF3D7_0625300' | 'MODERATE'   | 'PF3D7_0625300.1' | None | None | None | None          | None          |                     870 |                1015 |                    870 | 'SNP'      | None | [T]       |   8385.68 | []          | 'ERS740937' | [0, 15]  |      15 | 45 | '1/1'   | None | None | [455, 45, 0]   | [10]    | [1.0]   |      10 | None              | None                 |     0.0 | 38.1 | [10]       | [1.0]      |    60.0 | None           | None                | True                |   30.38 | None     | None    | None                | 'Core'          | 'A316E'                  | 'gCa/gAa'           | 'NON_SYNONYMOUS_CODING' | '1'            | 'MISSENSE'                   | 'protein_coding'    | 'PF3D7_0625300'  | 'MODERATE'         | 'PF3D7_0625300.1'    |    1.043 | None     |       12.64 | 'SNP'       | 'FS'         | 'snp'    | None         | None         | True           | False  | 'G'      | 'T'      |                    834 |               1015 |                   834 | 'SNP'     | True      | True     | 'Core'     | True     | 'NON_SYNONYMOUS_CODING' | 'SNP' | True         | 'KE01' |\n",
        "+---------------+---------+-----+-----+------------+------------------+-----------------+----------------+-------------------------+-----------------+--------------+-------------------+------+------+------+---------------+---------------+-------------------------+---------------------+------------------------+------------+------+-----------+-----------+-------------+-------------+----------+---------+----+---------+------+------+----------------+---------+---------+---------+-------------------+----------------------+---------+------+------------+------------+---------+----------------+---------------------+---------------------+---------+----------+---------+---------------------+-----------------+--------------------------+---------------------+-------------------------+----------------+------------------------------+---------------------+------------------+--------------------+----------------------+----------+----------+-------------+-------------+--------------+----------+--------------+--------------+----------------+--------+----------+----------+------------------------+--------------------+-----------------------+-----------+-----------+----------+------------+----------+-------------------------+-------+--------------+--------+\n",
        "| 'Pf3D7_06_v3' | 1030211 | 'G' | 'T' | [T]        | 'Core'           | True            | 'A316E'        | 'NON_SYNONYMOUS_CODING' | 'PF3D7_0625300' | 'MODERATE'   | 'PF3D7_0625300.1' | None | None | None | None          | None          |                     742 |                 595 |                    595 | 'SNP'      | None | [T]       |   8385.68 | []          | 'ERS740936' | [0, 11]  |      11 | 33 | '1/1'   | None | None | [309, 33, 0]   | [10]    | [1.0]   |      10 | None              | None                 |     0.0 | 38.1 | [10]       | [1.0]      |    60.0 | None           | None                | True                |   30.38 | None     | None    | None                | 'Core'          | 'A316E'                  | 'gCa/gAa'           | 'NON_SYNONYMOUS_CODING' | '1'            | 'MISSENSE'                   | 'protein_coding'    | 'PF3D7_0625300'  | 'MODERATE'         | 'PF3D7_0625300.1'    |    1.043 | None     |       12.64 | 'SNP'       | 'FS'         | 'snp'    | None         | None         | True           | False  | 'G'      | 'T'      |                    742 |                595 |                   595 | 'SNP'     | True      | True     | 'Core'     | True     | 'NON_SYNONYMOUS_CODING' | 'SNP' | True         | 'KH02' |\n",
        "+---------------+---------+-----+-----+------------+------------------+-----------------+----------------+-------------------------+-----------------+--------------+-------------------+------+------+------+---------------+---------------+-------------------------+---------------------+------------------------+------------+------+-----------+-----------+-------------+-------------+----------+---------+----+---------+------+------+----------------+---------+---------+---------+-------------------+----------------------+---------+------+------------+------------+---------+----------------+---------------------+---------------------+---------+----------+---------+---------------------+-----------------+--------------------------+---------------------+-------------------------+----------------+------------------------------+---------------------+------------------+--------------------+----------------------+----------+----------+-------------+-------------+--------------+----------+--------------+--------------+----------------+--------+----------+----------+------------------------+--------------------+-----------------------+-----------+-----------+----------+------------+----------+-------------------------+-------+--------------+--------+\n",
        "| 'Pf3D7_06_v3' | 1030211 | 'G' | 'T' | [T]        | 'Core'           | True            | 'A316E'        | 'NON_SYNONYMOUS_CODING' | 'PF3D7_0625300' | 'MODERATE'   | 'PF3D7_0625300.1' | None | None | None | None          | None          |                     742 |                 595 |                    595 | 'SNP'      | None | [T]       |   8385.68 | []          | 'ERS740940' | [0, 16]  |      16 | 47 | '1/1'   | None | None | [445, 47, 0]   | [10]    | [1.0]   |      10 | None              | None                 |     0.0 | 38.1 | [10]       | [1.0]      |    60.0 | None           | None                | True                |   30.38 | None     | None    | None                | 'Core'          | 'A316E'                  | 'gCa/gAa'           | 'NON_SYNONYMOUS_CODING' | '1'            | 'MISSENSE'                   | 'protein_coding'    | 'PF3D7_0625300'  | 'MODERATE'         | 'PF3D7_0625300.1'    |    1.043 | None     |       12.64 | 'SNP'       | 'FS'         | 'snp'    | None         | None         | True           | False  | 'G'      | 'T'      |                    742 |                595 |                   595 | 'SNP'     | True      | True     | 'Core'     | True     | 'NON_SYNONYMOUS_CODING' | 'SNP' | True         | 'GN01' |\n",
        "+---------------+---------+-----+-----+------------+------------------+-----------------+----------------+-------------------------+-----------------+--------------+-------------------+------+------+------+---------------+---------------+-------------------------+---------------------+------------------------+------------+------+-----------+-----------+-------------+-------------+----------+---------+----+---------+------+------+----------------+---------+---------+---------+-------------------+----------------------+---------+------+------------+------------+---------+----------------+---------------------+---------------------+---------+----------+---------+---------------------+-----------------+--------------------------+---------------------+-------------------------+----------------+------------------------------+---------------------+------------------+--------------------+----------------------+----------+----------+-------------+-------------+--------------+----------+--------------+--------------+----------------+--------+----------+----------+------------------------+--------------------+-----------------------+-----------+-----------+----------+------------+----------+-------------------------+-------+--------------+--------+\n",
        "..."
       ]
      }
     ],
     "prompt_number": 419
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(tbl_consensus_7G8\n",
      "    .addfield('is_clustered', lambda rec: rec['Consensus_distance_nearest'] <= 100)\n",
      "    .valuecounts('Consensus_mode', 'STR', 'is_clustered')\n",
      "    .displayall()\n",
      ")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>Consensus_mode</th>\n",
        "<th>STR</th>\n",
        "<th>is_clustered</th>\n",
        "<th>count</th>\n",
        "<th>frequency</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>5856</td>\n",
        "<td style='text-align: right'>0.3130044363675237</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>5836</td>\n",
        "<td style='text-align: right'>0.31193543214495695</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>2405</td>\n",
        "<td style='text-align: right'>0.12854775776364316</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>2233</td>\n",
        "<td style='text-align: right'>0.11935432144956973</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>1933</td>\n",
        "<td style='text-align: right'>0.10331925811106954</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td style='text-align: right'>446</td>\n",
        "<td style='text-align: right'>0.023838794163236945</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 413
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len(tbl_consensus_7G8)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 400,
       "text": [
        "18710"
       ]
      }
     ],
     "prompt_number": 400
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "consensus_distances_7G8 = tbl_consensus_7G8['Consensus_distance_nearest'].array()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 401
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ax = hist(consensus_distances_7G8, bins=arange(0, 100, 1))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "display_data",
       "png": "iVBORw0KGgoAAAANSUhEUgAAAcYAAAE7CAYAAABHbiHuAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAAMTQAADE0B0s6tTgAAHjdJREFUeJzt3X9wVPX97/HX5heR3azKlO+dVicp30yImBqmK0FgJDLO\nLaYOMyqYwWwk4+oMGgeqrmMJAgbH0VhpmqEDgqLdjhsuIVPTqXqdqS0jbgvU2EERgsaaqm39yQ9p\n92whu3HP/eN73X7AkM1udklCno8ZZjifs+ec9/nMZl97Pnt+OGzbtgUAACRJOaNdAAAAYwnBCACA\ngWAEAMBAMAIAYBg3wRgIBEa7hHGLvksffTcy9F/66Lv0jbTvCMYJgL5LH303MvRf+ui79E2YYAQA\n4FwYVjAeOHBAy5YtkyQdO3ZMjY2NuvXWW1VfX69//OMfkqTOzk4tWbJES5cu1e7duyVJp06d0sqV\nK1VfX6/ly5fr+PHj2dkLAAAyJC/ZC7Zt26YXXnhBTqdTkrRhwwbdcMMNqqmp0euvv66//OUvmjRp\nkoLBoLq6utTf36+6ujrNmzdPO3bsUHl5uVasWKGXX35ZW7Zs0Zo1a7K+UwAApCvpEWNJSYk2bdqk\nr2+Q8+abb+qzzz6Tz+fTiy++qDlz5ujtt9+Wx+NRfn6+XC6XSkpK1Nvbq/3796u6ulqSNH/+fO3b\nty+7ewMAwAglDcaFCxcqNzc3Mf3xxx/rwgsvVCAQ0Le//W1t27ZNkUhERUVFidc4nU5ZliXLshJH\nmk6nU+FwOAu7AABA5iQdSj3TRRddpGuvvVaSdO2116qtrU3f+973FIlEEq/5OihdLleiPRKJyO12\nD7nuQCBw1rOJvvrqK7W0tKRaLiQVFxfTd2mi70aG/ksffZe+r776KjFaeSafzyefzzfk8ikHo8fj\n0e7du3XDDTeou7tbZWVlqqysVFtbm6LRqPr7+9XX16fp06fL4/EoFAqpsrJSoVBIs2bNGnLdQxXc\n0tKi1atXp1ouAGACGkleDDsYHQ6HJKmpqUlr167Vjh075Ha71draqqKiIjU0NMjr9Soej8vv96ug\noEB1dXVatWqVvF6vCgoK1NramnahAACcC47x8tgpjhgBAMMx0rzgAn8AAAwEIwAABoIRAAADwQgA\ngIFgBADAQDACAGAgGAEAMBCMAAAYCEYAAAwEIwAAhpRvIj5WNNzeqKMnTiamv3XRBXruF1tGsSIA\nwPlg3Abj0RMnlTvtpv9Mf/DrUawGAHC+YCgVAAADwQgAgIFgBADAQDACAGAgGAEAMBCMAAAYCEYA\nAAwEIwAABoIRAAADwQgAgIFgBADAQDACAGAgGAEAMBCMAAAYhhWMBw4c0LJly05re/HFF3XLLbck\npjs7O7VkyRItXbpUu3fvliSdOnVKK1euVH19vZYvX67jx49nrnIAALIg6fMYt23bphdeeEFOpzPR\ndvjwYT3//POJ6SNHjigYDKqrq0v9/f2qq6vTvHnztGPHDpWXl2vFihV6+eWXtWXLFq1ZsyYrO/LO\n4UO6fvFtiWkeXAwASEfSI8aSkhJt2rRJtm1Lkr788ku1tbXpwQcfTLS9/fbb8ng8ys/Pl8vlUklJ\niXp7e7V//35VV1dLkubPn699+/Zlb0/yCpU77abEv6MnTmZvWwCA81bSYFy4cKFyc3MlSfF4XGvW\nrFFTU5MmT56ceI1lWSoqKkpMO51OWZYly7ISR5pOp1PhcDjT9QMAkFFJh1JNhw4d0t/+9jetX79e\n0WhU77//vlpaWnTVVVcpEokkXheJRFRUVCSXy5Voj0QicrvdQ64/EAgoEAgMOq+4uDiVUgEAE1RP\nT09itPJMPp9PPp9vyOVTCsbKykq99NJLkqSPP/5Yfr9fq1ev1pEjR9TW1qZoNKr+/n719fVp+vTp\n8ng8CoVCqqysVCgU0qxZs4Zc/1AFt7S0pFIqAGCCqqioUHt7e9rLDzsYHQ7HadO2bSfapk6dqoaG\nBnm9XsXjcfn9fhUUFKiurk6rVq2S1+tVQUGBWltb0y4UAIBzYVjBeOmll6qjo2PIttraWtXW1p72\nmsLCQm3cuDEDZQIAcG5wgT8AAAaCEQAAA8EIAICBYAQAwEAwAgBgIBgBADAQjAAAGAhGAAAMBCMA\nAAaCEQAAA8EIAICBYAQAwEAwAgBgIBgBADAQjAAAGAhGAAAMBCMAAAaCEQAAA8EIAICBYAQAwEAw\nAgBgIBgBADAQjAAAGAhGAAAMBCMAAAaCEQAAw7CC8cCBA1q2bJkk6Z133lF9fb2WLVumO+64Q8eO\nHZMkdXZ2asmSJVq6dKl2794tSTp16pRWrlyp+vp6LV++XMePH8/OXgAAkCFJg3Hbtm1au3atYrGY\nJOmxxx7TunXrFAwGtXDhQm3btk1Hjx5VMBhUR0eHnn32WbW2tioajWrHjh0qLy/X9u3bdeONN2rL\nli1Z3yEAAEYiaTCWlJRo06ZNsm1bkvSzn/1Ml112mSRpYGBAkyZN0ttvvy2Px6P8/Hy5XC6VlJSo\nt7dX+/fvV3V1tSRp/vz52rdvXxZ3BQCAkUsajAsXLlRubm5ieurUqZKk/fv3a/v27brttttkWZaK\niooSr3E6nbIsS5Zlyel0JtrC4XCm6wcAIKPy0lno5Zdf1tatW/X000/r4osvlsvlUiQSScyPRCIq\nKio6rT0SicjtdmemagAAsiTlYPzNb36jzs5OBYNBXXjhhZKkyspKtbW1KRqNqr+/X319fZo+fbo8\nHo9CoZAqKysVCoU0a9asIdcdCAQUCAQGnVdcXJxqqQCACainpyfxM96ZfD6ffD7fkMsPOxgdDofi\n8bgee+wxfec739GKFSskSVdddZVWrFihhoYGeb1exeNx+f1+FRQUqK6uTqtWrZLX61VBQYFaW1uH\n3MZQBbe0tAy3VADABFZRUaH29va0lx9WMF566aXq6OiQJL3++uuDvqa2tla1tbWntRUWFmrjxo1p\nFwcAwLnGBf4AABgIRgAADAQjAAAGghEAAAPBCACAgWAEAMBAMAIAYCAYAQAwEIwAABgIRgAADAQj\nAAAGghEAAAPBCACAgWAEAMBAMAIAYCAYAQAwEIwAABgIRgAADAQjAAAGghEAAAPBCACAgWAEAMBA\nMAIAYCAYAQAwEIwAABgIRgAADMMKxgMHDmjZsmWSpI8++kh1dXWqr6/X+vXrZdu2JKmzs1NLlizR\n0qVLtXv3bknSqVOntHLlStXX12v58uU6fvx4dvYCAIAMSRqM27Zt09q1axWLxSRJLS0t8vv92r59\nu2zb1q5du3TkyBEFg0F1dHTo2WefVWtrq6LRqHbs2KHy8nJt375dN954o7Zs2ZL1HQIAYCSSBmNJ\nSYk2bdqUODI8fPiwqqqqJEnV1dXau3evDh48KI/Ho/z8fLlcLpWUlKi3t1f79+9XdXW1JGn+/Pna\nt29fFncFAICRSxqMCxcuVG5ubmL664CUJKfTqXA4LMuyVFRUdFq7ZVmyLEtOp/O01wIAMJalfPJN\nTs5/FrEsS263Wy6XS5FIJNEeiURUVFR0WnskEpHb7c5AyQAAZE9eqgvMmDFD3d3dmj17tkKhkObO\nnavKykq1tbUpGo2qv79ffX19mj59ujwej0KhkCorKxUKhTRr1qwh1x0IBBQIBAadV1xcnGqpAIAJ\nqKenJ/Ez3pl8Pp98Pt+Qyw87GB0OhySpqalJ69atUywWU2lpqWpqauRwONTQ0CCv16t4PC6/36+C\nggLV1dVp1apV8nq9KigoUGtr65DbGKrglpaW4ZYKAJjAKioq1N7envbywwrGSy+9VB0dHZKk7373\nuwoGg994TW1trWpra09rKyws1MaNG9MuDgCAc40L/AEAMBCMAAAYCEYAAAwEIwAABoIRAAADwQgA\ngIFgBADAQDACAGAgGAEAMBCMAAAYCEYAAAwEIwAABoIRAABDys9jHC/eOXxI1y++LTH9rYsu0HO/\n2DJ6BQEAxoXzNhiVV6jcaTclJo9+8OtRLAYAMF4wlAoAgIFgBADAQDACAGAgGAEAMBCMAAAYCEYA\nAAwEIwAABoIRAAADwQgAgIFgBADAQDACAGBI616p8Xhca9as0YcffqicnBw98sgjys3NVVNTk3Jy\nclRWVqbm5mY5HA51dnZq586dysvLU2NjoxYsWJDhXQAAIHPSCsY//vGPOnnypHbs2KG9e/eqra1N\nAwMD8vv9qqqqUnNzs3bt2qWZM2cqGAyqq6tL/f39qqur07x581RQUJDp/QAAICPSGkotLCxUOByW\nbdsKh8PKz89XT0+PqqqqJEnV1dXau3evDh48KI/Ho/z8fLlcLpWUlKi3tzejOwAAQCaldcTo8XgU\njUZVU1OjEydOaOvWrXrjjTcS851Op8LhsCzLUlFR0WntlmWNvGoAALIkrWB85pln5PF4dN999+mz\nzz5TQ0ODBgYGEvMty5Lb7ZbL5VIkEkm0RyIRud3ukVcNAECWpBWMJ0+elNPplCS53W4NDAzo8ssv\nV3d3t2bPnq1QKKS5c+eqsrJSbW1tikaj6u/vV19fn8rKys663kAgoEAgMOi84uLidEoFAEwwPT09\nqq6uHnSez+eTz+cbcvm0gvGOO+7Q6tWr5fV6NTAwoPvvv18VFRVat26dYrGYSktLVVNTI4fDoYaG\nBnm9XsXjcfn9/iFPvBmq4JaWlnRKBQBMMBUVFWpvb097+bSC0e12a/Pmzd9oDwaD32irra1VbW1t\nOpsBAOCc4wJ/AAAMBCMAAAaCEQAAA8EIAIAhrZNvxqN3Dh/S9YtvS0x/66IL9NwvtoxeQQCAMWnC\nBKPyCpU77abE5NEPfj2KxQAAxiqGUgEAMBCMAAAYCEYAAAwEIwAABoIRAAADwQgAgIFgBADAQDAC\nAGAgGAEAMBCMAAAYCEYAAAwEIwAABoIRAAADwQgAgIFgBADAQDACAGAgGAEAMBCMAAAYCEYAAAwE\nIwAAhrx0F3zqqaf06quvKhaL6dZbb5XH41FTU5NycnJUVlam5uZmORwOdXZ2aufOncrLy1NjY6MW\nLFiQwfIBAMistILx9ddf15tvvqmOjg79+9//1jPPPKNXXnlFfr9fVVVVam5u1q5duzRz5kwFg0F1\ndXWpv79fdXV1mjdvngoKCjK9HwAAZERawbhnzx6Vl5fr7rvvlmVZ+vGPf6xf/epXqqqqkiRVV1dr\nz549ysnJkcfjUX5+vvLz81VSUqLe3l5dccUVGd0JAAAyJa1gPH78uD799FM99dRT+vvf/6677rpL\ntm0n5judToXDYVmWpaKiotPaLcsaedUAAGRJWsF48cUXq7S0VHl5eZo2bZomTZqkL774IjHfsiy5\n3W65XC5FIpFEeyQSkdvtHnnVAABkSVrBeOWVV+q5556Tz+fT559/rlOnTmnOnDnq7u7W7NmzFQqF\nNHfuXFVWVqqtrU3RaFT9/f3q6+tTWVnZWdcbCAQUCAQGnVdcXJxOqQCACaanp0fV1dWDzvP5fPL5\nfEMun1YwLliwQG+88YZuvvlmxeNxNTc365JLLtG6desUi8VUWlqqmpoaORwONTQ0yOv1Kh6Py+/3\nD3nizVAFt7S0pFMqAGCCqaioUHt7e9rLp325xgMPPPCNtmAw+I222tpa1dbWprsZAADOKS7wBwDA\nQDACAGAgGAEAMBCMAAAYCEYAAAwEIwAABoIRAAADwQgAgIFgBADAQDACAGAgGAEAMBCMAAAYCEYA\nAAwEIwAABoIRAAADwQgAgIFgBADAQDACAGAgGAEAMBCMAAAYCEYAAAwEIwAABoIRAABD3mgXMBY0\n3N6ooydOJqa/ddEFeu4XW0axIgDAaCEYJR09cVK50276z/QHvx7FagAAo4mhVAAADAQjAACGEQXj\nsWPHdM011+iDDz7QRx99pLq6OtXX12v9+vWybVuS1NnZqSVLlmjp0qXavXt3JmoGACBr0g7GWCym\nhx56SBdccIFs21ZLS4v8fr+2b98u27a1a9cuHTlyRMFgUB0dHXr22WfV2tqqaDSayfoBAMiotIPx\niSeeUF1dnaZOnSpJOnz4sKqqqiRJ1dXV2rt3rw4ePCiPx6P8/Hy5XC6VlJSot7c3M5UDAJAFaQVj\nV1eXpkyZoquvvlqSZNt2YuhUkpxOp8LhsCzLUlFR0WntlmWNsGQAALInrcs1urq65HA4tHfvXr37\n7rtqamrSl19+mZhvWZbcbrdcLpcikUiiPRKJyO12n3W9gUBAgUBg0HnFxcXplAoAmGB6enpUXV09\n6Dyfzyefzzfk8mkFY3t7e+L/y5Yt08MPP6wnnnhC3d3dmj17tkKhkObOnavKykq1tbUpGo2qv79f\nfX19KisrO+t6hyq4paUlnVIBABNMRUXFaTmVqoxc4O9wONTU1KR169YpFouptLRUNTU1cjgcamho\nkNfrVTwel9/vV0FBQSY2CQBAVow4GIPB4KD//1ptba1qa2tHupmMe+fwIV2/+DZJ0vt9H6h82ujW\nAwAYGybuBf55hcqddpNyp92k2FdfjXY1AIAxYuIGIwAAgyAYAQAwEIwAABgIRgAADAQjAAAGghEA\nAAPBCACAgWAEAMBAMAIAYCAYAQAwEIwAABgIRgAADAQjAAAGghEAAAPBCACAgWAEAMBAMAIAYMgb\n7QLGoncOH9L1i29LTH/rogv03C+2jF5BAIBzhmAcTF6hcqfdlJg8+sGvR7EYAMC5xFAqAAAGghEA\nAAPBCACAgWAEAMBAMAIAYCAYAQAwpHW5RiwW04MPPqhPPvlE0WhUjY2NKi0tVVNTk3JyclRWVqbm\n5mY5HA51dnZq586dysvLU2NjoxYsWJDhXci+M69r/Otf3tF/l81ITHOdIwCcP9IKxhdffFFTpkzR\nhg0b9M9//lM33HCDZsyYIb/fr6qqKjU3N2vXrl2aOXOmgsGgurq61N/fr7q6Os2bN08FBQWZ3o/s\nOuO6xv733uc6RwA4T6UVjDU1NbruuuskSfF4XHl5eTp8+LCqqqokSdXV1dqzZ49ycnLk8XiUn5+v\n/Px8lZSUqLe3V1dccUXm9gAAgAxK6zfGyZMny+l0yrIs3XPPPbr33nsVj8cT851Op8LhsCzLUlFR\n0WntlmWNvGoAALIk7VvCffrpp1qxYoXq6+u1aNEibdiwITHPsiy53W65XC5FIpFEeyQSkdvtPus6\nA4GAAoHAoPOKi4vTLRUAMIH09PSourp60Hk+n08+n2/I5dMKxqNHj+r2229Xc3Oz5syZI0maMWOG\nuru7NXv2bIVCIc2dO1eVlZVqa2tTNBpVf3+/+vr6VFZWdtb1DlVwS0tLOqUCACaYiooKtbe3p718\nWsG4detWhcNhbd68WZs3b5YkrVmzRo8++qhisZhKS0tVU1Mjh8OhhoYGeb1exeNx+f3+8XfiDQBg\nQkkrGNeuXau1a9d+oz0YDH6jrba2VrW1telsBgCAc47HTp0DDbc36uiJk4lprnsEgLGLYDwHjp44\nyXWPADBOcEs4AAAMBCMAAAaCEQAAA78xZsCZNxkf6ck1nKwDAKOHYMyEM24yPtKTazhZBwBGD8GY\nBWceQb7f94HKp41ePQCA4SMYs+GMI8jYexuGeDEAYCzh5BsAAAwcMZ4HOFkHADKHYBwFZ/4G+de/\nvKP/LpuRmE71N0lO1gGAzCEYR8MZv0H2v/f+Of1NkiNMADg7gnEC4ggTAM6Ok28AADBwxDgOZPrO\nOqli6BXAREIwjgcZvrNOqlIdek01SAleAGMJwXgeGu0jzDOD9A//95Eh6+E3TwBjCcF4PhrlI8xv\nyHA9HGECyCaCcRxK9V6s59u9W5MdYRKcAEaCYByPUr0Xa5LXj/SGA2cG0WgH71BDuamGJCELTDwE\nI1K+4cCgR6D/23/W14/UiIPX2L9Uh3H5/XPi4kvRxDVugvGdd3u1/f90SJIuuKBwlKuZ4M7x00PO\nDKdhBfUwgzPVD79kJzaNtQ/Tc13PWNv/keBL0cQ1boLxwF//pb4XP5Iknfr7a/pf//Vfo1wR0pUs\nyFIOuhEEdbIzaL+x7SQnEiVb35nD1NkO1kx/uCerL9UzkieSM/su2Xsh1fVN5L7NtHETjIXOKZry\nncskSf/659ujXA1GJFmQZfGINGnoZnrbSYapBw1iY1g628GS7MN10GHsIepL1p+p7k+mw2Q0nfml\n4cz3QqaH+Uc7OFN9b42l98K4CUYgI8baQ6RT/JKQ7MMv2YlTyX4fTnYEPOIvMSnuz5n1JQsTc/nx\nNmyc7CS4VNc/0uAc6f4l236qoyupvhdGIuvBGI/HtX79er333nvKz8/Xo48+quLi4mxvFjgnzvWl\nMMmCLemTWsbYF4NMn8hlfthm+g5NqX6pSFmS0YVk60/1vTisnxGGGB3I+BHbKD916LRSsr2B3//+\n94rFYuro6NCBAwf0+OOP68knn8z2ZoFz41wHTZa3d86veR3h/qRSb6aHhTN92dOIjfS9keLoxbB+\nFhin10tnPRj379+v+fPnS5JmzpypQ4cOZXuTANI1xo4okxqi3pSPRjMcLJk+4hnpl5asf+lJ8YvD\nWA7OrAejZVlyuVyJ6dzcXMXjceXkpPbEq9jJfyl89G+SpJP/DstZOCWjdQI4z4y3kE8m20eE2Tba\n20+Bw7ZtO5sbePzxxzVz5kz98Ic/lCRdc801eu211wZ9bSAQUCAQGHTeV199pUWLFmWtzvNZT0+P\nKioqRruMcYm+Gxn6L330Xfpeeukl5ebmDjrP5/PJ5/MNuXzWg/GVV17Rq6++qpaWFr311lt68skn\n9fTTT6e8nurqaoVCoSxUeP6j79JH340M/Zc++i59I+27rA+l/uAHP9CePXt0yy23SJJaWlqyvUkA\nANKW9WB0OBx6+OGHs70ZAAAyIrUzYAAAOM8RjAAAGAhGAAAMuevXr18/2kUM1/e///3RLmHcou/S\nR9+NDP2XPvoufSPpu6xfrgEAwHjCUCoAAAaCEQAAA8EIAICBYAQAwJD1O9+MBA85Tl0sFtODDz6o\nTz75RNFoVI2NjSotLVVTU5NycnJUVlam5uZmORyO0S51zDp27JgWL16sX/7yl8rJyaHvUvDUU0/p\n1VdfVSwW06233iqPx0P/DUM8HteaNWv04YcfKicnR4888ohyc3PpuyQOHDign/70pwoGg/roo48G\n7a/Ozk7t3LlTeXl5amxs1IIFC5Kv2B7Dfvvb39pNTU22bdv2W2+9ZTc2No5yRWPf888/bz/22GO2\nbdv2iRMn7Guuuca+66677O7ubtu2bfuhhx6yf/e7341miWNaNBq17777bvu6666z+/r67DvvvJO+\nG6Y//elP9p133mnbtm1HIhF748aNvPeG6bXXXrPvuece27Zte8+ePfaKFSvouySefvppe9GiRfbS\npUtt27YH/Vv94osv7EWLFtnRaNQOh8P2okWL7P7+/qTrHtNDqTzkOHU1NTX60Y9+JOl/voXm5eXp\n8OHDqqqqkvQ/d53fu3fvaJY4pj3xxBOqq6vT1KlTJYm+S8GePXtUXl6uu+++W3fddZeuvfZa9fT0\n0H/DUFhYqHA4LNu2FQ6HlZ+fT98lUVJSok2bNsn+/1ccDva3evDgQXk8HuXn58vlcqmkpES9vb1J\n1z2mg/FsDznG2U2ePFlOp1OWZemee+7Rvffee1qfTZ48WeFweBQrHLu6uro0ZcoUXX311ZIk27YT\nf3QSfZfM8ePHdejQIf385z/Xww8/rPvvv5/+GyaPx6NoNKqamho99NBDWrZsGX2XxMKFC0975qLZ\nX06nU+FwWJZlqaio6LR2y7KSrntM/8bocrkUiUQS0/F4XDk5YzrLx4RPP/1UK1asUH19vRYtWqQN\nG/7zpOxIJCK32z2K1Y1dXV1dcjgc2rt3r9599101NTXpyy+/TMyn74Z28cUXq7S0VHl5eZo2bZom\nTZqkL774IjGf/ju7Z555Rh6PR/fdd58+++wzNTQ0aGBgIDGfvkvOzAbLsuR2u7+RIcPtxzGdMh6P\nJ/Gwybfeekvl5eWjXNHYd/ToUd1+++164IEHtHjxYknSjBkz1N3dLUkKhUKaNWvWaJY4ZrW3tysY\nDCoYDOqyyy7TT37yE1199dX03TBdeeWV+sMf/iBJ+vzzz3Xq1CnNmTOH/huGkydPyul0SpLcbrcG\nBgZ0+eWX03cpGOxzrrKyUn/+858VjUYVDofV19ensrKypOsa00eMPOQ4dVu3blU4HNbmzZu1efNm\nSdKaNWv06KOPKhaLqbS0VDU1NaNc5fjgcDjU1NSkdevW0XfDsGDBAr3xxhu6+eabFY/H1dzcrEsu\nuYT+G4Y77rhDq1evltfr1cDAgO6//35VVFTQd8Pw9Zm6g/2tOhwONTQ0yOv1Kh6Py+/3q6CgIPk6\nbZt7pQIA8LUxPZQKAMC5RjACAGAgGAEAMBCMAAAYCEYAAAwEIwAABoIRAAADwQgAgIFgBADA8P8A\nTzpCZzMxaOgAAAAASUVORK5CYII=\n",
       "text": [
        "<matplotlib.figure.Figure at 0x172aaaa20>"
       ]
      }
     ],
     "prompt_number": 402
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "len(consensus_distances_7G8)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 403,
       "text": [
        "18709"
       ]
      }
     ],
     "prompt_number": 403
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "np.sum(consensus_distances_7G8>10)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 408,
       "text": [
        "12380"
       ]
      }
     ],
     "prompt_number": 408
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "arange(0, 30, 1)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 389,
       "text": [
        "array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,\n",
        "       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29])"
       ]
      }
     ],
     "prompt_number": 389
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tbl_consensus_7G8.selecteq('Consensus_distance_nearest', 1).display(10)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>CHROM</th>\n",
        "<th>POS</th>\n",
        "<th>REF</th>\n",
        "<th>ALT</th>\n",
        "<th>consensus_ALTs</th>\n",
        "<th>Consensus_RegionType</th>\n",
        "<th>Consensus_is_coding</th>\n",
        "<th>Consensus_AAChange</th>\n",
        "<th>Consensus_Effect</th>\n",
        "<th>Consensus_Gene</th>\n",
        "<th>Consensus_Impact</th>\n",
        "<th>Consensus_Transcript</th>\n",
        "<th>RPA</th>\n",
        "<th>RU</th>\n",
        "<th>STR</th>\n",
        "<th>Consensus_RPA_REF</th>\n",
        "<th>Consensus_RPA_ALT</th>\n",
        "<th>Consensus_distance_previous</th>\n",
        "<th>Consensus_distance_next</th>\n",
        "<th>Consensus_distance_nearest</th>\n",
        "<th>Consensus_mode</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>338277</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>D370</td>\n",
        "<td>SYNONYMOUS_CODING</td>\n",
        "<td>PF3D7_0108300</td>\n",
        "<td>LOW</td>\n",
        "<td>PF3D7_0108300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>338278</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>E371*</td>\n",
        "<td>STOP_GAINED</td>\n",
        "<td>PF3D7_0108300</td>\n",
        "<td>HIGH</td>\n",
        "<td>PF3D7_0108300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>351263</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>[G]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>PF3D7_0108600</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0108600.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>351264</td>\n",
        "<td>TA</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>PF3D7_0108600</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0108600.1</td>\n",
        "<td>[6, 5]</td>\n",
        "<td>A</td>\n",
        "<td>True</td>\n",
        "<td>6</td>\n",
        "<td>5</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>INDEL</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>372792</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>[G]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>PF3D7_0109300</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0109300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1199</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>372793</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>[A]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>PF3D7_0109300</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0109300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>372798</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>[G]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>PF3D7_0109300</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0109300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>372799</td>\n",
        "<td>TA</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>PF3D7_0109300</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0109300.1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>20811</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>INDEL</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>396509</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td>[C]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>19</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>396510</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td>[C]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n",
        "<p><strong>...</strong></p>"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 390
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "    - create new version of pos file with start, end, diff of new coordinates to be used for converting back\n",
      "    - create truth vcf from .Csnp file\n",
      "    - annotate truth vcf (convert to original coordinates then as 20150918_truth_vcfs)\n",
      "- create new table merging tbl_whole_genome_comparisons, pos file, truth vcf, accessible (overlap with filter-coords), distance of each GATK to nearest truth\n",
      "- plot histograms of distance of GATK variants to nearest truth and set sensible threshold\n",
      "- write function to do everything (including nucmer), with params for distance to assume TP, delta-filter i, \n",
      "- rerun everything with much lower VQSLOD, to sanity check many more FPs\n",
      "- create ROC curves based on new definitions of TP and FP\n",
      "- recreate everything from scratch, doing only what is necessary to get FDR and sens for SNP/INDEL, coding/non-coding, ensuring hets appropriately dealt with, and making as fast as possible (use numpy arrays throughout?)\n",
      "- check comparable results from 5 samples vcf and full vcf (how best to read in full vcf - talk to Alistair? maybe create small vcfs just from variant positions for that sample?)\n",
      "- turn pipeline into a single script which takes VCF, sample name, assembly file, TP_distance and delta_filter_i, and returns #TP, #FP, #FN, #inaccessible broken down by SNP/INDEL, coding/non-coding. Is this really worth it?"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Testing stuff"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "mutable_ref_dict = copy.deepcopy(ref_dict)\n",
      "print(mutable_ref_dict['Pf3D7_01_v3'])\n",
      "mutable_ref_dict['Pf3D7_01_v3'].seq.tomutable()\n",
      "print(mutable_ref_dict['Pf3D7_01_v3'])\n",
      "mutable_ref_dict['Pf3D7_01_v3'].seq = mutable_ref_dict['Pf3D7_01_v3'].seq.tomutable()\n",
      "print(mutable_ref_dict['Pf3D7_01_v3'])\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "ID: Pf3D7_01_v3\n",
        "Name: Pf3D7_01_v3\n",
        "Description: Pf3D7_01_v3\n",
        "Number of features: 0\n",
        "Seq('TGAACCCTAAAACCTAAACCCTAAACCCTAAACCCTGAACCCTAAACCCTGAAC...AGG', SingleLetterAlphabet())\n",
        "ID: Pf3D7_01_v3\n",
        "Name: Pf3D7_01_v3\n",
        "Description: Pf3D7_01_v3\n",
        "Number of features: 0\n",
        "Seq('TGAACCCTAAAACCTAAACCCTAAACCCTAAACCCTGAACCCTAAACCCTGAAC...AGG', SingleLetterAlphabet())\n",
        "ID: Pf3D7_01_v3\n",
        "Name: Pf3D7_01_v3\n",
        "Description: Pf3D7_01_v3\n",
        "Number of features: 0\n",
        "MutableSeq('TGAACCCTAAAACCTAAACCCTAAACCCTAAACCCTGAACCCTAAACCCTGAAC...AGG', SingleLetterAlphabet())\n"
       ]
      }
     ],
     "prompt_number": 166
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print(mutable_ref_dict['Pf3D7_01_v3'].id)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Pf3D7_01_v3\n"
       ]
      }
     ],
     "prompt_number": 172
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print(ref_dict['Pf3D7_01_v3'].seq[93150:93170])\n",
      "print(mutable_ref_dict['Pf3D7_01_v3'][93150:93170])\n",
      "print()\n",
      "print(ref_dict['Pf3D7_01_v3'].seq[94250:94370])\n",
      "print(mutable_ref_dict['Pf3D7_01_v3'][94250:94370])\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "GTGTATTTTTCACCCTTTCT\n",
        "GTGTATATTTCACCCTTTCT\n",
        "\n",
        "TTGTCAAATATATATATATATATATATATATAATAAATTATTTTATATTATAATTTATAATATATATTTATTATTTATATATTTTACCTTAATGGAAAAACTTTATTAGAAAAATATTTA\n",
        "TTGTCAAATATATATATATATATATATATATATATAATAAATTATTTTATATTATAATTTATAATATATATTTATTATTTATATATTTTACCTTAATGGAAAAACTTTATTAGAAAAATA\n"
       ]
      }
     ],
     "prompt_number": 115
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print(len(ref_dict['Pf3D7_01_v3']))\n",
      "print(len(mutable_ref_dict['Pf3D7_01_v3']))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "640851\n",
        "640732\n"
       ]
      }
     ],
     "prompt_number": 116
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print(ref_dict['Pf3D7_01_v3'].seq[-10:-1])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "AGGGTTTAG\n"
       ]
      }
     ],
     "prompt_number": 117
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print(mutable_ref_dict['Pf3D7_01_v3'][-10:-1])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "AGGGTTTAG\n"
       ]
      }
     ],
     "prompt_number": 118
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "mutable_ref_dict['Pf3D7_01_v3']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 124,
       "text": [
        "MutableSeq('TGAACCCTAAAACCTAAACCCTAAACCCTAAACCCTGAACCCTAAACCCTGAAC...AGG', SingleLetterAlphabet())"
       ]
      }
     ],
     "prompt_number": 124
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "ref_dict['Pf3D7_01_v3']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 125,
       "text": [
        "SeqRecord(seq=Seq('TGAACCCTAAAACCTAAACCCTAAACCCTAAACCCTGAACCCTAAACCCTGAAC...AGG', SingleLetterAlphabet()), id='Pf3D7_01_v3', name='Pf3D7_01_v3', description='Pf3D7_01_v3', dbxrefs=[])"
       ]
      }
     ],
     "prompt_number": 125
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "temp = ref_dict.copy()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 126
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for chrom in temp:\n",
      "    temp[chrom].seq = temp[chrom].seq.tomutable()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 129
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "temp['Pf3D7_01_v3']"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 130,
       "text": [
        "SeqRecord(seq=MutableSeq('TGAACCCTAAAACCTAAACCCTAAACCCTAAACCCTGAACCCTAAACCCTGAAC...AGG', SingleLetterAlphabet()), id='Pf3D7_01_v3', name='Pf3D7_01_v3', description='Pf3D7_01_v3', dbxrefs=[])"
       ]
      }
     ],
     "prompt_number": 130
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def second_allele_is_major(rec):\n",
      "    if (rec['gatk_AD'][int(rec['gatk_GT'][2]) - 1]) >= int(rec['gatk_AD'][int(rec['gatk_GT'][0]) -1]):\n",
      "        return True\n",
      "    else:\n",
      "        return False"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 50
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(tbl_whole_genome_comparisons['alistair_ann']\n",
      ".selecteq('Sample', '7G8')\n",
      ".selecteq('IsInGATK', True)\n",
      ".selecteq('RegionType', 'Core')\n",
      ".selectgt('gatk_VQSLOD', 0.0)\n",
      ".addfield('is_2nd_major', second_allele_is_major)\n",
      "# .sort(('CHROM', 'POS'))\n",
      "# .valuecounts('is_het', 'IsInTruth', 'is_2nd_major', 'mode')\n",
      ".valuecounts('is_het', 'mode')\n",
      ".displayall()\n",
      ")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>is_het</th>\n",
        "<th>mode</th>\n",
        "<th>count</th>\n",
        "<th>frequency</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>False</td>\n",
        "<td>INDEL</td>\n",
        "<td style='text-align: right'>26883</td>\n",
        "<td style='text-align: right'>0.6303904326415758</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>False</td>\n",
        "<td>SNP</td>\n",
        "<td style='text-align: right'>13664</td>\n",
        "<td style='text-align: right'>0.32041270957908313</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>True</td>\n",
        "<td>INDEL</td>\n",
        "<td style='text-align: right'>1807</td>\n",
        "<td style='text-align: right'>0.04237308007972799</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>True</td>\n",
        "<td>SNP</td>\n",
        "<td style='text-align: right'>291</td>\n",
        "<td style='text-align: right'>0.006823777699613085</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 53
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(tbl_whole_genome_comparisons['alistair_ann']\n",
      "    .selecteq('Sample', '7G8')\n",
      "    .selecteq('IsInGATK', True)\n",
      "    .selecteq('RegionType', 'Core')\n",
      "    .selectgt('gatk_VQSLOD', 0.0)\n",
      "#     .valuecounts('gatk_GT').displayall()\n",
      "    .sort(('CHROM', 'POS'))\n",
      "    .display(index_header=True)\n",
      ")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>0|CHROM</th>\n",
        "<th>1|POS</th>\n",
        "<th>2|REF</th>\n",
        "<th>3|ALT</th>\n",
        "<th>4|truth_ALTs</th>\n",
        "<th>5|Truth_RegionType</th>\n",
        "<th>6|Truth_is_coding</th>\n",
        "<th>7|Truth_AAChange</th>\n",
        "<th>8|Truth_Effect</th>\n",
        "<th>9|Truth_Gene</th>\n",
        "<th>10|Truth_Impact</th>\n",
        "<th>11|Truth_Transcript</th>\n",
        "<th>12|RPA</th>\n",
        "<th>13|RU</th>\n",
        "<th>14|STR</th>\n",
        "<th>15|Truth_RPA_REF</th>\n",
        "<th>16|Truth_RPA_ALT</th>\n",
        "<th>17|Truth_distance_previous</th>\n",
        "<th>18|Truth_distance_next</th>\n",
        "<th>19|Truth_distance_nearest</th>\n",
        "<th>20|Truth_mode</th>\n",
        "<th>21|ID</th>\n",
        "<th>22|gatk_ALTs</th>\n",
        "<th>23|gatk_QUAL</th>\n",
        "<th>24|gatk_FILTER</th>\n",
        "<th>25|SAMPLE</th>\n",
        "<th>26|gatk_AD</th>\n",
        "<th>27|gatk_DP</th>\n",
        "<th>28|GQ</th>\n",
        "<th>29|gatk_GT</th>\n",
        "<th>30|PGT</th>\n",
        "<th>31|PID</th>\n",
        "<th>32|PL</th>\n",
        "<th>33|gatk_AC</th>\n",
        "<th>34|gatk_AF</th>\n",
        "<th>35|gatk_AN</th>\n",
        "<th>36|gatk_BaseQRankSum</th>\n",
        "<th>37|gatk_ClippingRankSum</th>\n",
        "<th>38|gatk_FS</th>\n",
        "<th>39|GC</th>\n",
        "<th>40|gatk_MLEAC</th>\n",
        "<th>41|gatk_MLEAF</th>\n",
        "<th>42|gatk_MQ</th>\n",
        "<th>43|gatk_MQRankSum</th>\n",
        "<th>44|NEGATIVE_TRAIN_SITE</th>\n",
        "<th>45|POSITIVE_TRAIN_SITE</th>\n",
        "<th>46|gatk_QD</th>\n",
        "<th>47|gatk_RPA</th>\n",
        "<th>48|gatk_RU</th>\n",
        "<th>49|gatk_ReadPosRankSum</th>\n",
        "<th>50|gatk_RegionType</th>\n",
        "<th>51|SNPEFF_AMINO_ACID_CHANGE</th>\n",
        "<th>52|SNPEFF_CODON_CHANGE</th>\n",
        "<th>53|gatk_SNPEFF_EFFECT</th>\n",
        "<th>54|SNPEFF_EXON_ID</th>\n",
        "<th>55|gatk_SNPEFF_FUNCTIONAL_CLASS</th>\n",
        "<th>56|SNPEFF_GENE_BIOTYPE</th>\n",
        "<th>57|SNPEFF_GENE_NAME</th>\n",
        "<th>58|gatk_SNPEFF_IMPACT</th>\n",
        "<th>59|SNPEFF_TRANSCRIPT_ID</th>\n",
        "<th>60|gatk_SOR</th>\n",
        "<th>61|gatk_STR</th>\n",
        "<th>62|gatk_VQSLOD</th>\n",
        "<th>63|VariantType</th>\n",
        "<th>64|gatk_culprit</th>\n",
        "<th>65|gatk_set</th>\n",
        "<th>66|gatk_RPA_REF</th>\n",
        "<th>67|gatk_RPA_ALT</th>\n",
        "<th>68|GATK_is_coding</th>\n",
        "<th>69|is_het</th>\n",
        "<th>70|temp_REF</th>\n",
        "<th>71|temp_ALT</th>\n",
        "<th>72|GATK_distance_previous</th>\n",
        "<th>73|GATK_distance_next</th>\n",
        "<th>74|GATK_distance_nearest</th>\n",
        "<th>75|GATK_mode</th>\n",
        "<th>76|IsInTruth</th>\n",
        "<th>77|IsInGATK</th>\n",
        "<th>78|RegionType</th>\n",
        "<th>79|IsCoding</th>\n",
        "<th>80|Effect</th>\n",
        "<th>81|mode</th>\n",
        "<th>82|IsAccessible</th>\n",
        "<th>83|Sample</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>93157</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>[A]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>TRANSCRIPT</td>\n",
        "<td>Gene_PF3D7_0102000</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0102000</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>435</td>\n",
        "<td style='text-align: right'>744</td>\n",
        "<td style='text-align: right'>435</td>\n",
        "<td>SNP</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>3399.15</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[4, 111]</td>\n",
        "<td style='text-align: right'>115</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>1/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[3441, 209, 0]</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.2]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-0.252</td>\n",
        "<td style='text-align: right'>0.221</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>33.33</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.2]</td>\n",
        "<td style='text-align: right'>59.37</td>\n",
        "<td style='text-align: right'>3.38</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>29.56</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>Gene_PF3D7_0102000</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.951</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.13</td>\n",
        "<td>SNP</td>\n",
        "<td>MQRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>False</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>435</td>\n",
        "<td style='text-align: right'>90</td>\n",
        "<td style='text-align: right'>90</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>ERROR__False__True</td>\n",
        "<td>ERROR__TRANSCRIPT__INTRAGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>94257</td>\n",
        "<td>A</td>\n",
        "<td>AATAT</td>\n",
        "<td>[AATAT]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>[12, 14]</td>\n",
        "<td>AT</td>\n",
        "<td>True</td>\n",
        "<td>12</td>\n",
        "<td>14</td>\n",
        "<td style='text-align: right'>356</td>\n",
        "<td style='text-align: right'>331</td>\n",
        "<td style='text-align: right'>331</td>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>[AATATATATAT, AAT, AATAT, AATATATAT, AATATATATATATATAT]</td>\n",
        "<td style='text-align: right'>4524.72</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 0, 3, 20, 0, 0]</td>\n",
        "<td style='text-align: right'>23</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>3/3</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[1004, 689, 618, 621, 555, 546, 74, 66, 3, 0, 689, 618, 555, 66, 618, 689, 618, 555, 66, 618, 618]</td>\n",
        "<td>[2, 1, 4, 1, 2]</td>\n",
        "<td>[0.2, 0.1, 0.4, 0.1, 0.2]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.48</td>\n",
        "<td style='text-align: right'>1.27</td>\n",
        "<td style='text-align: right'>6.739</td>\n",
        "<td style='text-align: right'>14.29</td>\n",
        "<td>[2, 1, 3, 1, 2]</td>\n",
        "<td>[0.2, 0.1, 0.3, 0.1, 0.2]</td>\n",
        "<td style='text-align: right'>60.46</td>\n",
        "<td style='text-align: right'>1.73</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>27.88</td>\n",
        "<td>[5, 10, 6, 7, 9, 13]</td>\n",
        "<td>AT</td>\n",
        "<td style='text-align: right'>0.938</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.838</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>1.47</td>\n",
        "<td>MULTIALLELIC_COMPLEX.Other</td>\n",
        "<td>FS</td>\n",
        "<td>indel</td>\n",
        "<td>5</td>\n",
        "<td>7</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>A</td>\n",
        "<td>AATAT</td>\n",
        "<td style='text-align: right'>926</td>\n",
        "<td style='text-align: right'>165</td>\n",
        "<td style='text-align: right'>165</td>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>94588</td>\n",
        "<td>C</td>\n",
        "<td>CATATAT</td>\n",
        "<td>[CATATAT]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>[13, 16]</td>\n",
        "<td>AT</td>\n",
        "<td>True</td>\n",
        "<td>13</td>\n",
        "<td>16</td>\n",
        "<td style='text-align: right'>331</td>\n",
        "<td style='text-align: right'>405</td>\n",
        "<td style='text-align: right'>331</td>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>[CATATATATATATATATATAT, C, CATATATATATATATATATATAT]</td>\n",
        "<td style='text-align: right'>2384.07</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 0, 3, 14]</td>\n",
        "<td style='text-align: right'>19</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>2/3</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[956, 516, 457, 784, 341, 1286, 175, 157, 0, 116]</td>\n",
        "<td>[2, 2, 1]</td>\n",
        "<td>[0.2, 0.2, 0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-1.271</td>\n",
        "<td style='text-align: right'>-1.144</td>\n",
        "<td style='text-align: right'>2.628</td>\n",
        "<td style='text-align: right'>14.29</td>\n",
        "<td>[2, 2, 1]</td>\n",
        "<td>[0.2, 0.2, 0.1]</td>\n",
        "<td style='text-align: right'>60.83</td>\n",
        "<td style='text-align: right'>-2.294</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>26.2</td>\n",
        "<td>[5, 7, -3, 8]</td>\n",
        "<td>AT</td>\n",
        "<td style='text-align: right'>1.78</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.78</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>2.48</td>\n",
        "<td>MULTIALLELIC_COMPLEX.Other</td>\n",
        "<td>BaseQRankSum</td>\n",
        "<td>indel</td>\n",
        "<td>5</td>\n",
        "<td>8</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>CATATATATATATATAT</td>\n",
        "<td>CATATATATATATATATATATAT</td>\n",
        "<td style='text-align: right'>24</td>\n",
        "<td style='text-align: right'>40</td>\n",
        "<td style='text-align: right'>24</td>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>94993</td>\n",
        "<td>CATATATATAT</td>\n",
        "<td>C</td>\n",
        "<td>[C]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>[14, 9]</td>\n",
        "<td>AT</td>\n",
        "<td>True</td>\n",
        "<td>14</td>\n",
        "<td>9</td>\n",
        "<td style='text-align: right'>405</td>\n",
        "<td style='text-align: right'>1444</td>\n",
        "<td style='text-align: right'>405</td>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>[CATATATATATAT, C, CATATATATATATATATAT]</td>\n",
        "<td style='text-align: right'>5009.42</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 0, 84, 0]</td>\n",
        "<td style='text-align: right'>84</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>2/2</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[3659, 3659, 3659, 254, 254, 0, 3659, 3659, 254, 3659]</td>\n",
        "<td>[2, 4, 1]</td>\n",
        "<td>[0.2, 0.4, 0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.493</td>\n",
        "<td style='text-align: right'>0.394</td>\n",
        "<td style='text-align: right'>37.847</td>\n",
        "<td style='text-align: right'>9.52</td>\n",
        "<td>[2, 4, 1]</td>\n",
        "<td>[0.2, 0.4, 0.1]</td>\n",
        "<td style='text-align: right'>61.58</td>\n",
        "<td style='text-align: right'>1.38</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>27.68</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-0.197</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.996</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.831</td>\n",
        "<td>MULTIALLELIC_COMPLEX.Other</td>\n",
        "<td>FS</td>\n",
        "<td>indel</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>CATATATATAT</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>182</td>\n",
        "<td style='text-align: right'>140</td>\n",
        "<td style='text-align: right'>140</td>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>96836</td>\n",
        "<td>TTATAAA</td>\n",
        "<td>T</td>\n",
        "<td>[T]</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td>[4, 3]</td>\n",
        "<td>TATAAA</td>\n",
        "<td>True</td>\n",
        "<td>4</td>\n",
        "<td>3</td>\n",
        "<td style='text-align: right'>399</td>\n",
        "<td style='text-align: right'>460</td>\n",
        "<td style='text-align: right'>399</td>\n",
        "<td>INDEL</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>1035.56</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 7]</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td style='text-align: right'>22</td>\n",
        "<td>1/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[317, 22, 0]</td>\n",
        "<td>[4]</td>\n",
        "<td>[0.4]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[4]</td>\n",
        "<td>[0.4]</td>\n",
        "<td style='text-align: right'>60.93</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>31.85</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.051</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>4.72</td>\n",
        "<td>DELETION.NumRepetitions_1.EventLength_6</td>\n",
        "<td>FS</td>\n",
        "<td>indel</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>False</td>\n",
        "<td>TTATAAA</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>1691</td>\n",
        "<td style='text-align: right'>460</td>\n",
        "<td style='text-align: right'>460</td>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>INDEL</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n",
        "<p><strong>...</strong></p>"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 85
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "(tbl_whole_genome_comparisons['alistair_ann']\n",
      "    .selecteq('Sample', '7G8')\n",
      "    .selecteq('IsInGATK', True)\n",
      "    .selecteq('RegionType', 'Core')\n",
      "    .selectgt('gatk_VQSLOD', 0.0)\n",
      "#     .valuecounts('gatk_GT').displayall()\n",
      "    .sort(('CHROM', 'POS'))\n",
      "    .selecteq('IsInTruth', False)\n",
      "    .selecteq('mode', 'SNP')\n",
      "    .selecteq('is_het', True)\n",
      "    .display(50, index_header=True)\n",
      ")"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>0|CHROM</th>\n",
        "<th>1|POS</th>\n",
        "<th>2|REF</th>\n",
        "<th>3|ALT</th>\n",
        "<th>4|truth_ALTs</th>\n",
        "<th>5|Truth_RegionType</th>\n",
        "<th>6|Truth_is_coding</th>\n",
        "<th>7|Truth_AAChange</th>\n",
        "<th>8|Truth_Effect</th>\n",
        "<th>9|Truth_Gene</th>\n",
        "<th>10|Truth_Impact</th>\n",
        "<th>11|Truth_Transcript</th>\n",
        "<th>12|RPA</th>\n",
        "<th>13|RU</th>\n",
        "<th>14|STR</th>\n",
        "<th>15|Truth_RPA_REF</th>\n",
        "<th>16|Truth_RPA_ALT</th>\n",
        "<th>17|Truth_distance_previous</th>\n",
        "<th>18|Truth_distance_next</th>\n",
        "<th>19|Truth_distance_nearest</th>\n",
        "<th>20|Truth_mode</th>\n",
        "<th>21|ID</th>\n",
        "<th>22|gatk_ALTs</th>\n",
        "<th>23|gatk_QUAL</th>\n",
        "<th>24|gatk_FILTER</th>\n",
        "<th>25|SAMPLE</th>\n",
        "<th>26|gatk_AD</th>\n",
        "<th>27|gatk_DP</th>\n",
        "<th>28|GQ</th>\n",
        "<th>29|gatk_GT</th>\n",
        "<th>30|PGT</th>\n",
        "<th>31|PID</th>\n",
        "<th>32|PL</th>\n",
        "<th>33|gatk_AC</th>\n",
        "<th>34|gatk_AF</th>\n",
        "<th>35|gatk_AN</th>\n",
        "<th>36|gatk_BaseQRankSum</th>\n",
        "<th>37|gatk_ClippingRankSum</th>\n",
        "<th>38|gatk_FS</th>\n",
        "<th>39|GC</th>\n",
        "<th>40|gatk_MLEAC</th>\n",
        "<th>41|gatk_MLEAF</th>\n",
        "<th>42|gatk_MQ</th>\n",
        "<th>43|gatk_MQRankSum</th>\n",
        "<th>44|NEGATIVE_TRAIN_SITE</th>\n",
        "<th>45|POSITIVE_TRAIN_SITE</th>\n",
        "<th>46|gatk_QD</th>\n",
        "<th>47|gatk_RPA</th>\n",
        "<th>48|gatk_RU</th>\n",
        "<th>49|gatk_ReadPosRankSum</th>\n",
        "<th>50|gatk_RegionType</th>\n",
        "<th>51|SNPEFF_AMINO_ACID_CHANGE</th>\n",
        "<th>52|SNPEFF_CODON_CHANGE</th>\n",
        "<th>53|gatk_SNPEFF_EFFECT</th>\n",
        "<th>54|SNPEFF_EXON_ID</th>\n",
        "<th>55|gatk_SNPEFF_FUNCTIONAL_CLASS</th>\n",
        "<th>56|SNPEFF_GENE_BIOTYPE</th>\n",
        "<th>57|SNPEFF_GENE_NAME</th>\n",
        "<th>58|gatk_SNPEFF_IMPACT</th>\n",
        "<th>59|SNPEFF_TRANSCRIPT_ID</th>\n",
        "<th>60|gatk_SOR</th>\n",
        "<th>61|gatk_STR</th>\n",
        "<th>62|gatk_VQSLOD</th>\n",
        "<th>63|VariantType</th>\n",
        "<th>64|gatk_culprit</th>\n",
        "<th>65|gatk_set</th>\n",
        "<th>66|gatk_RPA_REF</th>\n",
        "<th>67|gatk_RPA_ALT</th>\n",
        "<th>68|GATK_is_coding</th>\n",
        "<th>69|is_het</th>\n",
        "<th>70|temp_REF</th>\n",
        "<th>71|temp_ALT</th>\n",
        "<th>72|GATK_distance_previous</th>\n",
        "<th>73|GATK_distance_next</th>\n",
        "<th>74|GATK_distance_nearest</th>\n",
        "<th>75|GATK_mode</th>\n",
        "<th>76|IsInTruth</th>\n",
        "<th>77|IsInGATK</th>\n",
        "<th>78|RegionType</th>\n",
        "<th>79|IsCoding</th>\n",
        "<th>80|Effect</th>\n",
        "<th>81|mode</th>\n",
        "<th>82|IsAccessible</th>\n",
        "<th>83|Sample</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>264949</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>92.87</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 3]</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>13</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>264937_T_A</td>\n",
        "<td>[123, 0, 13]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>47.61</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>30.96</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.833</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.29</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>284031</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C]</td>\n",
        "<td style='text-align: right'>5086.05</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[27, 126]</td>\n",
        "<td style='text-align: right'>153</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>284027_ATAG_A</td>\n",
        "<td>[5094, 0, 2595]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.25]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>-6.492</td>\n",
        "<td style='text-align: right'>-1.505</td>\n",
        "<td style='text-align: right'>0.811</td>\n",
        "<td style='text-align: right'>23.81</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.25]</td>\n",
        "<td style='text-align: right'>56.72</td>\n",
        "<td style='text-align: right'>5.95</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>33.24</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>4.26</td>\n",
        "<td>Core</td>\n",
        "<td>S959</td>\n",
        "<td>agT/agC</td>\n",
        "<td>SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>SILENT</td>\n",
        "<td>protein_coding</td>\n",
        "<td>AARP2</td>\n",
        "<td>LOW</td>\n",
        "<td>PF3D7_0106700.1</td>\n",
        "<td style='text-align: right'>0.831</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.188</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>388167</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>739.15</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[1, 21]</td>\n",
        "<td style='text-align: right'>22</td>\n",
        "<td style='text-align: right'>87</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>388167_A_G</td>\n",
        "<td>[478, 0, 87]</td>\n",
        "<td>[5]</td>\n",
        "<td>[0.833]</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>-0.315</td>\n",
        "<td style='text-align: right'>0.473</td>\n",
        "<td style='text-align: right'>7.404</td>\n",
        "<td style='text-align: right'>33.33</td>\n",
        "<td>[5]</td>\n",
        "<td>[0.833]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-0.473</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>22.4</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-0.315</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.092</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.38</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>111</td>\n",
        "<td style='text-align: right'>39</td>\n",
        "<td style='text-align: right'>39</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>392211</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T, AAAATA]</td>\n",
        "<td style='text-align: right'>872.3</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 18, 1]</td>\n",
        "<td style='text-align: right'>19</td>\n",
        "<td style='text-align: right'>12</td>\n",
        "<td>1/2</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[567, 12, 636, 584, 0, 705]</td>\n",
        "<td>[3, 3]</td>\n",
        "<td>[0.3, 0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.742</td>\n",
        "<td style='text-align: right'>-0.742</td>\n",
        "<td style='text-align: right'>6.381</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[2, 2]</td>\n",
        "<td>[0.2, 0.2]</td>\n",
        "<td style='text-align: right'>59.47</td>\n",
        "<td style='text-align: right'>-0.258</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>31.15</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-0.742</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.263</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.25</td>\n",
        "<td>MULTIALLELIC_MIXED</td>\n",
        "<td>BaseQRankSum</td>\n",
        "<td>indel</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>TAAATA</td>\n",
        "<td>AAAATA</td>\n",
        "<td style='text-align: right'>33</td>\n",
        "<td style='text-align: right'>666</td>\n",
        "<td style='text-align: right'>33</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_01_v3</td>\n",
        "<td style='text-align: right'>501815</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>54.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 0]</td>\n",
        "<td style='text-align: right'>0</td>\n",
        "<td style='text-align: right'>35</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[87, 0, 35]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>33.33</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>59.53</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>T308N</td>\n",
        "<td>aCt/aAt</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>2</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0113200</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0113200.1</td>\n",
        "<td style='text-align: right'>0.693</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>5.63</td>\n",
        "<td>SNP</td>\n",
        "<td>DP</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>190</td>\n",
        "<td style='text-align: right'>18</td>\n",
        "<td style='text-align: right'>18</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_02_v3</td>\n",
        "<td style='text-align: right'>177562</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>654.66</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[9, 19]</td>\n",
        "<td style='text-align: right'>28</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>177562_C_T</td>\n",
        "<td>[686, 0, 698]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>3.0</td>\n",
        "<td style='text-align: right'>-0.689</td>\n",
        "<td style='text-align: right'>1.691</td>\n",
        "<td style='text-align: right'>42.86</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>2.21</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>23.38</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-1.131</td>\n",
        "<td>Core</td>\n",
        "<td>H332</td>\n",
        "<td>caC/caT</td>\n",
        "<td>SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>SILENT</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0203900</td>\n",
        "<td>LOW</td>\n",
        "<td>PF3D7_0203900.1</td>\n",
        "<td style='text-align: right'>1.329</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.71</td>\n",
        "<td>SNP</td>\n",
        "<td>MQRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>1216</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_02_v3</td>\n",
        "<td style='text-align: right'>177563</td>\n",
        "<td>A</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C]</td>\n",
        "<td style='text-align: right'>651.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[11, 16]</td>\n",
        "<td style='text-align: right'>27</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>177562_C_T</td>\n",
        "<td>[683, 0, 703]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>-4.12</td>\n",
        "<td style='text-align: right'>-0.765</td>\n",
        "<td style='text-align: right'>1.632</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-0.222</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>24.14</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-1.357</td>\n",
        "<td>Core</td>\n",
        "<td>T333P</td>\n",
        "<td>Aca/Cca</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0203900</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0203900.1</td>\n",
        "<td style='text-align: right'>1.371</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.27</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_02_v3</td>\n",
        "<td style='text-align: right'>177564</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>637.66</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[12, 13]</td>\n",
        "<td style='text-align: right'>25</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>177562_C_T</td>\n",
        "<td>[669, 0, 720]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>1.22</td>\n",
        "<td style='text-align: right'>-0.843</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>42.86</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>0.571</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>25.51</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-2.149</td>\n",
        "<td>Core</td>\n",
        "<td>T333I</td>\n",
        "<td>aCa/aTa</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0203900</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0203900.1</td>\n",
        "<td style='text-align: right'>0.818</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.45</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>781</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_02_v3</td>\n",
        "<td style='text-align: right'>190061</td>\n",
        "<td>G</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>135.49</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[1, 3]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>65</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>190061_G_A</td>\n",
        "<td>[96, 0, 65]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>9.52</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>33.87</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-0.727</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.223</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.44</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>G</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>118</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_02_v3</td>\n",
        "<td style='text-align: right'>395849</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A, AATATATAT, TATATAT]</td>\n",
        "<td style='text-align: right'>2878.94</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 0, 0, 8]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>21</td>\n",
        "<td>0/3</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[193, 198, 242, 198, 242, 242, 0, 44, 44, 21]</td>\n",
        "<td>[4, 2, 1]</td>\n",
        "<td>[0.4, 0.2, 0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>1.56</td>\n",
        "<td style='text-align: right'>-0.365</td>\n",
        "<td style='text-align: right'>3.954</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[4, 2, 1]</td>\n",
        "<td>[0.4, 0.2, 0.1]</td>\n",
        "<td style='text-align: right'>61.04</td>\n",
        "<td style='text-align: right'>0.694</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>29.9</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.56</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>4</td>\n",
        "<td>NONE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0209600</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0209600.1</td>\n",
        "<td style='text-align: right'>1.042</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.83</td>\n",
        "<td>MULTIALLELIC_MIXED</td>\n",
        "<td>BaseQRankSum</td>\n",
        "<td>indel</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>AATATAT</td>\n",
        "<td>TATATAT</td>\n",
        "<td style='text-align: right'>1371</td>\n",
        "<td style='text-align: right'>25</td>\n",
        "<td style='text-align: right'>25</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTRON</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_02_v3</td>\n",
        "<td style='text-align: right'>561050</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G, *]</td>\n",
        "<td style='text-align: right'>2995.97</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[7, 28, 0]</td>\n",
        "<td style='text-align: right'>35</td>\n",
        "<td style='text-align: right'>73</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[600, 0, 73, 624, 157, 781]</td>\n",
        "<td>[1, 4]</td>\n",
        "<td>[0.1, 0.4]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-3.237</td>\n",
        "<td style='text-align: right'>-1.34</td>\n",
        "<td style='text-align: right'>1.533</td>\n",
        "<td style='text-align: right'>4.76</td>\n",
        "<td>[1, 4]</td>\n",
        "<td>[0.1, 0.4]</td>\n",
        "<td style='text-align: right'>60.07</td>\n",
        "<td style='text-align: right'>0.969</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>32.21</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.928</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.453</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.03</td>\n",
        "<td>MULTIALLELIC_SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>1502</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_02_v3</td>\n",
        "<td style='text-align: right'>612927</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A, *]</td>\n",
        "<td style='text-align: right'>2372.76</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[4, 3, 0]</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td style='text-align: right'>87</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>612919_T_A</td>\n",
        "<td>[114, 0, 87, 126, 96, 222]</td>\n",
        "<td>[3, 2]</td>\n",
        "<td>[0.3, 0.2]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>-0.55</td>\n",
        "<td style='text-align: right'>5.335</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[3, 2]</td>\n",
        "<td>[0.3, 0.2]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-1.3</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>26.78</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.55</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>3</td>\n",
        "<td>NONE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>RON6</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0214900.1</td>\n",
        "<td style='text-align: right'>1.751</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.19</td>\n",
        "<td>MULTIALLELIC_SNP</td>\n",
        "<td>MQRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTRON</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_02_v3</td>\n",
        "<td style='text-align: right'>673197</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>3409.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[131, 93]</td>\n",
        "<td style='text-align: right'>224</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>673185_GAGATCC_G</td>\n",
        "<td>[3442, 0, 27029]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-11.65</td>\n",
        "<td style='text-align: right'>2.24</td>\n",
        "<td style='text-align: right'>9.022</td>\n",
        "<td style='text-align: right'>23.81</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>59.1</td>\n",
        "<td style='text-align: right'>0.523</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>15.22</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-9.215</td>\n",
        "<td>Core</td>\n",
        "<td>K929E</td>\n",
        "<td>Aag/Gag</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0216300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0216300.1</td>\n",
        "<td style='text-align: right'>0.348</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.78</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>12</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_02_v3</td>\n",
        "<td style='text-align: right'>849359</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>1225.9</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[2, 8]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[190, 0, 102]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-1.733</td>\n",
        "<td style='text-align: right'>-0.898</td>\n",
        "<td style='text-align: right'>2.707</td>\n",
        "<td style='text-align: right'>4.76</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>1.73</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>28.89</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.751</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.276</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>5.23</td>\n",
        "<td>SNP</td>\n",
        "<td>MQRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>16</td>\n",
        "<td style='text-align: right'>33</td>\n",
        "<td style='text-align: right'>16</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_03_v3</td>\n",
        "<td style='text-align: right'>255196</td>\n",
        "<td>C</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A, CATATAT, CATAT]</td>\n",
        "<td style='text-align: right'>1899.01</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[16, 8, 0, 0]</td>\n",
        "<td style='text-align: right'>24</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>255194_T_A</td>\n",
        "<td>[232, 0, 644, 280, 668, 948, 280, 668, 948, 948]</td>\n",
        "<td>[1, 3, 1]</td>\n",
        "<td>[0.167, 0.5, 0.167]</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>2.05</td>\n",
        "<td style='text-align: right'>1.32</td>\n",
        "<td style='text-align: right'>4.195</td>\n",
        "<td style='text-align: right'>4.76</td>\n",
        "<td>[1, 1, 1]</td>\n",
        "<td>[0.167, 0.167, 0.167]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>0.153</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>31.65</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-0.827</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.259</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.63</td>\n",
        "<td>MULTIALLELIC_MIXED</td>\n",
        "<td>BaseQRankSum</td>\n",
        "<td>indel</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_03_v3</td>\n",
        "<td style='text-align: right'>311958</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>1587.96</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[2, 11]</td>\n",
        "<td style='text-align: right'>13</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>311932_ATC_A</td>\n",
        "<td>[419, 0, 3]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>1.09</td>\n",
        "<td style='text-align: right'>0.493</td>\n",
        "<td style='text-align: right'>1.731</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[4]</td>\n",
        "<td>[0.4]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-0.214</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>31.14</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.48</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.331</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.53</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>26</td>\n",
        "<td style='text-align: right'>222</td>\n",
        "<td style='text-align: right'>26</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_03_v3</td>\n",
        "<td style='text-align: right'>499025</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A, *]</td>\n",
        "<td style='text-align: right'>768.38</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[1, 3, 0]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>33</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>499025_T_A</td>\n",
        "<td>[123, 0, 33, 126, 42, 168]</td>\n",
        "<td>[1, 2]</td>\n",
        "<td>[0.125, 0.25]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td style='text-align: right'>3.757</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[1, 2]</td>\n",
        "<td>[0.125, 0.25]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-0.727</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>30.27</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.916</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.8</td>\n",
        "<td>MULTIALLELIC_SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>121</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_03_v3</td>\n",
        "<td style='text-align: right'>499029</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A, *]</td>\n",
        "<td style='text-align: right'>766.09</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[1, 3, 0]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>33</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>499025_T_A</td>\n",
        "<td>[123, 0, 33, 126, 42, 168]</td>\n",
        "<td>[1, 2]</td>\n",
        "<td>[0.1, 0.2]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-0.727</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td style='text-align: right'>3.757</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[1, 2]</td>\n",
        "<td>[0.1, 0.2]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-0.727</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>36.12</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.916</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.52</td>\n",
        "<td>MULTIALLELIC_SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>85</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_03_v3</td>\n",
        "<td style='text-align: right'>569904</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>763.03</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[1, 11]</td>\n",
        "<td style='text-align: right'>12</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>569874_TTA_T</td>\n",
        "<td>[557, 0, 3]</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.333]</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>-1.159</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>68.43</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>29.15</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.751</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>5</td>\n",
        "<td>NONE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0314100</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0314100.1</td>\n",
        "<td style='text-align: right'>0.269</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.43</td>\n",
        "<td>SNP</td>\n",
        "<td>MQ</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>30</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTRON</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_03_v3</td>\n",
        "<td style='text-align: right'>895101</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>4354.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[10, 42]</td>\n",
        "<td style='text-align: right'>52</td>\n",
        "<td style='text-align: right'>59</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[748, 0, 59]</td>\n",
        "<td>[9]</td>\n",
        "<td>[0.9]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>1.4</td>\n",
        "<td style='text-align: right'>-0.222</td>\n",
        "<td style='text-align: right'>3.427</td>\n",
        "<td style='text-align: right'>4.76</td>\n",
        "<td>[9]</td>\n",
        "<td>[0.9]</td>\n",
        "<td style='text-align: right'>60.93</td>\n",
        "<td style='text-align: right'>-0.29</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>24.88</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.4</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>1</td>\n",
        "<td>NONE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0321400</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0321400.1</td>\n",
        "<td style='text-align: right'>0.394</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.442</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td style='text-align: right'>667</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTRON</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>156511</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>2932.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[119, 80]</td>\n",
        "<td style='text-align: right'>199</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[2965, 0, 5119]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>10.72</td>\n",
        "<td style='text-align: right'>-1.899</td>\n",
        "<td style='text-align: right'>2.512</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>54.13</td>\n",
        "<td style='text-align: right'>-11.44</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>14.74</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-1.515</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>Gene_PF3D7_0402500</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.908</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.405</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>156519</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>2945.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[118, 82]</td>\n",
        "<td style='text-align: right'>200</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[2978, 0, 4625]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-9.239</td>\n",
        "<td style='text-align: right'>0.657</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>38.1</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>53.71</td>\n",
        "<td style='text-align: right'>-11.53</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>14.73</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-0.659</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>Gene_PF3D7_0402500</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.665</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.8</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>12</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>156531</td>\n",
        "<td>C</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>2998.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[118, 82]</td>\n",
        "<td style='text-align: right'>200</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[3031, 0, 4711]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-4.842</td>\n",
        "<td style='text-align: right'>0.111</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>52.38</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>52.43</td>\n",
        "<td style='text-align: right'>-12.02</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>14.99</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.4</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>Gene_PF3D7_0402500</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.717</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.976</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>12</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>156536</td>\n",
        "<td>A</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C]</td>\n",
        "<td style='text-align: right'>8168.68</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[120, 85]</td>\n",
        "<td style='text-align: right'>205</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[3384, 0, 4991]</td>\n",
        "<td>[5]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-11.7</td>\n",
        "<td style='text-align: right'>0.19</td>\n",
        "<td style='text-align: right'>0.841</td>\n",
        "<td style='text-align: right'>47.62</td>\n",
        "<td>[5]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>56.68</td>\n",
        "<td style='text-align: right'>-11.99</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>21.55</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.48</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>Gene_PF3D7_0402500</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.768</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.58</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>156545</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C]</td>\n",
        "<td style='text-align: right'>3676.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[122, 95]</td>\n",
        "<td style='text-align: right'>217</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[3709, 0, 4962]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-12.56</td>\n",
        "<td style='text-align: right'>-0.668</td>\n",
        "<td style='text-align: right'>0.499</td>\n",
        "<td style='text-align: right'>42.86</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>49.92</td>\n",
        "<td style='text-align: right'>-10.92</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>16.94</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.08</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>Gene_PF3D7_0402500</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.742</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.031</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>156546</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>3637.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[121, 96]</td>\n",
        "<td style='text-align: right'>217</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[3670, 0, 4923]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-11.2</td>\n",
        "<td style='text-align: right'>-1.78</td>\n",
        "<td style='text-align: right'>0.497</td>\n",
        "<td style='text-align: right'>42.86</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>49.83</td>\n",
        "<td style='text-align: right'>-11.04</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>16.76</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.428</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>Gene_PF3D7_0402500</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.735</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.058</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>156915</td>\n",
        "<td>G</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C]</td>\n",
        "<td style='text-align: right'>2608.45</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[111, 70]</td>\n",
        "<td style='text-align: right'>183</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[2638, 0, 4755]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-1.144</td>\n",
        "<td style='text-align: right'>0.73</td>\n",
        "<td style='text-align: right'>1.907</td>\n",
        "<td style='text-align: right'>14.29</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>53.45</td>\n",
        "<td style='text-align: right'>-11.42</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>14.41</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.01</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>Gene_PF3D7_0402500</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.862</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.165</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>G</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>20</td>\n",
        "<td style='text-align: right'>24</td>\n",
        "<td style='text-align: right'>20</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>INTRAGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>438123</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T, *]</td>\n",
        "<td style='text-align: right'>2564.15</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[17, 16, 0]</td>\n",
        "<td style='text-align: right'>33</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>438123_C_T</td>\n",
        "<td>[323, 0, 388, 373, 435, 808]</td>\n",
        "<td>[1, 2]</td>\n",
        "<td>[0.1, 0.2]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.162</td>\n",
        "<td style='text-align: right'>0.27</td>\n",
        "<td style='text-align: right'>1.068</td>\n",
        "<td style='text-align: right'>9.52</td>\n",
        "<td>[1, 3]</td>\n",
        "<td>[0.1, 0.3]</td>\n",
        "<td style='text-align: right'>58.05</td>\n",
        "<td style='text-align: right'>-0.991</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>34.19</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.126</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.494</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.376</td>\n",
        "<td>MULTIALLELIC_SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>16</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>535271</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A, TTTATTTATTTATTT]</td>\n",
        "<td style='text-align: right'>2182.9</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[2, 0, 5]</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td style='text-align: right'>40</td>\n",
        "<td>0/2</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[92, 98, 152, 0, 55, 40]</td>\n",
        "<td>[6, 1]</td>\n",
        "<td>[0.6, 0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>1.5</td>\n",
        "<td style='text-align: right'>1.5</td>\n",
        "<td style='text-align: right'>9.837</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[6, 1]</td>\n",
        "<td>[0.6, 0.1]</td>\n",
        "<td style='text-align: right'>64.28</td>\n",
        "<td style='text-align: right'>0.922</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>25.21</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-1.495</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.609</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.79</td>\n",
        "<td>MULTIALLELIC_MIXED</td>\n",
        "<td>BaseQRankSum</td>\n",
        "<td>indel</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>ATTATTTATTTATTT</td>\n",
        "<td>TTTATTTATTTATTT</td>\n",
        "<td style='text-align: right'>74</td>\n",
        "<td style='text-align: right'>207</td>\n",
        "<td style='text-align: right'>74</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>545705</td>\n",
        "<td>G</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>115.44</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[3, 1]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>545694_C_T</td>\n",
        "<td>[141, 0, 150]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>23.81</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.2]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-0.727</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>28.86</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.022</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.97</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>G</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>11</td>\n",
        "<td style='text-align: right'>159</td>\n",
        "<td style='text-align: right'>11</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>694915</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C]</td>\n",
        "<td style='text-align: right'>141.13</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[59, 12]</td>\n",
        "<td style='text-align: right'>71</td>\n",
        "<td style='text-align: right'>82</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[82, 0, 2330]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-0.637</td>\n",
        "<td style='text-align: right'>-1.082</td>\n",
        "<td style='text-align: right'>2.265</td>\n",
        "<td style='text-align: right'>14.29</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.2]</td>\n",
        "<td style='text-align: right'>66.97</td>\n",
        "<td style='text-align: right'>-3.015</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>1.91</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>5.38</td>\n",
        "<td>Core</td>\n",
        "<td>M337T</td>\n",
        "<td>aTg/aCg</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0415800</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0415800.1</td>\n",
        "<td style='text-align: right'>1.023</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.745</td>\n",
        "<td>SNP</td>\n",
        "<td>MQRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>866</td>\n",
        "<td style='text-align: right'>59</td>\n",
        "<td style='text-align: right'>59</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>831880</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>1011.55</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[3, 4]</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td style='text-align: right'>15</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>831880_T_A</td>\n",
        "<td>[82, 0, 15]</td>\n",
        "<td>[4]</td>\n",
        "<td>[0.4]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>1.48</td>\n",
        "<td style='text-align: right'>1.48</td>\n",
        "<td style='text-align: right'>4.749</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[4]</td>\n",
        "<td>[0.4]</td>\n",
        "<td style='text-align: right'>57.96</td>\n",
        "<td style='text-align: right'>0.555</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>27.34</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.382</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>3</td>\n",
        "<td>NONE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>BET3</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0418500.1</td>\n",
        "<td style='text-align: right'>2.093</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.12</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>160</td>\n",
        "<td style='text-align: right'>19</td>\n",
        "<td style='text-align: right'>19</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTRON</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>909434</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T, *]</td>\n",
        "<td style='text-align: right'>2650.57</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[1, 9, 0]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[243, 0, 0, 246, 27, 273]</td>\n",
        "<td>[1, 6]</td>\n",
        "<td>[0.1, 0.6]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-1.741</td>\n",
        "<td style='text-align: right'>-1.741</td>\n",
        "<td style='text-align: right'>3.282</td>\n",
        "<td style='text-align: right'>4.76</td>\n",
        "<td>[1, 7]</td>\n",
        "<td>[0.1, 0.7]</td>\n",
        "<td style='text-align: right'>57.71</td>\n",
        "<td style='text-align: right'>-0.696</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>34.59</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.39</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.469</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.058</td>\n",
        "<td>MULTIALLELIC_SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>148</td>\n",
        "<td style='text-align: right'>104</td>\n",
        "<td style='text-align: right'>104</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>916853</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>303.41</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[1, 6]</td>\n",
        "<td style='text-align: right'>7</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>916845_T_A</td>\n",
        "<td>[195, 0, 10]</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.2]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>1.13</td>\n",
        "<td style='text-align: right'>0.742</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.2]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>0.289</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>15.97</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.742</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.223</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.94</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>656</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_04_v3</td>\n",
        "<td style='text-align: right'>919308</td>\n",
        "<td>G</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>3722.6</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[167, 58]</td>\n",
        "<td style='text-align: right'>225</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>919304_A_G</td>\n",
        "<td>[1643, 0, 18796]</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.2]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>11.22</td>\n",
        "<td style='text-align: right'>1.05</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>4.76</td>\n",
        "<td>[2]</td>\n",
        "<td>[0.2]</td>\n",
        "<td style='text-align: right'>58.07</td>\n",
        "<td style='text-align: right'>-2.091</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>8.08</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-8.89</td>\n",
        "<td>Core</td>\n",
        "<td>S440N</td>\n",
        "<td>aGt/aAt</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>ApiAP2</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0420300.1</td>\n",
        "<td style='text-align: right'>0.692</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.037</td>\n",
        "<td>SNP</td>\n",
        "<td>DP</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>G</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>117223</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T, *]</td>\n",
        "<td style='text-align: right'>1497.4</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[5, 15, 0]</td>\n",
        "<td style='text-align: right'>20</td>\n",
        "<td style='text-align: right'>72</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[485, 0, 72, 187, 113, 291]</td>\n",
        "<td>[2, 2]</td>\n",
        "<td>[0.2, 0.2]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.764</td>\n",
        "<td style='text-align: right'>-0.085</td>\n",
        "<td style='text-align: right'>4.975</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[2, 2]</td>\n",
        "<td>[0.2, 0.2]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>0.736</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>34.82</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.736</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTRON</td>\n",
        "<td>4</td>\n",
        "<td>NONE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0502600</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>PF3D7_0502600.1</td>\n",
        "<td style='text-align: right'>1.036</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.79</td>\n",
        "<td>MULTIALLELIC_SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>874</td>\n",
        "<td style='text-align: right'>1307</td>\n",
        "<td style='text-align: right'>874</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTRON</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>216569</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C, *]</td>\n",
        "<td style='text-align: right'>3143.29</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[2, 25, 0]</td>\n",
        "<td style='text-align: right'>27</td>\n",
        "<td style='text-align: right'>22</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>216545_TATGATGATC_T</td>\n",
        "<td>[772, 0, 22, 806, 97, 903]</td>\n",
        "<td>[3, 4]</td>\n",
        "<td>[0.3, 0.4]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-0.417</td>\n",
        "<td style='text-align: right'>0.046</td>\n",
        "<td style='text-align: right'>2.416</td>\n",
        "<td style='text-align: right'>4.76</td>\n",
        "<td>[3, 3]</td>\n",
        "<td>[0.3, 0.3]</td>\n",
        "<td style='text-align: right'>59.73</td>\n",
        "<td style='text-align: right'>0.88</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>27.58</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>N541D</td>\n",
        "<td>Aat/Gat</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0505000</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0505000.1</td>\n",
        "<td style='text-align: right'>0.852</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.107</td>\n",
        "<td>MULTIALLELIC_SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>389456</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>117.29</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[1, 3]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>30</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>389454_T_A</td>\n",
        "<td>[148, 0, 30]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td style='text-align: right'>-0.727</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>62.65</td>\n",
        "<td style='text-align: right'>-0.727</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>29.32</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.727</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.916</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.017</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>389458</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>90.69</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[1, 2]</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td style='text-align: right'>33</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>389454_T_A</td>\n",
        "<td>[123, 0, 33]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-0.736</td>\n",
        "<td style='text-align: right'>0.736</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>63.51</td>\n",
        "<td style='text-align: right'>-0.736</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>30.23</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-0.736</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.223</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.62</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>477222</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C]</td>\n",
        "<td style='text-align: right'>4896.16</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[138, 47]</td>\n",
        "<td style='text-align: right'>185</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>477222_T_C</td>\n",
        "<td>[1091, 0, 22312]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-9.64</td>\n",
        "<td style='text-align: right'>0.229</td>\n",
        "<td style='text-align: right'>15.088</td>\n",
        "<td style='text-align: right'>9.52</td>\n",
        "<td>[5]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>55.61</td>\n",
        "<td style='text-align: right'>-9.362</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>15.54</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-5.291</td>\n",
        "<td>Core</td>\n",
        "<td>N507D</td>\n",
        "<td>Aat/Gat</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>2</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0511300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0511300.1</td>\n",
        "<td style='text-align: right'>0.739</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.132</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>207</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>477231</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[C]</td>\n",
        "<td style='text-align: right'>740.7</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[160, 31]</td>\n",
        "<td style='text-align: right'>191</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>477222_T_C</td>\n",
        "<td>[773, 0, 26502]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-8.134</td>\n",
        "<td style='text-align: right'>1.77</td>\n",
        "<td style='text-align: right'>7.843</td>\n",
        "<td style='text-align: right'>19.05</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>56.58</td>\n",
        "<td style='text-align: right'>-8.727</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.88</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-5.802</td>\n",
        "<td>Core</td>\n",
        "<td>N504D</td>\n",
        "<td>Aat/Gat</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>2</td>\n",
        "<td>MISSENSE</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PF3D7_0511300</td>\n",
        "<td>MODERATE</td>\n",
        "<td>PF3D7_0511300.1</td>\n",
        "<td style='text-align: right'>0.636</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.434</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>C</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>NON_SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>480018</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[GGTTTATTGTTTTTTATTTTTTTATATTTTTATTTTTTTTTCT, T]</td>\n",
        "<td style='text-align: right'>2180.46</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[2, 0, 9]</td>\n",
        "<td style='text-align: right'>11</td>\n",
        "<td style='text-align: right'>18</td>\n",
        "<td>0/2</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[236, 242, 288, 0, 45, 18]</td>\n",
        "<td>[4, 1]</td>\n",
        "<td>[0.667, 0.167]</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td style='text-align: right'>2.0</td>\n",
        "<td style='text-align: right'>0.589</td>\n",
        "<td style='text-align: right'>3.104</td>\n",
        "<td style='text-align: right'>19.05</td>\n",
        "<td>[4, 1]</td>\n",
        "<td>[0.667, 0.167]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-1.532</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>28.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-2.239</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.182</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.05</td>\n",
        "<td>MULTIALLELIC_MIXED</td>\n",
        "<td>BaseQRankSum</td>\n",
        "<td>indel</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>365</td>\n",
        "<td style='text-align: right'>335</td>\n",
        "<td style='text-align: right'>335</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>586032</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>590.91</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[4, 21]</td>\n",
        "<td style='text-align: right'>25</td>\n",
        "<td style='text-align: right'>16</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[408, 0, 16]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>0.751</td>\n",
        "<td style='text-align: right'>0.751</td>\n",
        "<td style='text-align: right'>2.53</td>\n",
        "<td style='text-align: right'>4.76</td>\n",
        "<td>[4]</td>\n",
        "<td>[0.4]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>1.3</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>17.38</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.751</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.207</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.32</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>924</td>\n",
        "<td style='text-align: right'>64</td>\n",
        "<td style='text-align: right'>64</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>620156</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>267.83</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[0, 0]</td>\n",
        "<td style='text-align: right'>0</td>\n",
        "<td style='text-align: right'>0</td>\n",
        "<td>0/1</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[78, 0, 0]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[5]</td>\n",
        "<td>[0.5]</td>\n",
        "<td style='text-align: right'>56.14</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>31.45</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.609</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.63</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>A</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>32</td>\n",
        "<td style='text-align: right'>212</td>\n",
        "<td style='text-align: right'>32</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>633219</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>13196.23</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[148, 138]</td>\n",
        "<td style='text-align: right'>286</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>633219_C_T</td>\n",
        "<td>[4855, 0, 11479]</td>\n",
        "<td>[5]</td>\n",
        "<td>[0.625]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>13.38</td>\n",
        "<td style='text-align: right'>-0.587</td>\n",
        "<td style='text-align: right'>7.153</td>\n",
        "<td style='text-align: right'>28.57</td>\n",
        "<td>[5]</td>\n",
        "<td>[0.625]</td>\n",
        "<td style='text-align: right'>53.96</td>\n",
        "<td style='text-align: right'>-6.976</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>18.99</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-8.314</td>\n",
        "<td>Core</td>\n",
        "<td>D1413</td>\n",
        "<td>gaC/gaT</td>\n",
        "<td>SYNONYMOUS_CODING</td>\n",
        "<td>1</td>\n",
        "<td>SILENT</td>\n",
        "<td>protein_coding</td>\n",
        "<td>PI3K</td>\n",
        "<td>LOW</td>\n",
        "<td>PF3D7_0515300.1</td>\n",
        "<td style='text-align: right'>1.172</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.43</td>\n",
        "<td>SNP</td>\n",
        "<td>MQRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td>C</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>2462</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>True</td>\n",
        "<td>SYNONYMOUS_CODING</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>674957</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[T]</td>\n",
        "<td style='text-align: right'>92.12</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[2, 2]</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>78</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>674957_G_T</td>\n",
        "<td>[78, 0, 78]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>1.03</td>\n",
        "<td style='text-align: right'>1.03</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>19.05</td>\n",
        "<td>[4]</td>\n",
        "<td>[0.4]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>23.03</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-1.026</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.303</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.578</td>\n",
        "<td>SNP</td>\n",
        "<td>ReadPosRankSum</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>G</td>\n",
        "<td>T</td>\n",
        "<td style='text-align: right'>641</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>686221</td>\n",
        "<td>G</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[AAAAAGAAAAT, G]</td>\n",
        "<td style='text-align: right'>852.79</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[40, 18, 0]</td>\n",
        "<td style='text-align: right'>58</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>686206_G_A</td>\n",
        "<td>[351, 0, 1613, 471, 1663, 2134]</td>\n",
        "<td>[1, 3]</td>\n",
        "<td>[0.1, 0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>3.47</td>\n",
        "<td style='text-align: right'>0.126</td>\n",
        "<td style='text-align: right'>7.249</td>\n",
        "<td style='text-align: right'>19.05</td>\n",
        "<td>[1, 3]</td>\n",
        "<td>[0.1, 0.3]</td>\n",
        "<td style='text-align: right'>59.47</td>\n",
        "<td style='text-align: right'>-0.059</td>\n",
        "<td>True</td>\n",
        "<td>True</td>\n",
        "<td style='text-align: right'>11.84</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-3.992</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.436</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.399</td>\n",
        "<td>MULTIALLELIC_MIXED</td>\n",
        "<td>BaseQRankSum</td>\n",
        "<td>indel</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>GAAAAGAAAAT</td>\n",
        "<td>AAAAAGAAAAT</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td style='text-align: right'>5</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>715205</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[G]</td>\n",
        "<td style='text-align: right'>383.97</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[10, 3]</td>\n",
        "<td style='text-align: right'>13</td>\n",
        "<td style='text-align: right'>20</td>\n",
        "<td>0/1</td>\n",
        "<td>1|0</td>\n",
        "<td>715204_T_G</td>\n",
        "<td>[20, 0, 227]</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-1.606</td>\n",
        "<td style='text-align: right'>0.592</td>\n",
        "<td style='text-align: right'>8.8</td>\n",
        "<td style='text-align: right'>14.29</td>\n",
        "<td>[3]</td>\n",
        "<td>[0.3]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-0.254</td>\n",
        "<td>True</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>16.0</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.12</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.51</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.368</td>\n",
        "<td>SNP</td>\n",
        "<td>QD</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>G</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td style='text-align: right'>250</td>\n",
        "<td style='text-align: right'>1</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>839160</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>234.66</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[3, 7]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>839160_T_A</td>\n",
        "<td>[266, 0, 105]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td style='text-align: right'>-1.58</td>\n",
        "<td style='text-align: right'>0.546</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.125]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-1.214</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>23.47</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>0.895</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.609</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>2.75</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>181</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_05_v3</td>\n",
        "<td style='text-align: right'>839164</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>[A]</td>\n",
        "<td style='text-align: right'>212.29</td>\n",
        "<td>[]</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>[3, 6]</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td style='text-align: right'>99</td>\n",
        "<td>0/1</td>\n",
        "<td>0|1</td>\n",
        "<td>839160_T_A</td>\n",
        "<td>[243, 0, 108]</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td style='text-align: right'>-1.203</td>\n",
        "<td style='text-align: right'>0.248</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td style='text-align: right'>0.0</td>\n",
        "<td>[1]</td>\n",
        "<td>[0.1]</td>\n",
        "<td style='text-align: right'>60.0</td>\n",
        "<td style='text-align: right'>-1.203</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>23.59</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>-0.248</td>\n",
        "<td>Core</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>None</td>\n",
        "<td>NONE</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>MODIFIER</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>1.402</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>3.0</td>\n",
        "<td>SNP</td>\n",
        "<td>FS</td>\n",
        "<td>snp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>T</td>\n",
        "<td>A</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td style='text-align: right'>102</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>SNP</td>\n",
        "<td>False</td>\n",
        "<td>True</td>\n",
        "<td>Core</td>\n",
        "<td>False</td>\n",
        "<td>INTERGENIC</td>\n",
        "<td>SNP</td>\n",
        "<td>True</td>\n",
        "<td>7G8</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n",
        "<p><strong>...</strong></p>"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 182
    }
   ],
   "metadata": {}
  }
 ]
}