{
 "metadata": {
  "name": "",
  "signature": "sha256:b1c1a536729960b5642d0ae92d1abef59ca0c3b0496e3d04446ecbbce0dc6660"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Setup"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%run _shared_setup.ipynb"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "python 3.4.1 |Anaconda 2.1.0 (x86_64)| (default, Sep 10 2014, 17:24:09) \n",
        "[GCC 4.2.1 (Apple Inc. build 5577)]\n",
        "numpy 1.9.2\n",
        "scipy 0.14.0\n",
        "pandas 0.14.1\n",
        "numexpr 2.3.1\n",
        "pysam 0.8.3\n",
        "petl 1.0.10\n",
        "petlx 1.0.3\n",
        "vcf 0.6.7\n",
        "h5py 2.3.1\n",
        "tables 3.1.1\n",
        "vcfplt 0.8\n",
        "mkdir: /nfs/team112_internal/production_files/Pf3k/methods/assembled_samples: File exists\r\n"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "tbl_pgv_metadata length = 5729\n",
        "tbl_pgv_locations length = 102\n",
        "tbl_pf3k_metadata length = 2512\n",
        "tbl_pf_solaris length = 10879"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "\n",
        "tbl_assembled_samples length = 11\n"
       ]
      }
     ],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "!mkdir -p {os.path.join(PROCESSED_ASSEMBLED_SAMPLES_DIR, 'fastq')}\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 2
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Check data"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def get_RG(bam_fn = '/nfs/team112_internal/production_files/Pf/4_0/PFprog1/PG0008_CW/PG0008_CW.bam'):\n",
      "    temp = !{samtools_exe} view -H {bam_fn} | grep '@RG'\n",
      "    return(temp[0])\n",
      "get_RG()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 3,
       "text": [
        "'@RG\\tID:4062_2_nonhuman\\tPL:ILLUMINA\\tPU:4062_2_nonhuman\\tLB:PG0008_CW_200_1\\tDS:PFprog1\\tSM:PG0008-CW\\tCN:SC'"
       ]
      }
     ],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "tbl_samples_to_process = (tbl_assembled_samples\n",
      "    .cutout('Notes')\n",
      "    .selectnotnone('bam_fn')\n",
      "#     .selectne('Sample name', 'Pf3D7_II')\n",
      "#     .selectne('Sample name', 'ITA4?')\n",
      "#     .addfield('bam_dir', lambda rec: os.path.dirname(rec['bam_fn']) if rec['bam_fn'] is not None else None)\n",
      "    .addfield('remapped_sam_fn', lambda rec: os.path.join(\n",
      "        PROCESSED_ASSEMBLED_SAMPLES_DIR,\n",
      "        'bams',\n",
      "        'bwa_mem',\n",
      "        os.path.basename(rec['bam_fn']).replace('.bam', '.bwa_mem.sam').replace('.cram', '.bwa_mem.sam')\n",
      "    ))\n",
      "    .addfield('sorted_bam_fn', lambda rec: rec['remapped_sam_fn'].replace('.sam', '.sorted.bam'))\n",
      "    .addfield('dedupped_bam_fn', lambda rec: rec['remapped_sam_fn'].replace('.sam', '.dedup.bam'))\n",
      "    .addfield('dedup_metrics_fn', lambda rec: rec['remapped_sam_fn'].replace('.sam', '.dedup.metrics'))\n",
      "    .addfield('targets_fn', lambda rec: rec['remapped_sam_fn'].replace('.sam', '.dedup.targets.list'))\n",
      "    .addfield('realigned_bam_fn', lambda rec: rec['remapped_sam_fn'].replace('.sam', '.realigned.bam'))\n",
      "    .addfield('recal_table_fn', lambda rec: rec['remapped_sam_fn'].replace('.sam', '.recal.table'))\n",
      "    .addfield('post_recal_table_fn', lambda rec: rec['remapped_sam_fn'].replace('.sam', '.post_recal.table'))\n",
      "    .addfield('recal_plots_fn', lambda rec: rec['remapped_sam_fn'].replace('.sam', '.recal.pdf'))\n",
      "    .addfield('recal_bam_fn', lambda rec: rec['remapped_sam_fn'].replace('.sam', '.recal.bam'))\n",
      "    .addfield('RG', lambda rec: get_RG(rec['bam_fn']).replace('\\t', '\\\\t'))\n",
      "    .addfield('gvcf_filestem', lambda rec: os.path.join(\n",
      "        PROCESSED_ASSEMBLED_SAMPLES_DIR,\n",
      "        'vcfs',\n",
      "        'gvcf',\n",
      "        'samples',\n",
      "        os.path.basename(rec['bam_fn']).replace('.bam', '.raw.snps.indels')\n",
      "    ))\n",
      "    .addfield('fastq_fn', lambda rec: os.path.join(\n",
      "        PROCESSED_ASSEMBLED_SAMPLES_DIR,\n",
      "        'fastq',\n",
      "        os.path.basename(rec['bam_fn']).replace('.bam', '.fastq').replace('.cram', '.fastq')\n",
      "    ))\n",
      "#     .head(4)\n",
      ")\n",
      "tbl_samples_to_process.displayall(index_header=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<table class='petl'>\n",
        "<thead>\n",
        "<tr>\n",
        "<th>0|Sample name</th>\n",
        "<th>1|Isolate code</th>\n",
        "<th>2|Cross parent?</th>\n",
        "<th>3|Origin</th>\n",
        "<th>4|Continent</th>\n",
        "<th>5|#contigs assemblies</th>\n",
        "<th>6|Both telomer repeats attached</th>\n",
        "<th>7|Icorned</th>\n",
        "<th>8|ox_code</th>\n",
        "<th>9|SRA accession</th>\n",
        "<th>10|coverage</th>\n",
        "<th>11|read length</th>\n",
        "<th>12|To be used for</th>\n",
        "<th>13|bam_fn</th>\n",
        "<th>14|remapped_sam_fn</th>\n",
        "<th>15|sorted_bam_fn</th>\n",
        "<th>16|dedupped_bam_fn</th>\n",
        "<th>17|dedup_metrics_fn</th>\n",
        "<th>18|targets_fn</th>\n",
        "<th>19|realigned_bam_fn</th>\n",
        "<th>20|recal_table_fn</th>\n",
        "<th>21|post_recal_table_fn</th>\n",
        "<th>22|recal_plots_fn</th>\n",
        "<th>23|recal_bam_fn</th>\n",
        "<th>24|RG</th>\n",
        "<th>25|gvcf_filestem</th>\n",
        "<th>26|fastq_fn</th>\n",
        "</tr>\n",
        "</thead>\n",
        "<tbody>\n",
        "<tr>\n",
        "<td>Dd2</td>\n",
        "<td>Dd2</td>\n",
        "<td>Yes</td>\n",
        "<td>Indochina</td>\n",
        "<td>SE Asia</td>\n",
        "<td style='text-align: right'>16</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>100bp reads</td>\n",
        "<td>PG0008-CW</td>\n",
        "<td>ERS010540</td>\n",
        "<td>80X</td>\n",
        "<td style='text-align: right'>76</td>\n",
        "<td>Training</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf/4_0/PFprog1/PG0008_CW/PG0008_CW.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0008_CW.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:4062_2_nonhuman\\tPL:ILLUMINA\\tPU:4062_2_nonhuman\\tLB:PG0008_CW_200_1\\tDS:PFprog1\\tSM:PG0008-CW\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/PG0008_CW.raw.snps.indels</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/PG0008_CW.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>HB3</td>\n",
        "<td>HB3</td>\n",
        "<td>Yes</td>\n",
        "<td>Honduras</td>\n",
        "<td>S America</td>\n",
        "<td style='text-align: right'>28</td>\n",
        "<td style='text-align: right'>3</td>\n",
        "<td>75bp</td>\n",
        "<td>PG0004-CW</td>\n",
        "<td>ERS010539</td>\n",
        "<td>122X</td>\n",
        "<td style='text-align: right'>76</td>\n",
        "<td>Training</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf/4_0/PFprog1/PG0004_CW/PG0004_CW.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0004_CW.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:4062_1_nonhuman\\tPL:ILLUMINA\\tPU:4062_1_nonhuman\\tLB:PG0004_CW_200_1\\tDS:PFprog1\\tSM:PG0004-CW\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/PG0004_CW.raw.snps.indels</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/PG0004_CW.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>7G8</td>\n",
        "<td>7G8</td>\n",
        "<td>Yes</td>\n",
        "<td>Brazil</td>\n",
        "<td>S America</td>\n",
        "<td style='text-align: right'>17</td>\n",
        "<td style='text-align: right'>9</td>\n",
        "<td>75bp</td>\n",
        "<td>PG0083-C</td>\n",
        "<td>ERR027099</td>\n",
        "<td>87X</td>\n",
        "<td style='text-align: right'>76</td>\n",
        "<td>Validation</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf/4_0/PFprog3/PG0083_C/PG0083_C.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0083_C.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:5459_2_nonhuman\\tPL:ILLUMINA\\tPU:5459_2_nonhuman\\tLB:PG0083_C_1361748\\tDS:PFprog3\\tSM:PG0083-C\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/PG0083_C.raw.snps.indels</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/PG0083_C.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>GB4</td>\n",
        "<td>GB4</td>\n",
        "<td>Yes</td>\n",
        "<td>Ghana</td>\n",
        "<td>W Africa</td>\n",
        "<td style='text-align: right'>26</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td>75bp</td>\n",
        "<td>PG0084-C</td>\n",
        "<td>ERR027100</td>\n",
        "<td>104X</td>\n",
        "<td style='text-align: right'>76</td>\n",
        "<td>Validation</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf/4_0/PFprog3/PG0084_C/PG0084_C.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/PG0084_C.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:5459_3_nonhuman\\tPL:ILLUMINA\\tPU:5459_3_nonhuman\\tLB:PG0084_C_1361749\\tDS:PFprog3\\tSM:PG0084-C\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/PG0084_C.raw.snps.indels</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/PG0084_C.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>ITA4?</td>\n",
        "<td>IT</td>\n",
        "<td>None</td>\n",
        "<td>Brazil?</td>\n",
        "<td>S America?</td>\n",
        "<td style='text-align: right'>27</td>\n",
        "<td style='text-align: right'>4</td>\n",
        "<td>76bp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>>200x</td>\n",
        "<td>many 37-300</td>\n",
        "<td>Training</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/GATKbuild/IT/14572_1#34.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/14572_1#34.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:1#34\\tPL:ILLUMINA\\tPU:141030_MS7_14572_A_MS2779432-500V2_1#34\\tLB:12062360\\tDS:ERP000234: A project to shotgun sequence the genome of the IT parasite line is underway as a component of the BioMalPar Consortium(European Union 6th Framework Program, Contract number LSHP-LT-2004-503578). This initiative is in collaboration with Chris Newbold of the Molecular Parasitology Group, Institute of Molecular Medicine, Oxford. Illumina sequencing will be used to aid assembly and to improve base accuracy. This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/\\tDT:2014-10-30T00:00:00+0000\\tSM:ERS557780\\tPG:BamIndexDecoder\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/14572_1#34.raw.snps.indels</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/14572_1#34.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Senegal</td>\n",
        "<td>SN01</td>\n",
        "<td>None</td>\n",
        "<td>Senegal</td>\n",
        "<td>W Africa</td>\n",
        "<td style='text-align: right'>36</td>\n",
        "<td style='text-align: right'>2</td>\n",
        "<td>250bp</td>\n",
        "<td>na</td>\n",
        "<td>?</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>76</td>\n",
        "<td>Training</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/cram/16503_1#1.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#1.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:1#1\\tPL:ILLUMINA\\tPU:150609_MS2_16503_A_MS2920703-500V2_1#1\\tLB:14013610\\tDS:ERP009847: Pf3k is a global collaboration using the latest sequencing technologies to provide a high-resolution view of natural variation in the malaria parasite Plasmodium falciparum. The Project is led by researchers at the Broad Institute, the University of Oxford and the Wellcome Trust Sanger Institute. https://www.malariagen.net/projects/parasite/pf3k This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/\\tDT:2015-06-09T00:00:00+0100\\tSM:ERS740935\\tPG:BamIndexDecoder\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/16503_1#1.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/16503_1#1.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Kilifi</td>\n",
        "<td>KE01</td>\n",
        "<td>None</td>\n",
        "<td>Kenya</td>\n",
        "<td>E Africa</td>\n",
        "<td style='text-align: right'>22</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>250bp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>100/250</td>\n",
        "<td>Validation</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/cram/16503_1#3.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#3.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:1#3\\tPL:ILLUMINA\\tPU:150609_MS2_16503_A_MS2920703-500V2_1#3\\tLB:14013612\\tDS:ERP009847: Pf3k is a global collaboration using the latest sequencing technologies to provide a high-resolution view of natural variation in the malaria parasite Plasmodium falciparum. The Project is led by researchers at the Broad Institute, the University of Oxford and the Wellcome Trust Sanger Institute. https://www.malariagen.net/projects/parasite/pf3k This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/\\tDT:2015-06-09T00:00:00+0100\\tSM:ERS740937\\tPG:BamIndexDecoder\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/16503_1#3.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/16503_1#3.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>KH2</td>\n",
        "<td>KH02</td>\n",
        "<td>None</td>\n",
        "<td>Camdodia</td>\n",
        "<td>SE Asia</td>\n",
        "<td style='text-align: right'>22</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>250bp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>250</td>\n",
        "<td>Validation</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/cram/16503_1#2.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#2.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:1#2\\tPL:ILLUMINA\\tPU:150609_MS2_16503_A_MS2920703-500V2_1#2\\tLB:14013611\\tDS:ERP009847: Pf3k is a global collaboration using the latest sequencing technologies to provide a high-resolution view of natural variation in the malaria parasite Plasmodium falciparum. The Project is led by researchers at the Broad Institute, the University of Oxford and the Wellcome Trust Sanger Institute. https://www.malariagen.net/projects/parasite/pf3k This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/\\tDT:2015-06-09T00:00:00+0100\\tSM:ERS740936\\tPG:BamIndexDecoder\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/16503_1#2.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/16503_1#2.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Gabon</td>\n",
        "<td>GA01</td>\n",
        "<td>None</td>\n",
        "<td>Gabon</td>\n",
        "<td>W Africa</td>\n",
        "<td style='text-align: right'>20</td>\n",
        "<td style='text-align: right'>8</td>\n",
        "<td>250bp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>250</td>\n",
        "<td>Training</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/cram/16503_1#5.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#5.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:1#5\\tPL:ILLUMINA\\tPU:150609_MS2_16503_A_MS2920703-500V2_1#5\\tLB:14013614\\tDS:ERP009847: Pf3k is a global collaboration using the latest sequencing technologies to provide a high-resolution view of natural variation in the malaria parasite Plasmodium falciparum. The Project is led by researchers at the Broad Institute, the University of Oxford and the Wellcome Trust Sanger Institute. https://www.malariagen.net/projects/parasite/pf3k This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/\\tDT:2015-06-09T00:00:00+0100\\tSM:ERS740939\\tPG:BamIndexDecoder\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/16503_1#5.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/16503_1#5.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Guinea</td>\n",
        "<td>GN01</td>\n",
        "<td>None</td>\n",
        "<td>Guinea</td>\n",
        "<td>W Africa</td>\n",
        "<td style='text-align: right'>51</td>\n",
        "<td style='text-align: right'>10</td>\n",
        "<td>250bp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>250</td>\n",
        "<td>Validation</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/cram/16503_1#6.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#6.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:1#6\\tPL:ILLUMINA\\tPU:150609_MS2_16503_A_MS2920703-500V2_1#6\\tLB:14013615\\tDS:ERP009847: Pf3k is a global collaboration using the latest sequencing technologies to provide a high-resolution view of natural variation in the malaria parasite Plasmodium falciparum. The Project is led by researchers at the Broad Institute, the University of Oxford and the Wellcome Trust Sanger Institute. https://www.malariagen.net/projects/parasite/pf3k This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/\\tDT:2015-06-09T00:00:00+0100\\tSM:ERS740940\\tPG:BamIndexDecoder\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/16503_1#6.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/16503_1#6.fastq</td>\n",
        "</tr>\n",
        "<tr>\n",
        "<td>Pf3D7_II</td>\n",
        "<td>Pf3D7_II</td>\n",
        "<td>None</td>\n",
        "<td>Amsterdam</td>\n",
        "<td>Africa/ Europa</td>\n",
        "<td style='text-align: right'>23</td>\n",
        "<td style='text-align: right'>6</td>\n",
        "<td>250bp</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td>None</td>\n",
        "<td style='text-align: right'>250</td>\n",
        "<td>None</td>\n",
        "<td>/nfs/pathogen003/tdo/Pfalciparum/PF3K/Reference12Genomes/Sharing_25Jun2015/cram/16503_1#4.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.sam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.sorted.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.dedup.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.dedup.metrics</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.dedup.targets.list</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.realigned.bam</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.post_recal.table</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.recal.pdf</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/bams/bwa_mem/16503_1#4.bwa_mem.recal.bam</td>\n",
        "<td>@RG\\tID:1#4\\tPL:ILLUMINA\\tPU:150609_MS2_16503_A_MS2920703-500V2_1#4\\tLB:14013613\\tDS:ERP009847: Pf3k is a global collaboration using the latest sequencing technologies to provide a high-resolution view of natural variation in the malaria parasite Plasmodium falciparum. The Project is led by researchers at the Broad Institute, the University of Oxford and the Wellcome Trust Sanger Institute. https://www.malariagen.net/projects/parasite/pf3k This data is part of a pre-publication release. For information on the proper use of pre-publication data shared by the Wellcome Trust Sanger Institute (including details of any publication moratoria), please see http://www.sanger.ac.uk/datasharing/\\tDT:2015-06-09T00:00:00+0100\\tSM:ERS740938\\tPG:BamIndexDecoder\\tCN:SC</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/vcfs/gvcf/samples/16503_1#4.cram</td>\n",
        "<td>/nfs/team112_internal/production_files/Pf3k/methods/assembled_samples/fastq/16503_1#4.fastq</td>\n",
        "</tr>\n",
        "</tbody>\n",
        "</table>\n"
       ],
       "metadata": {},
       "output_type": "display_data"
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "heading",
     "level": 1,
     "metadata": {},
     "source": [
      "Functions"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%%file first_last_100bp.py\n",
      "import sys\n",
      "from Bio import SeqIO\n",
      "\n",
      "iterator = SeqIO.parse(sys.stdin, \"fastq\")\n",
      "\n",
      "for firstRead in iterator:\n",
      "    firstRead_first100 = firstRead[0:100]\n",
      "    firstRead_first100.id = firstRead_first100.id.replace('#', '_first100#')\n",
      "    firstRead_first100.description = ''\n",
      "    firstRead_last100 = firstRead[150:250]\n",
      "    firstRead_last100.id = firstRead_last100.id.replace('#', '_last100#')\n",
      "    firstRead_last100.description = ''\n",
      "    secondRead = next(iterator)\n",
      "    secondRead_first100 = secondRead[0:100]\n",
      "    secondRead_first100.id = secondRead_first100.id.replace('#', '_first100#')\n",
      "    secondRead_first100.description = ''\n",
      "    secondRead_last100 = secondRead[150:250]\n",
      "    secondRead_last100.id = secondRead_last100.id.replace('#', '_last100#')\n",
      "    secondRead_last100.description = ''\n",
      "    SeqIO.write(firstRead_first100, sys.stdout, \"fastq\")\n",
      "    SeqIO.write(secondRead_first100, sys.stdout, \"fastq\")\n",
      "    SeqIO.write(firstRead_last100, sys.stdout, \"fastq\")\n",
      "    SeqIO.write(secondRead_last100, sys.stdout, \"fastq\")\n",
      "\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Overwriting first_last_100bp.py\n"
       ]
      }
     ],
     "prompt_number": 9
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def create_fastq(input_fn = tbl_samples_to_process.values('bam_fn')[0],\n",
      "                   fastq_fn = tbl_samples_to_process.values('fastq_fn')[0],\n",
      "#                    read_group_info = tbl_samples_to_process.values('RG')[0],\n",
      "#                    reference_fasta = REF_GENOME,\n",
      "                   rewrite=False):\n",
      "    \n",
      "    if not os.path.isfile(fastq_fn) or rewrite:\n",
      "        if input_fn.endswith('.bam'):\n",
      "            !{samtools_exe} bamshuf -uOn 128 {input_fn} tmp | \\\n",
      "            {samtools_exe} bam2fq - \\\n",
      "            > {fastq_fn} 2> /dev/null\n",
      "        elif input_fn.endswith('.cram'):\n",
      "            !{samtools_exe} view -b {input_fn} | \\\n",
      "            {samtools_exe} bamshuf -uOn 128 - tmp | \\\n",
      "            {samtools_exe} bam2fq - \\\n",
      "            > {fastq_fn} 2> /dev/null\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 12
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "create_fastq()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 13
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "create_fastq(tbl_samples_to_process.values('bam_fn')[8], tbl_samples_to_process.values('fastq_fn')[8])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 14
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}