{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Current plans for 2016\n",
    "\n",
    "\n",
    "## Copy all data to MacBook and create raw data files\n",
    "- Create _standard_imports (based on Docker), plotting_setup and _shared_setup (VCF_FNS, NPY_FNS, HDF5_FNS, GENOME_FASTA_FNS, etc.) in notebook 2016\n",
    "- Notebook to rsync all relevant data (3/4/5 and 5 samples vcfs, fasta, gff, mask bed, snpeff)\n",
    "- Notebook to merge vcfs, create npy and hdf5 files\n",
    "\n",
    "\n",
    "## Find problems with Interim 5 and create Interim 5.1\n",
    "- Create scatter plots of VQSLOD and input metrics of Release 4 vs Interim 5\n",
    "- Separate plots for training and other SNPs for all input variable and for VQSLOD\n",
    "- Breakdown of missingness in all input variables (ReadPosRankSum presumably very high?)\n",
    "- Do different peaks in VQSLOD scores correspond to presence/lack of ReadPosRankSum?\n",
    "- Redo VQSR without ReadPosRankSum - improvement in distribution of VQSLOD scores?\n",
    "- Manually run VQSR on Release 4 and Interim 5 call sets, with and without ReadPosRankSum. Improvement in VQSLOD scores? Still discrepancy between 4 and 5?\n",
    "- Run VQSR with GATK hard thresholds as training set (excluding RankSum variables) for 4 and 5\n",
    "- Manually run VQSR on intersection of SNPs between Release 4 and Interim 5 call sets\n",
    "- Run VQSR on intersection SNPs on subsets of input parameters\n",
    "- Once key problems fixed, manually create new vcf file, and give to Joe as interim\n",
    "\n",
    "\n",
    "## Build release 5 candidate\n",
    "- Follow up with Valentin about converting between haploid and diploid calls. Are these simple functions of PL? Is it possible to covert diploid to haploid, but not vice-versa?\n",
    "- Check with Cinzia on status of 16 lab samples\n",
    "- Add StrandAlleleCountsBySample annotation to HaplotypeCaller step. Check that this does indeed give filtered depths (i.e. always same or lower than AD).\n",
    "- Run with GATK 3.5+ (in haploid mode?)\n",
    "- Ensure we now have a clear merging strategy\n",
    "- Implement changes identified to fix VQSLOD problems\n",
    "- Create release 5 candidate inc 16 samples if ready\n",
    "\n",
    "\n",
    "## Single call set evaluation machinery\n",
    "- Sensitivity and FDR using “nucmer” truth set\n",
    "- Sensitivity and FDR using “unclustered nucmer” truth set\n",
    "- Sensitivity and FDR using “bwa/GATK” truth set\n",
    "- Mendelian error rate(s) in crosses\n",
    "- Genotype error rate(s) in mixtures\n",
    "- Duplicate discordance rate(s)\n",
    "- Mean heterozygosity or FWS in clonal lab samples\n",
    "- Ti/Tv ratio (SNP coding/noncoding only)\n",
    "- % frameshifts (indel coding only)\n",
    "- % coding/noncoding genome accessible\n",
    "\n",
    "\n",
    "## Evaluation metrics for release 3, 4 and 5\n",
    "- Run above evaluation metrics on all vcfs\n",
    "- Present results back to group\n",
    "\n",
    "\n",
    "## Release 5 candidate deltas\n",
    "- Inclusion of hyperhet filter\n",
    "- Core definition based on dust score and uniqueness\n",
    "- Genotyping by GATK GT, 5/2 rule, 5/3 rule\n",
    "- Run above evaluation metrics on all vcfs\n",
    "- Present results back to group\n",
    "\n",
    "\n",
    "## Two call set comparison machinery\n",
    "- Scatter of variant metrics (VQSLOD and other) for intersection variants in both call sets\n",
    "- Above metrics for variant intersection and setdiffs\n",
    "- Above metrics where callset 2 is re-thresholded to have same number of variants as callset 1\n",
    "- Histograms of setdiff variant metrics\n",
    "- Scatter of heterozygosity, missingness, etc for samples in both call sets\n",
    "- Comparisons of discordant unfiltered calls for 5 validation samples\n",
    "- Run above evaluation metrics on all vcfs\n",
    "- Present results back to group\n",
    "\n",
    "\n",
    "## Create evaluation samples set\n",
    "- To include two versions of 5 lab strains, crosses, mixtures and ~100 mixed field samples (diff read lengths)\n",
    "- Run above evaluation metrics on release 5 vs evaluation samples\n",
    "- Present results back to group\n",
    "\n",
    "\n",
    "## Explore use of filtered allele depths\n",
    "- If response at http://gatkforums.broadinstitute.org/gatk/discussion/comment/25870/#Comment_25870 add StrandAlleleCountsBySample to future builds\n",
    "- If not, re-create 5 samples call set\n",
    "\n",
    "\n",
    "## Decision on Docker strategy\n",
    "- Image on Docker hub that covers all the above\n",
    "- Discuss with Nick and Alistair forward Docker strategy\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.0+"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
